---
title: "Log"
subtitle: "Start Date: 2019-05-20"
author: "Christopher Tastad"
date: "Last Edit: `r format(Sys.time(), '%Y-%m-%d')`"
bibliography: library.bib
output:
    html_document:
        toc: yes
        toc_float: yes
        theme: paper
---


<!-- Style -->

<style type="text/css">

h1 {
  color: #333333;
}
h2 {
  color: #333333;
}
h3 {
  color: #333333;
}
h4 {
  color: #333333;
}
</style>

<!-- Content -->

```{r global_options, include=F}
knitr::opts_chunk$set(eval=F)
```

***

# Readme {.tabset .tabset-fade}

## Preamble

**Copyright**

This is the notebook of Christopher Tastad, a lab member of Tim Starr at the University of Minnesota. All considerations for ownership follow the discretion of the Starr Lab at the extension of the policies of the University of Minnesota.

* https://sites.google.com/a/umn.edu/starrlab/
* https://policy.umn.edu/research/copyright

**Layout**

Homepage (go here first)

* `notebook/_site/index.html`

This notebook was written in Rmarkdown and rendered using an automated knit script. For information on the implementation of this, refer to the "Knit" tab.

The layout is generally centered here on the `Log`, which is meant to be a full continuum of work and notation completed while in the Starr Lab. Some elements from the `Log` been reiterated on dedicated pages as special points of reference when a relevant context is needed such as a repeatedly used `method` or notable `results`. References used on each page are displayed at the bottom, and all references used across the notebook and bibliography at large (library.bib located at `~/ocpmi/notebook/`) are collated on the `References` page.

### Important!!

I did a major restructuring of my project directory on *2019-08-08*. I did update dir pointers on the `methods` page to the fullest extent possible. Many historic dir pointers in the log from *2019-08-08* and back are no longer correct and will not be updated. Please look at the "Dir Structure" tab to see the general paradigm going forward from this date.

***

## Dir Structure

```{r}
ocpmi/
├── data # location of any experimental or source dataset content
│   ├── bam
│   ├── gene_ref # gene reference sources for all pipelines
│   └── pt59
├── methods # code, packages, and source files for all pipelines
│   ├── EXCAVATOR2_Package_v1.1.2
│   ├── jason_cnv
│   ├── pbs # job scripts for msi
│   ├── seurat
│   ├── test_script.R
│   ├── tim
│   ├── varscan2
│   └── XCAVATOR
├── notebook # source files for project notebook
│   ├── csv
│   ├── ieee-with-url.csl
│   ├── index.Rmd # log source file
│   ├── library.bib # bibtex master reference citation file
│   ├── methods.Rmd # methods source file
│   ├── references.Rmd
│   ├── results.Rmd
│   ├── _site # notebook site dir - homepage location**
│   ├── _site.yml # yaml file for site knit
│   └── tmp
├── reference # reference documents
│   ├── background
│   ├── dir_res_proposal
│   ├── gene_markers
│   ├── presentations
│   ├── single_cell
│   └── wes
└── results # output for any code or pipeline
    ├── excavator2
    ├── seurat
    └── varscan2
```

***

## Knit

If placed in the `notebook/` directory, this script will render the complete set of webpages contained in the notebook from their &ast;.Rmd source to &ast;.html. With slight modification to the `setwd` function and addition to the \$PATH system variable, this function can be run as an automated system binary.

```{r}
#!/usr/bin/env Rscript

cur_dir <- getwd()
setwd("./")

rmarkdown::render("index.Rmd")
rmarkdown::render("results.Rmd")
rmarkdown::render("methods.Rmd")
rmarkdown::render("references.Rmd")
rmarkdown::render_site("index.Rmd")

fl <- Sys.glob("*.html")

ifelse(file.exists(fl), file.remove(fl))

setwd(cur_dir)
```

***

## Git

Version control was used throughout the development of this work and can be found at the University's private github server:

* https://github.umn.edu/tasta005/ocpmi.git

***

## Contact

Christopher Tastad, MS Candidate

* University: tasta005@umn.edu
* Personal: ctastad@gmail.com
* Mobile: 612-817-7989

***

# May

## 20

### Background mtg w/ Tim

* present: chris, tim

We discussed the outlook for my role in the lab. This included a considerable amount of background on the ocpmi project and potential directions I could take.

Focus Options

* ID of ovca stem cell
* what is the basis for clustering
* single cell seq methods paper
* compare mouse to human
* copy number variation on single cell
* rerun new version of cell ranger
* RHEL update

***

## 21

### TODO

```{r}
ID  Age   Project        Tags       R Due        Description                Urg
--- ----- -------------- ---------- - ---------- -------------------------- ----
326 10h   starrlab       computer n R 2019-05-21 write daily summary        10.5
327 27min starrlab       computer n              find umn git and hosting    1.9
312 20h   starrlab.ocpmi computer n              organize lab notebook [1]   2.7
315 18h   starrlab.ocpmi computer n              find ovca review re cur     1.9
                                                 treatment
316 18h   starrlab.ocpmi computer n              read starr paper re 1st pt  1.9
317 18h   starrlab.ocpmi computer n              read ocpmi summary          1.9
318 18h   starrlab.ocpmi computer n              install surat               1.9
319 18h   starrlab.ocpmi computer w              run surat tutorial w/ pt    1.9
                                                 59
320 18h   starrlab.ocpmi computer n              read single cell paper      1.9
                                                 zheng 2017
321 18h   starrlab.ocpmi computer n              find and read copy num      1.9
                                                 variation paper
322 18h   starrlab.ocpmi computer n              organize refs in email      1.9
                                                 inbox
323 18h   starrlab.ocpmi computer n              compile bibtex ref page     1.9

12 tasks
```


### Citation packages

#### Rmarkdown Default

This is the intrinsic formatting style for Rmarkdown citation management. It has the ability to call a BibTex file and carries the least effort to get it working. It's what I will probably go for across my whole setup. [@citations]

#### knitcitations

The knitcitations package might provide the best balance of feature set to ease of use. The only constraining factor is what seems like a few bugs that limit intended functionality. The ref() function doesn't work for instance. [@cboettig_2019]

#### RefManageR

This is a fully featured R package that seems to attempt to replace a complete citation manager. It comes across as very comprehensive but more complicated than I would like the approach to be. [@refmanager]

### Summary

#### Completed

* 327, 312:progress

The majority of the day was spent exploring the setup for my lab notebook. My intention is to create a living document using the Rmarkdown website knit feature [@rmarkdown_cheatsheet]. This will allow for a dynamic storage of both records and code work inline with one comprehensive document. A large portion of this time was spent giving consideration to how I will handle citations both in the document and for future publication. After significant exploration, my plan is to use the intrinsic Rmarkdown system [@citations] through use of a bibtex file. This will allow for simple reference in-place with the notebook along with using a standardized format for publication ready references.
Additionally, I plan on establishing a git repo to manage version control and historical record with the notebook as well using the umn github server [citation needed].

***

## 22

### TODO



```{r}
ID  Age Project        Tags       R Due        Description                  Urg
--- --- -------------- ---------- - ---------- ---------------------------- ----
326 15h starrlab       computer n R 2019-05-22 write daily summary          10.5
312 1d  starrlab.ocpmi computer n              organize lab notebook [1]    2.71
315 1d  starrlab.ocpmi computer n              find ovca review re cur      1.91
                                               treatment
316 1d  starrlab.ocpmi computer n              read starr paper re 1st pt   1.91
317 1d  starrlab.ocpmi computer n              read ocpmi summary           1.91
318 1d  starrlab.ocpmi computer n              install surat                1.91
319 1d  starrlab.ocpmi computer w              run surat tutorial w/ pt 59  1.91
320 1d  starrlab.ocpmi computer n              read single cell paper zheng 1.91
                                               2017
321 1d  starrlab.ocpmi computer n              find and read copy num       1.91
                                               variation paper
322 1d  starrlab.ocpmi computer n              organize refs in email inbox 1.91
323 1d  starrlab.ocpmi computer n              compile bibtex ref page      1.91

11 tasks
```


### wkly mtg w/ Tim

present: chris, tim

* Make sure to utilize notes field in bibtex file. Having an ongoing record of each paper read is critical.
* Consider separating data and results in file organization.
* Change notebook structure to follow layout of paper. Organize the work with a mindset toward writing the paper from the start, not the finish.
* Currently, we are still working to establish the standardized parameters of the data set. The goal is to arrive at a place where we can make biological conclusions.
    + How do we draw contrast from a data set of this nature?

### Notebook Structure

In my conversation with Tim, I decided I need to take an alternate structure to my lab notebook. My original intent was to emulate a development oriented documentation process. Instead, I will go a route that better follows the academic writing process with pages dedicated to methods, results, etc.

### Reference Management

While I would prefer to take a lean, text based approach to this, I am clearly beginning to waste some time in gear the system. I came across jabref, which seems to leverage a unix philosophy and can be found in the ubuntu repos.

### Summary

#### Completed

* 312:progress, 323:progress

My work today included continued focus on getting organized an taking a very deliberate approaching to my workflow before I begin. I met with Tim to begin our weekly meetings, and we covered some aspects of this workflow. Major take aways include the importance of annotation when reviewing literature. I had originally assumed I could handle references processing merely in a bare bibtex file. My feeling now is that I will begin to use a dedicated application for greater consistency in documentation.

***

## 23

### TODO



```{r}
ID  Age Project        Tags       R Due        Description                  Urg
--- --- -------------- ---------- - ---------- ---------------------------- ----
334 11h starrlab       computer n R 2019-05-23 write daily summary          10.6
327 23h starrlab       computer n              reorg notebook to follow      1.9
                                               paper structure
329 19h starrlab       computer n              setup jabref                  1.9
330 19h starrlab       lab n                   setup desk                    1.9
331 19h starrlab       computer n              create agenda field           1.9
332 19h starrlab       computer n              create lit tag                1.9
333 19h starrlab       computer n              create lab context            1.9
336 41s starrlab       computer n              create website knit script    1.9
312 2d  starrlab.ocpmi computer n              organize lab notebook [1]    2.71
315 2d  starrlab.ocpmi computer n              find ovca review re cur      1.91
                                               treatment
316 2d  starrlab.ocpmi computer n              read starr paper re 1st pt   1.91
317 2d  starrlab.ocpmi computer n              read ocpmi summary           1.91
318 2d  starrlab.ocpmi computer n              install surat                1.91
319 2d  starrlab.ocpmi computer w              run surat tutorial w/ pt 59  1.91
320 2d  starrlab.ocpmi computer n              read single cell paper zheng 1.91
                                               2017
321 2d  starrlab.ocpmi computer n              find and read copy num       1.91
                                               variation paper
322 2d  starrlab.ocpmi computer n              organize refs in email inbox 1.91
323 2d  starrlab.ocpmi computer n              compile bibtex ref page      1.91

18 tasks
```


### Citations

#### Custom CSL

At some point, I would like to setup a custom csl file to format my references. The goal here would be to add formatted features that are note readily available in some existing format style. [@2019], [@BibEntry2019May]

#### JabRef

I've started to figure out jabref. I think I will still look to my bibtex file as the primary source for citation management, but will make use of the interface and some of the features of jabref. Additionally dio2bib will play a central role in extracting bibtex content. [@2019b]

### GitHub

The setup for my git server is complete. It's a private server host by the university a /tasta005/ocpmi [@ocpmi_git]

### Summary

I was able to round out much of my organizational setup today. I feel confident about the citation management strategy I plan to use going forward which combines the use of dio2bib for bibtex creation, BibItNow for website extraction, and JabRef to centralize management. Additionally, I was able to connect my relevant directories to my git server, which will account for both a measure of backup and version control. Having gotten settled into these systems I think I will be able to begin incorporating my literature search into my daily activities.

#### Completed


```{r}
Created    Completed  Age Project        R Due        Description
---------- ---------- --- -------------- - ---------- --------------------------
2019-05-22 2019-05-23 1d  starrlab       R 2019-05-22 write daily summary
2019-05-22 2019-05-23 1d  starrlab                    setup desk
2019-05-23 2019-05-23 16h starrlab       R 2019-05-23 write daily summary
2019-05-20 2019-05-23 3d  starrlab.ocpmi              organize lab notebook
                                                        2019-05-21 record,
                                                      master, comms, refs, home,
                                                      docs
2019-05-22 2019-05-23 1d  starrlab                    create lab context
2019-05-22 2019-05-23 1d  starrlab                    create lit tag
2019-05-22 2019-05-23 1d  starrlab                    setup jabref
2019-05-20 2019-05-23 3d  starrlab.ocpmi              compile bibtex ref page
2019-05-20 2019-05-23 3d  starrlab.ocpmi              organize refs in email
                                                      inbox
2019-05-22 2019-05-23 1d  starrlab                    setup umn git
2019-05-22 2019-05-23 1d  starrlab                    create agenda field

11 tasks
```

***

## 24

### TODO


```{r}
ID  Age Project        Tag            Description                           Urg
--- --- -------------- -------------- ------------------------------------- ----
314 3d  starrlab.ocpmi computer lit n find ovca review re cur treatment     2.02
315 3d  starrlab.ocpmi computer lit n read starr paper re 1st pt            2.02
316 3d  starrlab.ocpmi computer lit n read ocpmi summary                    2.02
319 3d  starrlab.ocpmi computer lit n read single cell paper zheng 2017     2.02
320 3d  starrlab.ocpmi computer lit n find and read copy num variation      2.02
                                      paper
317 3d  starrlab.ocpmi computer n     install surat                         1.92
318 3d  starrlab.ocpmi computer w     run surat tutorial w/ pt 59           1.92
325 1d  starrlab       computer n     create website knit script            1.91
327 22h starrlab       comp w         make custom csl                        1.9
328 22h starrlab       lab n          complete lab training                  1.9
330 20h starrlab       w              agenda items                           1.8

11 tasks
```

### create_site Script

I created a small R script that will effectively knit each page of my lab notebook. This can be executed in the R console or in bash, but each file has to be saved. The save -> knit function that's performed by the IDE does not apply to the console executed render command.

```{r}
rmarkdown::render("index.Rmd")
rmarkdown::render("notes.Rmd")
rmarkdown::render("results.Rmd")
rmarkdown::render("methods.Rmd")
rmarkdown::render("references.Rmd")
rmarkdown::render_site("index.Rmd")
```


### Summary

Today was filled with primarily administrative work. A large bulk of my time was taken by lab safety training, which I was unable to fully complete. Along with this, I was finally able to install Seurat and tinker with the package tutorial. It seems like the version I downloaded may not be compatible with the tutorial code, so I will have to explore that more.

#### Completed


```{r}
Created    Completed  Age Project        Description
---------- ---------- --- -------------- ---------------------------------------
2019-05-23 2019-05-24 1d  starrlab       create website knit script
2019-05-24 2019-05-24 2h  starrlab       DEHS Radiation Safety Orientation
                                         Training
2019-05-24 2019-05-24 2h  starrlab       Implementation of NIH Guidelines
2019-05-24 2019-05-24 2h  starrlab       Bloodborne Pathogens: Advanced
2019-05-24 2019-05-24 2h  starrlab       DEHS Chemical Safety Training
2019-05-24 2019-05-24 2h  starrlab       Biological Safety in the Laboratory
2019-05-24 2019-05-24 2h  starrlab       Bloodborne Pathogens: Introduction and
                                         Advanced
2019-05-24 2019-05-24 2h  starrlab       DEHS Introduction to Research Safety
                                         Training
2019-05-24 2019-05-24 2h  starrlab       Animal Use Tutorial
2019-05-23 2019-05-24 1d  starrlab       complete lab training
2019-05-20 2019-05-24 4d  starrlab.ocpmi install surat

11 tasks
```

***

## 27

### TODO


```{r}
ID Age Project        Tags           Description                            Urg
-- --- -------------- -------------- -------------------------------------- ----
77 4d  starrlab       computer w     make custom csl                        1.92
79 3d  starrlab       computer n     GCP Human RSRCH Biomed Studies         1.92
80 3d  starrlab       computer n     CITI research ethics training          1.92
81 3d  starrlab       computer n     University Information Security        1.92
                                     Awareness Training
78 4d  starrlab                      agenda items                           1.02
91 15h starrlab.ocpmi computer lit n read papalexi 2018 [2]                  2.9
68 6d  starrlab.ocpmi computer lit n find ovca review re cur treatment      2.03
69 6d  starrlab.ocpmi computer lit w read starr paper re 1st pt             2.03
70 6d  starrlab.ocpmi computer lit w read ocpmi summary                     2.03
72 6d  starrlab.ocpmi computer lit w read single cell paper zheng 2017      2.03
73 6d  starrlab.ocpmi computer lit w find and read copy num variation paper 2.03
71 6d  starrlab.ocpmi computer w     run surat tutorial w/ pt 59            1.93

12 tasks
```


### @Lheureux2019

#### Evolution of Management in the Era of Precision Medicine  (1/2)

* State of field for ovarian cancer
* Current treatments and outcomes
* Ovca presents very late. The internal location of the cancer and general lack of screening options lead to most cases being diagnosed at a later stage than many other cancers would.
* Treatment options have largely focused on surgical debulking and platinum based chemotherapy.
* Risk factors are inherited: germ line mutations in breast cancer susceptible genes (e.g. brca), nulliparity, infertility, endometriosis, obesity, age, potentially perineal talc.
* Reducing factors that have been found to be potentially significant are gravidity, oral contraceptive use, regular use of aspirin, and nsaids.
* Currently, population screening in ovca has not been found to be effective.
* Epithelial ovarian cancer accounts for 95% of ovarian malignancies
* 5 eoc subtypes: high-grade serous, low-grade serous, clear cell, endometrioid, and mucinous
* HGSOC has the highest incidence with 70% of eoc cases.
* It is believed fallopian tube epithelium is the possible site for tumor initiation.
* Mutations in p53 are seen in +90% cases of HGSOC
* p53 mutations and brca inactivation lead to high numbers of copy variation.
* HGSOC has 4 prognostic subtypes: differentiated, immunoreactive, mesenchymal, and proliferative
* Immunoreactive is considered to have the best overall survival while mesenchymal has the worst.
* Ovca cells also over express folate and mesothelin which have been targets for antibody therapy.
* LGSOC is seen in about 10% of cases
* LGSOC is considered to develop in a pathway which is very distinct from HGSOC
* B-Raf and KRAS mutations are the most commonly present in LGSOC
* Hormone therapy is sometimes used due to the presence of estrogen/progesterone receptors.
* CNAs in p53 are rare, and brca inactivation seem to unrelated to LGSOC

### Summary

Just worked my way through the above summary today. Will continue and complete reading this lit tomorrow.


***

## 28

### TODO


```{r}
ID Age Project        Tags           Description                            Urg
-- --- -------------- -------------- -------------------------------------- ----
77 4d  starrlab       computer w     make custom csl                        1.92
79 3d  starrlab       computer n     GCP Human RSRCH Biomed Studies         1.92
80 3d  starrlab       computer n     CITI research ethics training          1.92
81 3d  starrlab       computer n     University Information Security        1.92
                                     Awareness Training
78 4d  starrlab                      agenda items [1]                       1.82
91 1d  starrlab.ocpmi computer lit n read papalexi 2018 [2]                 2.91
92 17h starrlab.ocpmi computer lit w read macintyre 2018 [2]                 2.9
68 7d  starrlab.ocpmi computer lit n find ovca review re cur treatment      2.04
69 7d  starrlab.ocpmi computer lit w read starr paper re 1st pt             2.04
70 7d  starrlab.ocpmi computer lit w read ocpmi summary                     2.04
72 7d  starrlab.ocpmi computer lit w read single cell paper zheng 2017      2.04
73 7d  starrlab.ocpmi computer lit w find and read copy num variation paper 2.04
71 7d  starrlab.ocpmi computer w     run surat tutorial w/ pt 59            1.94

13 tasks
```


### [@Lheureux2019]

#### Evolution of Management in the Era of Precision Medicine (2/2)

* OCCC or clear cell carcinoma represents about 5% of cases.
* The pathogenesis of OCCC is not well understood.
* A proposed mechanism is related to the level of iron content, promoting oxidative stress.
* Mutations in the SWI/SNF and PI3K pathway are represented in large proportions of cases.
* Activation of the PI3K or RTK pathway has resulted in better OS
* OCCCs are found to have considerably fewer CNAs and mutations in p53 or brca genes.
* Endometrioid cancers represent 10% of EOC cases.
* Current thought is that endometrioid and OCCC arise from similar pathways.
* Deregulation of WNT/b-catenin and PI3K pathways are essential in mouse models.
* Mucinous OC is a rare subtype.
* Could be classified as non-invasive and infiltrative.
* KRAS mutations are a common feature of this subtype.
* p53 and brca mutations do not seem to play a role.
* Treatment
* To date, there has been no randomized assessment of debulking surgery, platinum therapy, or radiation therapy.
* Surgery has been the bedrock of therapy.
* Surgery involves many approaches including primary, interval, and secondary debulking
* Chemotherapy
* carboplatin is as effective as cisplatin and is better tolerated.
* dose is important.
* pairing drugs does not improve outcomes.
* IP chemo is often used due to the tight localization of ovca in the abdomen
* hyperthermic IP chemo is currently being explored with reasonable results
* Treatment after recurrence follows several judgments of benefit.
* Recurrence after a period of over 6mo suggests platinum sensitivity where platinum based chemo would be reapplied
* Shorter intervals of recurrence employ non-platinum based therapies
* Current consensus is the use of carboplatin and paclitaxel together every 3wks
* Targeted therapy has been gradually introduced in recent years.
* Bevacizumab has shown the greatest promise among antiangiogenics being the only one in clinical use.
* PARP inhibitors
* Used as maintenance with platinum therapy at the time of recurrence.
* A trial showed that the key benefit in PFS was driven by the parp inhibitor maintenance therapy.
* Trial from 2018 led to first-line maintenance in brca mutated cases
* Stratified approach to systemic therapy can be used with bevacizumab after surgery with a switch to maintenance on the basis of brca mutations.
* Immunotherapy is currently being explored on several fronts though none are in clinical practice.
* ovca has a high expression of PD-L1.
* Checkpoint inhibitors targeting PD-L1/PD1 or CTLA-4 have shown only modest results.
* Autologous cell therapy is also being utilized.
* A variation of this therapy is exploring car-t cells targeting mesothelin
* Vaccines have not worked well due to the highly immunosuppressive cancer environment
* A vaccine targeting survivin is currently being explored
* Additionally, the concept of the patient-specific mutanome is being researched in an effort to develop patient-specific cancer vaccines.
* OCCC and LGSOC are relatively chemoresistant.
* Prevention and screening are currently limited.
* There is no validated screening method for standard practice.
* Ultrasound has produced false-negatives and false-positives and is not recommended.
* Screening high-risk populations has been shown to be effective but requires prior knowledge of family history.
* Some biomarkers have been utilized such as HE4 in combination with CA125.
* Another approach has been to evaluate DNA methylation in cfDNA.










### Summary

I have shifted focus to begin a comprehensive literature review. I've started by taking a look at a review of the current landscape of treatment and outcomes for ovca [@Lheureux2019]. This was helpful to fill many of the holes I have in my general knowledge of cancer biology with the more specific case studies of ovca. Much of this reinforced the review that Tim has given in our meetings, which was helpful. Along with this, I began to work my way through the Seurat tutorial. I was able to locate the v3 tutorial and pbmc data set and made it about half way through the written guide. I plan on getting fully through this tomorrow as well as meeting with Tim to discuss some questions I developed from the ovca review.

#### Completed


```{r}
Created    Completed  Age Project        Description
---------- ---------- --- -------------- ---------------------------------
2019-05-20 2019-05-28 7d  starrlab.ocpmi find ovca review re cur treatment

1 task
```

***

## 29

### TODO

```{r}
ID Age  Project        Tags           Description                           Urg
-- ---- -------------- -------------- ------------------------------------- ----
75 5d   starrlab                      agenda items [3]                      2.03
74 6d   starrlab       computer w     make custom csl                       1.93
76 5d   starrlab       computer n     GCP Human RSRCH Biomed Studies        1.93
77 5d   starrlab       computer n     CITI research ethics training         1.93
78 5d   starrlab       computer n     University Information Security       1.93
                                      Awareness Training
90 1d   starrlab       computer n     find alternative to jabref            1.91
92 22h  starrlab       computer n     fix god damn git issue                 1.9
97 1min starrlab       computer n     fix hostname issue w/ mesabi           1.9
88 2d   starrlab.ocpmi computer lit n read papalexi 2018 [2]                2.91
89 1d   starrlab.ocpmi computer lit w read macintyre 2018 [2]               2.91
91 1d   starrlab.ocpmi lit s          read sahin 2017 [1]                   2.71
66 8d   starrlab.ocpmi computer lit w read starr paper re 1st pt            2.04
67 8d   starrlab.ocpmi computer lit w read ocpmi summary                    2.04
69 8d   starrlab.ocpmi computer lit w read single cell paper zheng 2017     2.04
70 8d   starrlab.ocpmi computer lit w read venteicher - CNV paper           2.04
68 8d   starrlab.ocpmi computer w     run surat tutorial w/ pt 59           1.94
96 2min starrlab.ocpmi computer n     run seurat with pbmc data              1.9

17 tasks
```


### wkly mtg w/ Tim

present: chris, tim

```{r}


Description
--------------------------------------------------------------------------------
agenda items
  2019-05-27 validation in diff between lumpectomy and mastectomy
  2019-05-28 elaborate on lack of randomized controls for ovca treatments
  2019-05-28 discuss vacation
  2019-05-28 discuss chemotherapy at large
  2019-05-28 review parp inhibitor mechanism
  2019-05-28 are occc and lgsoc our general targets given their chemoresistance?
  2019-05-28 how do our questions differ from the findings of kondrashova 2017 -
RAD51 mutations acquired resistance

1 task
```

#### mtg notes

* mysql reference: single cell rnaseq analysis protocol
* figure out mesabi hostname issue
* explore starrt2 MSI path
    + [/home/starrt2/data_release/umgc/hiseq/170627_D00635_0259_ACB8R1ANXX/Starr_Project_037/Analysis/cellranger-Starr_037/filtered_gene_bc_matrices/GRCh38]
* There is no specific focus in the pathology we're focusing on. OCCC and LGSOC are being considered just as readily as other subtypes.
* Kondrashova 2017 (RAD51 mutations leading to chemoresistance) is potentially a good example of what our research questions will be.
* Tim's focus is more basic science oriented. Potential applications would be to identify when a pt would be resistant and that they should pursue other treatment from the outset.
* All of our findings from this data set will require wetlab validation
* This data set is incredibly unique. We should be able to discover something nobody else has.
* Data validation has generally been handled on the project so far.
    + Objective: determine cell type.
* There's no way to call copy number variation from this data. That would be novel.

### MSI Project Directory

I logged into the MSI account for the first time today. There is a relatively deep file structure which is composed of a combination of processed data from umgc/MSI and other collaborator work. I have a personal directory in this space along with access to other shared directories.

The root project directory is:

```{r}
/home/starrt2/
```

I have a personal directory under the root. At the same level, there is also a shared directory which includes some content that is under active collaboration.

```{r}
/home/starrt2/tasta005
```

Processed data can be found in the hiseq folder under data_release. This data is mirrored in the MSI file structure. Each file set is named by a unique ID which follows a naming convention given by umgc.

* Look for the leading 6 digit integer as a point of reference for the data_release folder.
* The Starr_Project sub folder is given a number which is considered the MSI and sample ID.

```{r}
/home/starrt2/data_release/umgc/hiseq
```


The Analysis directory contains the source data

* illumina-basicQC # quality control data
* cellranger-`*`
    + analysis # ?
    + filtered_gene_bc_matricies # data source that has been filtered by "true cell content"
    + raw_gene_bc_matrices # matrix data that includes unfiltered reads or all read content
    + web_summary.html # interactive cell ranger summary

An example of a complete data source location is below:

```{r}
/home/starrt2/data_release/umgc/hiseq/170627_D00635_0259_ACB8R1ANXX/Starr_Project_037/Analysis/cellranger-Starr_037/filtered_gene_bc_matrices/GRCh38
```

### Summary

I met with Tim today for our weekly meeting. We discussed some aspects of ovca content that I had been reading to develop my general knowledge. I major element of this conversation that will adjust what my intentions were going forward dealt with the nature of statistically validating the results of our pipeline. My original thought was that there was some need to dig into the backend of the Seurat/cellranger packages to both confirm and optimize the output. It sounds as though this has been done to a degree that Tim is satisfied with and that we can move on to explore questions that are Biological in nature.

Additionally, I did a little house keeping by establishing the first part of the methods page in my notebook. I explored the MSI project directory for the first time today and recorded some documentation about the file structure and data locations. My expectation is to pull what would be method oriented documentation from Notes -> Methods or Notes -> Results as these things need to be pinned for repeat use and reference.

#### Completed

```{r}
Created    Completed  Age Project  Description
---------- ---------- --- -------- ----------------------------
2019-05-28 2019-05-29 1d  starrlab fix god damn git issue
2019-05-29 2019-05-29 2h  starrlab fix hostname issue w/ mesabi

2 tasks
```

***

## 30

### TODO


```{r}
ID Age Project        Tags           Description                            Urg
-- --- -------------- -------------- -------------------------------------- ----
75 6d  starrlab                      agenda items [3]                       2.03
74 6d  starrlab       computer w     make custom csl                        1.93
76 5d  starrlab       computer n     GCP Human RSRCH Biomed Studies         1.93
77 5d  starrlab       computer n     CITI research ethics training          1.93
78 5d  starrlab       computer n     University Information Security        1.93
                                     Awareness Training
90 1d  starrlab       computer n     find alternative to jabref             1.91
88 3d  starrlab.ocpmi computer lit n read papalexi 2018 [2]                 2.92
89 2d  starrlab.ocpmi computer lit w read macintyre 2018 [2]                2.91
91 1d  starrlab.ocpmi lit s          read sahin 2017 [1]                    2.71
66 9d  starrlab.ocpmi computer lit w read starr paper re 1st pt             2.05
67 9d  starrlab.ocpmi computer lit w read ocpmi summary                     2.05
69 9d  starrlab.ocpmi computer lit w read single cell paper zheng 2017      2.05
70 9d  starrlab.ocpmi computer lit w read venteicher - CNV paper            2.05
68 9d  starrlab.ocpmi computer w     run surat tutorial w/ pt 59            1.95
95 20h starrlab.ocpmi computer n     run seurat with pbmc data               1.9

15 tasks
```

### Seurat Tutorial pbmc3k

I ran the v3 Seurat tutorial [@2019d] to completion using the pbmc dataset. This outline was very intuitive and worked through the pipeline seamlessly. There are several points that warrant further exploration.

The output of the v3 PBMC data tutorial can be found [here](/home/chris/starr_lab/ocpmi/code/seurat_tutorial/pbmc3k_tutorial.html).


### Summary

Briefly, I wrapped up safety training today, so that is all taken care of. The rest of my afternoon was dedicated to working my way through the default v3 Seurat tutorial. The tutorial comes packaged in a markdown file, so the presentation and documentation is very effective. To some extent, the documentation is so good I am a little concerned it encourages over use of the default settings. I haven't looked at any other processing that's been completed by Tim et al, so I'll have to keep an eye out for how adjustments have been made. Tim mentioned there has been a limited amount of customization, so this may stand as a short term area of focus. Generally, I think can consider today's work a reasonable summary of the process and syntax associated with the toolset. With this, I would like to spend some time digging into the nuances of the pipeline's analysis. I will probably spend most of tomorrow doing that. Much of the default analysis hinges on PCA dimensionality, so I feel I need to brush up on my knowledge in this area. Going forward I need to run the same pipeline using the new pt59 data and evaluate results.

#### Completed


```{r}
Created    Completed  Age Project        Description
---------- ---------- --- -------------- ---------------------------------------
2019-05-24 2019-05-30 6d  starrlab       GCP Human RSRCH Biomed Studies
2019-05-24 2019-05-30 6d  starrlab       CITI research ethics training
2019-05-24 2019-05-30 6d  starrlab       University Information Security
                                         Awareness Training
2019-05-29 2019-05-30 1d  starrlab.ocpmi run seurat with pbmc data

4 tasks
```

***

## 31

### TODO

```{r}
ID Age Project        Tags       Description                                Urg
-- --- -------------- ---------- ------------------------------------------ ----
73 8d  starrlab                  agenda items [3]                           2.04
72 8d  starrlab       computer w make custom csl                            1.94
83 3d  starrlab       computer n find alternative to jabref                 1.92
91 1d  starrlab       computer n review pca concepts                        1.91
92 1d  starrlab       computer n compare default seurat tutorial to         1.91
                                 starrlab protocol
93 1d  starrlab       computer n deep dive into default seurat analysis     1.91
94 1d  starrlab       computer n update notes regarding citation tools re   1.91
                                 mendeley
81 4d  starrlab.ocpmi lit n      read papalexi 2018 [2]                     2.82
82 4d  starrlab.ocpmi lit w      read macintyre 2018 [2]                    2.82
84 3d  starrlab.ocpmi lit s      read sahin 2017 [1]                        2.72
65 11d starrlab.ocpmi lit w      read starr paper re 1st pt                 1.96
66 11d starrlab.ocpmi lit w      read ocpmi summary                         1.96
67 11d starrlab.ocpmi computer w run surat tutorial w/ pt 59                1.96
68 11d starrlab.ocpmi lit w      read single cell paper zheng 2017          1.96
69 11d starrlab.ocpmi lit w      read venteicher - CNV paper                1.96

15 tasks
```

### Mendeley

At the risk of seeming indecisive, I've decided to switch citation managers. Given my use if the BibTex file format, it is effectively painless to switch. JabRef showed some promise but presented a few bugs related to database calls which would render much of the value of using it worthless. Mendeley has both a native linux application along with a web import tool for firefox. This is the only major citation tool I've come across that combines those two things. Here are a few key features of my use:

* All content will still be synced back to a BibTex file name `library.bib` located under the `starr_lab/ocpmi/notebook/` subdirectory
* There are watched folders which will draw sources from any pdf put into them at `starr_lab/ocpmi/ref/`.
* I've decided to use the IEEE with url csl style taken from the zotero style repository. [@2019]

### [@Winterhoff2019]

#### Single-Cell Sequencing in Ovarian Cancer: a new frontier in precision Medicine

Key Points

* Intratumor heterogeneity is an inherent property of ovarian cancer, which impacts clinical outcomes
* SC sequencing tech enables characterization of intratumor heterogeneity in ovarian cancer
* To date, single-cell studies on ovarian cancers are scarce, which limits the opportunity to investigate effects on clinical outcomes
* Larger data sets are needed to establish potential associations between the unique information captured by SC sequencing and clinically relevant outcomes in ovca

Notes

* They claim they found genes associated with ovca stem cells.
* When they evaluated based on molecular eoc cancer subtypes, all subtypes were present.
* Studying the immune microenvironment of a tumor becomes more feasible with scRNA sequencing
* Evaluation of cell plasticity as it relates to cell type transitions may provide insight into the development of chemoresistance
* Understanding the lineage of the cellular qualities of metastatic dissemination is made possible with scRNA data

### Using Ancestry to find the Stem Cell

Question: can we use sequence-based lineage tracing to identify a common progenitor cell expression profile?

A thought I had while looking at some lit tonight led me to wonder if it's worth exploring sequence-based cell lineage across individual cells rather than using differential expression patterns as a general means of characterization. I will need to do a dedicated lit search to explore this concept, but I imagine the use of single cell data for lineage tracing is relatively limited. At a minimum, Tim has alluded to the notion that our data set is unique among the ovca model, which would lend itself to a novel application of single cell lineage tracing. Regardless, this could be a relatively significant deviation from their current pipeline, so I will have to discuss it with Tim.

### Updated create_site.R Script

I made a minor update to the script I use to automate total site rendering to clean up the directory a bit. I don't have any use in the parent notebook/ directory for individual html files since they are generated in the site render. The `ifelse` at the end of the script clears these html files from the notebook directory.

```{r}
rmarkdown::render("index.Rmd")
rmarkdown::render("notes.Rmd")
rmarkdown::render("results.Rmd")
rmarkdown::render("methods.Rmd")
rmarkdown::render("references.Rmd")
rmarkdown::render_site("index.Rmd")

fl <- Sys.glob("*.html")

ifelse(file.exists(fl), file.remove(fl))
```

### Summary

I was able to explore a few different fronts today. I put some more work in to polishing my citation management. This is an area I have been woefully lacking in from past experience, so I do not discount the extra work applied to make sure this is at a publication-ready level.

Additionally, I worked my way through the earliest publication related to the ocpmi project [@Winterhoff2019]. Through this I was able to develop some insight to the early considerations this group gave to this data as they were only working with 66 individual cells. They offered a broad scope of the advantages single cell tech offers along with pointing me to a few sources that establish the current state of the field.

Last, I can't say what generated this thought specifically, but I came across what my limited knowledge of the field would call a good idea. All of our discussions on this data set up to this point have centered on the use of UMI counts and gene expression levels as a means to develop differentiated cell profiles. A consideration I would potentially like to explore is the possibility that we look at sequence level data to identify cell lineage in search of a common progenitor. I have little basis to say if this is a reasonable or even realistic scientific question, but I feel this could be a substantive, novel insight given the nature of our data set. It's quite possible this approach would require too great an investment in shifting the informatics work toward raw sequence analysis, but I will discuss it with Tim nonetheless.

#### Completed

```{r}
Created    Completed  Age Project        Description
---------- ---------- --- -------------- --------------------------
2019-05-20 2019-06-01 11d starrlab.ocpmi read starr paper re 1st pt
2019-05-28 2019-05-31 3d  starrlab       find alternative to jabref
2019-05-30 2019-05-31 1d  starrlab       update notes regarding citation tools re
                                         mendeley
2019-05-23 2019-05-31 8d  starrlab       make custom csl

4 tasks
```


***

# June

## 3

### TODO

```{r}
ID Age Project        Tags       Description                                Urg
-- --- -------------- ---------- ------------------------------------------ ----
52 10d starrlab                  agenda items [2]                           1.95
74 2d  starrlab       computer n add notes about cancer stem cell database  1.91
49 13d starrlab.ocpmi lit w      read tirosh 2016 [2]                       2.87
56 7d  starrlab.ocpmi lit n      read papalexi 2018 [2]                     2.84
57 6d  starrlab.ocpmi lit w      read macintyre 2018 [2]                    2.83
58 5d  starrlab.ocpmi lit s      read sahin 2017 [1]                        2.73
67 2d  starrlab.ocpmi lit n      do lit search on cell ancestry [1]         2.71
46 13d starrlab.ocpmi lit w      read ocpmi summary                         1.97
47 13d starrlab.ocpmi computer w run surat tutorial w/ pt 59                1.97
48 13d starrlab.ocpmi lit w      read single cell paper zheng 2017          1.97
63 3d  starrlab.ocpmi computer n review pca concepts                        1.92
64 3d  starrlab.ocpmi computer n compare default seurat tutorial to         1.92
                                 starrlab protocol
65 3d  starrlab.ocpmi computer w deep dive into default seurat analysis     1.92
69 2d  starrlab.ocpmi lit n      find cancer stem cell review               1.91

14 tasks
```

### Cancer Stem Cell Database

http://www.csctt.org/ [@csctt2019]

Tim sent me a link to a possible resource last week that could be useful in building references for exploring the cancer stem cell concept.

### LitSrch re Cancer Stem Cell

I've done some extremely high level review from the wikipedia article on CSCs [@csc_wiki]. This pointed me to several papers of relevance that I hope to look at in future exploration of the topic.

#### Notes

* While there has been significant research around the topic, CSCs remain a debated idea.
* There are currently a few models for CSC tumor propagation
    + The CSC model
        - This is also known as the hierarchical model
        - CSCs are a tumorigenic and biologically distinct subpopulation [@Beck2013].
        - Two defining features: ability to self-renew, capacity to generate pro-growth progeny
        - This model suggests only certain cells have the ability to drive the progression of cancer
    + Stochastic model
        - Also known as clonal evolution model
        - This model suggests the necessary mutations to achieve CSC status can happen to any cell at random.
        - All tumor cells retain the ability to develop plasticity and self-renew/differentiate [@Kreso2014].
        - Elements of resistance and fitness accumulate over time [@Beck2013].
        - CSCs with a growth advantage outperform other cells [@Barabe2007].
    + CSC Immunology model
        - Both stem cell and CSCs possess qualities that make them resistant to immune surveillance [@MacCalli2014].
        - CSCs may exist at all times but only propagate once 1) they shed dependence on niche factors or 2) they develop a means to escape immune surveillance or 3) the immune system loses tumorsuppressive capacity [@Bruttel2014].
* Debate
    + The existence of CSCs is still under debate because no consistently definable characteristics have been found [@Gupta2009].
    + The origin of CSCs is also a major topic.
        - Are these disregulated normal SCs
        - Are they instead a more specialized population that gained the ability to self-renew.
        - Is it something else entirely due to the modular plasticity cancer cells can show [@Nouri2017].
* Modeling
    + There has been some mathematical modeling taken to the growth of these cells.
    + The cell compartment method is one approach [@Ganguly2006].
    + Other models may exist and are worth exploring in the context of highlighting CSC mutation lineage.
* Origin
    + stem cell mutation
        - stem cell niche populations develop mutations which derail regular progression and can be past to progeny [@Wang2009].
    + adult stem cells
        - Stem cell populations in tissue with high rates of turnover experience frequent cell divisions. Increased likelihood of mutations leads to accumulated genetic defects that drive cancer initiation. [@Lopez-Lazaro2018]
    + De-differentiation
        - Cells are driven back to a state of stemness through some external factor.

### Summary

I focused mostly on lit search efforts today. I want to develop a better understanding of the current landscape of the CSC hypothesis, so I have taken a broad overview of different findings. I started with the wikipedia article on the subject and have branched out to several other major papers in the field. A few points of interest are the controversies around origin and characterization. There seems to be a relatively wide interpretation of CSCs in terms of typing and behavior. This is obviously due to the relatively abstract nature of stemness and cell development. Talking with Tim today, there are some limitations since we only have expression data. One consideration is that our pipeline is using a known reference sequence database to process the output. Given this, there may be unknown mutations that could serve as markers of both CSC identity and possible lineage association. Screening for these in a cell specific way could bring about interesting findings.

#### Completed

```{r}
Created    Completed  Age Project        Description
---------- ---------- --- -------------- ---------------------------------------
2019-06-01 2019-06-03 2d  starrlab       add notes about cancer stem cell
                                         database
2019-06-01 2019-06-03 2d  starrlab.ocpmi litsrch re cancer stem cell review

2 tasks
```

***

## 4

### TODO


```{r}
ID Age Project Tags   Description                                           Urg
-- --- ------- ------ ----------------------------------------------------- ----
49 2w  lab     lit    read tirosh 2016 [2]                                  2.78
55 8d  lab     lit    read papalexi 2018 [2]                                2.74
56 7d  lab     lit    read macintyre 2018 [2]                               2.74
48 2w  lab     lit    read zheng 2017 [1]                                   2.68
57 7d  lab     lit    read sahin 2017 [1]                                   2.64
65 3d  lab     lit    litsrch re genetic lineage tracing method [1]         2.62
52 11d lab            agenda items [3]                                      2.06
47 2w  lab     comp w run surat tutorial w/ pt 59                           1.98
62 5d  lab     comp n review pca concepts                                   1.93
63 4d  lab     comp n compare default seurat tutorial to starrlab protocol  1.92
64 4d  lab     comp w deep dive into default seurat analysis                1.92
46 2w  lab     lit    read ocpmi summary                                    1.88
77 20h lab     lit    find list of CSC markers                               1.8

13 tasks
```

### Summary

I didn't get a lot done today. I began to apply the vanilla Seurat analysis to the pt59 data set and found myself encountering many of the setup hurdles that come with using a remote system. I have to do some retooling so my environment can seamlessly run on MSI.

***

## 5

### TODO

```{r}
ID Age  Project Tags   Description                                          Urg
-- ---- ------- ------ ---------------------------------------------------- ----
49 2w   lab     lit    read tirosh 2016 [2]                                 2.78
55 9d   lab     lit    read papalexi 2018 [2]                               2.75
56 8d   lab     lit    read macintyre 2018 [2]                              2.74
76 2d   lab     lit    read konrashova 2017 [2]                             2.71
48 2w   lab     lit    read zheng 2017 [1]                                  2.68
57 8d   lab     lit    read sahin 2017 [1]                                  2.64
65 4d   lab     lit    litsrch re genetic lineage tracing method [1]        2.62
47 2w   lab     comp w run surat tutorial w/ pt 59                          1.98
62 6d   lab     comp n review pca concepts                                  1.93
63 5d   lab     comp n compare default seurat tutorial to starrlab protocol 1.93
64 5d   lab     comp w deep dive into default seurat analysis               1.93
84 20h  lab     comp n figure out remote vim setup                           1.9
89 1min lab     comp n send vacation dates to tim                            1.9
46 2w   lab     lit    read ocpmi summary                                   1.88
77 1d   lab     lit    find list of CSC markers                             1.81
52 12d  lab            agenda items                                         1.07

16 tasks
```


### wkly mtg w/ Tim

present: chris, tim

```{r}
Description
--------------------------------------------------------------------------------
agenda items
  2019-05-28 discuss vacation
  2019-06-01 can we use sequence-based lineage tracing to identify a common
progenitor cell expression profile
  2019-06-03 is the JGO article just a couple poster abstracts?
  2019-06-05 discuss seurat output

1 task
```

#### mtg notes

* We discussed the nature of the Seurat output to some degree
    + I had been wondering how Tim came about the specifications to the metrics he used in his Seurat analysis file.
    + He has largely used the default params other outside of a few specific cases
    + These specifications and their justifications are all detailed in the Single Cell RNAseq Analysis protocol document in my `methods` folder.
    + One thing he mentioned would be an area I could assist him with is the way we fine tune these params.
    + Isaac is working on this to some extent by evaluating the 3 methods available for `Identification of highly variable features`.
        - So far, he's found that there is a notable difference in the genes identified, but the clustering output is relatively similar.
    + Tim has created a custom analysis specific to our data set that applied cell typing to clusters.
    + Additionally, we discussed the broad strokes of concepts behind the idea of CSCs. There is a great deal of ambiguity here, and we didn't really arrive at any objective conclusions. To some extent, I just needed to work through the ideas with someone else. I will need to really formulate a sense of this for any comprehensive experimentation which leverages this concept.
    + This was a subpoint but an important realization: **the 10x do not include full coverage.** This was a mistaken assumption of mine in thinking we had more complete expression coverage. Instead, we are working with ~100bp lengths with poly-A ranges. It may be worth digging into this to make sure I understand the specifics.

### Remote MSI Setup

I've reached somewhat of a hang up in my development environment. I've yet to work in centos, and I'm generally unfamiliar with the MSI setup. So far, I've been able to make sense of my home directory in my personal instance and load the R console. Additionally, I've been able to get the installation of packages going and have a process running over night in tmux.

The challenge I see stems from the combination of developing in an Rstudio-like environment that allows for the execution of code snippets while being executed in the MSI environment. What I am looking at as I wrap the day is using Rserve on my MSI instance to connect to my local dev environment.




### Rserve

I've found the package `Rserve` which can establish a servable instance of R on a remote server. This holds some potential to allow me to develop locally and execute any part of my code in the MSI instance. There is documentation at the link below. One necessary element of making this work would be to find a suitable Rserve client that can run locally and communicate with the server.

https://www.rforge.net/Rserve/doc.html



https://www.msi.umn.edu/support/faq/how-can-i-install-r-packages-my-home-directory

https://satijalab.org/seurat/install.html

### Summary

I met with Tim today for our weekly meeting. We were able to discuss things that were mostly conceptual to aid in my continued content exploration. The rest of my time was spent getting a sense for the MSI environment and how I will be interacting with it. My hope is to actually develop in this space full time rather than just schedule jobs. My own computer is limited in its processing power, and using my MSI instance would be highly preferred. Still, there are some hurdles to doing this that I haven't dealt with previously, so I have been working to link my current setup to the operational environment of MSI. This will probably continue to take some time, but I hope to be fully ready tomorrow.

***

## 6

### TODO


```{r}
ID Age Project Tags   Description                                           Urg
-- --- ------- ------ ----------------------------------------------------- ----
49 2w  lab     lit    read tirosh 2016 [2]                                  2.79
54 10d lab     lit    read papalexi 2018 [2]                                2.75
55 9d  lab     lit    read macintyre 2018 [2]                               2.75
74 3d  lab     lit    read konrashova 2017 [2]                              2.72
48 2w  lab     lit    read zheng 2017 [1]                                   2.69
56 8d  lab     lit    read sahin 2017 [1]                                   2.64
63 5d  lab     lit    litsrch re genetic lineage tracing method [1]         2.63
47 2w  lab     comp w run surat tutorial w/ pt 59                           1.99
60 6d  lab     comp n review pca concepts                                   1.93
61 6d  lab     comp n compare default seurat tutorial to starrlab protocol  1.93
62 6d  lab     comp w deep dive into default seurat analysis                1.93
82 1d  lab     comp n figure out remote vim setup                           1.91
87 22h lab     comp n send vacation dates to tim                             1.9
88 18h lab     comp n update methods on msi dev setup                        1.9
89 18h lab     comp n spell check notebook                                   1.9
46 2w  lab     lit    read ocpmi summary                                    1.89
75 2d  lab     lit    find list of CSC markers                              1.81
51 13d lab            agenda items                                          1.07

18 tasks
```


### MSI Workflow

After a considerable amount of effort to create a synchronous workflow with my MSI instance, I've generally given up and settled on scripting locally and batching with MSI. I've mounted my home directory with sshfs and clone my ocpmi git repo so I can work out of my MSI instance for the most part. My only concern is if I run into resource intensive cases that require more ram. It does seem that MSI kills unbatched jobs that exceed a threshold of resource use since my process was killed each time it went over 10gb of ram.

### Notes to add

* git repo info
* sshfs setup and bash script
* msi setup info
* seurat install and docs methods

### Summary

I've wasted a large amount of time over the past two days. I spent a solid 10 hours trying to get my workflow in MSI established all to essentially go with what would have been my setup had I done nothing at all. I will generally resort to developing on my local machine with a connection to my MSI directory. Beyond this, I will need to take a look at the batching process for MSI and possible explore some of their training resources to gain a better understanding.

#### Completed

```{r}
Created    Completed  Age Project Description
---------- ---------- --- ------- -------------------------------
2019-06-05 2019-06-06 1d  lab     send vacation dates to tim
2019-06-04 2019-06-06 1d  lab     figure out remote vim setup
2019-06-06 2019-06-06 2h  lab     clone git repo to msi directory
2019-06-06 2019-06-06 2h  lab     exchange ssh keys with msi

4 tasks
```

***

## 10

### TODO

```{r}
ID Age Project Tags     Description                                         Urg
-- --- ------- -------- --------------------------------------------------- ----
48 2w  lab     lit work read tirosh 2016 [2]                                2.91
53 2w  lab     lit work read papalexi 2018 [2]                              2.88
54 13d lab     lit work read macintyre 2018 [2]                             2.87
70 7d  lab     lit work read konrashova 2017 [2]                            2.84
47 2w  lab     lit work read zheng 2017 [1]                                 2.81
55 13d lab     lit work read sahin 2017 [1]                                 2.77
61 9d  lab     lit work litsrch re genetic lineage tracing method [1]       2.75
45 2w  lab     lit work read ocpmi summary                                  2.01
46 2w  lab     w work   run surat tutorial w/ pt 59                         2.01
58 11d lab     n work   review pca concepts                                 1.96
59 10d lab     n work   compare default seurat tutorial to starrlab         1.95
                        protocol
60 10d lab     w work   deep dive into default seurat analysis              1.95
71 6d  lab     lit work find list of CSC markers                            1.93
80 4d  lab     n work   update methods on msi dev setup                     1.92
81 4d  lab     n work   spell check notebook                                1.92
83 3d  lab     n work   add notes re sshfs bash script                      1.92
50 2w  lab     work     agenda items                                        1.89

17 tasks
```

### Non-contemporaneous Notes - 6/6

#### Load/Install R pkgs

* In order to run R in my MSI instance, I need to load the module and install applications.

```{r}
module load R/3.6.0
R
```

* To install packages in my local home directory, need to define a path.

```{r}
.libPaths(new='~/R/x86_64-pc-linux-gnu-library/3.6.0')
dir.create('~/R/x86_64-pc-linux-gnu-library/3.6.0', showWarnings = FALSE, recursive = TRUE)
```

* Confirm the path is correct.

```{r}
.libPaths()
```

* Choose as CRAN mirror.

```{r}
getCRANmirrors()
chooseCRANmirror(ind=2)
```

* Packages should be able to be installed and loaded normally now.

#### sshfs Setup

I'm using `sshfs` to mount my MSI home folder on my local machine to make it more readily accessible. I already have ssh keys exchanged with MSI, so I can create an executable auto-mount bash script. This is located at `/home/chris/starr_lab/auto_mnt_sshfs` on my local machine.

* Auto-mount MSI sshfs

```{bash eval=F}
#!/bin/bash

sshfs tasta005@login.msi.umn.edu: ~/starr_lab/msi_mnt
```

### Summary

I made some more progress on developing a base implementation of the default Seurat analysis pipeline when evaluating the pt59 dataset. To a degree, I am mostly taking default params using the PBMC tutorial and running them with the new dataset. Outside of the default params, I have been using the file located at `/ocpmi/code/tim/seurat_analysis.R` as a reference to see what Tim has modified thus far. There are certain points where he has taken the base value and others where his code deviates heavily. The general flow of his implementation differs from the tutorial in the sense that his code is less step wise and more focused on the output while the tutorial is a bit more exploratory to exhibit features. This makes sense of course. An aspect of my own implementation will have to consider how distributed each analysis process to provide modularity.

A major point I came across today was that I forgot to review the RNAseq protocol located at `/ocpmi/methods/`. Tim and I had discussed this in our meeting last week, but I forgot to make a note about this. There is a good deal of documentation for his code and methods there, so I probably should have remembered to look at this more. I will dedicate some time tomorrow to do this.

Overall, I will spend most of tomorrow continuing to review the documentation and code for Tim's pipeline along with getting a sense of the veracity of my pt59 output.

#### Completed

```{r}
Created    Completed  Age Project Description
---------- ---------- --- ------- -------------------------------
2019-06-05 2019-06-10 4d  lab     spell check notebook
2019-06-05 2019-06-10 4d  lab     update methods on msi dev setup
2019-06-06 2019-06-10 3d  lab     add notes re sshfs bash script

3 tasks
```

***

## 11

### TODO

```{r}
ID Age Project Tags     Description                                         Urg
-- --- ------- -------- --------------------------------------------------- ----
48 3w  lab     lit work read tirosh 2016 [2]                                2.92
50 2w  lab     work     agenda items [3]                                     2.9
53 2w  lab     lit work read papalexi 2018 [2]                              2.88
54 2w  lab     lit work read macintyre 2018 [2]                             2.88
70 7d  lab     lit work read konrashova 2017 [2]                            2.84
47 3w  lab     lit work read zheng 2017 [1]                                 2.82
55 13d lab     lit work read sahin 2017 [1]                                 2.77
61 10d lab     lit work litsrch re genetic lineage tracing method [1]       2.75
45 3w  lab     lit work read ocpmi summary                                  2.02
46 3w  lab     w work   run surat tutorial w/ pt 59                         2.02
58 11d lab     n work   review pca concepts                                 1.96
59 11d lab     n work   compare default seurat tutorial to starrlab         1.96
                        protocol
60 11d lab     w work   deep dive into default seurat analysis              1.96
71 7d  lab     lit work find list of CSC markers                            1.94
89 18h lab     n work   evaluate list of highly var features pt59            1.9
91 18h lab     errand n bring monitor to work                                1.9
92 17h lab     n work   review rnaseq analysis protocol                      1.9
95 20s lab     n work   email tim re mtg resched                             1.9

18 tasks
```

### RNA Velocity

I came a cross a [thread](https://www.reddit.com/r/bioinformatics/comments/bz292v/_/) on the bioinformatics subreddit discussing the issues of the day for single cell analysis. One of the points that was discussed was the challenge around merging data sets and the way this might regress some relevant artifacts of the data. The concept of RNA velocity was brought up as a means to address this, and it may hold some relevance to how we analyze our data. [@LaManno2018]

### Render .R -> .html

I made an amazing discovery today. It is possible to render markdown output straight from a .R file. I'm not sure if I can apply certain yaml or formatting settings to this process yet, but this function alone will save the need to create two separate files when running code and producing a formatted output.

https://rmarkdown.rstudio.com/articles_report_from_r_script.html

A markdown report can be compiled from an R script by:

```{r}
rmarkdown::render("analysis.R") # will generate an html output
rmarkdown::render("analysis.R", "pdf_document")
```

The metadata and output can be modified through special comments.

```{r}
#' ---
#' title: "Crop Analysis Q3 2013"
#' author: "John Smith"
#' date: "May 3rd, 2014"
#' output: pdf_document
#' ---
```

### Summary

My day consisted of a continued focus on developing a complete working model of the general Seurat pipeline through the lens of the pt59 data. I haven't done much in terms of evaluation for the output and will need to dig into this more. The highly variable features and linear dimensional reduction are items I need to both confirm for efficacy and make sense of in their findings. Additionally, I merely clustered the data along 13 dimensions and did not apply any further analysis such as the exploration of highly variable features in the umap'd data.

I did make some unintended progress on formatting and organization of code after discovering that I can render markdown directly from an R script file. This is what I was striving for to some degree in my MSI setup from last week, so it is a relief to arrive at a workable solution.

Last, I plan on taking a bit of a dive into the concept of RNA velocity to see if it holds any relevance to evaluating our data set. As I have become acclimated to the content of this field, I have felt there is some aspect of single cell analysis that the group has not considered to explore concepts beyond cell typing such as CNV, cell lineage, and stem cell character. I don't say that with the authority to know if it is the case, but the adolescence of this field draws the general assumption.

#### Completed

```{r}
Created    Completed  Age Project Description
---------- ---------- --- ------- ---------------------------
2019-06-11 2019-06-11 5h  lab     email tim re mtg resched
2019-05-20 2019-06-11 3w  lab     run surat tutorial w/ pt 59

2 tasks
```

***

## 12

### TODO

```{r}


ID Age Project Tags     Description                                         Urg
-- --- ------- -------- --------------------------------------------------- ----
46 3w  lab     lit work read tirosh 2016 [2]                                2.92
48 2w  lab     work     agenda items [3]                                     2.9
51 2w  lab     lit work read papalexi 2018 [2]                              2.89
52 2w  lab     lit work read macintyre 2018 [2]                             2.88
68 9d  lab     lit work read konrashova 2017 [2]                            2.85
45 3w  lab     lit work read zheng 2017 [1]                                 2.82
53 2w  lab     lit work read sahin 2017 [1]                                 2.78
59 11d lab     lit work litsrch re genetic lineage tracing method [1]       2.76
90 1d  lab     lit work read manno 2018 [1]                                 2.71
44 3w  lab     lit work read ocpmi summary                                  2.02
56 13d lab     n work   review pca concepts                                 1.97
57 12d lab     n work   compare default seurat tutorial to starrlab         1.97
                        protocol
58 12d lab     w work   deep dive into default seurat analysis              1.97
69 8d  lab     lit work find list of CSC markers                            1.94
85 1d  lab     n work   evaluate list of highly var features pt59           1.91
87 1d  lab     errand n bring monitor to work                               1.91
88 1d  lab     n work   review rnaseq analysis protocol                     1.91
93 19h lab     n work   perform diff exp analysis on non-lin red pt59        1.9
                        output

18 tasks
```


### sshfs Unmount

I've run into the case where a server disconnect causes a hang any time I navigate to this mount point. To reset the mount point use `fusermount` to force the unmount first.

```{r}
fusermount -uz /data
```

### MSI Interactive Computing

Running all of my work locally has shown to be more resource consuming than originally anticipated. This has developed somewhat of a bottleneck, so I'm hoping I can transfer the larger rendering jobs to MSI so they can process in the background. A limitation in doing this is developing the ability to generate markdown since this would be the best output. It isn't necessary, but I would prefer this. The hangup seems to be the lack of a graphics environment. Obviously, a headless server will not have something like this loaded by default, but I have to see if I can get that working.

### RNA Velocity

I began to gloss over this paper today [@LaManno2018]. The concept is both relatively simple in general terms yet includes a good deal of complexity since it involves RNA splicing. I need to read further, but what I've read so far lends itself to the potential for some application in our data set. I'll have to discuss with Tim more tomorrow.

### pt59 analysis

I continued to round out my general analysis of the pt59 data. This analysis was furthered with my original approach of being blind to the Biological relevance. I've been able to generate outputs that follow the base pipeline but will need to evaluate markers specific to our cell types and genetic markers of interest.

### Summary

Today included some light work around finishing off my pt59 analysis. The code I've put together generally mirrors the PBMC tutorial with some minor variations. Key to this general approach is that I haven't customized the marker selection. This would be a good discussion with Tim in how he has identified this in the past in a consistent manner. Additionally, my computer is becoming more limited with every process I run. It may be worth breaking these things into smaller chunks of analysis to account for my resource limitations, but I'm going to return to getting things working on MSI first.

#### Completed

```{r}
Created    Completed  Age Project Description
---------- ---------- --- ------- ----------------------------------------------
2019-06-10 2019-06-12 2d  lab     bring monitor to work
2019-05-30 2019-06-12 13d lab     compare default seurat tutorial to starrlab
                                  protocol
2019-06-11 2019-06-12 1d  lab     perform diff exp analysis on non-lin red pt59
                                  output

3 tasks
```

***

## 13

### TODO

```{r}

ID Age Project Tags     Description                                   Urg
-- --- ------- -------- --------------------------------------------- ----
46 3w  lab     lit work read tirosh 2016 [2]                          2.93
48 2w  lab     work     agenda items [6]                              2.91
51 2w  lab     lit work read papalexi 2018 [2]                        2.89
52 2w  lab     lit work read macintyre 2018 [2]                       2.89
66 9d  lab     lit work read konrashova 2017 [2]                      2.85
45 3w  lab     lit work read zheng 2017 [1]                           2.83
53 2w  lab     lit work read sahin 2017 [1]                           2.78
58 12d lab     lit work litsrch re genetic lineage tracing method [1] 2.77
84 1d  lab     lit work read manno 2018 [1]                           2.71
44 3w  lab     lit work read ocpmi summary                            2.03
56 13d lab     n work   review pca concepts                           1.97
57 13d lab     w work   deep dive into default seurat analysis        1.97
67 9d  lab     lit work find list of CSC markers                      1.95
81 2d  lab     n work   evaluate list of highly var features pt59     1.91
83 2d  lab     n work   review rnaseq analysis protocol               1.91
88 20h lab     n work   clean up github tracked files                  1.9
89 17h lab     n work   put pt59 output on results page                1.9

17 tasks
```

### wkly mtg w/ Tim

present: chris, tim

```{r}

Description
------------------------------------------------------------------------------
agenda items
  2019-06-10 discuss application of different R files
  2019-06-10 compare pt59 data
  2019-06-11 job funding
  2019-06-12 discuss selection of genetic markers and highly variable features
  2019-06-13 assess current progress and direction
  2019-06-13 tirosh 2016 CNV method
  2019-06-13 how are you determining dimensionality

1 task
```

#### Notes

Direction and project selection

* When do I want my thesis to come together?
    + September is the goal right now.
* The CSC concept is probably not a suitable project since it is relatively hard to define without some type of in situ validation, which I won't have time for.
* The RNA velocity concept is interesting but probably outside my scope right now.
* The CNV project
    + Has been started to some degree by jason and would be a good candidate project.
    + He has a preliminary version of this that I should discuss with him.
    + They did not have the bulk exome data at this point, so there is potential for a greater reference in analysis.
    + I should get the exome data and develop a process to establish a CNV baseline.
    + I can use @Tirosh2016 as a general method reference in SC CNV analysis, but developing a novel, standardized approach would be valuable according to Tim.

Analysis of genetic markers and variable features

* Tim gave me a list of cell type gene reference lists which will come in handy when identifying clusters and variable features.
* We discussed Tim's R scripts. The `seurat_analysis.R` is meant to be a bit of a customization analysis after the general pipeline has run. The others serve their dedicated purpose, which is outlined in the RNAseq analysis protocol.

Selection of dimensionality

* Tim gave me an overview of his method for selection the dimensionality of the data sets. He uses the option to take a csv output of the PCA and rank by statistical significance. Currently, he uses the relatively arbitrary cutoff of 1e-10.

Cell Ranger Status

* It sounds like the Cell Ranger processing is a work in progress that is being conducted by Ying. They have 5 pt done, and the data is located at the root directory below.

```{r}
/scratch.global/zhan2142/starr/
```


ACTIONS:

* reach out to jason re his efforts so far.
* get exome data.
* read @Zare2017 for a comparison of CNV analysis in whole exome data.
* email yuk and explore grant opportunities.
* compare Tim's pt59 output.
* continue pt59 analysis by applying boolean script


### CNV

#### Points of contact

* Juan Abrahante <abrah023@umn.edu>
* Christy Henzler <chenzler@umn.edu>
* Jinhua Wang <wangjh@umn.edu>
* Ying Zhang <zhan2142@umn.edu>
* Andrew Nelson <nels2055@umn.edu>

#### Varscan2

Tim wants me to try to use Varscan2 as a method to assess the exome data [@Koboldt2012]. Jinhua had tried this in some fashion previously.

#### Contra

Juan had tried using contra in the past which did not work.

##### from: Juan 6/13/2019

Sure thing. In a nutshell back in 2013 I ran Contra with Tumor/Normal Samples and it worked, I got results... fast forward 6 years and the test files are not even working for me.


First, I attempted via Python 3, 2.7 and the original 2.6 version using the test set in my local computer. Then, I moved into the "Matrix" or in this case Mesabi.

With Help from Master Jedi Christy (Thank you Christy!) I created a conda environment for Contra via:

```{r}
conda create -n contra python=2.6
source activate /home/umgc-staff/abrah023/bin/miniconda3/envs/contra
```

change directory to folder containing Contra script (tried both Version 2.0.8 and 1.0.3).

```{r}

python contra.py --target ../0247401_D_BED_20090724_hg19_MERGED.bed --test ../P0667N_GATKrealigned_duplicates_marked.bam --control ../P0667T_GATKrealigned_duplicates_marked.bam --fasta human_g1k_v37.fasta --outfolder ../output2/
```

That was using the files that came with Contra and got an error:

```{r}


(contra) abrah023@ln0003 [~/AndyTemp/CONTRA.v2.0.8] % python contra.py --target ../0247401_D_BED_20090724_hg19_MERGED.bed --test ../P0667N_GATKrealigned_duplicates_marked.bam --control ../P0667T_GATKrealigned_duplicates_marked.bam --fasta human_g1k_v37.fasta --o ../outputr/

target		: ../0247401_D_BED_20090724_hg19_MERGED.bed

test		: ../P0667N_GATKrealigned_duplicates_marked.bam

control		: ../P0667T_GATKrealigned_duplicates_marked.bam

outfolder	: ../outputr/

numBin		: [20]

minreaddepth	: 10

minNBases	: 10

sam		: False

pval		: 0.05

sampleName	: No-SampleName

nomultimapped	: False

plot		: False

bedInput		: False

minExon		: 2000

largeDeletion	: False

removeDups	: False

Creating Output Folder :  Done.

Traceback (most recent call last):

  File "contra.py", line 625, in <module>

    main()

  File "contra.py", line 570, in main

    get_genome(params.TEST, genomeFile)

  File "/panfs/roc/groups/8/umgc-staff/abrah023/AndyTemp/CONTRA.v2.0.8/scripts/get_chr_length.py", line 31, in get_genome

    raw_header = subprocess.Popen(args, stdout = subprocess.PIPE).communicate()[0]

  File "/home/umgc-staff/abrah023/bin/miniconda3/envs/contra/lib/python2.6/subprocess.py", line 623, in __init__

    errread, errwrite)

  File "/home/umgc-staff/abrah023/bin/miniconda3/envs/contra/lib/python2.6/subprocess.py", line 1141, in _execute_child

    raise child_exception

OSError: [Errno 2] No such file or directory

```

### Summary

I had my weekly meeting with Tim today. We had a productive conversation where we were able to assess and re-gear my progress. We came to the conclusion that I would focus on the CNV approach and were able to put together a large ranger of reference points around getting this started. There are several actions that came from our meeting, and I will have a good amount to focus on. The primary focus beyond content review will be developing a method for evaluating CNV within the whole exome data.

I was also able to get some house keeping done around documentation.

#### Completed

```{r}
Created    Completed  Age  Project Description
---------- ---------- ---- ------- ---------------------------------------------
2019-06-11 2019-06-13 2d   lab     read manno 2018
                                     2019-06-11 rna velocity of single cells
2019-06-03 2019-06-13 9d   lab     find list of CSC markers
2019-06-01 2019-06-13 12d  lab     litsrch re genetic lineage tracing method
                                     2019-06-01 can we determine a common
                                   ancestor among cell ancestry
2019-06-12 2019-06-13 1d   lab     clean up github tracked files
2019-06-13 2019-06-13 2min lab     organize cnv reference content

5 tasks
```

***

## 14

### TODO

```{r}

ID Age  Project Tags     Description                                        Urg
-- ---- ------- -------- -------------------------------------------------- ----
44 3w   lab     lit work read tirosh 2016 [2]                               2.94
49 2w   lab     lit work read papalexi 2018 [2]                              2.9
50 2w   lab     lit work read macintyre 2018 [2]                             2.9
60 11d  lab     lit work read konrashova 2017 [2]                           2.86
43 3w   lab     lit work read zheng 2017 [1]                                2.84
80 1d   lab     lit work read zare 2017 [1]                                 2.71
42 3w   lab     lit work read ocpmi summary                                 2.04
53 2w   lab     n work   review pca concepts                                1.98
54 2w   lab     w work   deep dive into default seurat analysis             1.98
69 4d   lab     n work   evaluate list of highly var features pt59          1.92
70 4d   lab     n work   review rnaseq analysis protocol                    1.92
46 3w   lab     work     agenda items                                       1.92
71 2d   lab     n work   put pt59 output on results page                    1.91
78 1d   lab     n work   reach out to jason re his CNV method               1.91
79 1d   lab     w work   get exome data                                     1.91
81 1d   lab     n work   email yuk re grant opportunities                   1.91
82 1d   lab     n work   explore grant opportunities                        1.91
83 1d   lab     n work   compare tims pt59 output                          1.91
84 1d   lab     n work   continue pt59 analysis with cell type selection    1.91
                         boolean script
85 1d   lab     n work   get graphics function to work on msi               1.91
93 3min lab     n work   send reminder to tim re rhel access                 1.9

21 tasks

```

### More CNV refernce

#### from: Jinhua 6/14/19

If it is for single cell exome DNA copy number call, you may try this tool: SCOPE,   https://www.biorxiv.org/content/10.1101/594267v1.full

For low coverage whole exome CNV,  you may try: EXCAVATOR,  http://sourceforge.net/projects/excavatortool/,  and also VARscan 2.

#### from: Andrew 6/14/19

You should now have access to nelsona2, where the somatic DNA exome data is located.  Rebecca and Christy have done the primary work on these samples, so please reach out to them if you have questions about the exome data.  Also, there are directories on there of sequencing from the Illumina TST170 panel, a focused DNAseq panel that has copy number calling built into the stock pipeline (using an algorithm called Craft).

Tim has cross walk files; as you will find out there are multiple different sample identifiers for each surgical specimen.

***

## 17

### TODO

```{r}

ID  Age  Project Tags     Description                                       Urg
--- ---- ------- -------- ------------------------------------------------- ----
 44 3w   lab     lit work read tirosh 2016 [2]                              2.95
 49 3w   lab     lit work read papalexi 2018 [2]                            2.92
 50 2w   lab     lit work read macintyre 2018 [2]                           2.91
 60 2w   lab     lit work read konrashova 2017 [2]                          2.88
 43 3w   lab     lit work read zheng 2017 [1]                               2.85
 80 4d   lab     lit work read zare 2017 [1]                                2.72
 42 3w   lab     lit work read ocpmi summary                                2.05
 53 2w   lab     n work   review pca concepts                                  2
 54 2w   lab     w work   deep dive into default seurat analysis            1.99
 69 6d   lab     n work   evaluate list of highly var features pt59         1.93
 70 6d   lab     n work   review rnaseq analysis protocol                   1.93
 46 3w   lab     work     agenda items                                      1.93
 71 4d   lab     n work   put pt59 output on results page                   1.92
 78 4d   lab     n work   reach out to jason re his CNV method              1.92
 79 4d   lab     w work   get exome data                                    1.92
 81 4d   lab     n work   email yuk re grant opportunities                  1.92
 82 4d   lab     n work   explore grant opportunities                       1.92
 83 4d   lab     n work   compare tim's pt59 output                         1.92
 84 4d   lab     n work   continue pt59 analysis with cell type selection   1.92
                          boolean script
 85 3d   lab     n work   get graphics function to work on msi              1.92
 86 3d   lab     w work   get scripts from jason re cnv analysis            1.92
 93 2d   lab     n work   send reminder to tim re rhel access               1.91
 94 2d   lab     n work   explore nelsona2 directory                        1.91
103 4min lab     n work   email tim re short mtg to discuss updated grad     1.9
                          plan

24 tasks
```

### @Tirosh2016

This paper is the only other case Tim is aware of where a group was able to effectively show CNV in single cell expression data.

* This group took human oligodendrogliomas and isolated ~4300 cells for sequencing.
* They constructed a developmental program from genome-wide expression signatures.
* The analysis produced the conclusion that these cells display 2 distinct developmental programs.
* Subclonal point mutation and insitu hybridization was also done as verification of these results.

#### CNV Method

* Their CNV approach was to define a window along a chromosomal mapping. By doing this, they effectively created "bins" of expression level to define relative changes in expression.
* They identified two cell types that aligned with a genetic character (high values at 1p and 19q) which they defined as baseline expression.
* This was used to define the baseline for microglia and oligdendrocytes.
* PCA comprised the main statistical approach.

### Minor Notebook Restructure

I realized I have a home page for no reason. I've replaced what is currently a blank placeholder page with what is my notes page since that is effectively the main page of my notebook. In doing so, I added a preamble to the begging of this document as well.

### Jason's CNV Method

Jason sent a very detailed overview over the weekend of his work on the CNV project. This includes code, references, and some analysis. I will need to look at this in greater depth and potentially set time to discuss it with him further. The files are located at the directory below.

* `/starr_lab/ocpmi/methods/jason_cnv`

### Summary

I did some house keeping on my documentation today. I've received a considerable amount of data and references regarding the CNV project, so I put some effort into organizing and clarifying my direction with this. My primary goals right now are covering the literature re the comparison for CNV techniques in whole exome data (@Zare2017) and then getting up to speed on Jason's past work. I hope to have a better mastery of the content to attack the project by the time I return from France.

#### Completed

```{r}

Created    Completed  Age Project Description
---------- ---------- --- ------- ----------------------------------------------
2019-06-13 2019-06-17 4d  lab     email yuk re grant opportunities
2019-06-15 2019-06-17 2d  lab     send reminder to tim re rhel access
2019-06-17 2019-06-17 2h  lab     email tim re short mtg to discuss updated grad
                                  plan
2019-05-20 2019-06-17 3w  lab     read tirosh 2016
                                    2019-06-03
                                  https://doi.org/10.1038/nature20123
                                    2019-06-03 CNV analysis in glioma
2019-06-13 2019-06-17 4d  lab     reach out to jason re his CNV method
2019-06-13 2019-06-17 4d  lab     get exome data
2019-06-13 2019-06-17 4d  lab     explore grant opportunities
2019-06-13 2019-06-17 4d  lab     get scripts from jason re cnv analysis

8 tasks
```

## 18

### mtg w/ Tim

present: tim, chris

I met briefly with Tim before leaving on my trip. We discussed the potential change in trajectory given the update that I will need to take 3 credits of directed research instead of 1 in the fall. He felt this generally won't adjust what our current plan is, and I tend to aggree.

Additionally, I introduced a concept that Yuk had brought up regarding the possiblity of a "long-distance, part-time PhD" given that I may move in the fall and secure a full time job. Tim was open to this, and I will keep it on the backburner for when that time comes after I've completed my masters degree.

Last, we discussed some of the smaller CNV analysis papers I've found that take a gene prep platform approach to evaluating CNV in single cells. The key distinction here is that these are likely done with DNA, but we both agreed that they may lack significance since I haven't found one published in a journal of significant impact. It may be something to explore for expanding my knowledge of methods though.

## 19

### @Zare2017

This paper explored a comparative analysis between current CNV analysis tools that operate in the context of whole exome sequencing which in wide use in the field. The group narrowed the landscape down to just 5 tools that meet a few criteria.

* There are 3 approaches to identifying CNVs
    + read count (read depth)
    + paired-end
    + assembly
* All tools the group evaluated used the read depth method since the pair-end and assembly methods are rendered ineffective with low read coverage like is seen in exome data. All CNV detection tools for WES are based on this method as a result.
* Ambiguities in alignment can create bias in the RD approach.
* The hybridization process and an uneven distribution of reads in exonic regions leads to bias.
* There are reads in the WES data that will be overlapping noise which would otherwise be delineated in the scRNA data. I will have to account for this.
* As of 2016, there are 15 seq-based CNV tools.
* B allele frequencies (BAF) is used to address tumor complexity in some of these tools.
* Group used both read and synthesized data.
* Look at table 1 for feature breakdown
* Look at table 4 for performance summary

ADTEx

* Newest of the tools they evaluated
* Has ability to leverage BAF
* Uses Discrete Wavelet Transform as a preprocessing step to reduce noise.
* Two Hidden Markov Models are used in detection.
* CBS is used in segmentation

Contra

* used for very small target regions.
* normalizes read count and log ratios for each base.
* better for reducing effect of GC bias.
* addresses problem of very low coverage and GC bias.

cn.MOPS

* Can use multiple samples for each genomics region to create a better estimate
* Uses a non-overlapping sliding window.
* Creates a null-hypothesis to measure distance to observed value.
* Can reduce false discovery rate

ExomeCNV

* Specifically design for WES data.
* Can use BAF data
* Counts overlapping reads from exons by using these counts for tumor and normal.
* Can also detect loss of heterzygosity.

VarScan2

* Also specifically designed for WES data from tumor-normal.
* Does not use a sliding window.
* Calculates tumor to normal read count ratios of high quality base that fulfill minimum coverage requirements.
* Using Fischers test, consecutive bases do not change significantly are binned together.
* Does not utilize a segmentation algorithm, CBS is advised.

Results

* The tools show low consistency.
* only about 25% of the true amplified genes and deleted genes were identified by all tools in the gene-based comparison.
* In the segment-based comparison, the sensitivities and FDRs were all comparably similar to the gene-based comparison.
* There was no consistency among tools for the size and number of detected CNVs as well.
* Due to contra's approach, it excels with clean data but may not be suitable for messy cancer data.
* Mis-detection of poor coverage sequences is a central cause of the poor performance.
* Because of the poor performance across the tools, there is a great need for improvement in preprocessing techniques.
* A point made that may hold relevance to my project is that tumor subclonality has not been addressed by any tool. The granularity offered by our scRNA data could help with this.

### items to clarify

* More specifically, what is the source of the exome data?
    + Is it at the site of the tumor?
    + Is it the same sample as the scRNA?
    + Is it general pt seq data?
* Difference between CBS and HMM algorithms.
* GC bias
* Do lit search on newer tools.
* B allele frequency (BAF). Does our dataset have these values.
* What is phred base quality
* VarSimlab, for generating synthetic CNV
* cghMCR to identify CNV genes using Refseq gene identifications.
* GenomicRanges package to detect overlapping regions.

***

# July

## 8

### TODO

```{r}

ID Age  Project  Tags     Description                                       Urg
-- ---- -------- -------- ------------------------------------------------- ----
46 6w   lab      lit work read papalexi 2018 [2]                            3.03
47 5w   lab      lit work read macintyre 2018 [2]                           3.02
72 2w   lab      w work   what is the source of the exome data [3]             3
79 8min lab      w work   items to clarify [6]                               2.9
44 6w   lab      work     agenda items [1]                                  2.85
41 6w   lab      w work   review frameworks for ccgd database               2.16
42 6w   lab      lit work read ocpmi summary                                2.16
43 6w   lab      w work   explore shiny framework for database              2.16
45 6w   lab      w work   get admin access to server                        2.14
48 5w   lab      n work   review pca concepts                               2.11
49 5w   lab      w work   deep dive into default seurat analysis            2.11
57 3w   lab      n work   evaluate list of highly var features pt59         2.05
58 3w   lab      n work   review rnaseq analysis protocol                   2.05
59 3w   lab      n work   put pt59 output on results page                   2.04
61 3w   lab      n work   compare tim's pt59 output                         2.04
62 3w   lab      n work   continue pt59 analysis with cell type selection   2.04
                          boolean script
63 3w   lab      n work   get graphics function to work on msi              2.04
66 3w   lab      n work   explore nelsona2 directory                        2.03
70 2w   lab      n work   begin comparison of cnv tools                     2.01
71 2w   lab      n work   clear uncached files from git                     2.01
73 2w   lab      w work   look up version tracking in document editing for     2
                          rmarkdown
74 2w   lab      n work   go back to june 19 and review items to clarify       2
69 2w   lab.ccgd n work   email oit points of contact to get rhel server    2.81
                          access [1]

23 tasks
```

### CCGD upgrade

I reached out to oialinux@umn.edu today regarding getting server access to the ccgd application. From looking at the ccgd project document, I found a few oit points of contact: Sonya Sustacek (ssustace), John Trammell (tram0004), and Tom Krell (sp?).


### Summary

I'm back. Feeling pretty rusty, so I've been reacclimating myself to the work I left 3wks ago. I started up the process of getting ccgd upgrade going and should be able to work on that in tandem with the cnv work. Additionally, I need to begin the process of really testing and applying the cnv protocols I reviewed on 6/19.

#### Completed

```{r}
Created    Completed  Age P Project  Description
---------- ---------- --- - -------- -------------------------------------------
2019-06-13 2019-07-08 3w  H lab      read zare 2017
                                       2019-06-13 comparison of CNV analysis in
                                     whole exome data
2019-06-17 2019-07-08 2w    lab      clear uncached files from git
2019-06-17 2019-07-08 3w    lab.ccgd email oit points of contact to get rhel
                                     server access
3 tasks
```

***

## 9

### TODO

```{r}
ID Age Project Tags     Description                                         Urg
-- --- ------- -------- --------------------------------------------------- ----
45 6w  lab     lit work read papalexi 2018 [2]                              3.04
46 6w  lab     lit work read macintyre 2018 [2]                             3.03
69 2w  lab     w work   what is the source of the exome data [3]            3.01
75 21h lab     w work   items to clarify [6]                                 2.9
44 6w  lab     work     agenda items [1]                                    2.85
41 7w  lab     w work   review frameworks for ccgd database                 2.17
42 7w  lab     lit work read ocpmi summary                                  2.17
43 7w  lab     w work   explore shiny framework for database                2.17
47 5w  lab     n work   review pca concepts                                 2.11
48 5w  lab     w work   deep dive into default seurat analysis              2.11
56 4w  lab     n work   evaluate list of highly var features pt59           2.05
57 4w  lab     n work   review rnaseq analysis protocol                     2.05
58 3w  lab     n work   put pt59 output on results page                     2.04
60 3w  lab     n work   compare tim's pt59 output                           2.04
61 3w  lab     n work   continue pt59 analysis with cell type selection     2.04
                        boolean script
62 3w  lab     n work   get graphics function to work on msi                2.04
65 3w  lab     n work   explore nelsona2 directory                          2.03
68 3w  lab     n work   begin comparison of cnv tools                       2.02
70 2w  lab     n work   go back to june 19 and review items to clarify      2.01
76 10s lab     n work   contact hst rep re ccgd access                       1.9

20 tasks
```

### CCGD access

I was informed by OIT that the server is actually owned by HST (Health Science Technology?) and that I need to reach out to them. The contacts I was given are:

* Colby Reese reese007@umn.edu
* Tom Kell kell@umn.edu
* Nathan Huff nrhuff@umn.edu
* Jeffrey Sik sikxx005@umn.edu

I emailed Tom since he was the only one listed in the original CCGD documentation.

### Summary

I am still going through the process of reviewing my work to reestablish the momentum and direction I had 3 weeks ago. I have been looking at Jason's code to make sense of his approach. I will probably reach out to him depending on what comes from my meeting with Tim tomorrow so that we can discuss his method. I need to clarify the picture of how I will implement the proper tool sets for both the exome and scRNA data.

#### Completed

```{r}
Created    Completed  Age Project Description
---------- ---------- --- ------- ----------------------------------------------
2019-07-09 2019-07-09 5h  lab     contact hst rep re ccgd access
2019-06-19 2019-07-09 2w  lab     go back to june 19 and review items to clarify

2 tasks
```

***

## 10

### TODO

```{r}
ID Age P Project Tags     Description                                  Urg
-- --- - ------- -------- -------------------------------------------- ----
72 18h H lab     n work   review zare notes before meeting              7.9
45 6w    lab     lit work read papalexi 2018 [2]                       3.04
46 6w    lab     lit work read macintyre 2018 [2]                      3.04
63 3w    lab     w work   what is the source of the exome data [3]     3.02
68 1d    lab     w work   items to clarify [6]                         2.91
44 6w    lab     work     agenda items [1]                             2.86
41 7w    lab     w work   review frameworks for ccgd database          2.17
42 7w    lab     lit work read ocpmi summary                           2.17
43 7w    lab     w work   explore shiny framework for database         2.17
47 5w    lab     n work   review pca concepts                          2.12
56 3w    lab     n work   get graphics function to work on msi         2.04
59 3w    lab     n work   explore nelsona2 directory                   2.04
62 3w    lab     n work   clarify direction for cnv tool selection     2.02

13 tasks
```

### wkly mtg w/ Tim

present: chris, tim

* Since I have been away, I did not have any work to discuss with Tim.
* We realigned my direction around the CNV project approach.
* We have variant allele freq (var). I should look into this.
* Reach out to people who processed wes data re ability get baf data. Talk to Rebecca Larue first.
  - Ask Rebecca which pipeline she used for methods purposes.
* Read papers on excavator (@Magi2013), varscan2, scope (@Wang2019) and assess which would be best. Run on one dataset.
* We discussed the source exome data since I wanted to clarify my understanding of this. The wes tissue can effectively be assumed to be the same source as the scRNA for each pt.

### Summary

The conversation with Tim today was helpful in regaining some direction on the CNV work. Right now, my immediate goals are to establish the toolset for wes analysis, develop a reliable wes output for reference, and then evaluate the method I use to approach the scRNA cnv analysis. I need more detail about the development of the wes data, so reaching out to Rebecca will be important at this point. Identifying supporting information for the wes data such as BAFs and VAFs could be determining factors in the pipeline I use.

#### Completed

```{r}
Created    Completed  Age P Project Description
---------- ---------- --- - ------- --------------------------------------------
2019-06-19 2019-07-10 3w    lab     what is the source of the exome data
                                      2019-06-19 is it at the tumor site?
                                      2019-06-19 is it the same sample as the
                                    scRNA?
                                      2019-06-19 is it just general pt exome
                                    data?
2019-07-09 2019-07-10 20h H lab     review zare notes before meeting
2019-06-17 2019-07-10 3w    lab     clarify direction for cnv tool selection

3 tasks
```

***

## 11

### TODO

```{r}
ID Age P Project Tags     Description                                       Urg
-- --- - ------- -------- ------------------------------------------------- ----
67 23h H lab     n work   contact rebecca re wes processing                  7.9
68 23h M lab     n work   create summarizing outline cnv project direction   5.8
63 2d    lab     w work   items to clarify [6]                              2.91
65 23h   lab     n work   read wang 2019 [1]                                 2.7
66 23h   lab     n work   read magi 2013 [1]                                 2.7
41 7w    lab     lit work read ocpmi summary                                2.18
46 6w    lab     n work   review pca concepts                               2.13
44 6w  L lab     lit work read papalexi 2018 [2]                            2.05
57 3w    lab     n work   explore nelsona2 directory                        2.04
45 6w  L lab     lit work read macintyre 2018 [2]                           2.04
69 23h   lab     w work   ask rebecca about wes processing method            1.9
43 6w  L lab     work     agenda items [1]                                  1.86
40 7w  L lab     w work   review frameworks for ccgd database               1.18
42 7w  L lab     w work   explore shiny framework for database              1.18
54 3w  L lab     n work   get graphics function to work on msi              1.05

15 tasks
```

### Summary

I reached out to Rebecca re a conversation about the background of the exome data processing. While I can investigate different tool sets to get a sense of what I will use, There are a few things she may be able to help me with that could be determining factors. I also began investigating the SCOPE tool today at the suggestion of Jinhua. I didn't have any background on this pipeline and was surprised to find it's purpose is to address single cell data, albeit with DNA. They indicate they did some validation with RNA data from 10x, so it remains to be seen if I will be able to use this.

#### Completed

```{r}
Created    Completed  Age P Project Description
---------- ---------- --- - ------- ---------------------------------
2019-07-10 2019-07-11 1d  H lab     contact rebecca re wes processing

1 task
```

***

## 12

### TODO

```{r}
ID Age P Project Tags     Description                                       Urg
-- --- - ------- -------- ------------------------------------------------- ----
67 1d  M lab     n work   create summarizing outline cnv project direction  5.81
63 3d    lab     w work   items to clarify [6]                              2.92
65 1d    lab     n work   read wang 2019 [1]                                2.71
66 1d    lab     n work   read magi 2013 [1]                                2.71
41 7w    lab     lit work read ocpmi summary                                2.18
46 6w    lab     n work   review pca concepts                               2.13
44 6w  L lab     lit work read papalexi 2018 [2]                            2.05
57 3w    lab     n work   explore nelsona2 directory                        2.05
45 6w  L lab     lit work read macintyre 2018 [2]                           2.05
68 19h   lab     w work   email other hst staff                              1.9
69 1h    lab     n work   take a look at genomespace                         1.9
43 7w  L lab     work     agenda items [1]                                  1.87
40 7w  L lab     w work   review frameworks for ccgd database               1.18
42 7w  L lab     w work   explore shiny framework for database              1.18
54 4w  L lab     n work   get graphics function to work on msi              1.05

15 tasks
```


### Collaborator mtg

present: josh, andy, isaac, shobhana, borris, jason, tim, ying

* exome data
    + freebase was used to call variants
    + there are some issues in variant filtering for calls in processing the wes data.
    + Andy took pt pm1 (pt54) and did a cross refernce across the 4 bams available
        - Coverage variability is high
        - Filtering poor calls stands as a significant hurdle
        - Currently, we need to establish a base line for "common" mutation calls
        - the tst170 was thought to be a higher standard, but the consistency does not seem to be there
        - Andy and Tim are going to create a simplified list of common genes to identify mutations
* scRNA
    + all single cell data has been rerun through cell ranger v3 and seurat.
    + we are switching to nova seq from hiseq.
    + the location and naming of this are somewhat ambiguous
    + the new v3 cell ranger data might be named "v2"...

### CNV tool lit search

#### EXCAVATOR

#### SCOPE

#### Awesome single cell repo

I found a really comprehensive list of single cell analysis tools at the github below.

* https://github.com/seandavi/awesome-single-cell

#### InferCNV

I also found a tool that was inspired by the @Tirosh2016 paper which explicitly aims to identify cnv in scRNA data. This looks very promising, and I will probably attempt this tool set first when getting into the sc data. The tool look like it could potentially be very simple. As this may help in getting started, it may be a drawback with the added complexity of our data set.

* https://github.com/broadinstitute/infercnv/wiki

### Summary

I spent the day evaluating CNV tools. While this was relatively productive, a brief conversation with Tim did draw attention to the fact I need to stop aiming and shoot. With the general info I've gathered, I can begin to run some analysis. My first step will be to run EXCAVATOR with a single pt data set followed by VarScan2 with the same data. My hope is that these pipelines come together with some ease and offer some level of a tutorial or test case. If not, I will focus on which of the two I can fully execute more quickly.

Additionally, my content search led me to an application I have an early excitement for, and that is InferCNV. I have really only glanced at the wiki for this tool, but it seems as if it was made specifically for our purposes. It looks relatively simple to use, and I am hoping that it can allow me to generate some data right away.

#### Completed

```{r}
Created    Completed  Age Project Description
---------- ---------- --- ------- ----------------------------------------------
2019-07-11 2019-07-12 1d  lab     email other hst staff
2019-07-08 2019-07-12 4d  lab     items to clarify
                                    2019-07-08 varsimlab, for generating
                                  synthetic cnv
                                    2019-07-08 cghmcr to identify cnv genes
                                  using refseq gene id
                                    2019-07-08 genomicranges package to detect
                                  overlapping regions
2019-07-12 2019-07-12 8h  lab     take a look at genomespace
2019-05-20 2019-07-12 7w  lab     read ocpmi summary
2019-07-10 2019-07-12 2d  lab     read wang 2019
                                    2019-07-10 scope wes tool
2019-07-10 2019-07-12 2d  lab     read magi 2013
                                    2019-07-10 excavator wes tool

7 tasks
```

***

## 15

### TODO

```{r}
ID Age P Project Tags   Description                                         Urg
-- --- - ------- ------ --------------------------------------------------- ----
60 2d  H lab     n work run excavator                                       7.91
59 4d  M lab     n work create summarizing outline cnv project direction    5.82
61 2d    lab     n work summarize value of excavator                        1.91
62 2d    lab     n work summarize features of scope                         1.91
42 7w  L lab     work   agenda items [1]                                    1.88
40 7w  L lab     w work review frameworks for ccgd database                  1.2
41 7w  L lab     w work explore shiny framework for database                 1.2
50 4w  L lab     n work get graphics function to work on msi                1.07

8 tasks
```

### WES analysis with EXCAVATOR/XCAVATOR

I will be taking the first wes cnv approach using the excavator/xcavator tool. This group published EXCAVATOR in @Magi2013 and have since expanded on it @Magi2017. I will be looking at XCAVATOR first, given its recency to see if it can bring anything to fruition.

Excavator leverages perl, bash, R, and fortran (of all things), and xcavator runs off of the same underlying architecture. I've installed the source files for the application using the scripts they provided at the location below.

* `/home/chris/starr_lab/ocpmi/methods/code/xcavator/source/lib`

Documentation can be found here:

* `/home/chris/starr_lab/ocpmi/methods/code/xcavator/source/docs/EXCAVATOR Manual.pdf`

#### XCAVATOR Implementation

As stated in @Magi2017

> All the computational approaches described in this paper have been packaged in the XCAVATOR software tool. XCAVATOR is a collection of perl, bash, R and fortran codes and its computational architecture has been derived from the EXCAVATOR tool that we published in 2013 [34] for the detection of CNVs/sCNA from whole-exome sequencing data. Our tool takes as input WGS data as BAM files and gives as output plots reporting raw, normalized, segmented and called data and a list of detected CNVs in tab-delimited and VCF format.

>The tool allows to analyze WGS data with three different experimental designs: “pooling”, “paired” and “nocontrol”. In “paired” mode each test sample is compared with its matched control and it is the best scheme to detect sCNA from pairs of tumor and matched normal samples. The “paired” mode has been used to analyze the TCGA Benchmark 4 dataset. In “nocontrol” mode the RCs of each test sample are normalized to two copies and this scheme is best suited for population genomics studies and has been used to analyze the 929 WGS experiments sequenced by 1000GP consortium. In “pooling” scheme, each test sample is compared to a pool control samples by summing the RC of each window across all the control samples. XCAVATOR can run on any unix system (desktop and servers) and allows the user to set the number of processor to analyze multiple samples in parallel. On a desktop computer with a 2.5 GHz cpu and 8 GB of ram, by using four cores, it takes four hours to perform the analysis of ten WGS samples sequenced at 60x. The XCAVATOR tool is freely available at https://sourceforge.net/projects/xcavator/.

### Summary

I put time into getting xcavator implemented. I am going with the updated version of excavator that was introduced in 2017. I had not noticed in my initial glance at the tool that it utilized perl and fortran. I don't believe this will impact any of the coding necessary since the pipeline seems to rely on modification of reference files, so my limited knowledge of perl and fortran shouldn't be an issue.

***

## 16

### TODO

```{r}
ID Age P Project Tags   Description                                         Urg
-- --- - ------- ------ --------------------------------------------------- ----
60 3d  H lab     n work run excavator                                       7.92
59 5d  M lab     n work create summarizing outline cnv project direction    5.83
61 3d    lab     n work summarize value of excavator                        1.92
62 3d    lab     n work summarize features of scope                         1.92
42 7w  L lab     work   agenda items [1]                                    1.89
40 8w  L lab     w work review frameworks for ccgd database                 1.21
41 8w  L lab     w work explore shiny framework for database                1.21
50 4w  L lab     n work get graphics function to work on msi                1.08

8 tasks
```

### Interactive Computing

It is possible to run an interactive session with the MSI lab space using the `isub` command. Additionally, `qsub` allows for scheduling of jobs in the lab space.

This is useful in the event I want to do development inside MSI rather than just batch jobs. More documentation at the link below:

* https://www.msi.umn.edu/content/interactive-queue-use-isub

***

## 17

### TODO

```{r}
ID Age P Project Tags    Description                                        Urg
-- --- - ------- ------- -------------------------------------------------- ----
60 4d  H lab     n work  run excavator                                      7.92
59 6d  M lab     n work  create summarizing outline cnv project direction   5.83
71 23h M lab     n phone call any hst number                                 5.8
61 4d    lab     n work  summarize value of excavator                       1.92
62 4d    lab     n work  summarize features of scope                        1.92
42 7w  L lab     work    agenda items [1]                                    1.9
40 8w  L lab     w work  review frameworks for ccgd database                1.21
41 8w  L lab     w work  explore shiny framework for database               1.21
50 4w  L lab     n work  get graphics function to work on msi               1.08

9 tasks
```

### wkly mtg w/ Tim

present: tim, chris

We had a limited discussion today. I updated Tim on progress with getting started in xcavator and getting access to the ccgd server space. Tim suggested I approach sonya given her past involvement with the project.

### XCAVATOR Progress

I'm continuing the process of getting xcavator setup and running. As a pipeline, xcavator is less mature than something like Seurat, especially with its documentation. I need to discuss a handful of items with either Rebecca or Christy regarding the processing of the wes data.

### Summary

Most of my time today was spent making sense of the xcavator workflow. There are some gaps in the documentation that I may have to parse out, but I am getting a sense for the parameters.

#### Completed

```{r}
Created    Completed  Age Project Description
---------- ---------- --- ------- ----------------------------------------------
2019-07-17 2019-07-17 23h lab     reach out to ssustace@umn.edu re ccgd server
                                  access

1 task
```

***

## 18

### TODO

```{r}
ID Age P Project Tags   Description                                         Urg
-- --- - ------- ------ --------------------------------------------------- ----
59 5d  H lab     n work run excavator                                       7.93
58 7d  M lab     n work create summarizing outline cnv project direction    5.84
73 21h   lab     w work mtg with rebecca [3]                                 2.9
60 5d    lab     n work summarize value of excavator                        1.93
61 5d    lab     n work summarize features of scope                         1.93
42 7w  L lab     work   agenda items [1]                                     1.9
75 17h   lab     n work find all sfconflicts                                 1.9
40 8w  L lab     w work review frameworks for ccgd database                 1.22
41 8w  L lab     w work explore shiny framework for database                1.22
50 4w  L lab     n work get graphics function to work on msi                1.09

10 tasks
```

### create_site to notebook_knit

I've renamed my `create_site` file to `notebook_knit` due to a conflict with a different executable. The location of this file is now at `/usr/bin` on my local machine. The updated script is below.

```{r}
#!/usr/bin/env Rscript

cur_dir <- getwd()
setwd("/home/chris/starr_lab/ocpmi/notebook")

rmarkdown::render("index.Rmd")
rmarkdown::render("results.Rmd")
rmarkdown::render("methods.Rmd")
rmarkdown::render("references.Rmd")
rmarkdown::render_site("index.Rmd")

fl <- Sys.glob("*.html")

ifelse(file.exists(fl), file.remove(fl))

setwd(cur_dir)

```

A key element to make this executable was to include it in an environment directory (i.e. `/usr/bin`) and also make it executable for all users with the command below.

```{r}
sudo chmod a+x notebook_knit.R
```

### XCAVATOR Window Initialize

The first command seems to take too much memory, so I installed the pipeline on MSI today. The setup should mirror my local installation for now. The goal right now is just to get the code to run. I'm attempting to execute the `ReferenceWindowInitialize.pl` command.

Overall, I'm using the default params and referencing the hg19.fa fasta file I downloaded from here:

* http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/

Here is the `SourceTarget.txt` file on the MSI instance:

```{r}
/panfs/roc/groups/11/starrt2/tasta005/ocpmi/methods/XCAVATOR/data/ucsc.hg19.bw /panfs/roc/groups/11/starrt2/tasta005/ocpmi/methods/XCAVATOR/data/hg19.fa

```

Here is the command itself:

```{r}
perl ReferenceWindowInitialize.pl -v SourceTarget.txt test_window1 100 hg19
```


#### Error

I've encountered an error attempting to execute `ReferenceWindowInitialize.pl`.

I am initially just putting any data that will work into the pipeline to get the code running. As far as I can tell, I should be able to use a random fasta file. I chose hg19 since that is identified as a reference genome.

The output of the error is below. The key issues seems to be the creation of the test directory and whatever `*_chromosome.txt` is.

```{r}
Program folder is: /panfs/roc/groups/11/starrt2/tasta005/ocpmi/methods/XCAVATOR
Creating Consecutive and non-overlapping windows of 100 bp Size
sh: R: command not found
Calculating Mappability and GC content...
cat: /panfs/roc/groups/11/starrt2/tasta005/ocpmi/methods/XCAVATOR/data/targets/hg19/test_window1/*_chromosome.txt: No such file or directory
...done!

```

### Summary

Thinking about the test data I've chosen, I might need to reevaluate. The hg19 dataset isn't wes data, and I don't know if that is an issue for the pipeline. It's also entirely possible I'm missing a setting that is essential to the initialization step. Another point of consideration is that I need a dataset that also has associated bam files, which I am not aware of for hg19. I could take a pt from the ocpmi data set, but they do not have basic fasta files that I am aware of yet, and I don't know if the fastq files we do actually have are compatible with the pipeline.

#### Completed

```{r}
Created    Completed  Age P Project Description
---------- ---------- --- - ------- --------------------
2019-07-16 2019-07-18 2d  M lab     call any hst number
2019-07-17 2019-07-18 23h   lab     find all sfconflicts

2 tasks
```

***

## 21

### TODO

```{r}
ID Age P Project Tags   Description                                         Urg
-- --- - ------- ------ --------------------------------------------------- ----
58 8d  H lab     n work run excavator                                       7.94
57 11d M lab     n work create summarizing outline cnv project direction    5.86
72 4d    lab     w work mtg with rebecca [5]                                2.92
59 8d    lab     n work summarize value of excavator                        1.94
60 8d    lab     n work summarize features of scope                         1.94
68 5d    lab     w work get ahold of someone from hst re ccgd database      1.93
81 2d    lab     n work test fastq compatiblity with xcavator               1.91
82 2d    lab     n work trouble shoot xcavator error from june 18           1.91
86 44s   lab     n work access and explore ccgd server                       1.9
40 8w  L lab     w work review frameworks for ccgd database                 0.24
41 8w  L lab     w work explore shiny framework for database                0.23
42 8w  L lab            agenda [1]                                          0.12

12 tasks
```

### CCGD Access

I was finally able to get through on the access portion of the ccgd project. Sonya played an important role in initiating access, which was ultimately completed by Satish.

#### Server site

* [ccgd-starrlab.oit.umn.edu](ccgd-starrlab.oit.umn.edu)

The working directory for the website seems to be located at:

* `/swadm/var/www/html`

***

## 22

### TODO

```{r}
ID Age P Project Tags   Description                                         Urg
-- --- - ------- ------ --------------------------------------------------- ----
58 9d  H lab     n work run excavator                                       7.95
57 11d M lab     n work create summarizing outline cnv project direction    5.86
71 4d    lab     w work mtg with rebecca [5]                                2.92
59 9d    lab     n work summarize value of excavator                        1.95
60 9d    lab     n work summarize features of scope                         1.95
79 3d    lab     w work check status of access to ccgd server               1.92
80 3d    lab     n work test fastq compatiblity with xcavator               1.92
81 3d    lab     n work trouble shoot xcavator error from june 18           1.92
85 21h   lab     n work put satish on contact list                           1.9
87 19h   lab     n work enter daily summary from yesterday                   1.9
40 8w  L lab     w work review frameworks for ccgd database                 0.24
41 8w  L lab     w work explore shiny framework for database                0.24
42 8w  L lab            agenda [1]                                          0.12

13 tasks
```


### Folding elements

Following an example from the Seurat pbmc tutorial, I am incorporating folding page elements in my notebook to reduce length and clutter.

```{r eval=F}
<details>
  <summary>****</summary>
</details>
```

### XCAVATOR vs EXCAVATOR2

I've been working through the error I'm getting in running the `ReferenceWindowInitialize.pl` script to no avail. There is some ambiguity in the file setup for the script arguments. It asks for a bed file but does not specify where that is supposed to be referenced. As a result, I'm going to see if I can easily transition to using EXCAVATOR instead.

* https://sourceforge.net/projects/excavator2tool/

### Summary

The XCAVATOR package has been giving me some difficulty with the inconsistencies in its documentation. Part of the issue could stem from my lack of knowledge regarding the file types necessary, but there is not a clear indication that I am wrong either. There script requirements do not fully line up, and the lack of a complete test dataset compounds the issue. I'm going to try my luck with the EXCAVATOR2 package since it is probably more mature in its function and support.

#### Completed

```{r}
Created    Completed  Age Project Description
---------- ---------- --- ------- -------------------------------------
2019-07-18 2019-07-22 4d  lab     check status of access to ccgd server
2019-07-21 2019-07-22 1d  lab     put satish on contact list
2019-07-21 2019-07-22 1d  lab     enter daily summary from yesterday
2019-07-18 2019-07-22 3d  lab     trouble shoot xcavator error from june 18

3 tasks
```

***

## 23

### TODO

```{r}
ID Age  P Project Tags   Description                                        Urg
-- ---- - ------- ------ -------------------------------------------------- ----
66 5d     lab     w work mtg with rebecca [5]                               2.93
75 1min   lab     n work find vcf files in dna_project                       1.9
76 1min   lab     n work begin pilot with data from RL                       1.9
78 18s    lab     n work explore application of vcf files in cnv pipeline    1.9
40 9w   L lab     w work review frameworks for ccgd database                0.25
41 9w   L lab     w work explore shiny framework for database               0.25
42 8w   L lab            agenda [1]                                         0.13

7 tasks
```

### Call w/ Rebecca

Topic: wes data processing and prep

I was able to finally connect with Rebecca and discuss the nature of the wes data processing that took place earlier this year. It sounds like there has been some distance between when they were actually working on this and now. Nonetheless, she was able to share some general information on their process, which will be very helpful.

```{r}
Description
------------------------------------------------------------
mtg with rebecca
  2019-07-17 what is the reference assembly for the wes data
  2019-07-17 file naming conventions
  2019-07-17 what is a reasonable window size
  2019-07-18 what is the nature of our controls
  2019-07-18 do we have fasta files

1 task
```

#### Notes

* All processing was done in MSI scratch due to the large resource use.
* The data is stored in 2nd tier storage right now as a result.
  BAM files might also be in 2nd tier storage.
* The location of the data is `/nelsona2/shared/ovarian_Jan2019/DNA_project`.
* The directory `/ovarian_Jan2019/singlesamples`, which I had thought was the directory of interest, contains the data for the RNAseq project.
* The sc and wes data were mapped using different reference genomes. sc - GR38 and wes - GR37. **This could be a problem.**
    - We might need to remap the wes data to GR38.
* I need to do a search of `/DNA_project` to see if there are any usable sequence files in there.
* RL is going to move a small dataset from aws so that I can begin a pilot analysis.
* The Freebayes params may be relevant, I should look into those.

#### Process

*Pipeline (in-house) &rarr; Mapping (GR37) &rarr; Preprocessing (`ScanIndel-ovarian.py`) &rarr; Freebayes (variance calling) &rarr; Selective Capture (ovarian specific) &rarr; VCF file output &rarr; Selective Filtering (`fb_VCF_filter.R`) &rarr; ClinVar (Christy)*

Project dir: `/nelsona2/shared/ovarian_Jan2019/DNA_project`

* Important scripts are located in `/DNA_project/scanindel_mod`.
* The probe pull down inclued some non-coding regions.
    - The assumption was that there may be a relevant impact on the coding regions in some cases, so these elements should be recovered and selectively filtered out.
* ClinVar was used to apply annotation to variance that matches existing clinical reference.

***

## 24

### TODO

```{r}
ID Age P Project Tags   Description                                         Urg
-- --- - ------- ------ --------------------------------------------------- ----
74 21h   lab     n work find vcf files in dna_project                        1.9
75 21h   lab     n work begin pilot with data from RL                        1.9
77 21h   lab     n work explore application of vcf files in cnv pipeline     1.9
78 20h   lab     n work explore freebayes params                             1.9
42 8w  L lab            agenda [5]                                          0.33
40 9w  L lab     w work review scripts for ccgd database                    0.25
41 9w  L lab     w work explore shiny framework for database                0.25

7 tasks
```

### wkly mtg w/ Tim

present: chris, tim

```{r}
Description
---------------------------------------------------------------
agenda
  2019-07-08 status of ccgd project
  2019-07-23 expand on the rnaseq project
  2019-07-23 discuss call with rebecca
  2019-07-24 cosmic update
  2019-07-24 can we use the preprocessed data for cnv pipelines

1 task
```

#### Notes

Our meeting covered the two major branches of the ccgd and ocpmi project. First, I updated Tim on my new access to the ccgd server.

* Looking at the current status, the next steps are to determine the process for updating the server os.
    - I need to reach out to OIT to determine their policy on upgrades.
* Also, I need to begin the backup process to make sure all elements can be recovered in the event of any issues in the upgrade.
    - This is also a conversation to be had with OIT.

The updates on the ocpmi project centered on my conversation with Rebecca.

* As I had highlighted, the processing work done by RL/CH on the wes data will not be relevant to my cnv pipeline.
    - I need to reach out to them and clarify where their raw data is currently since I assume this is also on aws.
* With the roadblocks I've experienced with E/Xcavator, Tim suggested I reach out to the developers. Which I will do after a trial of Excavator2
    - The plan for my approach in the event I continue to see issues can be seen below
* A point of clarification is that hg19 and GRCh37 are the same. This means that any data I use with upstream processing done by Rebecca should follow with hg19 as the reference.

### CNV Plan

Fallback strategy for cnv pipeline:

*Try Excavator2 &rarr; Contact Devs &rarr; Resort to VarScan2*

### Excavator2 Troubleshooting

#### BED and FASTA

Continuing from the error on 7/18, I found I was not referencing the bed file correctly. This was not as clear in the Xcavator documentation. I took generic bed file from ucsc genome browser.

* http://genome.ucsc.edu/cgi-bin/hgTables?command=start
* Dir: `/EXCAVATOR2_Package_v1.1.2/data/hg19.bed`

Looking at the background on fasta reference, I decided to take the g1k fasta file from 1000 genome project.

* ftp://ftp-trace.ncbi.nih.gov/1000genomes/ftp/technical/reference/
* Dir: `/EXCAVATOR2_Package_v1.1.2/data/human_g1k_v37.fasta`

The `TargetPerla.pl` script executes after I arranged these bed and fasta files.

#### libpng12.deb

This produced an error re a missing object

```{r}
/home/chris/Seafile/library/work/research/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/lib/OtherLibrary/./bigWigAverageOverBed: error while loading shared libraries: libpng12.so.0: cannot open shared object file: No such file or directory
Error in file(file, "rt") : cannot open the connection
Calls: read.table -> file
In addition: Warning message:
In file(file, "rt") :
  cannot open file '/home/chris/Seafile/library/work/research/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/targets/hg19/test_target2/MAP/Mapout.txt': No such file or directory
Execution halted
```

* Needed to install legacy version of `libpng12.deb`
* https://askubuntu.com/questions/978294/how-to-fix-libpng12-so-0-cannot-open-shared-object-file-no-such-file-or-direc/978338

```{r}
wget -q -O /tmp/libpng12.deb http://mirrors.kernel.org/ubuntu/pool/main/libp/libpng/libpng12-0_1.2.54-1ubuntu1_amd64.deb
sudo dpkg -i /tmp/libpng12.deb
sudo rm /tmp/libpng12.deb
```

#### BEDTools

Was getting another error re another missing object: bedtools. The documentation says nothing about it, but I needed to install `bedtools`.

```{r}
sudo apt install bedtools
```

#### Unresolved

Current error. Not sure what this is yet. My current thought is that there is an issue with the formatting between the fasta and bed file. I believe the g1k fasta I downloaded is a multi fasta, and I don't know if that's an issue.

```{r}
WARNING. chromosome (chr22) was not found in the FASTA file. Skipping.
```

### Summary

I made some progress working through the Excavator2 pipeline today. I am still encountering hang ups that are probably related to file formatting and binary file setup. I need to reach out to RL/CH regarding the change in source files I need since we didn't fully cover this when I spoke to Rebecca. Given the quick jump I made in troubleshooting today, I hope to be able to continue that progress with the pipeline tomorrow.

***

## 25

### TODO

```{r}
ID Age  P Project Tags   Description                                        Urg
-- ---- - ------- ------ -------------------------------------------------- ----
73 1d     lab     n work find vcf files in dna_project                      1.91
74 1d     lab     n work begin pilot with data from RL                      1.91
76 1d     lab     n work explore application of vcf files in cnv pipeline   1.91
77 1d     lab     n work explore freebayes params                           1.91
79 21h    lab     n work add tasks from wkly mtg                             1.9
81 2min   lab     n work email rebecca re bam files                          1.9
82 1min   lab     n work contact oit re update policy and process            1.9
83 37s    lab     n work start back up process for ccgd server               1.9
42 8w   L lab            agenda [5]                                         0.34
40 9w   L lab     w work review scripts for ccgd database                   0.26
41 9w   L lab     w work explore shiny framework for database               0.26

11 tasks
```

### BAM file search

I contacted Rebecca re the location of the wes bam files today since we missed that point in our conversation.

### Excavator2 Trbl (cont.)

#### Unresolved - Solved

I was able to address the last error given on 7/24 by using a different fasta reference file. I believe the issue was that the fasta and bed files contained different chromosome notation as indicated by section 1.6 "Before you start" of the Excavator2 docs. Probably should have read that.

As a result, the fasta file I'm using now also comes from UCSC

* http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/
* Dir: `/EXCAVATOR2_Package_v1.1.2/data/hg19.fa`

### ExcavatorDataPrepare.pl

The next step in running the Excavator2 pipeline is setting up `EXCAVATORDataPrepare.pl` to process sample BAM files. I don't have a clear set of files I know will work at this point, so I will need to get that sorted out. The BAMs are referenced using the `ExperimentalFilePrepare.window.txt` file which includes 1) an abs path to the BAMs 2) the path to the sample output 3) the sample name.

### Summary

I was able to finally run the first of 3 execution scripts with the Excavator2 pipeline. Having lined all the necessary items up for that first step, I need to arrange a set of pilot BAM files that will allow me to continue with the test run of the pipeline. Currently, I'm looking at pulling a 1kgp dataset that was referenced in the Excavator2 (@DAurizio2016). This may be an issue since the 1kgp has restructured their ftp since the Excavator2 publication.

***

## 26

### TODO

```{r}
ID Age  P Project Tags   Description                                        Urg
-- ---- - ------- ------ -------------------------------------------------- ----
69 2d     lab     n work find vcf files in dna_project                      1.91
70 2d     lab     n work begin pilot with data from RL                      1.91
72 2d     lab     n work explore application of vcf files in cnv pipeline   1.91
73 2d     lab     n work explore freebayes params                           1.91
75 1d     lab     n work add tasks from wkly mtg                            1.91
78 1d     lab     n work contact oit re update policy and process           1.91
79 1d     lab     n work start back up process for ccgd server              1.91
83 1min   lab     n work continue to troubleshoot excavator2 pipeline        1.9
84 1min   lab     n work find pilot bam files for excavator2 test run        1.9
42 9w   L lab            agenda [5]                                         0.35
40 9w   L lab     w work review scripts for ccgd database                   0.26
41 9w   L lab     w work explore shiny framework for database               0.26

12 tasks
```

### 1kgp ftp

The 1kgp ftp has moved to a new site.

* ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000_genomes_project/data

### BAM Selection

Currently I am taking 2 of the genome samples referenced in Supplementary Table 2 of @DAurizio2016.

* NA19131
* NA10847

Both BAMs were downloaded as the phase 1 1kgp sample from the IGSR

* http://www.internationalgenome.org/data-portal/sample

### Excavator2 Trbl (cont.)

#### Missing BAM Index

I realized I was missing the BAM index included on the IGSR.

```{r}
[main_samview] random alignment retrieval only works for indexed BAM or CRAM files.
```

#### Chr notation

It seems like the chromosome notation has become an issue again. The UCSC and 1kgp gene notations use "chr1" and "1" respectively making them incompatible without a variety of data wrangling.

```{r}
[main_samview] region "chr1" specifies an unknown reference name. Continue anyway.
```

### Summary

I hit another formatting snag today that will require me to go back and find different reference files. I believe I generally have the `ExcavatorDataPrepare.pl` script setup correctly, but the chromosome naming scheme needs to be consistent across the reference and sample files. This should be relatively easy to fix. I just need to find fasta, bed, and bam files that match.

***

## 27

### New fasta and bed

To address the chromosome naming discrepancy, I retrieved new fasta and bed files from the 1kgp so that the source will be consistent.

* http://www.internationalgenome.org/category/grch37/

More specifically:

fasta

* http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/
  Dir: `/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/human_g1k_v37.fasta`

phase 1/2 bed files

* http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/exome_pull_down_targets_phases1_and_2/
* Dir: `/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/20120518.consensus.annotation.bed`

***

## 28

### New fasta and bed again

In addition, I reran the pipeline with a new reference due to the chromosome annotation issue. This includes some BAMs from UCSC that are smaller in size.

* http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeUwRepliSeq/
* Dir: `/home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc`

Here's the updated `SourceTarget.txt`

```{r}
/home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bPctSignalRep1.bigWig /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/hg19.fa

```


### ExcavatorPrepareData.pl

I've was able to run the `ExcavatorPrepareData.pl` script with the following parameters. I'm not entirely sure this worked based on the output, but it doesn't seem explicitly that it didn't work either.

#### ExperimentalFilePrepare.50000.txt

This is the tab delimited file that identifies the bam files, of which were

```{r}
/home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/pilot/test_bam test_bam

```


Final `ExperimentalFilePrepare.50000.txt`

```{r}
perl EXCAVATORDataPrepare.pl ExperimentalFilePrepare.50000.txt --processors 4 --target test_target2 --assembly hg19
```


<details>
  <summary>**ExcavatorDataPrepare.pl Output**</summary>

```{r}
perl EXCAVATORDataPrepare.pl ExperimentalFilePrepare.50000.txt --processors 4 --target test_target2 --assembly hg19
Program folder is: /home/chris/Seafile/library/work/research/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2
Preparing Multiprocessor Analysis...
Starting Multiprocessor Analysis!
Creating '/home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/pilot/test_bam/RC' folder...
Creating '/home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/pilot/test_bam/RCNorm' folder...
Creating '/home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/pilot/test_bam/Images' folder...
Working on sample test_bam.
Creating Read Count Data...
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
Removing temporary files...
Read Count done!
Normalizing Read Count Data...
Loading required package: lattice
Loading required package: survival
Loading required package: Formula
Loading required package: ggplot2

Attaching package: ‘Hmisc’

The following objects are masked from ‘package:base’:

    format.pval, units

null device
          1
null device
          1
null device
          1
null device
          1
null device
          1
Normalization done!
Performed RC calculations and Normalization on samples:
```

</details>

***

## 29

### TODO

```{r}
ID Age P Project Tags   Description                                         Urg
-- --- - ------- ------ --------------------------------------------------- ----
66 5d    lab     n work begin pilot with data from RL                       1.93
67 5d    lab     n work reach out to christy re her wes work                1.93
68 5d    lab     n work explore application of vcf files in cnv pipeline    1.93
69 5d    lab     n work explore freebayes params                            1.93
72 3d    lab     n work contact oit re update policy and process            1.92
73 3d    lab     n work start back up process for ccgd server               1.92
76 2d    lab     n work continue to troubleshoot excavator2 pipeline        1.91
77 2d    lab     n work find pilot bam files for excavator2 test run        1.91
78 2d    lab     w work summarize excv2 process in methods.Rmd              1.91
40 9w  L lab     w work review scripts for ccgd database                    0.28
41 9w  L lab     w work explore shiny framework for database                0.28
42 9w  L lab            agenda [1]                                          0.16

12 tasks
```

### Excv2 trbl (cont.)

I've run the final script for the Excavator2 (excv2) pipeline to no avail. I'm at a bit of a loss and will likely transition to VarScan2 today unless I get a decent lead from reaching out the Excv2 authors.

```{r}
perl EXCAVATORDataAnalysis.pl ExperimentalFileAnalysis.50000.txt --processors 4 --target test_target --assembly hg19 --output /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/results --mode pooling
```

<details>
  <summary>**ExcavatorDataAnalysis Failed Output**</summary>

```{r}
Checking output folders...
'/home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/results' folder ready!
Checking output subfolders...
'/home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/results/Results' folder ready!
'/home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/results/Plots' folder ready!
'/home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/results/.tmp' folder ready!
Creating Pooling Control!
Warning message:
In dir.create(file.path(DataFolder, "Control")) :
  '/home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/results/Control' already exists
Warning message:
In dir.create(file.path(DataFolder, "Control", "RCNorm")) :
  '/home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/results/Control/RCNorm' already exists
Pooling Control Created!
Preparing Multiprocessor Analysis...
Multiprocessor Analysis Complete
Starting Multiprocessor Analysis!
My Children: 10574
Starting Segmentation and Calling...
Read 23 items
Error in JointSegIn(DataSeq1, muk, mi, smu, sepsilon, Pos1, omega, eta,  :
  NA/NaN/Inf in foreign function call (arg 3)
Execution halted
Segmentation and Calling Complete
Error in file(file, "rt") : cannot open the connection
Calls: read.table -> file
In addition: Warning message:
In file(file, "rt") :
  cannot open file '/home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/results/Results/g1b/HSLMResults_g1b.txt': No such file or directory
Execution halted
Multiprocessor Analysis Complete!

```

</details>

### VarScan2

While waiting to hear back from the excv2 authors, I've decided to see where VarScan2 can take me.

Project Home

* https://dkoboldt.github.io/varscan/index.html
* git: https://github.com/dkoboldt/varscan
* dir: `/home/chris/starr_lab/ocpmi/methods/varscan2`
* man: https://dkoboldt.github.io/varscan/using-varscan.html

I need to build a pileup file with `samtools`. I don't know how long that will take, so I'm letting that run on msi. Currently, I am planning to use the UCSC hg19 fasta reference with BAMs from the 1kgp (NA10847, NA19131).

### Summary

I've hit an impasse with excv2 and am putting it on hold without new information from the authors. I've reached out to them for a pilot dataset since I believe file inconsistencies are playing a role in the pipeline not running to completion. So as to not waste time, I've started the process of testing out VarScan2. I've downloaded the .jar file and am getting setup to run on msi. Taking a cursory glance, it seems like VarScan2 has a similar executional structure as excv2 in that it is also implemented through command line arguments. I expect to spend all of tomorrow focused on seeing how far I can get with this.


#### Completed

```{r}
Created    Completed  Age Project Description
---------- ---------- --- ------- --------------------------------------------
2019-07-25 2019-07-29 4d  lab     email rebecca re bam files
2019-07-23 2019-07-29 6d  lab     reach out to christy re her wes work
2019-07-26 2019-07-29 3d  lab     find pilot bam files for excavator2 test run
2019-07-29 2019-07-29 6h  lab     reach out to excv2 devs re test dataset
2019-07-26 2019-07-29 2d  lab     summarize excv2 process in methods.Rmd
2019-07-26 2019-07-29 3d  lab     continue to troubleshoot excavator2 pipeline

6 tasks
```

***

## 30

### TODO

```{r}
ID Age P Project Tag    Description                                      Urg
-- --- - ------- ------ ------------------------------------------------ ----
65 6d    lab     n work begin pilot with data from RL                    1.93
66 6d    lab     n work explore application of vcf files in cnv pipeline 1.93
67 6d    lab     n work explore freebayes params                         1.93
70 4d    lab     n work contact oit re update policy and process         1.92
71 4d    lab     n work start back up process for ccgd server            1.92
76 18h   lab     n work begin varscan2 testing                            1.9
40 10w L lab     w work review scripts for ccgd database                 0.28
41 10w L lab     w work explore shiny framework for database             0.28

8 tasks
```

### vs2 pileup file

VarScan2 (vs2) requires a pileup to operate. I'm running that on both msi and my personal machine as it seems to take a long time.

```{r}
samtools mpileup -f [reference sequence] [BAM file(s)]
```

***

## 31

### TODO

```{r}
ID Age P Project Tags   Description                                         Urg
-- --- - ------- ------ --------------------------------------------------- ----
64 7d    lab     n work begin pilot with data from RL                       1.94
65 7d    lab     n work explore application of vcf files in cnv pipeline    1.94
66 7d    lab     n work explore freebayes params                            1.94
69 5d    lab     n work contact oit re update policy and process            1.93
70 5d    lab     n work start back up process for ccgd server               1.93
75 1d    lab     n work begin varscan2 testing                              1.91
40 10w L lab     w work review scripts for ccgd database                    0.29
41 10w L lab     w work explore shiny framework for database                0.29

8 tasks
```

### Collaborator mtg

* Speaking with Ying, it sounds like RL and CH may not have saved bam files in their wes processing. I may need to generate my own files from the raw data
    - looking at the directory, Tim was able to find fastq files for this
* Tim has generally completed the cellranger3 reprocessing.
    - Apparently, the updated pipeline identifies a different resolution of rna content in immune cells. As a result, the new processing gives an elevated immune cell content in the Seurat output.

### excv2 w/ 1kgp GR38

**excv2 pilot**

* dir: `~/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output`
* samples: `NA10847`, `NA19131` via 1kgp phase3
* gene_ref: `hg38.fa`, `hg38.bed` via UCSC; `GCA_000001405.15_GRCh38.bw` via excv2

*msi install &rarr; GR38 ref &rarr; `TargetPerla.pl` &rarr; **`EXCAVATORDataPrepare.pl`** &rarr; `EXCAVATORDataAnalysis.pl` &rarr; cnv output*

### vs2 Pilot w/ 1kgp GR38

To get started on the vs2 pipeline, I've decided to work with the same 1kgp bam files I have available for excv2. In the event both work, I'll hopefully be able to compare the results.

**vs2 pilot**

* dir: `~/ocpmi/methods/varscan2`
* samples: `NA10847`, `NA19131` via 1kgp phase3
* gene_ref: `hg38.fa`, via UCSC

*BAMs &rarr; **`pileup` (samtools)** &rarr; [bam + pileup] `copynumber` (vs2) &rarr; cnv output?*

### BAM Creation

I need to create new bam files from our source wes data since these files were likely lost to scratch. I'll be doing a pilot run with two files that I will then run through each of excv2 and vs2.

**First Samples**

* src: `/home/nelsona2/data_release/umgc/hiseq/181205_D00635_0425_BCD07EANXX/Winterhoff_Project_001_SI`
* dir: `~/ocpmi/bam/ocpmi/pilot`
* samples: `PM-2-T17-0307-3_S9_R1_001.fastq`, `PM-15-T17-0910-1_S2_R1_001.fastq`
* gene_ref: `hg38.fa`, via UCSC

*fastq &rarr; **`index` gene ref (bwa bwtsw)** &rarr; align (bwa mem) + `sort` (samtools) &rarr; output.bam*

#### bwa

I'll be using bwa for sequence alignment.

* man: http://bio-bwa.sourceforge.net/bwa.shtml

Command for reference index creation using the bwtsw algorithm:

```{r}
bwa index -a bwtsw reference.fa
```

Command for direct bwa to bam output using the mem algorithm:

```{r}
bwa mem genome.fa reads.fastq | samtools sort -o output.bam -

# multiple threads

bwa mem -t 8 genome.fa reads.fastq | samtools sort -@8 -o output.bam -
```

### Summary

Currently, I have the vs2 and excv2 pipelines running in parallel. I've adjusted the approach of the excv2 pipeline to use the GR38 reference. I learned I will need to create new bam files since these were not preserved in scratch. To some extent this frees me to use the newer reference since I won't be relying on RL/CH's outputs.

***

# August

## 1

### TODO

```{r}
ID Age  P Project Tags   Description                                   Urg
-- ---- - ------- ------ --------------------------------------------- ----
64 8d     lab     w work explore freebayes params                      1.94
79 2min   lab     n work run pm2 and pm15 bams through vs2/excv2        1.9
80 1min   lab     n work evaluate excv2 output                          1.9
81 1min   lab     n work run vs2 copy number function on 1kgp bams      1.9
40 10w  L lab     w work review scripts for ccgd database              0.29
41 10w  L lab     w work explore shiny framework for database          0.29

6 tasks
```

### BAM Creation

I believe I've finished the process of converting a couple fastq files to bam files. I'm going to run these through the vs2 and excv2 pipelines to see if they will take.

**Pilot**

* src: `/home/nelsona2/data_release/umgc/hiseq/181205_D00635_0425_BCD07EANXX/Winterhoff_Project_001_SI`
* samples: `PM-2-T17-0307-3_S9_R1_001.fastq`, `PM-15-T17-0910-1_S2_R1_001.fastq`
* gene_ref: `hg38.fa`, via UCSC
* output: `pm2.bam`, `pm15.bam`; `~/ocpmi/data/bam/wes/pilot`

**Process**

*fastq &rarr; `index` gene ref (bwa bwtsw) &rarr; align (bwa mem) + `sort` (samtools) &rarr;*  ***output.bam***

Since it seems like I have this process down, I'm going to begin the transfer of all fastq files at the src dir. The script below should allow me to iterate the bwa conversion process across all the files.

```{r}
for i in ./*.fastq;
do
    bwa mem -t 8 ~/ocpmi/gene_ref/hg38.fa $i.fastq | samtools sort -@8 -o $i.bam -;
done

```

### vs2 Pilot w/ 1kgp GR38

I was able to create pile up files that will allow me to continue along the vs2 pipeline pilot test. I'm going to test these two using the `copynumber` function listed in the vs2 man. That function uses inputs from tumor and normal samples, so I will arbitrarily assign normal to `NA10847` and tumor to `NA19131`.

* samples: `NA10847`, `NA19131` via 1kgp phase3
* output: `na10847.mpileup` (normal), `na19131.mpileup` (tumor)
* src: `~/ocpmi/bam/1kgp`
* gene_ref: `hg38.fa`, via UCSC


**Start of day**

*BAMs &rarr; `pileup` (samtools) &rarr; **[bam + pileup] `copynumber` (vs2)** &rarr; output.copynumber &rarr; `copyCaller` (vs2) &rarr; segmentation &rarr; ?*


#### Trbl

I ran the vs2 `copynumber` function on my `na10847.mpileup` (normal), `na19131.mpileup` (tumor) samples. For some reason, only chr1 was processed.

```{r}
java -jar VarScan.v2.4.4.jar copynumber ~/ocpmi/bam/1kgp/na10847.mpileup ~/ocpmi/bam/1kgp/na19131.mpileup test_output
```

<details>
  <summary>**copynumber Output**</summary>

```{r}
Normal Pileup: /home/starrt2/tasta005/ocpmi/bam/1kgp/na10847.mpileup
Tumor Pileup: /home/starrt2/tasta005/ocpmi/bam/1kgp/na19131.mpileup
NOTICE: While dual input files are still supported, using a single mpileup file (normal-tumor) with the --mpileup 1 setting is strongly recommended.
Min coverage:   10
Min avg qual:   15
P-value thresh: 0.01
53515932 positions in tumor
53395419 positions shared in normal
10800306 had sufficient coverage for comparison
120744 raw copynumber segments with size > 10
120740 good copynumber segments with depth > 10

```

</details>

For the region that as processed, I was able to run `copyCaller` on `test_output.copynumber` and get an output.

```{r}
java -jar VarScan.v2.4.4.jar copyCaller test_output.copynumber
```

<details>
  <summary>**copyCaller Output**</summary>

241480 raw regions parsed
89442 met min depth
89442 met min size
11573 regions (1131082 bp) were called amplification (log2 > 0.25)
58753 regions (5816994 bp) were called neutral
19116 regions (1843613 bp) were called deletion (log2 <-0.25)
140 regions (7388 bp) were called homozygous deletion (normal cov >= 20 and tumor cov <= 5)

</details>

**End of day**

*BAMs &rarr; `pileup` (samtools) &rarr; [bam + pileup] `copynumber` (vs2) &rarr; output.copynumber &rarr; **`copyCaller` (vs2)** &rarr; segmentation &rarr; ?*

### excv2 w/ 1kgp GR38

I ran the EXCAVATORDataAnalysis.pl script over night, but it seems like there was an error with the output. I don't have any results figures even though msi gave an exit_status=0. I will try to trouble shoot in isub today.

**excv2 pilot**

* dir: `~/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output`
* samples: `NA10847`, `NA19131` via 1kgp phase3
* gene_ref: `hg38.fa`, `hg38.bed` via UCSC; `GCA_000001405.15_GRCh38.bw` via excv2

**Start of day**

*msi install &rarr; GR38 ref &rarr; `TargetPerla.pl` &rarr; `EXCAVATORDataPrepare.pl` &rarr; **`EXCAVATORDataAnalysis.pl`** &rarr; cnv output*

#### Trbl

Running the following script produced an error regarding a JointSegIn Fortran function. I found a thread that points to issues with the bed file.

* https://sourceforge.net/p/excavator2tool/tickets/1/


```{r}
perl EXCAVATORDataAnalysis.pl ExperimentalFileAnalysis.50000.txt --processors 8 --target test_target_38 --assembly hg38 --output ~/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/analysis --mode pooling

```


<details>
  <summary>**EXCAVATORDataAnalysis.pl failed output**</summary>
```{r}
Checking output folders...
'/home/starrt2/tasta005/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/analysis' folder ready!
Checking output subfolders...
Creating '/home/starrt2/tasta005/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/analysis/Results' folder...
Creating '/home/starrt2/tasta005/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/analysis/Plots' folder...
'/home/starrt2/tasta005/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/analysis/.tmp' folder ready!
Creating Pooling Control!
Pooling Control Created!
Preparing Multiprocessor Analysis...
Multiprocessor Analysis Complete
Starting Multiprocessor Analysis!
My Children: 29388
Starting Segmentation and Calling...
Read 23 items
Error in JointSegIn(DataSeq1, muk, mi, smu, sepsilon, Pos1, omega, eta,  :
  NA/NaN/Inf in foreign function call (arg 3)
Calls: JointSegIn -> .Fortran
Execution halted
Segmentation and Calling Complete
Error in file(file, "rt") : cannot open the connection
Calls: read.table -> file
In addition: Warning message:
In file(file, "rt") :
  cannot open file '/home/starrt2/tasta005/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/analysis/Results/NA19131/HSLMResults_NA19131.txt': No such file or directory
Execution halted
Multiprocessor Analysis Complete!
```


</details>

I'll go back and and rerun the pipeline after sorting the bedfile.

```{r}
sort -k1,1 -k2,2n *.bed | bedtools merge
```

**End of day**

*msi install &rarr; GR38 ref &rarr; bedfile sort &rarr; `TargetPerla.pl` &rarr; **`EXCAVATORDataPrepare.pl`** &rarr; `EXCAVATORDataAnalysis.pl` &rarr; cnv output*


### Summary

Today, I continued the 3 major threads I had started yesterday. The excv2 pipeline has produced a comparable error that leads me to a final attempt at getting the software to work. I may need to save the bed file doing the sort function, but I'm not sure about this. I'll run it one additional time after saving a sorted bed file to see if it makes a difference. If it doesn't, I will be done with excv2.

The bam file creation looks to have worked fine, so I am extending that process to the entire set of fastq files. They are being transferred and extracted tonight.

The vs2 process has worked to some degree at least producing actual copy number data. This is limited on the level that it only applied to chr1. It's possible the pileup creation is the issue here. I need to apply some additional processing to smooth this out the actual copy number data though in addition to segmentation. I haven't actually found what their suggested visualization method is either.

#### Completed

```{r}
Created    Completed  Age Project Description
---------- ---------- --- ------- ----------------------------------------------
2019-07-25 2019-08-01 7d  lab     contact oit re update policy and process
2019-07-25 2019-08-01 7d  lab     start back up process for ccgd server
2019-07-31 2019-08-01 1d  lab     call msi re isub closeout
2019-07-23 2019-08-01 9d  lab     begin pilot with data from RL
2019-07-29 2019-08-01 2d  lab     begin varscan2 testing
2019-07-23 2019-08-01 9d  lab     explore application of vcf files in cnv
                                  pipeline
2019-08-01 2019-08-01 4h  lab     evaluate excv2 output
2019-08-01 2019-08-01 4h  lab     explore process of job parallelization on msi

8 tasks
```

***

## 2

### TODO

```{r}
ID Age P Project Tags   Description                                   Urg
-- --- - ------- ------ --------------------------------------------- ----
64 9d    lab     w work explore freebayes params                      1.95
79 23h   lab     n work run pm2 and pm15 bams through vs2/excv2        1.9
80 23h   lab     n work run vs2 copy number function on 1kgp bams      1.9
81 18h   lab     n work rerun excv2 with gr38 sorted bed               1.9
82 18h   lab     n work trbl vs2 only processing chr1                  1.9
42 10w L lab     work   agenda [3]                                    1.18
40 10w L lab     w work review scripts for ccgd database               0.3
41 10w L lab     w work explore shiny framework for database           0.3

8 tasks
```

### Annotation

For consistency in documenting the wide range of information, I want to have a systematic means of tracking this accross experimentation. Going forward I will try to use this general convention where files are listed by category followed by a directory if necessary.

* samples: `file`; `/path/to/file/`
* output: `/path/to/output/`
* gene_ref: `hg38.fa`, via UCSC; `~/ocpmi/gene_ref`

### BAM Creation

Looking at the bam files I created, I noticed they're all 92B in size. This is probably an indication something went wrong in their creation. The bwa script seems to work fine on its own, so I imagine the issue is the bash script I created.

* samples: `*.fastq`; `~/ocpmi/bam/wes`
* output: `~/ocpmi/bam/wes`
* gene_ref: `hg38.fa`, via UCSC; `~/ocpmi/gene_ref`

I'm trying a new script I found on biostars that uses gnu-parallel. I'll run this overnight on all files in the `/wes` directory and see where that gets me.

* https://www.biostars.org/p/231491/

```{r}
cd ~/ocpmi/bam/wes
module load bwa samtools parallel

ls *.fastq | parallel -j 2 'bwa mem ~/ocpmi/gene_ref/hg38.fa {} | samtools sort -o {.}.bam'

```

### vs2 Pilot w/ 1kgp GR38

* samples: `test_output.copynumber`; `~/ocpmi/methods/varscan2`
* output: `copyCall_test_output.txt`; `~/ocpmi/methods/varscan2`
* gene_ref: `hg38.fa`, via UCSC; `~/ocpmi/gene_ref`

**Start of day**

*BAMs &rarr; `pileup` (samtools) &rarr; [bam + pileup] `copynumber` (vs2) &rarr; output.copynumber &rarr; **`copyCaller` (vs2)** &rarr; segmentation &rarr; ?*

```{r}
java -jar VarScan.v2.4.4.jar copyCaller test_output.copynumber > copyCall_test_output.txt 2>&1
```

I set created a file containing the output for this function but didn't take time to make sense of it. Running segmentation will be a little more invovled since this isn't an included functionality in vs2, and I don't have a plan of attach currently. The vs2 man suggests using the `DNAcopy` package from BioC, which employs the CBS algorithm.

* https://www.bioconductor.org/packages/2.3/bioc/html/DNAcopy.html

**End of day**

*BAMs &rarr; `pileup` (samtools) &rarr; [bam + pileup] `copynumber` (vs2) &rarr; output.copynumber &rarr; `copyCaller` (vs2) &rarr; **segmentation** &rarr; ?*

### excv2 w/ 1kgp GR38

* samples: `NA10847`, `NA19131` via 1kgp phase3; `~/ocpmi/bam/1kgp`
* output: `~/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output`
* gene_ref: `hg38.fa`, `hg38_sorted.bed` via UCSC; `GCA_000001405.15_GRCh38.bw` via excv2

**Start of day**

*sorted bed + GR38 ref &rarr; `TargetPerla.pl` &rarr; `EXCAVATORDataPrepare.pl` &rarr; `EXCAVATORDataAnalysis.pl` &rarr;* ***cnv output***

It seems like this actually worked. Unbelievable. In hindsight, I did a variety of things wrong, but it seems like I would be able to employ this pipeline across a full scale of samples now.

### Summary

I'm continuing to make progress on the three tandem processes I have going. I believe I can say the excv2 pipeline officially produced a complete result today making that the first of my time in the lab. woot. The bam files I thought I had created last night turned out to be faulty, so I am currently re-running that process with a new script. Additionally, I have a `copyCaller` output from vs2 that I could possibly feed into a segmentation algorithm of my choosing. Taken together, I think I'll be able to run the full data set soon.

#### Completed

```{r}
Created    Completed  Age Project Description
---------- ---------- --- ------- ----------------------------------------------
2019-08-01 2019-08-02 1d  lab     reply to oit help ticket re upgrad process
2019-08-01 2019-08-02 1d  lab     update exit codes table
2019-08-01 2019-08-02 1d  lab     extract wes fastq
2019-08-01 2019-08-02 1d  lab     convert wes fastq to bam
2019-08-01 2019-08-02 1d  lab     transfer wes fastq from nelsona2
2019-08-01 2019-08-02 1d  lab     rerun excv2 with gr38 sorted bed
2019-08-01 2019-08-02 1d  lab     run vs2 copy number function on 1kgp bams
2019-08-02 2019-08-02 3h  lab     learn if theres a remote access to msi for at
                                  ups

8 tasks
```

***

## 5

### TODO

```{r}
ID Age  P Project Tags   Description                                        Urg
-- ---- - ------- ------ -------------------------------------------------- ----
68 4d     lab     n work run pm2 and pm15 bams through vs2/excv2            1.92
69 3d     lab     n work trbl vs2 only processing chr1                      1.92
70 2d     lab     n work add some pbs scripts to methods                    1.91
71 2d     lab     n work evaluate excv2 1kgp gr38 output                    1.91
72 2d     lab     n work make readme for notebook                           1.91
73 2d     lab     w work add excv2 1kgp output to results page              1.91
74 1d     lab     n work add citation for parallel and other pipeline tools 1.91
                         bwa, samtools, etc
76 1d     lab     n work add tree diagrams to relevant dirs                 1.91
79 7min   lab     n work explore controls for cnv                            1.9
80 1min   lab     w work call oit re open tickets                            1.9
42 10w  L lab     work   agenda [3]                                          1.2
40 10w  L lab     w work review scripts for ccgd database                   0.32
41 10w  L lab     w work explore shiny framework for database               0.32

13 tasks


```

### BAM Creation

I ran the process through the weekend with an extended wall time. It took several goes in excess of 24 hrs. I’m not sure if there is a more computationally efficient method, but I do need to confirm my process with Tim or a tutorial.

### vs2 Pilot: ocpmi pm2/15

* samples: `pm2.bam`, `pm15.bam`; `~/ocpmi/bam/wes/pilot`
* output: `~/ocpmi/bam/wes/pilot`
* gene_ref: `hg38.fa`, via UCSC; `~/ocpmi/gene_ref`

**Start of day**

*BAMs &rarr; `pileup` (samtools) &rarr; [bam + pileup] `copynumber` (vs2) &rarr; output.copynumber &rarr; **`copyCaller` (vs2)** &rarr; segmentation &rarr; ?*

I’ve discovered my pileup files didn’t get processed completely which explains why only chr1 is represented. I’ve started a new job to rerun the process with an updated bash script as well. Also, according to the thread below, another issue presenting itself is that the \*.cram files I thought were aligned with hg38.fa actually used a different reference, which would cause the mpileup process to fail. As a result, I'm rerunning the process with the pilot bam files I created.

* https://www.biostars.org/p/93643/

```{r}
cd ~/ocpmi/bam/wes/pilot

for i in *.bam; do
    samtools mpileup -f ~/ocpmi/gene_ref/hg38.fa "$i" > ${i%%.*}.mpileup
done

```

**End of day**

*BAMs &rarr; **`pileup` (samtools)** &rarr; [bam + pileup] `copynumber` (vs2) &rarr; output.copynumber &rarr; `copyCaller` (vs2) &rarr; segmentation &rarr; ?*

### Summary

I spent much of the day doing documentation work on the 3 major processes I've been running since last week. Given the number of concurrent things I had running, I didn't want to lose track of anything.

Additionally, I've discovered some issues with my vs2 pilot and have restarted that process. Looking ahead, I should have a variety of things to discuss with Tim re the BAM files and excv2 pipeline. I need to evaluate that stuff.

#### Completed

```{r}
Created    Completed  Age   Project Description
---------- ---------- ----- ------- -----------------------------
2019-08-05 2019-08-05 5h    lab     reply to oit re ccgd upgrade
2019-08-01 2019-08-05 4d    lab     trbl vs2 only processing chr1
2019-08-05 2019-08-05 16min lab     rerun pileup_create

3 tasks
```

***

## 6

### TODO

```{r}
ID Age P Project Tags Description                                           Urg
-- --- - ------- ---- ----------------------------------------------------- ----
68 4d    lab     work add some pbs scripts to methods                       1.82
69 4d    lab     work evaluate excv2 1kgp gr38 output                       1.82
70 4d    lab     work make readme for notebook                              1.82
71 4d    lab     work add excv2 1kgp output to results page                 1.82
72 3d    lab     work add citation for parallel and other pipeline tools    1.82
                      bwa, samtools, etc
74 3d    lab     work add tree diagrams to relevant dirs                    1.82
77 1d    lab     work explore controls for cnv                              1.81
78 1d    lab     work check process of bam creation (youtube)               1.81
79 1d    lab     work update mnt_msi docs                                   1.81
81 1d    lab     work update notebook_knit.R docs                           1.81
42 10w L lab     work agenda [4]                                            1.21
40 11w L lab     work review scripts for ccgd database                      0.23
41 11w L lab     work explore shiny framework for database                  0.23

13 tasks
```

### vs2 Pilot: ocpmi pm2/15

* samples: `pm2.bam`, `pm15.bam`; `~/ocpmi/bam/wes/pilot`
* output: `~/ocpmi/bam/wes/pilot`
* gene_ref: `hg38.fa`, via UCSC; `~/ocpmi/gene_ref`

**Start of day**

*__&ast;.bam `mpileup` (samtools)__ &rarr; &ast;.mpileup `copynumber` (vs2) &rarr; output.copynumber `copyCaller` (vs2) &rarr; output.copynumber.called &rarr; segmentation (DNAcopy) &rarr; visualization &rarr; ?*

#### vs2 copyCaller Script

```{r}
cd ~/ocpmi/bam/wes/pilot
java -jar ~/ocpmi/methods/varscan2/VarScan.v2.4.4.jar copyCaller pilot_output.copynumber --output-file pilot_output.copynumber.called
```

A primary difference for the vs2 package is that it doesn't include a segmentation algorithm. The documentation suggests CBS via the DNAcopy BioC library.

* https://www.bioconductor.org/packages/2.3/bioc/html/DNAcopy.html

**End of day**

*&ast;.bam `mpileup` (samtools) &rarr; &ast;.mpileup `copynumber` (vs2) &rarr; output.copynumber `copyCaller` (vs2) &rarr; __output.copynumber.called__ &rarr; segmentation (DNAcopy) &rarr; visualization &rarr; ?*

### Project folder sync

I got sick of pushing and pulling back and forth between msi and my personal machine, so I decided to fully transition what I had left from the personal machine to msi. This is largely my notebook structure and ancillary files that are ignored by git.

What this amounted to was doing an `rsync` command to bring the folders current with each other along with pointing some files to the mount location of my sshfs for msi.

### Summary

Today consisted of a large amount of house keeping.

***

## 7

### TODO

```{r}
ID Age P Project Tags Description                                           Urg
-- --- - ------- ---- ----------------------------------------------------- ----
68 4d    lab     work add some pbs scripts to methods                       1.82
69 4d    lab     work evaluate excv2 1kgp gr38 output                       1.82
70 4d    lab     work make readme for notebook                              1.82
71 4d    lab     work add excv2 1kgp output to results page                 1.82
72 3d    lab     work add citation for parallel and other pipeline tools    1.82
                      bwa, samtools, etc
74 3d    lab     work add tree diagrams to relevant dirs                    1.82
77 1d    lab     work explore controls for cnv                              1.81
78 1d    lab     work check process of bam creation (youtube)               1.81
79 1d    lab     work update mnt_msi docs                                   1.81
81 1d    lab     work update notebook_knit.R docs                           1.81
42 10w L lab     work agenda [4]                                            1.21
40 11w L lab     work review scripts for ccgd database                      0.23
41 11w L lab     work explore shiny framework for database                  0.23

13 tasks
```

### vs2 Pilot: ocpmi pm2/15

**Start of day**

*&ast;.bam `mpileup` (samtools) &rarr; &ast;.mpileup `copynumber` (vs2) &rarr; output.copynumber `copyCaller` (vs2) &rarr; __output.copynumber.called__ &rarr; segmentation (DNAcopy) &rarr; visualization &rarr; ?*

### Summary

Today was another day for housekeeping. I've made a significant modification to the layout of my notebook by instituting a .tabset parameter. Things were starting to feel somewhat unruly with the page length and continuous flow at this point, so tabbing sections should help. Additionally, I made sure to go back and get citations for the recent tools I've incorporated into my workflow.

I expected to be meeting with Tim where I thought we would be able to discuss a few things that would define the next phase of my project with the wes data. I found he will be out for the week though, and will need to work on incremental items till then. I might just run the excv2 pipeline with the bam files I've created either way.

#### Completed

```{r}
Created    Completed  Age Project Description
---------- ---------- --- ------- ----------------------------------------------
2019-08-04 2019-08-07 3d  lab     add citation for parallel and other pipeline
                                  tools bwa, samtools, etc
2019-08-07 2019-08-07 1h  lab     install dnacopy on msi
2019-08-07 2019-08-07 1h  lab     add tabset formatting to notebook

3 tasks
```

***

## 8

### TODO

```{r}
ID Age P Project Tags Description                                 Urg
-- --- - ------- ---- ------------------------------------------- ----
83 17h H lab     work review dnacopy [1]                           8.6
68 5d    lab     work add some pbs scripts to methods             1.83
69 5d    lab     work evaluate excv2 1kgp gr38 output             1.83
70 5d    lab     work make readme for notebook                    1.83
71 5d    lab     work add excv2 1kgp output to results page       1.83
73 4d    lab     work add tree diagrams to relevant dirs          1.82
76 2d    lab     work explore controls for cnv                    1.81
77 2d    lab     work check process of bam creation (youtube)     1.81
78 2d    lab     work update mnt_msi docs                         1.81
80 2d    lab     work update notebook_knit.R docs                 1.81
85 14s   lab     work run excv2 across ocpmi bams                  1.8
42 10w L lab     work agenda [6]                                  1.22
40 11w L lab     work review scripts for ccgd database            0.23
41 11w L lab     work explore shiny framework for database        0.23

14 tasks
```

### vs2 Pilot: ocpmi pm2/15

* samples: `pm2.bam`, `pm15.bam`; `~/ocpmi/bam/wes/pilot`
* output: `~/ocpmi/methods/varscan2/pilot`
* gene_ref: `hg38.fa`, via UCSC; `~/ocpmi/gene_ref`

**Start of day**

*&ast;.bam `mpileup` (samtools) &rarr; &ast;.mpileup `copynumber` (vs2) &rarr; output.copynumber `copyCaller` (vs2) &rarr; __output.copynumber.called__ &rarr; segmentation (DNAcopy) &rarr; visualization &rarr; ?*

I moved the location of the associated files to the vs2 methods directory to get them out of the gene data folder they were in temporarily.

I made an R script that utilizes the `DNAcopy` BioC package for segmentation by the CBS algorithm. Simply, this will take the output of the vs2 `copyCaller` function and build a `cna.object` to perform segmentation and visualization. There are several plotting options with this function. This one currently employs the whole genome view.

* CNA: https://www.rdocumentation.org/packages/DNAcopy/versions/1.46.0/topics/CNA
* plot.DNAcopy: https://www.rdocumentation.org/packages/DNAcopy/versions/1.46.0/topics/plot.DNAcopy

**dnacopy_seg_plot.R**

```{r}
library(DNAcopy)

cn <- read.table("pilot_output.copynumber.called",header=T)
cna.object <- CNA(genomdat = cbind(cn$normal_depth, cn$tumor_depth), chrom = cn$chrom, maploc = cn$chr_start, data.type = "logratio", sampleid=c("pm2","pm15"))

# Smoothing object
cna.object.smooth <- smooth.CNA(cna.object)

#Segmentation at default parameters
cna.object.smooth.segd <- segment(cna.object.smooth, verbose=1)

#Plot whole studies
pdf('rplot.pdf')
plot(cna.object.smooth.segd, plot.type="w")
dev.off()

```

One point of clarity I need is related to the `genomdat` param in the `CNA` function. I'm not sure if this should point to read depth data or the log ratio.

**End of day**

*&ast;.bam `mpileup` (samtools) &rarr; &ast;.mpileup `copynumber` &rarr; output.copynumber `copyCaller` &rarr; output.copynumber.called &rarr; segmentation (DNAcopy) &rarr; __visualization (DNAcopy)__*

### Summary

I was able to complete the vs2 pipeline today. There are some minor elements I may need to confirm in the form of how the controls relate to samples in the vs2 `copynumber` function. Otherwise, I think I may be in a position to run both pipelines across the full wes dataset. I will likely get this started tomorrow.

I also cleaned up my directory structure. This may have been a huge mistake in some sense because I will have to go reassign a bunch of directory pointers, but things were getting super unorganized.

#### Completed

```{r}
Created    Completed  Age P Project Description
---------- ---------- --- - ------- --------------------------------------------
2019-08-06 2019-08-08 2d    lab     update mnt_msi docs
2019-08-06 2019-08-08 2d    lab     update notebook_knit.R docs
2019-08-07 2019-08-08 23h H lab     review dnacopy
                                      2019-08-07
                                    https://www.rdocumentation.org/packages/DNA-
                                    copy/versions/1.46.0/topics/CNA
2019-08-02 2019-08-08 6d    lab     add some pbs scripts to methods
2019-08-02 2019-08-08 6d    lab     evaluate excv2 1kgp gr38 output

5 tasks
```

***

## 10

### excv2 Pilot redo: 1kgp GRCh38

* samples: `NA10847`; `NA19131` via 1kgp GRCh38
* output: `~/ocpmi/results/excavator2/pilot`
* gene_ref: `hg38.fa`, `hg38_sorted.bed` via UCSC; `~/ocpmi/data/gene_ref/ucsc/hg38`

In performing this directory restructuring, I came across the realization I may have mixed up file association for the version of the excv2 pilot that worked. The config files for excv2 seem to be pointing to older versions of the pilot I had attempted but didn't work. I'm rerunning the pilot with the file set I believe worked in the new directory setup.

***

## 11

### excv2 Pilot redo: 1kgp GRCh38

* samples: `NA10847`; `NA19131` via 1kgp GRCh38
* output: `~/ocpmi/results/excavator2/pilot`
* gene_ref: `hg38.fa`, `hg38_sorted.bed` via UCSC; `~/ocpmi/data/gene_ref/ucsc/hg38`

I'm generally continuing this process, and it isn't working for some inexplicable reason. I have been making minor adjustments to the config files to replicate the original successful run as best as possible to no avail. The most ridiculous part is the fact I'm getting the `JointSegIn` error from *2019-08-01*. I will pretty much just keep tinkering with no clue what is going on.

***

## 12

### TODO

```{r}
ID Age P Project Tags Description                                       Urg
-- --- - ------- ---- ------------------------------------------------- ----
68 9d    lab     work make readme for notebook                          1.85
69 9d    lab     work add excv2 1kgp output to results page             1.85
71 8d    lab     work add tree diagrams to relevant dirs                1.84
74 6d    lab     work explore controls for cnv                          1.83
75 6d    lab     work check process of bam creation (youtube)           1.83
80 3d    lab     work run excv2 and vs2 across ocpmi bams               1.82
81 3d    lab     work create loop for excv2 and vs2 to run all bams     1.82
82 3d    lab     work update dir links to bam, pbs, gene_ref            1.82
85 1d    lab     work update support txt files for excv2                1.81
42 11w L lab     work agenda [6]                                        1.24
40 11w L lab     work review scripts for ccgd database                  0.25
41 11w L lab     work explore shiny framework for database              0.25

12 tasks
```

### excv2 Pilot redo: 1kgp GRCh38

* samples: `NA10847` (control); `NA19131` via 1kgp GRCh38
* output: `~/ocpmi/results/excavator2/pilot`
* gene_ref: `hg38.fa`, `hg38_sortmerge.bed` via UCSC; `~/ocpmi/data/gene_ref/ucsc/hg38`

I was able to fully execute the excv2 pipeline again with the pilot 1kgp samples. The issue I was having in this case was an incorrect bam pointer in the `ExperimentalFilePrepare.50000.txt` file for sample `NA19131`. This had the unfortunate result of generating the same error as the sorted bed file error I solved on *2019-08-01*.

### Summary

Order has been restored in the excv2 pipeline, and I can go into my next conversation with Tim with the expectation that we can determine experimental parameters for the full scale wes analysis. In addition, I did some more documentation restructuring which extended from my directory restructure on *2019-08-08*. This included a `README.Rmd` for both my `Log` and the notebook dir itself. I also updated the `methods` to reflect any dir adjustments and will need to update any outdated reference as I go.

#### Completed

```{r}
Created    Completed  Age Project Description
---------- ---------- --- ------- -------------------------------------
2019-08-02 2019-08-12 10d lab     make readme for notebook
2019-08-11 2019-08-12 1d  lab     update support txt files for excv2
2019-08-04 2019-08-12 8d  lab     add tree diagrams to relevant dirs
2019-08-02 2019-08-12 10d lab     add excv2 1kgp output to results page
2019-08-12 2019-08-12 17s lab     rerun excv2 with new dir setup

5 tasks
```

***

## 13

### TODO

```{r}
ID Age P Project Tags Description                                       Urg
-- --- - ------- ---- ------------------------------------------------- ----
71 7d    lab     work explore controls for cnv                          1.84
72 7d    lab     work check process of bam creation (youtube)           1.84
77 5d    lab     work run excv2 and vs2 across ocpmi bams               1.83
78 4d    lab     work create loop for excv2 and vs2 to run all bams     1.82
79 4d    lab     work update dir links to bam, pbs, gene_ref            1.82
84 22h   lab     work index ocpmi bams                                   1.8
42 11w L lab     work agenda [6]                                        1.24
40 12w L lab     work review scripts for ccgd database                  0.26
41 12w L lab     work explore shiny framework for database              0.26

9 tasks
```

### excv2 Pilot: pm2/15

* samples: `pm2` (control); `pm15`; `~/ocpmi/data/bam/wes/pilot`
* output: `~/ocpmi/results/excavator2/pilot1`
* gene_ref: `hg38.fa`, `hg38_sortmerge.bed` via UCSC; `~/ocpmi/data/gene_ref/ucsc/hg38`

I reran a pilot test of the excv2 with the same samples as the pilot for vs2. Everything worked as expected, which is nice.

### BAM indexing

* samples: `*.bam`; `~/ocpmi/data/bam/wes`
* output: `~/ocpmi/data/bam/wes`

I forgot to index the bam files I had created from *2019-08-05*. I will perform this function using gnu parallel and samtools.

* https://www.biostars.org/p/114921/

```{r}
parallel samtools index ::: *.bam
```

### excv2 create_configs.sh File Builder

The excv2 pipeline requires repetitive config files to be setup for each experimental case. As a result, I've started to write a script that will allow me to automate the process of creating the excv2 config files for when I apply the pipeline to a wider set of data. This should take directory pointers and generate the two text files for the 2nd and 3rd steps of the excv2 process.

```{r}
#!/bin/bash

> ExperimentalFilePrepare.txt
> ExperimentalFileAnalysis.txt
prepDir="data/prepare/test"
bamDir="/home/chris/ocpmi/data/bam/wes"

for i in *.bam; do
    echo "$bamDir/$i" "$prepDir/${i%.*}" "${i%.*}" >> ExperimentalFilePrepare.txt
    echo "T" "$prepDir/${i%.*}" "${i%.*}" >> ExperimentalFileAnalysis.txt
    sed -i -e 's/\/home\/chris/~/g' ./ExperimentalFilePrepare.txt
done
```

### Summary

I was able to smoothly run the excv2 pipeline on another set of test data today. Overall, I feel like I have a mastery of the process at this point and should be able to apply the package in any case. I've begun to write a script that should automate the process to some degree so that the relatively disjointed steps can be strung together more effectively.

Additionally, I heard back from Bob

#### Completed

```{r}

Created    Completed  Age Project Description
---------- ---------- --- ------- ---------------------------------------
2019-08-12 2019-08-13 2d  lab     contact oit about meeting again
2019-08-08 2019-08-13 5d  lab     update dir links to bam, pbs, gene_ref
2019-08-12 2019-08-13 1d  lab     index ocpmi bams
2019-08-05 2019-08-13 8d  lab     check process of bam creation (youtube)

4 tasks
```

***

## 14

### TODO

```{r}

ID Age P Project Tags Description                                       Urg
-- --- - ------- ---- ------------------------------------------------- ----
71 8d    lab     work explore controls for cnv                          1.84
76 6d    lab     work run excv2 and vs2 across ocpmi bams               1.83
77 5d    lab     work create loop for excv2 and vs2 to run all bams     1.83
82 20h   lab     work create output for pm2 and pm15 with vs2            1.8
83 12s   lab     work submit application for new vm                      1.8
42 11w L lab     work agenda [8]                                        1.25
40 12w L lab     work review scripts for ccgd database                  0.27
41 12w L lab     work explore shiny framework for database              0.27

8 tasks
```

### excv2 create_configs.sh File Builder

I expanded on the script I started yesterday. This script now effectively automates the entire excv2 process in one command. It requires having created a reference target completed with the `TargetPerla.pl` command already, but that process does not need to be completed if using the same reference genome, which I am.

```{r}
#!/bin/bash

# This script is meant to prep the configuration and execution of the Excavator2 pipeline. It first auto generates two text files that will serve as configs for the 2nd and 3rd pipeline scripts. It then creates directories for each sample at the prepDir site. The relevant variables are the directory locations for the bam files, the DataPrepare output, and the results output defined by the "vars.txt" file. Both full_send.sh and vars.txt need to be in the Excavator2 working directory.

# This file assumes a reference target has been created for repeat use using the TargetPerla.pl function. The name of this target should also be given in "vars.txt".

source vars.txt
cd "$HOME/$bamDir"

> ExperimentalFilePrepare.txt
> ExperimentalFileAnalysis.txt

# Loop to iterate through each bam
count=1
for i in *.bam; do
    echo "$PWD/$i" "$prepDir/${i%.*}" "${i%.*}" >> ExperimentalFilePrepare.txt
    echo "T$count" "$prepDir/${i%.*}" "${i%.*}" >> ExperimentalFileAnalysis.txt
    mkdir -p ~/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/$prepDir/${i%.*}
    count=$[count+1]
done

# Function to remove leading path from sshfs local mount path
sed -i -e 's/\/home\/chris/~/g' ./ExperimentalFilePrepare.txt
sed -i '$s/T[[:digit:]]*/C1/g' ./ExperimentalFileAnalysis.txt

# Working directory of the Excavator2 pipeline
mv ExperimentalFile* ~/ocpmi/methods/EXCAVATOR2_Package_v1.1.2

perl EXCAVATORDataPrepare.pl ExperimentalFilePrepare.txt --processors 8 --target $target --assembly hg38
perl EXCAVATORDataAnalysis.pl ExperimentalFileAnalysis.txt --processors 8 --target $target --assembly hg38 --output $HEAD/$reDir --mode pooling

```

### Call w/ Bob re CCGD upgrade process

* Don't upgrade in place
    - new vm and name/ip
    - can add alias for web address redirect
* Management is important
    - Fully vs Jointly managed
        + Full = OS admin performed by OIT (preferred)
        + Joint = OS admin by starrlab, root access
* Bob sent links re package mgmt and vm req
* Access to the server needs to happen over the vpn for now. Being on the U's network gives me a `port 22: Connection refused` error, and ale01 seems to have an issue with duo.

**actions**

* I need to discuss these items with Tim, specifically: the management format Tim prefers, which department the server would be hosted under, and if there were site analytics available.
* Clarify duo login issue with ale01.

### mtg w/ Tim

present: tim, chris

```{r}
Description
agenda
  2019-07-26 what pull down method was used for the wes data
  2019-08-01 check my bam process
  2019-08-02 determine controls for wes data
  2019-08-05 ccgd site analytics - how long could site be down
  2019-08-07 discuss excv2 output
  2019-08-07 progress on vs2 pilot
  2019-08-14 management preference for ccgd
```

#### mtg notes

**ocpmi wes**

* Tim agreed that the output of the excv2 and vs2 pilots turned out well enough. The two outputs for excv2 show effectively no cnv in the cases of the 1kgp genomes (where there should be none) and a high degree in our ocpmi samples (where there should be a lot). Additionally, the vs2 output, while odd, shows a similar result reaffirming the output of excv2 in the case of the ocpmi samples.
* We face an issue in running the wider set of ocpmi samples in that we don't have a dedicated control(s). I was somewhat concerned about this, and Tim confirmed that we don't have pt specific samples that would serve a tumor-normal experimental design. Our best bet is to find and use a publicly available genome that doesn't introduce any additional variables to our setup. The 1kgp genomes seem like the best option.

**alignment**

* I wanted to confirm my process in creating the bam files. What I've done is probably acceptable with `bwa`, but Tim did mention the `star` tool that Mayank and others have brought up. I will probably rerun the alignment with `star` then.

**ccgd**

* I gave Tim an update on the process of the ccgd upgrade. I was able to talk with Bob of OIT earlier today and have a general direction in the upgrade process.
* Points of clarification were the management format Tim prefers, which department the server would be hosted under, and if there were site analytics available.
    - A fully managed server is preferred.
    - Tim thinks the dept abbreviation is OGC. Looking at this further, it needs to be HST.
    - There are analytics which are referenced in the project doc.

**actions**

* realign bam files with star
* run comparison with 1kgp and ocpmi files
* look for differences 1kgp comparison in exv2 paper
* do search for best suited cnv controls
* get analytics for ccgd
* submit vm application for ccgd
* determine ports that need to be open ccgd

### Summary

I had a couple meetings today which allowed me to address several standing questions for both the ocpmi and ccgd projects.

After speaking with Bob, I have a sense of what needs to be done to transition the server from an OS implementation standpoint. From there I will be able to do application testing to see if there are any params that need to be adjusted, but I'm hoping that only amounts to installing legacy packages to address existing scripts.

On the ocpmi front, I need to work on determining what my controls should be. I will be running a small experiment to size up `NA10847` as a candidate control. Otherwise, I need to explore what the best approach will be in this experimental setup.

#### Completed

```{r}
Created    Completed  Age Project Description
---------- ---------- --- ------- -----------------------------
2019-08-05 2019-08-14 10d lab     explore controls for cnv
2019-08-14 2019-08-14 1d  lab     determine open ports for ccgd

2 tasks
```

***

## 15

### TODO

```{r}
ID Age P Project Tags Description                                           Urg
-- --- - ------- ---- ----------------------------------------------------- ----
67 7d    lab     work run excv2 and vs2 across ocpmi bams                   1.84
68 6d    lab     work create loop for excv2 and vs2 to run all bams         1.83
72 1d    lab     work create output for pm2 and pm15 with vs2               1.81
73 21h   lab     work clarify duo login issues with ale01                    1.8
74 21h   lab     work realign bam files with star                            1.8
75 21h   lab     work run test comparison with 1kgp and ocpmi files          1.8
76 21h   lab     work look for differences in 1kgp comparison in exv2 paper  1.8
77 21h   lab     work do search for best suited cnv controls                 1.8
78 21h   lab     work get site analytics for ccgd                            1.8
79 21h   lab     work submit vm application for ccgd                         1.8
80 19h   lab     work copy na10847 back to og dir                            1.8
81 19h   lab     work add mkdir for prepDir to fullsend.sh                   1.8
82 19h   lab     work test full send with ocpmi pilot samples                1.8
40 12w L lab     work review scripts for ccgd database                      0.27
41 12w L lab     work explore shiny framework for database                  0.27
42 11w L lab     work agenda                                                0.25

16 tasks
```

### excv2 full_send.sh and vars.sh

I continued the expansion of the script I started on *2019-08-13* by bringing a variable source file into the mix. I am now about 95% of the way to turning the entire pipeline into a single command with variable modifiers simplifying the process.

**Cur full_send.sh**

* dir: `~/ocpmi/methods/EXCAVATOR2_Package_v1.1.2`

```{r}
#!/bin/bash
# This script is meant to prep the configuration and execution of the Excavator2 pipeline. It first auto generates two text files that will serve as configs for the 2nd and 3rd pipeline scripts. It then creates directories for each sample at the prepDir site.

source vars.sh
cd $HOME$bamDir

# Overwrite any existing config files
> ExperimentalFilePrepare.txt
> ExperimentalFileAnalysis.txt

# Loop to iterate through each bam to create string for FilePrepare and FileAnalysis configs
count=1
for i in *.{bam,cram}; do
    echo "$PWD/$i" "data/prepare/$expName/${i%.*}" "${i%.*}" >> ExperimentalFilePrepare.txt
    echo "T$count" "data/prepare/$expName/${i%.*}" "${i%.*}" >> ExperimentalFileAnalysis.txt
    mkdir -p $HOME$excv2Dir/data/prepare/$expName/${i%.*}
    count=$[count+1]
done

# Function to remove leading path from sshfs local mount path and create control
sed -i -e 's/\/home\/chris/~/g' ./ExperimentalFilePrepare.txt
sed -i '$s/T[[:digit:]]*/C1/g' ./ExperimentalFileAnalysis.txt

# Relocate configs to Excavator working dir
mv ExperimentalFile* $HOME$excv2Dir
mkdir $HOME$resDir/$expName

# Generate SourceTarget.txt and execute step 1 TargetPerla.pl
cd $HOME$excv2Dir
if [ "$runTp"="1" ]; then
    echo "$HOME$bigWig" "$HOME$genRef" > SourceTarget.txt
    perl TargetPerla.pl SourceTarget.txt $HOME$bedFile $target $window $refAssm
fi

# Executing steps 2 and 3 of the Excavator pipeline
perl EXCAVATORDataPrepare.pl ExperimentalFilePrepare.txt --processors $numProc --target $target --assembly $refAssm
perl EXCAVATORDataAnalysis.pl ExperimentalFileAnalysis.txt --processors $numProc --target $target --assembly $refAssm --output $HOME$resDir/$expName --mode $anMode

```

**vars.sh**

* dir: `~/ocpmi/methods/EXCAVATOR2_Package_v1.1.2`

This is a source file that is read into `full_send.sh` for variable modification. I have generally designed this file for anyone to be able to approach Excavator2 with this command. Using version `1.1.2` and modifying these variables, the goals is that `full_send.sh` can execute the entire process from sorted bed file to figures.

I'm testing that today with the ocpmi vs 1kgp comparison.

```{r}
# This is the config file for full_send.sh
#

# Experiment name
expName="ocpmi_1kgp_compare"

# Choose to run step 1 TargetPerla.pl [1,0] (not required if desired targets already built)
runTp="1"

# The target name as output from the TargetPerla.sh script
target="hg38_target"

# Reference assembly to use [hg19,hg38]
refAssm="hg38"

# Window size
window="50000"

# Location of bed file
bedFile="/ocpmi/data/gene_ref/ucsc/hg38/hg38_sortmerge.bed"

# Location of BigWig file
bigWig="/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/GCA_000001405.15_GRCh38.bw"

# Location of genome reference fasta
genRef="/ocpmi/data/gene_ref/ucsc/hg38/hg38.fa"

# Number of processors
numProc="8"

# Step 3 EXCAVATORDataAnalysis.pl sample analysis mode [pooling, paired]
anMode="pooling"

# Location of the Excavator working directory
excv2Dir="/ocpmi/methods/EXCAVATOR2_Package_v1.1.2"

# Location of the bam files
bamDir="/ocpmi/data/bam/wes/pilot/ocmpi_1kgp_compare"

# Location of results directory [format will be /path/to/dir/$expName]
resDir="/ocpmi/results/excavator2"

```

### excv2 ocpmi vs 1kgp pilot

* samples: `NA10847` (control) via 1kgp, `pm2` (expt) via ocpmi; `/ocpmi/data/bam/wes/pilot/ocmpi_1kgp_compare`
* output: `/ocpmi/results/excavator2/ocpmi_1kgp_compare`
* gene_ref: `hg38.fa` via UCSC, `GCA_000001405.15_GRCh38.bw` via excv2; `~/ocpmi/gene_ref`
    - .fa: `~/ocpmi/data/gene_ref/ucsc/hg38`, .bw: `~/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data`

I am running another pilot to see what the results of comparing an ocpmi sample against the 1kgp control used by @DAurizio2016a. My hope would be that we would see a notable amount of variance between these samples and that this would server as a data point in support of use of `NA10847` as our standard control.

**Modified Pipeline**

*&ast;.bed `sort` (bedtools) &rarr; `full_send.sh` &rarr; cnv output*

### Summary

I think my `full_send.sh` script is effectively working. I'm running a complete test across 2 samples that also happen to be the next analysis I wanted to run. There are some minor tweaks I need to perform, but the script should generally be good to go.

The next major step I need to peform is to rerun my seq alignment on my ocpmi bams with star. That will allow me to start full scale analysis once my controls are set.

#### Running

```{r}
ID Started    Active Age Project Tags Description
-- ---------- ------ --- ------- ---- ------------------------------------------
72 2019-08-15 1h     1d  lab     work run test comparison with 1kgp and ocpmi
                                      files

1 task
```

#### Completed

```{r}
Created    Completed  Age Project Description
---------- ---------- --- ------- ---------------------------------------------
2019-08-08 2019-08-15 7d  lab     create loop for excv2 and vs2 to run all bams
2019-08-13 2019-08-15 2d  lab     create output for pm2 and pm15 with vs2
2019-08-14 2019-08-15 1d  lab     copy na10847 back to og dir
2019-08-14 2019-08-15 1d  lab     add mkdir for prepDir to fullsend.sh

4 tasks
```

***

## 18

### excv2 full_send.sh

I think I have the final version of the full_send script for the excv2 pipeline.

Features added:

* Removed granular control on not running individual steps. This was causing issues in being able to generate the final output.
* Created loop to rename output to give leading zeros for easy renaming.
* Fixed minor syntax issues around file movement and naming.

```{r}
#!/bin/bash
# This script is meant to prep the configuration and execution of the Excavator2 pipeline. It auto generates two text files that serve as configs for the pipeline scripts. Proper use of this script is paired with the full_send_vars.sh variable file to indicate pointers and variable names. Both should be placed in the Excavator2 working directory with this script being executed from there.

source full_send_vars.sh
cd $HOME$bamDir

# Overwrite any existing config files
rm -f ExperimentalFile*
target="${expName}_target"
runTp1=1 # step 1 TargetPerla.pl
runDp2=1 # step 2 EXCAVATORDataPrepare.pl
runDa3=1 # step 3 EXCAVATORDataAnalysis.pl

# Loop to iterate through each bam to create string for FilePrepare and FileAnalysis configs
count=1
for i in *.{bam,cram}; do
    echo "$PWD/$i" "data/prepare/$expName/${i%.*}" "${i%.*}" >> ExperimentalFilePrepare.$expName.txt
    echo "T$count" "data/prepare/$expName/${i%.*}" "${i%.*}" >> ExperimentalFileAnalysis.$expName.txt
    mkdir -p $HOME$excv2Dir/data/prepare/$expName/${i%.*}
    count=$[count+1]
done

# Function to remove leading path from sshfs local mount path and create control
sed -i -e 's/\/home\/chris/~/g' ./ExperimentalFilePrepare.$expName.txt
sed -i '$s/T[[:digit:]]*/C1/g' ./ExperimentalFileAnalysis.$expName.txt

# Relocate configs to Excavator working dir
mv ExperimentalFile* $HOME$excv2Dir
mkdir -p $HOME$resDir/$expName
mkdir -p $HOME$excv2Dir/config_files/$expName

# Generate SourceTarget.txt and execute step 1 TargetPerla.pl
cd $HOME$excv2Dir
if [[ $runTp1 = 1 ]]; then
    echo "$HOME$bigWig" "$HOME$genRef" > SourceTarget.$expName.txt
    perl TargetPerla.pl SourceTarget.$expName.txt $HOME$bedFile $target $window $refAssm;
else
    echo "Skipping step 1 TargetPerla.pl"
fi

# Executing step 2 EXCAVATORDataPrepare.pl
if [[ $runDp2 = 1 ]]; then
    perl EXCAVATORDataPrepare.pl -v ExperimentalFilePrepare.$expName.txt --processors $numProc --target $target --assembly $refAssm
else
    echo "Skipping step 2 EXCAVATORDataPrepare.pl"
fi

# Executing step 3 EXCAVATORDataAnalysis.pl
if [[ $runDa3 = 1 ]]; then
    perl EXCAVATORDataAnalysis.pl -v ExperimentalFileAnalysis.$expName.txt --processors $numProc --target $target --assembly $refAssm --output $HOME$resDir/$expName --mode $anMode
else
    echo "Skipping step 3 EXCAVATORDataAnalysis.pl"
fi

# Copy config files for record
cp ParameterFile.txt ParameterFile.$expName.txt
cp full_send_vars.sh full_send_vars.$expName.sh
mv *.{$expname}.* $HOME$excv2Dir/config_files/$expName

# Cleanup bad file naming convention
cd $HOME$resDir/$expName/Plots
for i in */ ; do
    cd $i
    for j in {1..9}; do
        mv PlotResults_chr$j.pdf PlotResults_chr0$j.pdf
    done
    cd ../
done

```

#### full_send_vars.sh

```{r}
################################################################################
# This is the config file for full_send.sh
#
# Use of this config file should be paired with full_send.sh.
# Both should be placed in the same directory and executed there.
#
# After setting all variables, the function can be executed with the following
# command in the Excavator2 working directory
#
# ./full_send.sh
#
# This function will overwrite any existing config or target files, so care
# should be taken to use a unique experiment name variable for every case.
################################################################################


################################################################################
# Analysis variables
################################################################################
# Experiment name
expName="ocpmi_v_1kgp_full_50k1"
# Window size
window="50000"
# Number of processors for parallelization
numProc="8"
# Step 3 EXCAVATORDataAnalysis.pl sample analysis mode [pooling, paired]
anMode="pooling"

################################################################################
# Gene reference target files
################################################################################
# Reference assembly to use [hg19,hg38]
refAssm="hg38"
# Location of genome reference fasta
genRef="/ocpmi/data/gene_ref/ucsc/hg38/hg38.fa"
# Location of bed file
bedFile="/ocpmi/data/gene_ref/ucsc/hg38/hg38_sortmerge.bed"
# Location of BigWig file
bigWig="/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/GCA_000001405.15_GRCh38.bw"

################################################################################
# Location of working and data directories
################################################################################
# Location of the Excavator working directory
excv2Dir="/ocpmi/methods/EXCAVATOR2_Package_v1.1.2"
# Location of the bam files
bamDir="/ocpmi/data/bam/wes"
# Location of results directory [format will be /path/to/dir/$expName]
resDir="/ocpmi/results/excavator2"


```

### excv2 ocpmi vs 1kgp pilot

I'm running several different permutations of my pilot bam samples to get a sense of how these results come out against the `NA10847` control. To a certain extent, I am just taking a shotgut approach to have a variety of cases to compare. Part of this includes testing the minor tweaking to the `full_send.sh` script with known runs. There are several failed iterations that are not shown.

#### ocpmi_vs_1kgp_2pm_50k2 {.tabset .tabfade}

##### Discussion

*Completed 2019-08-19*

* name: `ocpmi_vs_1kgp_2pm_50k2`
* samples: `NA10847` (control), `2pm` (expt) via ; `~/ocpmi/data/bam/wes/pilot/ocmpi_1kgp_compare`
* excv2_vars: `50000` (window); `pooling` (analysis mode)
* output: `~/ocpmi/results/excavator2`
* gene_ref: `hg38.fa`, `hg38_sortmerge.bed` via UCSC; `~/ocpmi/gene_ref`
* targets: `GCA_000001405.15_GRCh38.bw`, via excv2

This analysis completed successfully. There are some indications of variation at certain points (chr5, 11, 14, 17, 18, 19, 21, 22) though these sectors are very large. This may be an indication I need to adjust my window size since whole arms seem to be up or down regulated.

***

##### Figures

```{r, out.width = "100%", out.height = "600px", echo = F, eval=T}
knitr::include_graphics("/home/chris/ocpmi/results/excavator2/ocpmi_vs_1kgp_2pm_50k2/Plots/pm2/combined.pdf")
```

***

#### ocpmi_vs_1kgp_15pm_50k {.tabset .tabfade}

##### Discussion

*Completed 2019-08-19*

* name: `ocpmi_vs_1kgp_15pm_50k`
* samples: `NA10847` (control), `15pm` (expt) via ; `~/ocpmi/data/bam/wes/pilot/ocmpi_1kgp_compare`
* excv2_vars: `50000` (window); `pooling` (analysis mode)
* output: `~/ocpmi/results/excavator2`
* gene_ref: `hg38.fa`, `hg38_sortmerge.bed` via UCSC; `~/ocpmi/gene_ref`
* targets: `GCA_000001405.15_GRCh38.bw`, via excv2

This run did not complete successfully. This is somewhat of a concern since I don't have a good indication of why this could not run to completion. I tried a few different variations on the experimental setup and also attempted to run it interactively with no positive output. I'm hoping that rerunning the alignment will address this in some way.

***

## 19

### TODO

```{r}
ID Active Age  P Project Tag  Description                                   Urg
-- ------ ---- - ------- ---- --------------------------------------------- ----
71        5d   H lab     work realign bam files with star                   7.83
72 3d     5d     lab     work run test comparison with 1kgp and ocpmi files 5.83
66        11d    lab     work run excv2 and vs2 across ocpmi bams           1.86
70        5d     lab     work clarify duo login issues with ale01           1.83
73        5d     lab     work look for differences in 1kgp comparison in    1.83
                              exv2 paper
74        5d     lab     work do search for best suited cnv controls        1.83
75        5d     lab     work get site analytics for ccgd                   1.83
76        5d     lab     work submit vm application for ccgd                1.83
77        4d     lab     work test full send with ocpmi pilot samples       1.82
78        3d     lab     work add note re sorted bed file to vars.sh        1.82
79        3d     lab     work add discussion notes to expts on resutls page 1.82
81        2d     lab     work create branch for full_send                   1.81
82        21h    lab     work add expt name annotation convention to         1.8
                              methods page
84        1min   lab     work find way to combine pdfs                       1.8
85        2s     lab     work record results from weekend analysis           1.8

17 tasks
```

### STAR Alignment: ocpmi

* samples: `*.fastq` via ocpmi; `~/ocpmi/data/bam/wes`
* star_vars: `100` (overhang); `Within` (unmapped); `Standard` (SAM attr)
* output: `~/ocpmi/data/bam/wes/star_aligned`
* gene_ref: `hg38.fa`, `hg38.gtf` via UCSC; `~/ocpmi/gene_ref`

* man: [file:///mnt/msi/tasta005/ocpmi/reference/gene_markers/STARmanual.pdf](file:///mnt/msi/tasta005/ocpmi/reference/gene_markers/STARmanual.pdf)

I am redoing the sequence alignment on my fastq files out of fear that bwa is dated and does not have specific enough parameters to deal with our highliy variable dataset (@Dobin2013). The package requires a gtf file in order to index the gene reference. I retrieved this from the ucsc genome table browser (@Karolchik2004)

I found a helpful walkthrough on getting star going.

* https://hbctraining.github.io/Intro-to-rnaseq-hpc-O2/lessons/03_alignment.html

#### bam_create_star.pbs

```{r}
#!/bin/bash -l
#PBS -l walltime=24:00:00,nodes=1:ppn=8,mem=64gb
#PBS -m abe
#PBS -M tasta005@umn.edu
#PBS -N bam_create
#PBS -o logs/output
#PBS -e logs/error

module load gcc star

cd ~/ocpmi/data/gene_ref/ucsc/hg38

STAR --runThreadN 8 \
    --runMode genomeGenerate \
    --genomeDir ~/ocpmi/data/gene_ref/ucsc/hg38 \
    --genomeFastaFiles ~/ocpmi/data/gene_ref/ucsc/hg38/hg38.fa \
    --sjdbGTFfile ~/ocpmi/data/gene_ref/ucsc/hg38/hg38.gtf \
    --sjdbOverhang 100

cd ~/ocpmi/data/bam/wes

for i in *.fastq; do
STAR --genomeDir ~/ocpmi/data/gene_ref/ucsc/hg38 \
    --runThreadN 8 \
    --readFilesIn $i \
    --outFileNamePrefix ./star_aligned \
    --outSAMtype BAM SortedByCoordinate \
    --outSAMunmapped Within \
    --outSAMattributes Standard
done
```

### excv2 ocpmi_v_1kgp_full_50k

* name: `ocpmi_v_1kgp_full_50k`
* samples: `NA10847` (control), `*.bam` (expt) via ocpmi; `~/ocpmi/data/bam/wes`
* excv2_vars: `50000` (window); `pooling` (analysis mode)
* output: `~/ocpmi/results/excavator2`
* gene_ref: `hg38.fa`, `hg38_sortmerge.bed` via UCSC; `~/ocpmi/gene_ref`
* targets: `GCA_000001405.15_GRCh38.bw`, via excv2

Even though I am doing realignment for the ocpmi samples, I figured I could run this in the background. This is the whole deal. All ocpmi bam files with the complete excv2 process to see how things go.

### Summary

#### Running

```{r}
ID Started    Active Age P Project Tags Description
-- ---------- ------ --- - ------- ---- -----------------------------------
70 2019-08-19 20h    5d  H lab     work realign bam files with star
66 2019-08-19 20h    12d   lab     work run excv2 and vs2 across ocpmi bams

2 tasks
```

#### Completed

```{r}
Created    Completed  Age Project Description
---------- ---------- --- ------- ---------------------------------------------
2019-08-14 2019-08-19 5d  lab     run test comparison with 1kgp and ocpmi files
2019-08-14 2019-08-19 5d  lab     test full send with ocpmi pilot samples
2019-08-15 2019-08-19 4d  lab     add note re sorted bed file to vars.sh
2019-08-17 2019-08-19 2d  lab     create branch for full_send

4 tasks
```

***

## 20

### TODO

```{r}
ID Active Age P Project Tag  Description                                    Urg
-- ------ --- - ------- ---- ---------------------------------------------- ----
70 17h    5d  H lab     work realign bam files with star                    11.8
66 16h    11d   lab     work run excv2 and vs2 across ocpmi bams            5.86
71        5d    lab     work look for differences in 1kgp comparison in     1.83
                             exv2 paper
72        5d    lab     work do search for best suited cnv controls         1.83
73        5d    lab     work get site analytics for ccgd                    1.83
74        5d    lab     work submit vm application for ccgd                 1.83
75        4d    lab     work add discussion notes to expts on resutls page  1.82
77        1d    lab     work add expt name annotation convention to methods 1.81
                             page
79        20h   lab     work find way to combine pdfs                        1.8
80        20h   lab     work record results from weekend analysis            1.8
81        17h   lab     work add general parameters to methods for star      1.8
                             alignment
82        16h   lab     work create full_send.sh for varscan                 1.8
41        12w L lab     work agenda                                         1.08
                               2019-08-19 discuss possibility of grant

14 tasks
```

### STAR Alignment: ocpmi

* samples: `*.fastq` via ocpmi; `~/ocpmi/data/bam/wes`
* star_vars: `100` (overhang); `Within` (unmapped); `Standard` (SAM attr)
* output: `~/ocpmi/data/bam/wes/star_aligned`
* gene_ref: `hg38.fa`, `hg38.gtf` via UCSC; `~/ocpmi/gene_ref`

The process from yesterday ended in an error, so I'm troubleshooting today. The issue stems from the fact the indexing process is not running to completion. After doing some research, it seems that STAR doesn't do anything to identify working memory and is likely exceeding the allotted amount. I'm going to rerun the process with 64GB to see if this resolves the issue.

### excv2 ocpmi_v_1kgp_full_50k1

* name: `ocpmi_v_1kgp_full_50k1`
* samples: `NA10847` (control), `*.bam` (expt) via ocpmi; `~/ocpmi/data/bam/wes`
* excv2_vars: `50000` (window); `pooling` (analysis mode)
* output: `~/ocpmi/results/excavator2`
* gene_ref: `hg38.fa`, `hg38_sortmerge.bed` via UCSC; `~/ocpmi/gene_ref`
* targets: `GCA_000001405.15_GRCh38.bw`, via excv2

I foolishly did not set an adequate wall time for this process yesterday, so I needed to restart it. I initiated a new process at 24hrs, but that might also have been inadequate. We'll see.

### figs_finish_excv2.sh

I wanted to automate the final processing of the figure output for the excv2 pipeline, so I wrote a bash script to do that. This creates two items:

* A `combined.pdf` which is a string of all the independent figures.
* An image of a 5x5 matrix of each figure on one page.

```{r}
#!/bin/sh

dir=$PWD
cd $dir

pdfunite Plot* combined.pdf
pdfnup Plot* --nup 5x5 --outfile tmp.pdf
pdftoppm tmp.pdf fig_matrix.png -png
rm -rf tmp.pdf
```

### Summary

I was able to get a good deal done today. Most of that was documentation related, but I feel that was lacking from my work over the weekend. My large processes to redo seq alignment and run excv2 across all current bam files are still running. Expect them to run for a while still.

#### Running

```{r}
ID Started    Active Age P Project Tags Description
-- ---------- ------ --- - ------- ---- ---------------------------
70 2019-08-19 22h    6d  H lab     work realign bam files with star
66 2019-08-19 22h    12d   lab     work run excv2 across ocpmi bams

2 tasks
```

#### Completed

```{r}
Created    Completed  Age Project Description
---------- ---------- --- ------- ----------------------------------------------
2019-08-15 2019-08-20 4d  lab     add discussion notes to expts on resutls page
2019-08-19 2019-08-20 1d  lab     record results from weekend analysis
2019-08-18 2019-08-20 1d  lab     add expt name annotation convention to methods
                                  page
2019-08-19 2019-08-20 22h lab     add general parameters to methods for star
                                  alignment
2019-08-20 2019-08-20 2h  lab     fold pilot expts on methods page
2019-08-19 2019-08-20 1d  lab     find way to combine pdfs

6 tasks
```

***

## 21

### TODO

```{r}
ID Active Age P Project Tags      Description                               Urg
-- ------ --- - ------- --------- ----------------------------------------- ----
65 1d     12d   lab     work      run excv2 across ocpmi bams               5.87
68 1d     6d  H lab     work      realign bam files with star               11.8
79        11h   lab     ccgd work is a java license necessary                1.9
80        11h   lab     ccgd work how much drive space is necessary          1.9
81        11h   lab     ccgd work determine firewall config ports open       1.9
82        11h   lab     ccgd work determine security level for vm            1.9
69        6d    lab     work      look for differences in 1kgp comparison   1.83
                                  in exv2 paper
70        6d    lab     work      do search for best suited cnv controls    1.83
71        6d    lab     work      get site analytics for ccgd               1.83
72        6d    lab     work      submit vm application for ccgd            1.83
74        1d    lab     work      create full_send.sh for varscan           1.81
78        20h   lab     work      replace full_send files with new ones      1.8
                                  after excv2 process completes
83        9h    lab     work      remove base16 terminal theme               1.8
41        12w L lab     work      agenda [5]                                1.29
40        3mo L lab     work      review scripts for ccgd database           0.3

15 tasks
```
### excv2 ocpmi_v_1kgp_full_50k1 {.tabset .tabfade}

#### Discussion

* name: `ocpmi_v_1kgp_full_50k1`
* samples: `NA10847` (control), `*.bam` (expt) via ocpmi; `~/ocpmi/data/bam/wes`
* excv2_vars: `50000` (window); `pooling` (analysis mode)
* output: `~/ocpmi/results/excavator2`
* gene_ref: `hg38.fa`, `hg38_sortmerge.bed` via UCSC; `~/ocpmi/gene_ref`
* targets: `GCA_000001405.15_GRCh38.bw`, via excv2

#### Figures {.tabset .tabfade}

##### Collated

        PM-14-T17-0867-1_S29_R1_001 No Output

        PM-14-T17-0867-1_S29_R2_001 No Output

        PM-15-T17-0910-1_S2_R1_001 No Output

        PM-15-T17-0910-1_S2_R2_001 No Output

        PM-16-T17-0964-3_S6_R1_001 No Output

        PM-16-T17-0964-3_S6_R2_001 No Output

        PM-1-T17-0146-3_S1_R1_001 No Output

        PM-1-T17-0146-3_S1_R2_001 No Output

        PM-1-T18-0648-17_S32_R1_001 No Output

![PM-1-T18-0648-17_S32_R2_001](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-1-T18-0648-17_S32_R2_001/fig_matrix-1.png)

![PM-24-T17-1392-4_S10_R1_001](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-24-T17-1392-4_S10_R1_001/fig_matrix-1.png)

![PM-24-T17-1392-4_S10_R2_001](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-24-T17-1392-4_S10_R2_001/fig_matrix-1.png)

![PM-25-T17-1476-1_S18_R1_001](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-25-T17-1476-1_S18_R1_001/fig_matrix-1.png)

![PM-25-T17-1476-1_S18_R2_001](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-25-T17-1476-1_S18_R2_001/fig_matrix-1.png)

![PM-26-T17-1472-1_S14_R1_001](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-26-T17-1472-1_S14_R1_001/fig_matrix-1.png)

![PM-26-T17-1472-1_S14_R2_001](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-26-T17-1472-1_S14_R2_001/fig_matrix-1.png)

        PM-27-T17-1564-2_S22_R1_001 No Output

        PM-27-T17-1564-2_S22_R2_001 No Output

        PM-2-T17-0307-3_S9_R1_001 No Output

        PM-2-T17-0307-3_S9_R2_001 No Output

        PM-31-T18-0112-4_S3_R1_001 No Output

        PM-31-T18-0112-4_S3_R2_001 No Output

        PM-32-T18-0094-03_S26_R1_001 No Output

        PM-32-T18-0094-03_S26_R2_001 No Output

        PM-32-T18-0097-03_S30_R1_001 No Output

![PM-32-T18-0097-03_S30_R2_001](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-32-T18-0097-03_S30_R2_001/fig_matrix-1.png)

        PM-37-T18-0421-13_S7_R1_001 No Output

        PM-37-T18-0421-13_S7_R2_001 No Output

        PM-38-T18-0426-06_S11_R1_001 No Output

        PM-38-T18-0426-06_S11_R2_001 No Output

        PM-39-T18-0449-18_S15_R1_001 No Output

        PM-39-T18-0449-18_S15_R2_001 No Output

        PM-3-T17-0315-2_S5_R1_001 No Output

![PM-3-T17-0315-2_S5_R2_001](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-3-T17-0315-2_S5_R2_001/fig_matrix-1.png)

![PM-40-T18-0451-07_S19_R1_001](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-40-T18-0451-07_S19_R1_001/fig_matrix-1.png)

![PM-40-T18-0451-07_S19_R2_001](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-40-T18-0451-07_S19_R2_001/fig_matrix-1.png)

![PM-41-T18-0468-01_S23_R1_001](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-41-T18-0468-01_S23_R1_001/fig_matrix-1.png)

![PM-41-T18-0468-01_S23_R2_001](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-41-T18-0468-01_S23_R2_001/fig_matrix-1.png)

        PM-42-T18-0486-16_S27_R1_001 No Output

        PM-42-T18-0486-16_S27_R2_001 No Output

        PM-43-T18-0525-02_S4_R1_001 No Output

![PM-43-T18-0525-02_S4_R2_001](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-43-T18-0525-02_S4_R2_001/fig_matrix-1.png)

![PM-44-T18-0445-19_S31_R1_001](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-44-T18-0445-19_S31_R1_001/fig_matrix-1.png)

![PM-44-T18-0445-19_S31_R2_001](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-44-T18-0445-19_S31_R2_001/fig_matrix-1.png)

        PM-47-T18-0569-07_S12_R1_001 No Output

        PM-47-T18-0569-07_S12_R2_001 No Output

        PM-49-T18-0605-11_S20_R1_001 No Output

        PM-49-T18-0605-11_S20_R2_001 No Output

        PM-4-T17-0325-3_S13_R1_001 No Output

![PM-4-T17-0325-3_S13_R2_001](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-4-T17-0325-3_S13_R2_001/fig_matrix-1.png)

        PM-50-T18-0595-05_S16_R1_001 No Output

        PM-50-T18-0595-05_S16_R2_001 No Output

        PM-51-T18-0606-08_S24_R1_001 No Output

        PM-51-T18-0606-08_S24_R2_001 No Output

        PM-52-T18-0523-02_S8_R1_001 No Output

        PM-52-T18-0523-02_S8_R2_001 No Output

        PM-55-T18-0639-02_S28_R1_001 No Output

![PM-55-T18-0639-02_S28_R2_001](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-55-T18-0639-02_S28_R2_001/fig_matrix-1.png)

        PM-6-T17-0382-2_S17_R1_001 No Output

        PM-6-T17-0382-2_S17_R2_001 No Output

        PM-7-T17-0576-3_S25_R1_001 No Output

        PM-7-T17-0576-3_S25_R2_001 No Output

        PM-9-T17-0591-1_S21_R1_001 No Output

        PM-9-T17-0591-1_S21_R2_001 No Output

##### PM-14-T17-0867-1_S29_R1_001

**No output**

##### PM-14-T17-0867-1_S29_R2_001

**No output**

##### PM-15-T17-0910-1_S2_R1_001

**No output**

##### PM-15-T17-0910-1_S2_R2_001

**No output**

##### PM-16-T17-0964-3_S6_R1_001

**No output**

##### PM-16-T17-0964-3_S6_R2_001

**No output**

##### PM-1-T17-0146-3_S1_R1_001

**No output**

##### PM-1-T17-0146-3_S1_R2_001

**No output**

##### PM-1-T18-0648-17_S32_R1_001

**No output**

##### PM-1-T18-0648-17_S32_R2_001

**Figure Matrix**

![](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-1-T18-0648-17_S32_R2_001/fig_matrix-1.png)

***

**Individual Output**

```{r, out.width = "100%", out.height = "600px", echo = F, eval=T}
knitr::include_graphics("/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-1-T18-0648-17_S32_R2_001/combined.pdf")
```

##### PM-24-T17-1392-4_S10_R1_001

**Figure Matrix**

![](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-24-T17-1392-4_S10_R1_001/fig_matrix-1.png)

***

**Individual Output**

```{r, out.width = "100%", out.height = "600px", echo = F, eval=T}
knitr::include_graphics("/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-24-T17-1392-4_S10_R1_001/combined.pdf")
```

##### PM-24-T17-1392-4_S10_R2_001

**Figure Matrix**

![](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-24-T17-1392-4_S10_R2_001/fig_matrix-1.png)

***

**Individual Output**

```{r, out.width = "100%", out.height = "600px", echo = F, eval=T}
knitr::include_graphics("/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-24-T17-1392-4_S10_R2_001/combined.pdf")
```

##### PM-25-T17-1476-1_S18_R1_001

**Figure Matrix**

![](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-25-T17-1476-1_S18_R1_001/fig_matrix-1.png)

***

**Individual Output**

```{r, out.width = "100%", out.height = "600px", echo = F, eval=T}
knitr::include_graphics("/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-25-T17-1476-1_S18_R1_001/combined.pdf")
```

##### PM-25-T17-1476-1_S18_R2_001

**Figure Matrix**

![](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-25-T17-1476-1_S18_R2_001/fig_matrix-1.png)

***

**Individual Output**

```{r, out.width = "100%", out.height = "600px", echo = F, eval=T}
knitr::include_graphics("/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-25-T17-1476-1_S18_R2_001/combined.pdf")
```

##### PM-26-T17-1472-1_S14_R1_001

**Figure Matrix**

![](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-26-T17-1472-1_S14_R1_001/fig_matrix-1.png)

***

**Individual Output**

```{r, out.width = "100%", out.height = "600px", echo = F, eval=T}
knitr::include_graphics("/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-26-T17-1472-1_S14_R1_001/combined.pdf")
```

##### PM-26-T17-1472-1_S14_R2_001

**Figure Matrix**

![](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-26-T17-1472-1_S14_R2_001/fig_matrix-1.png)

***

**Individual Output**

```{r, out.width = "100%", out.height = "600px", echo = F, eval=T}
knitr::include_graphics("/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-26-T17-1472-1_S14_R2_001/combined.pdf")
```

##### PM-27-T17-1564-2_S22_R1_001

**No output**

##### PM-27-T17-1564-2_S22_R2_001

**No output**

##### PM-2-T17-0307-3_S9_R1_001

**No output**

##### PM-2-T17-0307-3_S9_R2_001

**No output**

##### PM-31-T18-0112-4_S3_R1_001

**No output**

##### PM-31-T18-0112-4_S3_R2_001

**No output**

##### PM-32-T18-0094-03_S26_R1_001

**No output**

##### PM-32-T18-0094-03_S26_R2_001

**No output**

##### PM-32-T18-0097-03_S30_R1_001

**No output**

##### PM-32-T18-0097-03_S30_R2_001

**Figure Matrix**

![](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-32-T18-0097-03_S30_R2_001/fig_matrix-1.png)

***

**Individual Output**

```{r, out.width = "100%", out.height = "600px", echo = F, eval=T}
knitr::include_graphics("/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-32-T18-0097-03_S30_R2_001/combined.pdf")
```

##### PM-37-T18-0421-13_S7_R1_001

**No output**

##### PM-37-T18-0421-13_S7_R2_001

**No output**

##### PM-38-T18-0426-06_S11_R1_001

**No output**

##### PM-38-T18-0426-06_S11_R2_001

**No output**

##### PM-39-T18-0449-18_S15_R1_001

**No output**

##### PM-39-T18-0449-18_S15_R2_001

**No output**

##### PM-3-T17-0315-2_S5_R1_001

**No output**

##### PM-3-T17-0315-2_S5_R2_001

**Figure Matrix**

![](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-3-T17-0315-2_S5_R2_001/fig_matrix-1.png)

***

**Individual Output**

```{r, out.width = "100%", out.height = "600px", echo = F, eval=T}
knitr::include_graphics("/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-3-T17-0315-2_S5_R2_001/combined.pdf")
```

##### PM-40-T18-0451-07_S19_R1_001

**Figure Matrix**

![](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-40-T18-0451-07_S19_R1_001/fig_matrix-1.png)

***

**Individual Output**

```{r, out.width = "100%", out.height = "600px", echo = F, eval=T}
knitr::include_graphics("/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-40-T18-0451-07_S19_R1_001/combined.pdf")
```

##### PM-40-T18-0451-07_S19_R2_001

**Figure Matrix**

![](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-40-T18-0451-07_S19_R2_001/fig_matrix-1.png)

***

**Individual Output**

```{r, out.width = "100%", out.height = "600px", echo = F, eval=T}
knitr::include_graphics("/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-40-T18-0451-07_S19_R2_001/combined.pdf")
```

##### PM-41-T18-0468-01_S23_R1_001

**Figure Matrix**

![](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-41-T18-0468-01_S23_R1_001/fig_matrix-1.png)

***

**Individual Output**

```{r, out.width = "100%", out.height = "600px", echo = F, eval=T}
knitr::include_graphics("/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-41-T18-0468-01_S23_R1_001/combined.pdf")
```

##### PM-41-T18-0468-01_S23_R2_001

**Figure Matrix**

![](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-41-T18-0468-01_S23_R2_001/fig_matrix-1.png)

***

**Individual Output**

```{r, out.width = "100%", out.height = "600px", echo = F, eval=T}
knitr::include_graphics("/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-41-T18-0468-01_S23_R2_001/combined.pdf")
```

##### PM-42-T18-0486-16_S27_R1_001

**No output**

##### PM-42-T18-0486-16_S27_R2_001

**No output**

##### PM-43-T18-0525-02_S4_R1_001

**No output**

##### PM-43-T18-0525-02_S4_R2_001

**Figure Matrix**

![](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-43-T18-0525-02_S4_R2_001/fig_matrix-1.png)

***

**Individual Output**

```{r, out.width = "100%", out.height = "600px", echo = F, eval=T}
knitr::include_graphics("/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-43-T18-0525-02_S4_R2_001/combined.pdf")
```

##### PM-44-T18-0445-19_S31_R1_001

**Figure Matrix**

![](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-44-T18-0445-19_S31_R1_001/fig_matrix-1.png)

***

**Individual Output**

```{r, out.width = "100%", out.height = "600px", echo = F, eval=T}
knitr::include_graphics("/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-44-T18-0445-19_S31_R1_001/combined.pdf")
```

##### PM-44-T18-0445-19_S31_R2_001

**Figure Matrix**

![](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-44-T18-0445-19_S31_R2_001/fig_matrix-1.png)

***

**Individual Output**

```{r, out.width = "100%", out.height = "600px", echo = F, eval=T}
knitr::include_graphics("/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-44-T18-0445-19_S31_R2_001/combined.pdf")
```

##### PM-47-T18-0569-07_S12_R1_001

**No output**

##### PM-47-T18-0569-07_S12_R2_001

**No output**

##### PM-49-T18-0605-11_S20_R1_001

**No output**

##### PM-49-T18-0605-11_S20_R2_001

**No output**

##### PM-4-T17-0325-3_S13_R1_001

**No output**

##### PM-4-T17-0325-3_S13_R2_001

**Figure Matrix**

![](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-4-T17-0325-3_S13_R2_001/fig_matrix-1.png)

***

**Individual Output**

```{r, out.width = "100%", out.height = "600px", echo = F, eval=T}
knitr::include_graphics("/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-4-T17-0325-3_S13_R2_001/combined.pdf")
```

##### PM-50-T18-0595-05_S16_R1_001

**No output**

##### PM-50-T18-0595-05_S16_R2_001

**No output**

##### PM-51-T18-0606-08_S24_R1_001

**No output**

##### PM-51-T18-0606-08_S24_R2_001

**No output**

##### PM-52-T18-0523-02_S8_R1_001

**No output**

##### PM-52-T18-0523-02_S8_R2_001

**No output**

##### PM-55-T18-0639-02_S28_R1_001

**No output**

##### PM-55-T18-0639-02_S28_R2_001

**Figure Matrix**

![](/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-55-T18-0639-02_S28_R2_001/fig_matrix-1.png)

***

**Individual Output**

```{r, out.width = "100%", out.height = "600px", echo = F, eval=T}
knitr::include_graphics("/home/chris/ocpmi/results/excavator2/ocpmi_v_1kgp_full_50k1/Plots/PM-55-T18-0639-02_S28_R2_001/combined.pdf")
```

##### PM-6-T17-0382-2_S17_R1_001

**No output**

##### PM-6-T17-0382-2_S17_R2_001

**No output**

##### PM-7-T17-0576-3_S25_R1_001

**No output**

##### PM-7-T17-0576-3_S25_R2_001

**No output**

##### PM-9-T17-0591-1_S21_R1_001

**No output**

##### PM-9-T17-0591-1_S21_R2_001

**No output**

### Collaborator mtg

present: tim, andy, ying, chris

* is the 10847 the baseline genome or is it immortalized?
* exome data is agilent v6 wes data
* what did 1kgp use for their genome generation?
* andy might have some clinical data that was also generated by agilinet v6
    + he's going to reach out to the people for that.
* check out cnvrf: this is used by the clinical pathology dept
* naming convention for wes ocpmi samples
    + [PM-pathology ID]-[Bionet year retrieved]-[Bionet ID]-[Bionet subID]-[Illumina ID]

### wkly mtg w/ Tim

present: tim, chris

* grant
* update on ccgd

### STAR Alignment: ocpmi

* samples: `*.fastq` via ocpmi; `~/ocpmi/data/bam/wes`
* star_vars: `100` (overhang); `Within` (unmapped); `Standard` (SAM attr)
* output: `~/ocpmi/data/bam/wes/star_aligned`
* gene_ref: `hg38.fa`, `hg38.gtf` via UCSC; `~/ocpmi/gene_ref`

This process completed early this morning. There isn't any indication that an error occured or that the process was interrupted, but I have no idea where the output is. I'm going to run the process again making sure the dir prefix isn't the issue.

<details>
  <summary>**star_alignedLog.final.out**</summary>
```{r}
                                 Started job on |	Aug 20 22:15:35
                             Started mapping on |	Aug 20 22:16:25
                                    Finished on |	Aug 20 22:20:18
       Mapping speed, Million of reads per hour |	294.04

                          Number of input reads |	19030610
                      Average input read length |	51
                                    UNIQUE READS:
                   Uniquely mapped reads number |	6009979
                        Uniquely mapped reads % |	31.58%
                          Average mapped length |	49.72
                       Number of splices: Total |	305612
            Number of splices: Annotated (sjdb) |	288296
                       Number of splices: GT/AG |	298136
                       Number of splices: GC/AG |	2481
                       Number of splices: AT/AC |	271
               Number of splices: Non-canonical |	4724
                      Mismatch rate per base, % |	0.82%
                         Deletion rate per base |	0.02%
                        Deletion average length |	1.45
                        Insertion rate per base |	0.01%
                       Insertion average length |	1.11
                             MULTI-MAPPING READS:
        Number of reads mapped to multiple loci |	6361438
             % of reads mapped to multiple loci |	33.43%
        Number of reads mapped to too many loci |	5314146
             % of reads mapped to too many loci |	27.92%
                                  UNMAPPED READS:
       % of reads unmapped: too many mismatches |	0.00%
                 % of reads unmapped: too short |	6.84%
                     % of reads unmapped: other |	0.23%
                                  CHIMERIC READS:
                       Number of chimeric reads |	0
                            % of chimeric reads |	0.00%
```

</details>

<details>
  <summary>**star_alignedLog.out**</summary>

```{r}
STAR version=STAR_2.5.3a
STAR compilation time,server,dir=Fri Mar 17 11:35:46 EDT 2017 florence.cshl.edu:/sonas-hs/gingeras/nlsas_norepl/user/dobin/STAR/STAR.sandbox/source
##### DEFAULT parameters:
versionSTAR                       20201
versionGenome                     20101   20200
parametersFiles                   -
sysShell                          -
runMode                           alignReads
runThreadN                        1
runDirPerm                        User_RWX
runRNGseed                        777
genomeDir                         ./GenomeDir/
genomeLoad                        NoSharedMemory
genomeFastaFiles                  -
genomeChainFiles                  -
genomeSAindexNbases               14
genomeChrBinNbits                 18
genomeSAsparseD                   1
genomeSuffixLengthMax             18446744073709551615
genomeFileSizes                   0
readFilesIn                       Read1   Read2
readFilesCommand                  -
readMatesLengthsIn                NotEqual
readMapNumber                     18446744073709551615
readNameSeparator                 /
inputBAMfile                      -
bamRemoveDuplicatesType           -
bamRemoveDuplicatesMate2basesN    0
limitGenomeGenerateRAM            31000000000
limitIObufferSize                 150000000
limitOutSAMoneReadBytes           100000
limitOutSJcollapsed               1000000
limitOutSJoneRead                 1000
limitBAMsortRAM                   0
limitSjdbInsertNsj                1000000
outTmpDir                         -
outTmpKeep                        None
outStd                            Log
outReadsUnmapped                  None
outQSconversionAdd                0
outMultimapperOrder               Old_2.4
outSAMtype                        SAM
outSAMmode                        Full
outSAMstrandField                 None
outSAMattributes                  Standard
outSAMunmapped                    None
outSAMorder                       Paired
outSAMprimaryFlag                 OneBestScore
outSAMreadID                      Standard
outSAMmapqUnique                  255
outSAMflagOR                      0
outSAMflagAND                     65535
outSAMattrRGline                  -
outSAMheaderHD                    -
outSAMheaderPG                    -
outSAMheaderCommentFile           -
outBAMcompression                 1
outBAMsortingThreadN              0
outSAMfilter                      None
outSAMmultNmax                    18446744073709551615
outSAMattrIHstart                 1
outSJfilterReads                  All
outSJfilterCountUniqueMin         3   1   1   1
outSJfilterCountTotalMin          3   1   1   1
outSJfilterOverhangMin            30   12   12   12
outSJfilterDistToOtherSJmin       10   0   5   10
outSJfilterIntronMaxVsReadN       50000   100000   200000
outWigType                        None
outWigStrand                      Stranded
outWigReferencesPrefix            -
outWigNorm                        RPM
outFilterType                     Normal
outFilterMultimapNmax             10
outFilterMultimapScoreRange       1
outFilterScoreMin                 0
outFilterScoreMinOverLread        0.66
outFilterMatchNmin                0
outFilterMatchNminOverLread       0.66
outFilterMismatchNmax             10
outFilterMismatchNoverLmax        0.3
outFilterMismatchNoverReadLmax    1
outFilterIntronMotifs             None
clip5pNbases                      0
clip3pNbases                      0
clip3pAfterAdapterNbases          0
clip3pAdapterSeq                  -
clip3pAdapterMMp                  0.1
winBinNbits                       16
winAnchorDistNbins                9
winFlankNbins                     4
winAnchorMultimapNmax             50
winReadCoverageRelativeMin        0.5
winReadCoverageBasesMin           0
scoreGap                          0
scoreGapNoncan                    -8
scoreGapGCAG                      -4
scoreGapATAC                      -8
scoreStitchSJshift                1
scoreGenomicLengthLog2scale       -0.25
scoreDelBase                      -2
scoreDelOpen                      -2
scoreInsOpen                      -2
scoreInsBase                      -2
seedSearchLmax                    0
seedSearchStartLmax               50
seedSearchStartLmaxOverLread      1
seedPerReadNmax                   1000
seedPerWindowNmax                 50
seedNoneLociPerWindow             10
seedMultimapNmax                  10000
alignIntronMin                    21
alignIntronMax                    0
alignMatesGapMax                  0
alignTranscriptsPerReadNmax       10000
alignSJoverhangMin                5
alignSJDBoverhangMin              3
alignSJstitchMismatchNmax         0   -1   0   0
alignSplicedMateMapLmin           0
alignSplicedMateMapLminOverLmate    0.66
alignWindowsPerReadNmax           10000
alignTranscriptsPerWindowNmax     100
alignEndsType                     Local
alignSoftClipAtReferenceEnds      Yes
alignEndsProtrude                 0   ConcordantPair
chimSegmentMin                    0
chimScoreMin                      0
chimScoreDropMax                  20
chimScoreSeparation               10
chimScoreJunctionNonGTAG          -1
chimMainSegmentMultNmax           10
chimJunctionOverhangMin           20
chimOutType                       SeparateSAMold
chimFilter                        banGenomicN
chimSegmentReadGapMax             0
sjdbFileChrStartEnd               -
sjdbGTFfile                       -
sjdbGTFchrPrefix                  -
sjdbGTFfeatureExon                exon
sjdbGTFtagExonParentTranscript    transcript_id
sjdbGTFtagExonParentGene          gene_id
sjdbOverhang                      100
sjdbScore                         2
sjdbInsertSave                    Basic
quantMode                         -
quantTranscriptomeBAMcompression    1
quantTranscriptomeBan             IndelSoftclipSingleend
twopass1readsN                    18446744073709551615
twopassMode                       None
##### Command Line:
STAR --genomeDir /home/starrt2/tasta005/ocpmi/data/gene_ref/ucsc/hg38 --runThreadN 8 --readFilesIn PM-9-T17-0591-1_S21_R2_001.fastq --outFileNamePrefix ./star_aligned --outSAMtype BAM SortedByCoordinate --outSAMunmapped Within --outSAMattributes Standard
##### Initial USER parameters from Command Line:
outFileNamePrefix                 ./star_aligned
###### All USER parameters from Command Line:
genomeDir                     /home/starrt2/tasta005/ocpmi/data/gene_ref/ucsc/hg38     ~RE-DEFINED
runThreadN                    8     ~RE-DEFINED
readFilesIn                   PM-9-T17-0591-1_S21_R2_001.fastq        ~RE-DEFINED
outFileNamePrefix             ./star_aligned     ~RE-DEFINED
outSAMtype                    BAM   SortedByCoordinate        ~RE-DEFINED
outSAMunmapped                Within        ~RE-DEFINED
outSAMattributes              Standard        ~RE-DEFINED
##### Finished reading parameters from all sources

##### Final user re-defined parameters-----------------:
runThreadN                        8
genomeDir                         /home/starrt2/tasta005/ocpmi/data/gene_ref/ucsc/hg38
readFilesIn                       PM-9-T17-0591-1_S21_R2_001.fastq
outFileNamePrefix                 ./star_aligned
outSAMtype                        BAM   SortedByCoordinate
outSAMattributes                  Standard
outSAMunmapped                    Within

-------------------------------
##### Final effective command line:
STAR   --runThreadN 8   --genomeDir /home/starrt2/tasta005/ocpmi/data/gene_ref/ucsc/hg38   --readFilesIn PM-9-T17-0591-1_S21_R2_001.fastq      --outFileNamePrefix ./star_aligned   --outSAMtype BAM   SortedByCoordinate      --outSAMattributes Standard      --outSAMunmapped Within

##### Final parameters after user input--------------------------------:
versionSTAR                       20201
versionGenome                     20101   20200
parametersFiles                   -
sysShell                          -
runMode                           alignReads
runThreadN                        8
runDirPerm                        User_RWX
runRNGseed                        777
genomeDir                         /home/starrt2/tasta005/ocpmi/data/gene_ref/ucsc/hg38
genomeLoad                        NoSharedMemory
genomeFastaFiles                  -
genomeChainFiles                  -
genomeSAindexNbases               14
genomeChrBinNbits                 18
genomeSAsparseD                   1
genomeSuffixLengthMax             18446744073709551615
genomeFileSizes                   0
readFilesIn                       PM-9-T17-0591-1_S21_R2_001.fastq
readFilesCommand                  -
readMatesLengthsIn                NotEqual
readMapNumber                     18446744073709551615
readNameSeparator                 /
inputBAMfile                      -
bamRemoveDuplicatesType           -
bamRemoveDuplicatesMate2basesN    0
limitGenomeGenerateRAM            31000000000
limitIObufferSize                 150000000
limitOutSAMoneReadBytes           100000
limitOutSJcollapsed               1000000
limitOutSJoneRead                 1000
limitBAMsortRAM                   0
limitSjdbInsertNsj                1000000
outFileNamePrefix                 ./star_aligned
outTmpDir                         -
outTmpKeep                        None
outStd                            Log
outReadsUnmapped                  None
outQSconversionAdd                0
outMultimapperOrder               Old_2.4
outSAMtype                        BAM   SortedByCoordinate
outSAMmode                        Full
outSAMstrandField                 None
outSAMattributes                  Standard
outSAMunmapped                    Within
outSAMorder                       Paired
outSAMprimaryFlag                 OneBestScore
outSAMreadID                      Standard
outSAMmapqUnique                  255
outSAMflagOR                      0
outSAMflagAND                     65535
outSAMattrRGline                  -
outSAMheaderHD                    -
outSAMheaderPG                    -
outSAMheaderCommentFile           -
outBAMcompression                 1
outBAMsortingThreadN              0
outSAMfilter                      None
outSAMmultNmax                    18446744073709551615
outSAMattrIHstart                 1
outSJfilterReads                  All
outSJfilterCountUniqueMin         3   1   1   1
outSJfilterCountTotalMin          3   1   1   1
outSJfilterOverhangMin            30   12   12   12
outSJfilterDistToOtherSJmin       10   0   5   10
outSJfilterIntronMaxVsReadN       50000   100000   200000
outWigType                        None
outWigStrand                      Stranded
outWigReferencesPrefix            -
outWigNorm                        RPM
outFilterType                     Normal
outFilterMultimapNmax             10
outFilterMultimapScoreRange       1
outFilterScoreMin                 0
outFilterScoreMinOverLread        0.66
outFilterMatchNmin                0
outFilterMatchNminOverLread       0.66
outFilterMismatchNmax             10
outFilterMismatchNoverLmax        0.3
outFilterMismatchNoverReadLmax    1
outFilterIntronMotifs             None
clip5pNbases                      0
clip3pNbases                      0
clip3pAfterAdapterNbases          0
clip3pAdapterSeq                  -
clip3pAdapterMMp                  0.1
winBinNbits                       16
winAnchorDistNbins                9
winFlankNbins                     4
winAnchorMultimapNmax             50
winReadCoverageRelativeMin        0.5
winReadCoverageBasesMin           0
scoreGap                          0
scoreGapNoncan                    -8
scoreGapGCAG                      -4
scoreGapATAC                      -8
scoreStitchSJshift                1
scoreGenomicLengthLog2scale       -0.25
scoreDelBase                      -2
scoreDelOpen                      -2
scoreInsOpen                      -2
scoreInsBase                      -2
seedSearchLmax                    0
seedSearchStartLmax               50
seedSearchStartLmaxOverLread      1
seedPerReadNmax                   1000
seedPerWindowNmax                 50
seedNoneLociPerWindow             10
seedMultimapNmax                  10000
alignIntronMin                    21
alignIntronMax                    0
alignMatesGapMax                  0
alignTranscriptsPerReadNmax       10000
alignSJoverhangMin                5
alignSJDBoverhangMin              3
alignSJstitchMismatchNmax         0   -1   0   0
alignSplicedMateMapLmin           0
alignSplicedMateMapLminOverLmate    0.66
alignWindowsPerReadNmax           10000
alignTranscriptsPerWindowNmax     100
alignEndsType                     Local
alignSoftClipAtReferenceEnds      Yes
alignEndsProtrude                 0   ConcordantPair
chimSegmentMin                    0
chimScoreMin                      0
chimScoreDropMax                  20
chimScoreSeparation               10
chimScoreJunctionNonGTAG          -1
chimMainSegmentMultNmax           10
chimJunctionOverhangMin           20
chimOutType                       SeparateSAMold
chimFilter                        banGenomicN
chimSegmentReadGapMax             0
sjdbFileChrStartEnd               -
sjdbGTFfile                       -
sjdbGTFchrPrefix                  -
sjdbGTFfeatureExon                exon
sjdbGTFtagExonParentTranscript    transcript_id
sjdbGTFtagExonParentGene          gene_id
sjdbOverhang                      100
sjdbScore                         2
sjdbInsertSave                    Basic
quantMode                         -
quantTranscriptomeBAMcompression    1
quantTranscriptomeBan             IndelSoftclipSingleend
twopass1readsN                    18446744073709551615
twopassMode                       None
----------------------------------------

WARNING: --limitBAMsortRAM=0, will use genome size as RAM limit for BAM sorting
Finished loading and checking parameters
Reading genome generation parameters:
versionGenome                 20201        ~RE-DEFINED
genomeFastaFiles              /home/starrt2/tasta005/ocpmi/data/gene_ref/ucsc/hg38/hg38.fa        ~RE-DEFINED
genomeSAindexNbases           14     ~RE-DEFINED
genomeChrBinNbits             18     ~RE-DEFINED
genomeSAsparseD               1     ~RE-DEFINED
sjdbOverhang                  100     ~RE-DEFINED
sjdbFileChrStartEnd           -        ~RE-DEFINED
sjdbGTFfile                   /home/starrt2/tasta005/ocpmi/data/gene_ref/ucsc/hg38/hg38.gtf     ~RE-DEFINED
sjdbGTFchrPrefix              -     ~RE-DEFINED
sjdbGTFfeatureExon            exon     ~RE-DEFINED
sjdbGTFtagExonParentTranscripttranscript_id     ~RE-DEFINED
sjdbGTFtagExonParentGene      gene_id     ~RE-DEFINED
sjdbInsertSave                Basic     ~RE-DEFINED
genomeFileSizes               3358284875   25799931897        ~RE-DEFINED
Genome version is compatible with current STAR version
Number of real (reference) chromosomes= 455
1	chr1	248956422	0
2	chr10	133797422	249036800
3	chr11	135086622	382992384
4	chr11_KI270721v1_random	100316	518258688
5	chr12	133275309	518520832
6	chr13	114364328	651952128
7	chr14	107043718	766509056
8	chr14_GL000009v2_random	201709	873725952
9	chr14_GL000225v1_random	211173	873988096
10	chr14_KI270722v1_random	194050	874250240
11	chr14_GL000194v1_random	191469	874512384
12	chr14_KI270723v1_random	38115	874774528
13	chr14_KI270724v1_random	39555	875036672
14	chr14_KI270725v1_random	172810	875298816
15	chr14_KI270726v1_random	43739	875560960
16	chr15	101991189	875823104
17	chr15_KI270727v1_random	448248	978059264
18	chr16	90338345	978583552
19	chr16_KI270728v1_random	1872759	1069023232
20	chr17	83257441	1071120384
21	chr17_GL000205v2_random	185591	1154482176
22	chr17_KI270729v1_random	280839	1154744320
23	chr17_KI270730v1_random	112551	1155268608
24	chr18	80373285	1155530752
25	chr19	58617616	1236008960
26	chr1_KI270706v1_random	175055	1294729216
27	chr1_KI270707v1_random	32032	1294991360
28	chr1_KI270708v1_random	127682	1295253504
29	chr1_KI270709v1_random	66860	1295515648
30	chr1_KI270710v1_random	40176	1295777792
31	chr1_KI270711v1_random	42210	1296039936
32	chr1_KI270712v1_random	176043	1296302080
33	chr1_KI270713v1_random	40745	1296564224
34	chr1_KI270714v1_random	41717	1296826368
35	chr2	242193529	1297088512
36	chr20	64444167	1539309568
37	chr21	46709983	1603796992
38	chr22	50818468	1650720768
39	chr22_KI270731v1_random	150754	1701576704
40	chr22_KI270732v1_random	41543	1701838848
41	chr22_KI270733v1_random	179772	1702100992
42	chr22_KI270734v1_random	165050	1702363136
43	chr22_KI270735v1_random	42811	1702625280
44	chr22_KI270736v1_random	181920	1702887424
45	chr22_KI270737v1_random	103838	1703149568
46	chr22_KI270738v1_random	99375	1703411712
47	chr22_KI270739v1_random	73985	1703673856
48	chr2_KI270715v1_random	161471	1703936000
49	chr2_KI270716v1_random	153799	1704198144
50	chr3	198295559	1704460288
51	chr3_GL000221v1_random	155397	1902903296
52	chr4	190214555	1903165440
53	chr4_GL000008v2_random	209709	2093481984
54	chr5	181538259	2093744128
55	chr5_GL000208v1_random	92689	2275409920
56	chr6	170805979	2275672064
57	chr7	159345973	2446589952
58	chr8	145138636	2605973504
59	chr9	138394717	2751201280
60	chr9_KI270717v1_random	40062	2889613312
61	chr9_KI270718v1_random	38054	2889875456
62	chr9_KI270719v1_random	176845	2890137600
63	chr9_KI270720v1_random	39050	2890399744
64	chr1_KI270762v1_alt	354444	2890661888
65	chr1_KI270766v1_alt	256271	2891186176
66	chr1_KI270760v1_alt	109528	2891448320
67	chr1_KI270765v1_alt	185285	2891710464
68	chr1_GL383518v1_alt	182439	2891972608
69	chr1_GL383519v1_alt	110268	2892234752
70	chr1_GL383520v2_alt	366580	2892496896
71	chr1_KI270764v1_alt	50258	2893021184
72	chr1_KI270763v1_alt	911658	2893283328
73	chr1_KI270759v1_alt	425601	2894331904
74	chr1_KI270761v1_alt	165834	2894856192
75	chr2_KI270770v1_alt	136240	2895118336
76	chr2_KI270773v1_alt	70887	2895380480
77	chr2_KI270774v1_alt	223625	2895642624
78	chr2_KI270769v1_alt	120616	2895904768
79	chr2_GL383521v1_alt	143390	2896166912
80	chr2_KI270772v1_alt	133041	2896429056
81	chr2_KI270775v1_alt	138019	2896691200
82	chr2_KI270771v1_alt	110395	2896953344
83	chr2_KI270768v1_alt	110099	2897215488
84	chr2_GL582966v2_alt	96131	2897477632
85	chr2_GL383522v1_alt	123821	2897739776
86	chr2_KI270776v1_alt	174166	2898001920
87	chr2_KI270767v1_alt	161578	2898264064
88	chr3_JH636055v2_alt	173151	2898526208
89	chr3_KI270783v1_alt	109187	2898788352
90	chr3_KI270780v1_alt	224108	2899050496
91	chr3_GL383526v1_alt	180671	2899312640
92	chr3_KI270777v1_alt	173649	2899574784
93	chr3_KI270778v1_alt	248252	2899836928
94	chr3_KI270781v1_alt	113034	2900099072
95	chr3_KI270779v1_alt	205312	2900361216
96	chr3_KI270782v1_alt	162429	2900623360
97	chr3_KI270784v1_alt	184404	2900885504
98	chr4_KI270790v1_alt	220246	2901147648
99	chr4_GL383528v1_alt	376187	2901409792
100	chr4_KI270787v1_alt	111943	2901934080
101	chr4_GL000257v2_alt	586476	2902196224
102	chr4_KI270788v1_alt	158965	2902982656
103	chr4_GL383527v1_alt	164536	2903244800
104	chr4_KI270785v1_alt	119912	2903506944
105	chr4_KI270789v1_alt	205944	2903769088
106	chr4_KI270786v1_alt	244096	2904031232
107	chr5_KI270793v1_alt	126136	2904293376
108	chr5_KI270792v1_alt	179043	2904555520
109	chr5_KI270791v1_alt	195710	2904817664
110	chr5_GL383532v1_alt	82728	2905079808
111	chr5_GL949742v1_alt	226852	2905341952
112	chr5_KI270794v1_alt	164558	2905604096
113	chr5_GL339449v2_alt	1612928	2905866240
114	chr5_GL383530v1_alt	101241	2907701248
115	chr5_KI270796v1_alt	172708	2907963392
116	chr5_GL383531v1_alt	173459	2908225536
117	chr5_KI270795v1_alt	131892	2908487680
118	chr6_GL000250v2_alt	4672374	2908749824
119	chr6_KI270800v1_alt	175808	2913468416
120	chr6_KI270799v1_alt	152148	2913730560
121	chr6_GL383533v1_alt	124736	2913992704
122	chr6_KI270801v1_alt	870480	2914254848
123	chr6_KI270802v1_alt	75005	2915303424
124	chr6_KB021644v2_alt	185823	2915565568
125	chr6_KI270797v1_alt	197536	2915827712
126	chr6_KI270798v1_alt	271782	2916089856
127	chr7_KI270804v1_alt	157952	2916614144
128	chr7_KI270809v1_alt	209586	2916876288
129	chr7_KI270806v1_alt	158166	2917138432
130	chr7_GL383534v2_alt	119183	2917400576
131	chr7_KI270803v1_alt	1111570	2917662720
132	chr7_KI270808v1_alt	271455	2918973440
133	chr7_KI270807v1_alt	126434	2919497728
134	chr7_KI270805v1_alt	209988	2919759872
135	chr8_KI270818v1_alt	145606	2920022016
136	chr8_KI270812v1_alt	282736	2920284160
137	chr8_KI270811v1_alt	292436	2920808448
138	chr8_KI270821v1_alt	985506	2921332736
139	chr8_KI270813v1_alt	300230	2922381312
140	chr8_KI270822v1_alt	624492	2922905600
141	chr8_KI270814v1_alt	141812	2923692032
142	chr8_KI270810v1_alt	374415	2923954176
143	chr8_KI270819v1_alt	133535	2924478464
144	chr8_KI270820v1_alt	36640	2924740608
145	chr8_KI270817v1_alt	158983	2925002752
146	chr8_KI270816v1_alt	305841	2925264896
147	chr8_KI270815v1_alt	132244	2925789184
148	chr9_GL383539v1_alt	162988	2926051328
149	chr9_GL383540v1_alt	71551	2926313472
150	chr9_GL383541v1_alt	171286	2926575616
151	chr9_GL383542v1_alt	60032	2926837760
152	chr9_KI270823v1_alt	439082	2927099904
153	chr10_GL383545v1_alt	179254	2927624192
154	chr10_KI270824v1_alt	181496	2927886336
155	chr10_GL383546v1_alt	309802	2928148480
156	chr10_KI270825v1_alt	188315	2928672768
157	chr11_KI270832v1_alt	210133	2928934912
158	chr11_KI270830v1_alt	177092	2929197056
159	chr11_KI270831v1_alt	296895	2929459200
160	chr11_KI270829v1_alt	204059	2929983488
161	chr11_GL383547v1_alt	154407	2930245632
162	chr11_JH159136v1_alt	200998	2930507776
163	chr11_JH159137v1_alt	191409	2930769920
164	chr11_KI270827v1_alt	67707	2931032064
165	chr11_KI270826v1_alt	186169	2931294208
166	chr12_GL877875v1_alt	167313	2931556352
167	chr12_GL877876v1_alt	408271	2931818496
168	chr12_KI270837v1_alt	40090	2932342784
169	chr12_GL383549v1_alt	120804	2932604928
170	chr12_KI270835v1_alt	238139	2932867072
171	chr12_GL383550v2_alt	169178	2933129216
172	chr12_GL383552v1_alt	138655	2933391360
173	chr12_GL383553v2_alt	152874	2933653504
174	chr12_KI270834v1_alt	119498	2933915648
175	chr12_GL383551v1_alt	184319	2934177792
176	chr12_KI270833v1_alt	76061	2934439936
177	chr12_KI270836v1_alt	56134	2934702080
178	chr13_KI270840v1_alt	191684	2934964224
179	chr13_KI270839v1_alt	180306	2935226368
180	chr13_KI270843v1_alt	103832	2935488512
181	chr13_KI270841v1_alt	169134	2935750656
182	chr13_KI270838v1_alt	306913	2936012800
183	chr13_KI270842v1_alt	37287	2936537088
184	chr14_KI270844v1_alt	322166	2936799232
185	chr14_KI270847v1_alt	1511111	2937323520
186	chr14_KI270845v1_alt	180703	2938896384
187	chr14_KI270846v1_alt	1351393	2939158528
188	chr15_KI270852v1_alt	478999	2940731392
189	chr15_KI270851v1_alt	263054	2941255680
190	chr15_KI270848v1_alt	327382	2941779968
191	chr15_GL383554v1_alt	296527	2942304256
192	chr15_KI270849v1_alt	244917	2942828544
193	chr15_GL383555v2_alt	388773	2943090688
194	chr15_KI270850v1_alt	430880	2943614976
195	chr16_KI270854v1_alt	134193	2944139264
196	chr16_KI270856v1_alt	63982	2944401408
197	chr16_KI270855v1_alt	232857	2944663552
198	chr16_KI270853v1_alt	2659700	2944925696
199	chr16_GL383556v1_alt	192462	2947809280
200	chr16_GL383557v1_alt	89672	2948071424
201	chr17_GL383563v3_alt	375691	2948333568
202	chr17_KI270862v1_alt	391357	2948857856
203	chr17_KI270861v1_alt	196688	2949382144
204	chr17_KI270857v1_alt	2877074	2949644288
205	chr17_JH159146v1_alt	278131	2952527872
206	chr17_JH159147v1_alt	70345	2953052160
207	chr17_GL383564v2_alt	133151	2953314304
208	chr17_GL000258v2_alt	1821992	2953576448
209	chr17_GL383565v1_alt	223995	2955411456
210	chr17_KI270858v1_alt	235827	2955673600
211	chr17_KI270859v1_alt	108763	2955935744
212	chr17_GL383566v1_alt	90219	2956197888
213	chr17_KI270860v1_alt	178921	2956460032
214	chr18_KI270864v1_alt	111737	2956722176
215	chr18_GL383567v1_alt	289831	2956984320
216	chr18_GL383570v1_alt	164789	2957508608
217	chr18_GL383571v1_alt	198278	2957770752
218	chr18_GL383568v1_alt	104552	2958032896
219	chr18_GL383569v1_alt	167950	2958295040
220	chr18_GL383572v1_alt	159547	2958557184
221	chr18_KI270863v1_alt	167999	2958819328
222	chr19_KI270868v1_alt	61734	2959081472
223	chr19_KI270865v1_alt	52969	2959343616
224	chr19_GL383573v1_alt	385657	2959605760
225	chr19_GL383575v2_alt	170222	2960130048
226	chr19_GL383576v1_alt	188024	2960392192
227	chr19_GL383574v1_alt	155864	2960654336
228	chr19_KI270866v1_alt	43156	2960916480
229	chr19_KI270867v1_alt	233762	2961178624
230	chr19_GL949746v1_alt	987716	2961440768
231	chr20_GL383577v2_alt	128386	2962489344
232	chr20_KI270869v1_alt	118774	2962751488
233	chr20_KI270871v1_alt	58661	2963013632
234	chr20_KI270870v1_alt	183433	2963275776
235	chr21_GL383578v2_alt	63917	2963537920
236	chr21_KI270874v1_alt	166743	2963800064
237	chr21_KI270873v1_alt	143900	2964062208
238	chr21_GL383579v2_alt	201197	2964324352
239	chr21_GL383580v2_alt	74653	2964586496
240	chr21_GL383581v2_alt	116689	2964848640
241	chr21_KI270872v1_alt	82692	2965110784
242	chr22_KI270875v1_alt	259914	2965372928
243	chr22_KI270878v1_alt	186262	2965635072
244	chr22_KI270879v1_alt	304135	2965897216
245	chr22_KI270876v1_alt	263666	2966421504
246	chr22_KI270877v1_alt	101331	2966945792
247	chr22_GL383583v2_alt	96924	2967207936
248	chr22_GL383582v2_alt	162811	2967470080
249	chrX_KI270880v1_alt	284869	2967732224
250	chrX_KI270881v1_alt	144206	2968256512
251	chr19_KI270882v1_alt	248807	2968518656
252	chr19_KI270883v1_alt	170399	2968780800
253	chr19_KI270884v1_alt	157053	2969042944
254	chr19_KI270885v1_alt	171027	2969305088
255	chr19_KI270886v1_alt	204239	2969567232
256	chr19_KI270887v1_alt	209512	2969829376
257	chr19_KI270888v1_alt	155532	2970091520
258	chr19_KI270889v1_alt	170698	2970353664
259	chr19_KI270890v1_alt	184499	2970615808
260	chr19_KI270891v1_alt	170680	2970877952
261	chr1_KI270892v1_alt	162212	2971140096
262	chr2_KI270894v1_alt	214158	2971402240
263	chr2_KI270893v1_alt	161218	2971664384
264	chr3_KI270895v1_alt	162896	2971926528
265	chr4_KI270896v1_alt	378547	2972188672
266	chr5_KI270897v1_alt	1144418	2972712960
267	chr5_KI270898v1_alt	130957	2974023680
268	chr6_GL000251v2_alt	4795265	2974285824
269	chr7_KI270899v1_alt	190869	2979266560
270	chr8_KI270901v1_alt	136959	2979528704
271	chr8_KI270900v1_alt	318687	2979790848
272	chr11_KI270902v1_alt	106711	2980315136
273	chr11_KI270903v1_alt	214625	2980577280
274	chr12_KI270904v1_alt	572349	2980839424
275	chr15_KI270906v1_alt	196384	2981625856
276	chr15_KI270905v1_alt	5161414	2981888000
277	chr17_KI270907v1_alt	137721	2987130880
278	chr17_KI270910v1_alt	157099	2987393024
279	chr17_KI270909v1_alt	325800	2987655168
280	chr17_JH159148v1_alt	88070	2988179456
281	chr17_KI270908v1_alt	1423190	2988441600
282	chr18_KI270912v1_alt	174061	2990014464
283	chr18_KI270911v1_alt	157710	2990276608
284	chr19_GL949747v2_alt	729520	2990538752
285	chr22_KB663609v1_alt	74013	2991325184
286	chrX_KI270913v1_alt	274009	2991587328
287	chr19_KI270914v1_alt	205194	2992111616
288	chr19_KI270915v1_alt	170665	2992373760
289	chr19_KI270916v1_alt	184516	2992635904
290	chr19_KI270917v1_alt	190932	2992898048
291	chr19_KI270918v1_alt	123111	2993160192
292	chr19_KI270919v1_alt	170701	2993422336
293	chr19_KI270920v1_alt	198005	2993684480
294	chr19_KI270921v1_alt	282224	2993946624
295	chr19_KI270922v1_alt	187935	2994470912
296	chr19_KI270923v1_alt	189352	2994733056
297	chr3_KI270924v1_alt	166540	2994995200
298	chr4_KI270925v1_alt	555799	2995257344
299	chr6_GL000252v2_alt	4604811	2996043776
300	chr8_KI270926v1_alt	229282	3000762368
301	chr11_KI270927v1_alt	218612	3001024512
302	chr19_GL949748v2_alt	1064304	3001286656
303	chr22_KI270928v1_alt	176103	3002597376
304	chr19_KI270929v1_alt	186203	3002859520
305	chr19_KI270930v1_alt	200773	3003121664
306	chr19_KI270931v1_alt	170148	3003383808
307	chr19_KI270932v1_alt	215732	3003645952
308	chr19_KI270933v1_alt	170537	3003908096
309	chr19_GL000209v2_alt	177381	3004170240
310	chr3_KI270934v1_alt	163458	3004432384
311	chr6_GL000253v2_alt	4677643	3004694528
312	chr19_GL949749v2_alt	1091841	3009413120
313	chr3_KI270935v1_alt	197351	3010723840
314	chr6_GL000254v2_alt	4827813	3010985984
315	chr19_GL949750v2_alt	1066390	3015966720
316	chr3_KI270936v1_alt	164170	3017277440
317	chr6_GL000255v2_alt	4606388	3017539584
318	chr19_GL949751v2_alt	1002683	3022258176
319	chr3_KI270937v1_alt	165607	3023306752
320	chr6_GL000256v2_alt	4929269	3023568896
321	chr19_GL949752v1_alt	987100	3028549632
322	chr6_KI270758v1_alt	76752	3029598208
323	chr19_GL949753v2_alt	796479	3029860352
324	chr19_KI270938v1_alt	1066800	3030908928
325	chrM	16569	3032219648
326	chrUn_KI270302v1	2274	3032481792
327	chrUn_KI270304v1	2165	3032743936
328	chrUn_KI270303v1	1942	3033006080
329	chrUn_KI270305v1	1472	3033268224
330	chrUn_KI270322v1	21476	3033530368
331	chrUn_KI270320v1	4416	3033792512
332	chrUn_KI270310v1	1201	3034054656
333	chrUn_KI270316v1	1444	3034316800
334	chrUn_KI270315v1	2276	3034578944
335	chrUn_KI270312v1	998	3034841088
336	chrUn_KI270311v1	12399	3035103232
337	chrUn_KI270317v1	37690	3035365376
338	chrUn_KI270412v1	1179	3035627520
339	chrUn_KI270411v1	2646	3035889664
340	chrUn_KI270414v1	2489	3036151808
341	chrUn_KI270419v1	1029	3036413952
342	chrUn_KI270418v1	2145	3036676096
343	chrUn_KI270420v1	2321	3036938240
344	chrUn_KI270424v1	2140	3037200384
345	chrUn_KI270417v1	2043	3037462528
346	chrUn_KI270422v1	1445	3037724672
347	chrUn_KI270423v1	981	3037986816
348	chrUn_KI270425v1	1884	3038248960
349	chrUn_KI270429v1	1361	3038511104
350	chrUn_KI270442v1	392061	3038773248
351	chrUn_KI270466v1	1233	3039297536
352	chrUn_KI270465v1	1774	3039559680
353	chrUn_KI270467v1	3920	3039821824
354	chrUn_KI270435v1	92983	3040083968
355	chrUn_KI270438v1	112505	3040346112
356	chrUn_KI270468v1	4055	3040608256
357	chrUn_KI270510v1	2415	3040870400
358	chrUn_KI270509v1	2318	3041132544
359	chrUn_KI270518v1	2186	3041394688
360	chrUn_KI270508v1	1951	3041656832
361	chrUn_KI270516v1	1300	3041918976
362	chrUn_KI270512v1	22689	3042181120
363	chrUn_KI270519v1	138126	3042443264
364	chrUn_KI270522v1	5674	3042705408
365	chrUn_KI270511v1	8127	3042967552
366	chrUn_KI270515v1	6361	3043229696
367	chrUn_KI270507v1	5353	3043491840
368	chrUn_KI270517v1	3253	3043753984
369	chrUn_KI270529v1	1899	3044016128
370	chrUn_KI270528v1	2983	3044278272
371	chrUn_KI270530v1	2168	3044540416
372	chrUn_KI270539v1	993	3044802560
373	chrUn_KI270538v1	91309	3045064704
374	chrUn_KI270544v1	1202	3045326848
375	chrUn_KI270548v1	1599	3045588992
376	chrUn_KI270583v1	1400	3045851136
377	chrUn_KI270587v1	2969	3046113280
378	chrUn_KI270580v1	1553	3046375424
379	chrUn_KI270581v1	7046	3046637568
380	chrUn_KI270579v1	31033	3046899712
381	chrUn_KI270589v1	44474	3047161856
382	chrUn_KI270590v1	4685	3047424000
383	chrUn_KI270584v1	4513	3047686144
384	chrUn_KI270582v1	6504	3047948288
385	chrUn_KI270588v1	6158	3048210432
386	chrUn_KI270593v1	3041	3048472576
387	chrUn_KI270591v1	5796	3048734720
388	chrUn_KI270330v1	1652	3048996864
389	chrUn_KI270329v1	1040	3049259008
390	chrUn_KI270334v1	1368	3049521152
391	chrUn_KI270333v1	2699	3049783296
392	chrUn_KI270335v1	1048	3050045440
393	chrUn_KI270338v1	1428	3050307584
394	chrUn_KI270340v1	1428	3050569728
395	chrUn_KI270336v1	1026	3050831872
396	chrUn_KI270337v1	1121	3051094016
397	chrUn_KI270363v1	1803	3051356160
398	chrUn_KI270364v1	2855	3051618304
399	chrUn_KI270362v1	3530	3051880448
400	chrUn_KI270366v1	8320	3052142592
401	chrUn_KI270378v1	1048	3052404736
402	chrUn_KI270379v1	1045	3052666880
403	chrUn_KI270389v1	1298	3052929024
404	chrUn_KI270390v1	2387	3053191168
405	chrUn_KI270387v1	1537	3053453312
406	chrUn_KI270395v1	1143	3053715456
407	chrUn_KI270396v1	1880	3053977600
408	chrUn_KI270388v1	1216	3054239744
409	chrUn_KI270394v1	970	3054501888
410	chrUn_KI270386v1	1788	3054764032
411	chrUn_KI270391v1	1484	3055026176
412	chrUn_KI270383v1	1750	3055288320
413	chrUn_KI270393v1	1308	3055550464
414	chrUn_KI270384v1	1658	3055812608
415	chrUn_KI270392v1	971	3056074752
416	chrUn_KI270381v1	1930	3056336896
417	chrUn_KI270385v1	990	3056599040
418	chrUn_KI270382v1	4215	3056861184
419	chrUn_KI270376v1	1136	3057123328
420	chrUn_KI270374v1	2656	3057385472
421	chrUn_KI270372v1	1650	3057647616
422	chrUn_KI270373v1	1451	3057909760
423	chrUn_KI270375v1	2378	3058171904
424	chrUn_KI270371v1	2805	3058434048
425	chrUn_KI270448v1	7992	3058696192
426	chrUn_KI270521v1	7642	3058958336
427	chrUn_GL000195v1	182896	3059220480
428	chrUn_GL000219v1	179198	3059482624
429	chrUn_GL000220v1	161802	3059744768
430	chrUn_GL000224v1	179693	3060006912
431	chrUn_KI270741v1	157432	3060269056
432	chrUn_GL000226v1	15008	3060531200
433	chrUn_GL000213v1	164239	3060793344
434	chrUn_KI270743v1	210658	3061055488
435	chrUn_KI270744v1	168472	3061317632
436	chrUn_KI270745v1	41891	3061579776
437	chrUn_KI270746v1	66486	3061841920
438	chrUn_KI270747v1	198735	3062104064
439	chrUn_KI270748v1	93321	3062366208
440	chrUn_KI270749v1	158759	3062628352
441	chrUn_KI270750v1	148850	3062890496
442	chrUn_KI270751v1	150742	3063152640
443	chrUn_KI270752v1	27745	3063414784
444	chrUn_KI270753v1	62944	3063676928
445	chrUn_KI270754v1	40191	3063939072
446	chrUn_KI270755v1	36723	3064201216
447	chrUn_KI270756v1	79590	3064463360
448	chrUn_KI270757v1	71251	3064725504
449	chrUn_GL000214v1	137718	3064987648
450	chrUn_KI270742v1	186739	3065249792
451	chrUn_GL000216v2	176608	3065511936
452	chrUn_GL000218v1	161147	3065774080
453	chrX	156040895	3066036224
454	chrY	57227415	3222274048
455	chrY_KI270740v1_random	37240	3279683584
--sjdbOverhang = 100 taken from the generated genome
Started loading the genome: Tue Aug 20 22:15:38 2019

Genome: size given as a parameter = 3358284875
SA: size given as a parameter = 25799931897
/SAindex: size given as a parameter = 1
Read from SAindex: genomeSAindexNbases=14  nSAi=357913940
nGenome=3358284875;  nSAbyte=25799931897
GstrandBit=32   SA number of indices=6254528944
Shared memory is not used for genomes. Allocated a private copy of the genome.
Genome file size: 3358284875 bytes; state: good=1 eof=0 fail=0 bad=0
Loading Genome ... done! state: good=1 eof=0 fail=0 bad=0; loaded 3358284875 bytes
SA file size: 25799931897 bytes; state: good=1 eof=0 fail=0 bad=0
Loading SA ... done! state: good=1 eof=0 fail=0 bad=0; loaded 25799931897 bytes
Loading SAindex ... done: 1565873619 bytes
Finished loading the genome: Tue Aug 20 22:16:25 2019

Processing splice junctions database sjdbN=389747,   sjdbOverhang=100
alignIntronMax=alignMatesGapMax=0, the max intron size will be approximately determined by (2^winBinNbits)&ast;winAnchorDistNbins=589824
Created thread # 1
Created thread # 2
Created thread # 3
Created thread # 4
Created thread # 5
Created thread # 6
Created thread # 7
BAM sorting: 318141 mapped reads
BAM sorting bins genomic start loci:
1	2	62852992
2	6	49862793
3	34	43419570
4	36	8211397
5	36	8254983
6	36	8393812
7	36	8401030
8	36	8438658
9	40	126740
10	40	132554
11	40	172778
12	51	1312024
13	56	145997340
14	324	9728
15	428	110875
16	428	154288
Thread #2 end of input stream, nextChar=-1
Completed: thread #3
Completed: thread #5
Completed: thread #1
Completed: thread #4
Completed: thread #2
Completed: thread #6
Completed: thread #0
Joined thread # 1
Joined thread # 2
Joined thread # 3
Joined thread # 4
Joined thread # 5
Joined thread # 6
Completed: thread #7
Joined thread # 7
Aug 20 22:19:38 ..... started sorting BAM
Max memory needed for sorting = 613034726
ALL DONE!
```

</details>

<details>
  <summary>**star_alignedLog.progress.out**</summary>

```{r}
           Time    Speed        Read     Read   Mapped   Mapped   Mapped   Mapped Unmapped Unmapped Unmapped Unmapped
                    M/hr      number   length   unique   length   MMrate    multi   multi+       MM    short    other
Aug 20 22:17:26    283.9     4811171       51    31.4%     49.7     0.8%    33.6%    27.9%     0.0%     6.8%     0.2%
Aug 20 22:18:27    362.7    12291502       51    31.6%     49.7     0.8%    33.4%    27.9%     0.0%     6.8%     0.2%
Aug 20 22:19:27    365.1    18457682       51    31.6%     49.7     0.8%    33.4%    27.9%     0.0%     6.8%     0.2%
ALL DONE!
```

</details>

### STAR Alignment redo: ocpmi

* samples: `*.fastq` via ocpmi; `~/ocpmi/data/bam/wes`
* star_vars: `100` (overhang); `Within` (unmapped); `Standard` (SAM attr)
* output: `~/ocpmi/data/bam/wes/`
* gene_ref: `hg38.fa`, `hg38.gtf` via UCSC; `~/ocpmi/gene_ref`

### figs_finish_excv2.sh

I started a bash script that would automate the steps necessary to process the excv2 output. Given the number of samples I have, this would be relatively pretty tedious, so I'm hoping this will be a good supplement to `full_send.sh`

### Summary

#### Running

```{r}
ID Started    Active Age  P Project Tags Description
-- ---------- ------ ---- - ------- ---- ---------------------------
67 2019-08-19 1d     7d   H lab     work realign bam files with star
81 2019-08-21 37s    1min   lab     work rerun test excv2 full send

2 tasks
```


#### Completed

```{r}
Created    Completed  Age Project Description
---------- ---------- --- ------- ----------------------------------------------
2019-08-08 2019-08-21 13d lab     run excv2 across ocpmi bams
2019-08-20 2019-08-21 1d  lab     replace full_send files with new ones after
                                  excv2 process completes
2019-08-21 2019-08-21 16h lab     remove base16 terminal theme

3 tasks
```

***

## 22

### TODO

```{r}
ID Active Age P Project Tags      Description                               Urg
-- ------ --- - ------- --------- ----------------------------------------- ----
80 1min   16h   lab     work      rerun excv2 across all bwa bams            5.8
76        1d    lab     ccgd work is a java license necessary               1.91
77        1d    lab     ccgd work how much drive space is necessary         1.91
78        1d    lab     ccgd work determine firewall config ports open      1.91
79        1d    lab     ccgd work determine security level for vm           1.91
67        7d    lab     work      look for differences in 1kgp comparison   1.84
                                  in exv2 paper
68        7d    lab     work      do search for best suited cnv controls    1.84
69        7d    lab     work      get site analytics for ccgd               1.84
70        7d    lab     work      submit vm application for ccgd            1.84
72        2d    lab     work      create full_send.sh for varscan           1.81
81        23s   lab     work      trblsht star alignment                     1.8
41        3mo L lab     work      agenda [5]                                1.29
40        3mo L lab     work      review scripts for ccgd database          0.31

13 tasks
```

### figs_finish_excv2.sh

This is a continuation of the bash script I started yesterday. I've expanded it quite a bit to make it effectively pump and dump for the markdown output needed to go to the `results` page. It needs to be run from the `Plots` results directory.

```{r}
#!/bin/bash

# Establish path for function working dir
dir=$PWD

# Get experiment name from parent dir
cd ..
expName1=$(pwd | awk -F'/' '{print $NF}')
cd $dir

# Source variable file from experiment
source /home/chris/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/config_files/$expName1/full_send_vars.$expName1.txt

# Text for header in figures markdown file
header=$(cat <<EOF
### $expName1 {.tabset .tabfade}

#### Discussion

* name: \`$expName1\`
* samples: \`SAMPLE\` (control), \`SAMPLE\` (expt) via ; \`$bamDir\`
* excv2_vars: \`$window\` (window); \`$anMode\` (analysis mode)
* output: \`$resDir/$expName1\`
* gene_ref: \`hg38.fa\`, \`hg38_sortmerge.bed\` via UCSC; \`~/ocpmi/gene_ref\`
* targets: \`GCA_000001405.15_GRCh38.bw\` via excv2

#### Figures {.tabset .tabfade}

##### Collated

EOF
)

echo "${header}" > figs.md
> collated.md

# Loop to created collated figures and output formatted markdown
for i in */; do
    if [ "$(ls -A $i)" ]; then
        cd $i
        # Create figs
        pdfunite Plot* combined.pdf
        pdfnup Plot* --nup 5x5 --outfile tmp.pdf
        pdftoppm tmp.pdf fig_matrix -png
        rm -rf tmp.pdf
        # Assign variable names for figs
        sampleName=${PWD##*/}
        figMatrix=$(find $PWD -name "fig_matrix*" | grep $sampleName)
        combinedPdf=$(find $PWD -name "combined.pdf" | grep $sampleName)
        # Formatted output to individual sample entry
        sampleMd=$(cat <<EOF

##### ${sampleName}

**Figure Matrix**

![]($figMatrix)

***

**Individual Output**

\`\`\`{r, out.width = "100%", out.height = "600px", echo = F, eval=T}
knitr::include_graphics("${combinedPdf}")
\`\`\`
EOF
)
        # Formatted output to collated entry for all samples
        collatedMd=$(cat << EOF

![${sampleName}]($figMatrix)
EOF
)
        # Pass to markdown files
        echo "${sampleMd}" >> ../figs.md
        echo "${collatedMd}" >> ../collated.md
        cd ../
    else
        cd $i
        sampleName=${PWD##*/}
        # Formatted output for no output
        noOutput=$(cat <<EOF

##### ${sampleName}

**No output**
EOF
)
        # Pass to markdown files
        echo "${noOutput}" >> ../figs.md
        echo "
        ${sampleName} No Output" >> ../collated.md
        cd ../
    fi
done

# Insert contents of collated.md at top of sample entries
cd $dir
sed -i -e '/##### Collated/r collated.md' figs.md
```

### Summary

Holy crap it feels like I've wasted an enormous amount of time. The last two days have been largely preoccupied with figuring out how I'm going to handle figure processing. A point I hadn't considered was the sheer quantity of figures generated by the excv2 pipeline. I think I generally have this under control now, but the actual size of these images may be an issue for my notebook. We'll have to see.

From my conversations with the collaborators yesterday and the current progress of the project, I think I need to do a mind dump and regear my direction.

#### Running

```{r}
ID Started    Active Age Project Tags Description
-- ---------- ------ --- ------- ---- -------------------------------
73 2019-08-22 7h     23h lab     work rerun excv2 across all bwa bams

1 task
```

#### Completed

```{r}
Created    Completed  Age   P Project Description
---------- ---------- ----- - ------- -------------------------------------
2019-08-21 2019-08-22 23h     lab     rerun test excv2 full send
2019-08-14 2019-08-22 8d    H lab     realign bam files with star
2019-08-21 2019-08-22 1d      lab     is a java license necessary
2019-08-21 2019-08-22 1d      lab     how much drive space is necessary
2019-08-21 2019-08-22 1d      lab     determine firewall config ports open
2019-08-21 2019-08-22 1d      lab     determine security level for vm
2019-08-22 2019-08-22 42min   lab     find lightweight browser for notebook
2019-08-14 2019-08-22 8d      lab     submit vm application for ccgd
2019-05-20 2019-08-22 3mo   L lab     review scripts for ccgd database
2019-08-22 2019-08-22 20s     lab     finish figs_finish_excv2.sh

10 tasks
```

***

## 24

### STAR alignment: ocpmi

**ocpmi**

* samples: `*.fastq` via ocpmi; `~/ocpmi/data/bam/wes`
* job: `16806914.mesabim1`
* star_vars: `100` (overhang); `SortedByCoordinate` (SAM attr)
* output: `~/ocpmi/data/bam/wes/star_aligned`
* gene_ref: `hg38.fa`, `hg38.gtf` via UCSC; `~/ocpmi/gene_ref`

**1kgp: NA10847**

* samples: `NA10847_SRR070823.fastq` via ocpmi; `~/ocpmi/data/bam/wes`
* job: `16811628.mesabim1`
* star_vars: `100` (overhang); `SortedByCoordinate` (SAM attr)
* output: `~/ocpmi/data/bam/wes/star_aligned`
* gene_ref: `hg38.fa`, `hg38.gtf` via UCSC; `~/ocpmi/gene_ref`

I was finally able to complete realignment with star. A github tutorial gave me an insight on how to structure my file naming scheme in my for loop since this was the issue. My previous iteration was overwriting each file.

A major issue here was that I had not addressed the paired reads in my alignment. This was a major oversight that I'm glad I caught and should now be addressed in the new alignment.

* https://gist.github.com/ipurusho/f6a6e53e0aa798c44e09c87bdc8b74fd

**bam_star_align.pbs**

```{r}
#!/bin/bash -l
#PBS -l walltime=4:00:00,nodes=1:ppn=8,mem=32gb
#PBS -m abe
#PBS -M tasta005@umn.edu
#PBS -N bam_star_align
#PBS -o logs/output
#PBS -e logs/error

module load gcc star

cd ~/ocpmi/data/bam/wes

## For looped
for i in *_R1_001.fastq; do
STAR --runMode alignReads \
    --outSAMtype BAM SortedByCoordinate \
    --genomeDir ~/ocpmi/data/gene_ref/ucsc/hg38 \
    --readFilesIn $i ${i%_R1_001.fastq}_R2_001.fastq \
    --runThreadN 8 \
    --outFileNamePrefix ${i%_R1_001.fastq}
done

### Individual files
#STAR --runMode alignReads \
#    --outSAMtype BAM SortedByCoordinate \
#    --genomeDir ~/ocpmi/data/gene_ref/ucsc/hg38 \
#    --readFilesIn read_1.fastq read_2.fastq \
#    --runThreadN 8 \
#    --outFileNamePrefix NA10847_SRR070823

```

### bwa_ocpmi_full1

* name: `bwa_ocpmi_full1_50000_2019-08-22_06:19`
* samples: `NA10847` (control), `*.bam` (expt) via ; `/ocpmi/data/bam/wes/bwa_aligned`
* excv2_vars: `50000` (window); `pooling` (analysis mode)
* output: `/ocpmi/results/excavator2/bwa_ocpmi_full1_50000_2019-08-22_06:19`
* gene_ref: `hg38.fa`, `hg38_sortmerge.bed` via UCSC; `~/ocpmi/gene_ref`
* targets: `GCA_000001405.15_GRCh38.bw` via excv2

I made some adjustments to the `excv2_full_send.sh` script hoping this would have an impact on the outcomes I saw from *2019-08-21*. Unfortunately, the outcome was effectively the same with several samples not producing an output.

***

## 26

### TODO

```{r}
ID Age P Project Tags Description                                           Urg
-- --- - ------- ---- ----------------------------------------------------- ----
65 11d   lab     work clarify duo login issues with ale01                   1.86
66 11d   lab     work look for differences in 1kgp comparison in exv2 paper 1.86
67 11d   lab     work do search for best suited cnv controls                1.86
69 6d    lab     work create full_send.sh for varscan                       1.83
72 3d    lab     work followup with andy re possible cnv controls           1.82
73 3d    lab     work check if im using the immortalized or baseline 10847  1.82
74 3d    lab     work do mind dump. regear direction                        1.82
75 3d    lab     work run vs2 across all bwa bams                           1.82
76 3d    lab     work add documentation re ccgd hosting app stats           1.82
77 1d    lab     work setup termux with umn vpn                             1.81
78 1d    lab     work review resulst of 8/24 star alignment                 1.81
82 21s   lab     work run excv2_full_send across all star bams               1.8
40 3mo L lab     work agenda [5]                                            1.32

13 tasks
```

### star bams indexing

* samples: `*.bam`; `/ocpmi/data/bam/wes/star_aligned`
* job: `16827658.mesabim1`
* output: `/ocpmi/data/bam/wes/star_aligned`

### excv2 star_aligned all bams

* name: `star_ocpmi_full4_50000_2019-08-26_17:59`
* job: `16828021.mesabim1`
* samples: `NA10847` (control), `*.bam` (expt) via ; `/ocpmi/data/bam/wes/star_aligned`
* excv2_vars: `50000` (window); `pooling` (analysis mode)
* output: `~/ocpmi/results/excavator2`
* gene_ref: `hg38.fa`, `hg38_sortmerge.bed` via UCSC; `~/ocpmi/gene_ref`
* targets: `GCA_000001405.15_GRCh38.bw` via excv2

*&ast;.bed `sort` (bedtools) &rarr; **`full_send.sh`** &rarr; cnv output &rarr; `figs_finish_excv2.sh`*

### Updated wes data!!

Apparently, I've been using the wrong data set this whole time. The directory Tim had given me pointed to a set of failed RNAseq data. Andy sent me an email detailing the location of the correct dataset. I'll need to start the process over.

* `/panfs/roc/groups/3/nelsona2/data_release/umgc/novaseq/181103_A00223_0048_AHGMHLDMXX/Nelson3_Project_003`

### STAR align: wes1

* samples: `sample` via ocpmi; `~/ocpmi/data/bam/wes1`
* job: `16828650.mesabim1`
* star_vars: `100` (overhang); `Standard` (SAM attr)
* output: `~/ocpmi/data/bam/wes1/star_aligned`
* gene_ref: `hg38.fa`, `hg38.gtf` via UCSC; `~/ocpmi/gene_ref`

*gene_ref, gtf &rarr; `genomeGenerate` indexing &rarr; **`alignReads` + `sort`** &rarr; output.bam &rarr; index `samtools`*

This is the new set of data that Andy referred to today - referred to by him as Winterhoff Project 002. I will need to do a full work up on this data and effectively start fresh with the analysis.

I'm leaving the files gzipped to conserve space and adjusted the star function by adding the `readFilesCommand`.

### Summary

We had a major development today in that the dataset I've been working with up to this point has been incorrect. I'll need to restart the process and discuss the nature of this with Tim. I immediately started work on the new data set as I finish out a few items with the old one just to see the results.

#### Running

```{r}
ID Started    Active Age Project Tags Description
-- ---------- ------ --- ------- ---- ----------------------------------------
82 2019-08-26 4h     4h  lab     work run excv2_full_send across all star bams
83 2019-08-26 1h     1h  lab     work align new wes1 fastq files

2 tasks
```

#### Completed

```{r}
Created    Completed  Age Project Description
---------- ---------- --- ------- ----------------------------------------------
2019-08-21 2019-08-26 4d  lab     rerun excv2 across all bwa bams
2019-08-22 2019-08-26 4d  lab     trblsht star alignment
2019-08-22 2019-08-26 4d  lab     possibly realign 10847 with star
2019-08-23 2019-08-26 3d  lab     rerun star align with updated readFilesIn
                                  syntax

4 tasks
```

***

## 27

### TODO

```{r}
ID Active Age   P Project Tags Description                                Urg
-- ------ ----- - ------- ---- ------------------------------------------ ------
81 19h    19h     lab     work align new wes1 fastq files                    5.8
80 22h    22h     lab     work run excv2_full_send across all star bams      5.8
65        12d     lab     work look for differences in 1kgp comparison in   1.87
                               exv2 paper
66        12d     lab     work do search for best suited cnv controls       1.87
68        7d      lab     work create full_send.sh for varscan              1.84
71        4d      lab     work followup with andy re possible cnv           1.82
                               controls
72        4d      lab     work check if im using the immortalized or        1.82
                               baseline 10847
73        4d      lab     work do mind dump. regear direction               1.82
74        4d      lab     work run vs2 across all bwa bams                  1.82
75        4d      lab     work add documentation re ccgd hosting app        1.82
                               stats
77        2d      lab     work review resulst of 8/24 star alignment        1.81
82        19h     lab     work run excv2 on wes1 ocpmi files                 1.8
83        18h     lab     work summary from yesterday                        1.8
84        54min   lab     work add code block below pipeline summaries       1.8
39        3mo   L lab     work agenda [2]                                   1.22
64        12d   L lab     work clarify duo login issues with ale01        -0.134
76        2d    L lab     work setup termux with umn vpn                  -0.189

17 tasks
```

### STAR align: wes1

* samples: `sample` via ocpmi; `~/ocpmi/data/bam/wes1`
* job: `xxxx`
* star_vars: `100` (overhang); `Standard` (SAM attr)
* output: `~/ocpmi/data/bam/wes1/star_aligned`
* gene_ref: `hg38.fa`, `hg38.gtf` via UCSC; `~/ocpmi/gene_ref`

**Start of day**

*gene_ref, gtf &rarr; `genomeGenerate` indexing &rarr; **`alignReads` + `sort`** &rarr; output.bam &rarr; index `samtools`*

### excv2 star_aligned all bams

* name: `star_ocpmi_full4_50000_2019-08-26_17:59`
* job: `16828021.mesabim1`
* samples: `NA10847` (control), `*.bam` (expt) via ; `/ocpmi/data/bam/wes/star_aligned`
* excv2_vars: `50000` (window); `pooling` (analysis mode)
* output: `~/ocpmi/results/excavator2`
* gene_ref: `hg38.fa`, `hg38_sortmerge.bed` via UCSC; `~/ocpmi/gene_ref`
* targets: `GCA_000001405.15_GRCh38.bw` via excv2

**Start of day**

*&ast;.bed `sort` (bedtools) &rarr; `full_send.sh` &rarr; **cnv output** &rarr; `figs_finish_excv2.sh`*

I was able to complete this process, yet it comes with a sense of futility. This data has been identified as incorrect, so the results are no longer relavent.

**End of day**

*&ast;.bed `sort` (bedtools) &rarr; `full_send.sh` &rarr; cnv output &rarr; `**figs_finish_excv2.sh**`*

### Summary

My primary focus right now is to reestablish the pipeline on the new wes data set. I am getting that going and have transferred new files to my directory. It seems current hurdles will center around getting the resource allocation correct in alignment with star.

#### Completed

```{r}
Created    Completed  Age Project Description
---------- ---------- --- ------- -------------------------------------------
2019-08-26 2019-08-27 1d  lab     summary from yesterday
2019-08-26 2019-08-27 2d  lab     run excv2_full_send across all star bams
2019-08-23 2019-08-27 5d  lab     add documentation re ccgd hosting app stats
2019-08-25 2019-08-27 3d  lab     review resulst of 8/24 star alignment
2019-08-27 2019-08-27 1d  lab     add code block below pipeline summaries

5 tasks
```


***

## 28

### TODO

```{r}
ID Age P Project Tags Description                                         Urg
-- --- - ------- ---- --------------------------------------------------- ------
78 1d  H lab     work align new wes1 fastq files                            7.81
65 2w    lab     work look for differences in 1kgp comparison in exv2       1.88
                      paper
66 2w    lab     work do search for best suited cnv controls                1.88
68 8d    lab     work create full_send.sh for varscan                       1.84
71 6d    lab     work followup with andy re possible cnv controls           1.83
72 6d    lab     work check if im using the immortalized or baseline        1.83
                      10847
73 5d    lab     work do mind dump. regear direction                        1.83
74 5d    lab     work run vs2 across all bwa bams                           1.83
79 1d    lab     work run excv2 on wes1 ocpmi files                         1.81
81 1d    lab     work add results of excv2 bwa bams to results page         1.81
82 23h   lab     work req new vm for lab notebooks                           1.8
39 3mo L lab     work agenda [2]                                            1.23
64 2w  L lab     work clarify duo login issues with ale01                 -0.123
75 3d  L lab     work setup termux with umn vpn                           -0.184

14 tasks
```

### wkly mtg w/ Tim

present: chris, tim

```{r}
Description
-----------------------------------------------------
agenda
  2019-08-26 redid seq alignment w/ star
  2019-08-26 whats the deal with the updated wes data
  2019-08-28 lab notebooks vm
  2019-08-28 update on ccgd

1 task
```

Our meeting was relatively brief today. I updated Tim on my general progress with ccgd and the new data. We clarified the status of the wes data since there has been some ambiguity coming from Andy and msi with the integrity of the data. As of now, I can reliably say I have the correct dataset from the `Nelson3_Project_003` data_release dir.

### STAR align: wes1

* samples: `sample` via ocpmi; `~/ocpmi/data/bam/wes1`
* job: `xxxx`
* star_vars: `100` (overhang); `Standard` (SAM attr)
* output: `~/ocpmi/data/bam/wes1/star_aligned`
* gene_ref: `hg38.fa`, `hg38.gtf` via UCSC; `~/ocpmi/gene_ref`

**Start of day**

*gene_ref, gtf &rarr; `genomeGenerate` indexing &rarr; **`alignReads` + `sort`** &rarr; output.bam &rarr; index `samtools`*

This has continued to be a thorn in my side. Based on the error output I'm getting, I seem to need to thread the needle between striving for the large resource allocation the aligner needs and putting together a resource request that will actually initiate on msi. I'm waiting to hear back about a ticket with msi to see if there's a better approach to strategically requesting resources.

***

# References

