---
title: "Log"
author: "Christopher Tastad"
date: \today
bibliography: library.bib
output:
    html_document:
        toc: yes
        toc_float: yes
        theme: paper
---


<!-- Style -->

<style type="text/css">

h1 {
  color: #333333;
}
h2 {
  color: #333333;
}
h3 {
  color: #333333;
}
h4 {
  color: #333333;
}
</style>

<!-- Content -->

# Preamble

This is the notebook of Christopher Tastad, a lab member of Tim Starr at the University of Minnesota. All considerations for ownership follow the discretion of the Starr Lab at the extension of the policies of the University of Minnesota.

The layout is generally centered here on the `Log`, which is meant to be a full continuum of work and notation completed while in the Starr lab. Some elements from the `Log` been reiterated on dedicated pages as special points of reference when a relevant context is needed such as a repeatedly used `method` or notable `results`. References used on each page are displayed at the bottom, and all references used across the notebook and bibliography at large (library.bib located at `starr_lab/ocpmi/notebook/`) are collated on the `References` page.

### Git

Version control was used throughout the development of this work and can be found at the University's private github server:

https://github.umn.edu/tasta005/ocpmi.git

### Contact

Christopher Tastad, MS Candidate

* University: tasta005@umn.edu
* Personal: ctastad@gmail.com
* Mobile: 612-817-7989

```{r global_options}
knitr::opts_chunk$set(eval=F)
```

# May 2019

## 20

### Background mtg w/ Tim

* present: chris, tim

We discussed the outlook for my role in the lab. This included a considerable amount of background on the ocpmi project and potential directions I could take.

Focus Options

* ID of ovca stem cell
* what is the basis for clustering
* single cell seq methods paper
* compare mouse to human
* copy number variation on single cell
* rerun new version of cell ranger
* RHEL update

***

## 21

### TODO

```{r}
ID  Age   Project        Tags       R Due        Description                Urg
--- ----- -------------- ---------- - ---------- -------------------------- ----
326 10h   starrlab       computer n R 2019-05-21 write daily summary        10.5
327 27min starrlab       computer n              find umn git and hosting    1.9
312 20h   starrlab.ocpmi computer n              organize lab notebook [1]   2.7
315 18h   starrlab.ocpmi computer n              find ovca review re cur     1.9
                                                 treatment
316 18h   starrlab.ocpmi computer n              read starr paper re 1st pt  1.9
317 18h   starrlab.ocpmi computer n              read ocpmi summary          1.9
318 18h   starrlab.ocpmi computer n              install surat               1.9
319 18h   starrlab.ocpmi computer w              run surat tutorial w/ pt    1.9
                                                 59
320 18h   starrlab.ocpmi computer n              read single cell paper      1.9
                                                 zheng 2017
321 18h   starrlab.ocpmi computer n              find and read copy num      1.9
                                                 variation paper
322 18h   starrlab.ocpmi computer n              organize refs in email      1.9
                                                 inbox
323 18h   starrlab.ocpmi computer n              compile bibtex ref page     1.9

12 tasks
```


### Citation packages

#### Rmarkdown Default

This is the intrinsic formatting style for Rmarkdown citation management. It has the ability to call a BibTex file and carries the least effort to get it working. It's what I will probably go for across my whole setup. [@citations]

#### knitcitations

The knitcitations package might provide the best balance of feature set to ease of use. The only constraining factor is what seems like a few bugs that limit intended functionality. The ref() function doesn't work for instance. [@cboettig_2019]

#### RefManageR

This is a fully featured R package that seems to attempt to replace a complete citation manager. It comes across as very comprehensive but more complicated than I would like the approach to be. [@refmanager]

### Summary

#### Completed

* 327, 312:progress

The majority of the day was spent exploring the setup for my lab notebook. My intention is to create a living document using the Rmarkdown website knit feature [@rmarkdown_cheatsheet]. This will allow for a dynamic storage of both records and code work inline with one comprehensive document. A large portion of this time was spent giving consideration to how I will handle citations both in the document and for future publication. After significant exploration, my plan is to use the intrinsic Rmarkdown system [@citations] through use of a bibtex file. This will allow for simple reference in-place with the notebook along with using a standardized format for publication ready references.
Additionally, I plan on establishing a git repo to manage version control and historical record with the notebook as well using the umn github server [citation needed].

***

## 22

### TODO



```{r}
ID  Age Project        Tags       R Due        Description                  Urg
--- --- -------------- ---------- - ---------- ---------------------------- ----
326 15h starrlab       computer n R 2019-05-22 write daily summary          10.5
312 1d  starrlab.ocpmi computer n              organize lab notebook [1]    2.71
315 1d  starrlab.ocpmi computer n              find ovca review re cur      1.91
                                               treatment
316 1d  starrlab.ocpmi computer n              read starr paper re 1st pt   1.91
317 1d  starrlab.ocpmi computer n              read ocpmi summary           1.91
318 1d  starrlab.ocpmi computer n              install surat                1.91
319 1d  starrlab.ocpmi computer w              run surat tutorial w/ pt 59  1.91
320 1d  starrlab.ocpmi computer n              read single cell paper zheng 1.91
                                               2017
321 1d  starrlab.ocpmi computer n              find and read copy num       1.91
                                               variation paper
322 1d  starrlab.ocpmi computer n              organize refs in email inbox 1.91
323 1d  starrlab.ocpmi computer n              compile bibtex ref page      1.91

11 tasks
```


### wkly mtg w/ Tim

present: chris, tim

* Make sure to utilize notes field in bibtex file. Having an ongoing record of each paper read is critical.
* Consider separating data and results in file organization.
* Change notebook structure to follow layout of paper. Organize the work with a mindset toward writing the paper from the start, not the finish.
* Currently, we are still working to establish the standardized parameters of the data set. The goal is to arrive at a place where we can make biological conclusions.
    + How do we draw contrast from a data set of this nature?

### Notebook Structure

In my conversation with Tim, I decided I need to take an alternate structure to my lab notebook. My original intent was to emulate a development oriented documentation process. Instead, I will go a route that better follows the academic writing process with pages dedicated to methods, results, etc.

### Reference Management

While I would prefer to take a lean, text based approach to this, I am clearly beginning to waste some time in gear the system. I came across jabref, which seems to leverage a unix philosophy and can be found in the ubuntu repos.

### Summary

#### Completed

* 312:progress, 323:progress

My work today included continued focus on getting organized an taking a very deliberate approaching to my workflow before I begin. I met with Tim to begin our weekly meetings, and we covered some aspects of this workflow. Major take aways include the importance of annotation when reviewing literature. I had originally assumed I could handle references processing merely in a bare bibtex file. My feeling now is that I will begin to use a dedicated application for greater consistency in documentation.

***

## 23

### TODO



```{r}
ID  Age Project        Tags       R Due        Description                  Urg
--- --- -------------- ---------- - ---------- ---------------------------- ----
334 11h starrlab       computer n R 2019-05-23 write daily summary          10.6
327 23h starrlab       computer n              reorg notebook to follow      1.9
                                               paper structure
329 19h starrlab       computer n              setup jabref                  1.9
330 19h starrlab       lab n                   setup desk                    1.9
331 19h starrlab       computer n              create agenda field           1.9
332 19h starrlab       computer n              create lit tag                1.9
333 19h starrlab       computer n              create lab context            1.9
336 41s starrlab       computer n              create website knit script    1.9
312 2d  starrlab.ocpmi computer n              organize lab notebook [1]    2.71
315 2d  starrlab.ocpmi computer n              find ovca review re cur      1.91
                                               treatment
316 2d  starrlab.ocpmi computer n              read starr paper re 1st pt   1.91
317 2d  starrlab.ocpmi computer n              read ocpmi summary           1.91
318 2d  starrlab.ocpmi computer n              install surat                1.91
319 2d  starrlab.ocpmi computer w              run surat tutorial w/ pt 59  1.91
320 2d  starrlab.ocpmi computer n              read single cell paper zheng 1.91
                                               2017
321 2d  starrlab.ocpmi computer n              find and read copy num       1.91
                                               variation paper
322 2d  starrlab.ocpmi computer n              organize refs in email inbox 1.91
323 2d  starrlab.ocpmi computer n              compile bibtex ref page      1.91

18 tasks
```


### Citations

#### Custom CSL

At some point, I would like to setup a custom csl file to format my references. The goal here would be to add formatted features that are note readily available in some existing format style. [@2019], [@BibEntry2019May]

#### JabRef

I've started to figure out jabref. I think I will still look to my bibtex file as the primary source for citation management, but will make use of the interface and some of the features of jabref. Additionally dio2bib will play a central role in extracting bibtex content. [@2019b]

### GitHub

The setup for my git server is complete. It's a private server host by the university a /tasta005/ocpmi [@ocpmi_git]

### Summary

I was able to round out much of my organizational setup today. I feel confident about the citation management strategy I plan to use going forward which combines the use of dio2bib for bibtex creation, BibItNow for website extraction, and JabRef to centralize management. Additionally, I was able to connect my relevant directories to my git server, which will account for both a measure of backup and version control. Having gotten settled into these systems I think I will be able to begin incorporating my literature search into my daily activities.

#### Completed


```{r}
Created    Completed  Age Project        R Due        Description
---------- ---------- --- -------------- - ---------- --------------------------
2019-05-22 2019-05-23 1d  starrlab       R 2019-05-22 write daily summary
2019-05-22 2019-05-23 1d  starrlab                    setup desk
2019-05-23 2019-05-23 16h starrlab       R 2019-05-23 write daily summary
2019-05-20 2019-05-23 3d  starrlab.ocpmi              organize lab notebook
                                                        2019-05-21 record,
                                                      master, comms, refs, home,
                                                      docs
2019-05-22 2019-05-23 1d  starrlab                    create lab context
2019-05-22 2019-05-23 1d  starrlab                    create lit tag
2019-05-22 2019-05-23 1d  starrlab                    setup jabref
2019-05-20 2019-05-23 3d  starrlab.ocpmi              compile bibtex ref page
2019-05-20 2019-05-23 3d  starrlab.ocpmi              organize refs in email
                                                      inbox
2019-05-22 2019-05-23 1d  starrlab                    setup umn git
2019-05-22 2019-05-23 1d  starrlab                    create agenda field

11 tasks
```

***

## 24

### TODO


```{r}
ID  Age Project        Tag            Description                           Urg
--- --- -------------- -------------- ------------------------------------- ----
314 3d  starrlab.ocpmi computer lit n find ovca review re cur treatment     2.02
315 3d  starrlab.ocpmi computer lit n read starr paper re 1st pt            2.02
316 3d  starrlab.ocpmi computer lit n read ocpmi summary                    2.02
319 3d  starrlab.ocpmi computer lit n read single cell paper zheng 2017     2.02
320 3d  starrlab.ocpmi computer lit n find and read copy num variation      2.02
                                      paper
317 3d  starrlab.ocpmi computer n     install surat                         1.92
318 3d  starrlab.ocpmi computer w     run surat tutorial w/ pt 59           1.92
325 1d  starrlab       computer n     create website knit script            1.91
327 22h starrlab       comp w         make custom csl                        1.9
328 22h starrlab       lab n          complete lab training                  1.9
330 20h starrlab       w              agenda items                           1.8

11 tasks
```

### create_site Script

I created a small R script that will effectively knit each page of my lab notebook. This can be executed in the R console or in bash, but each file has to be saved. The save -> knit function that's performed by the IDE does not apply to the console executed render command.

```{r}
rmarkdown::render("index.Rmd")
rmarkdown::render("notes.Rmd")
rmarkdown::render("results.Rmd")
rmarkdown::render("methods.Rmd")
rmarkdown::render("references.Rmd")
rmarkdown::render_site("index.Rmd")
```


### Summary

Today was filled with primarily administrative work. A large bulk of my time was taken by lab safety training, which I was unable to fully complete. Along with this, I was finally able to install Seurat and tinker with the package tutorial. It seems like the version I downloaded may not be compatible with the tutorial code, so I will have to explore that more.

#### Completed


```{r}
Created    Completed  Age Project        Description
---------- ---------- --- -------------- ---------------------------------------
2019-05-23 2019-05-24 1d  starrlab       create website knit script
2019-05-24 2019-05-24 2h  starrlab       DEHS Radiation Safety Orientation
                                         Training
2019-05-24 2019-05-24 2h  starrlab       Implementation of NIH Guidelines
2019-05-24 2019-05-24 2h  starrlab       Bloodborne Pathogens: Advanced
2019-05-24 2019-05-24 2h  starrlab       DEHS Chemical Safety Training
2019-05-24 2019-05-24 2h  starrlab       Biological Safety in the Laboratory
2019-05-24 2019-05-24 2h  starrlab       Bloodborne Pathogens: Introduction and
                                         Advanced
2019-05-24 2019-05-24 2h  starrlab       DEHS Introduction to Research Safety
                                         Training
2019-05-24 2019-05-24 2h  starrlab       Animal Use Tutorial
2019-05-23 2019-05-24 1d  starrlab       complete lab training
2019-05-20 2019-05-24 4d  starrlab.ocpmi install surat

11 tasks
```

***

## 27

### TODO


```{r}
ID Age Project        Tags           Description                            Urg
-- --- -------------- -------------- -------------------------------------- ----
77 4d  starrlab       computer w     make custom csl                        1.92
79 3d  starrlab       computer n     GCP Human RSRCH Biomed Studies         1.92
80 3d  starrlab       computer n     CITI research ethics training          1.92
81 3d  starrlab       computer n     University Information Security        1.92
                                     Awareness Training
78 4d  starrlab                      agenda items                           1.02
91 15h starrlab.ocpmi computer lit n read papalexi 2018 [2]                  2.9
68 6d  starrlab.ocpmi computer lit n find ovca review re cur treatment      2.03
69 6d  starrlab.ocpmi computer lit w read starr paper re 1st pt             2.03
70 6d  starrlab.ocpmi computer lit w read ocpmi summary                     2.03
72 6d  starrlab.ocpmi computer lit w read single cell paper zheng 2017      2.03
73 6d  starrlab.ocpmi computer lit w find and read copy num variation paper 2.03
71 6d  starrlab.ocpmi computer w     run surat tutorial w/ pt 59            1.93

12 tasks
```


###[@Lheureux2019]

#### Evolution of Management in the Era of Precision Medicine  (1/2)

* State of field for ovarian cancer
* Current treatments and outcomes
* Ovca presents very late. The internal location of the cancer and general lack of screening options lead to most cases being diagnosed at a later stage than many other cancers would.
* Treatment options have largely focused on surgical debulking and platinum based chemotherapy.
* Risk factors are inherited: germ line mutations in breast cancer susceptible genes (e.g. brca), nulliparity, infertility, endometriosis, obesity, age, potentially perineal talc.
* Reducing factors that have been found to be potentially significant are gravidity, oral contraceptive use, regular use of aspirin, and nsaids.
* Currently, population screening in ovca has not been found to be effective.
* Epithelial ovarian cancer accounts for 95% of ovarian malignancies
* 5 eoc subtypes: high-grade serous, low-grade serous, clear cell, endometrioid, and mucinous
* HGSOC has the highest incidence with 70% of eoc cases.
* It is believed fallopian tube epithelium is the possible site for tumor initiation.
* Mutations in p53 are seen in +90% cases of HGSOC
* p53 mutations and brca inactivation lead to high numbers of copy variation.
* HGSOC has 4 prognostic subtypes: differentiated, immunoreactive, mesenchymal, and proliferative
* Immunoreactive is considered to have the best overall survival while mesenchymal has the worst.
* Ovca cells also over express folate and mesothelin which have been targets for antibody therapy.
* LGSOC is seen in about 10% of cases
* LGSOC is considered to develop in a pathway which is very distinct from HGSOC
* B-Raf and KRAS mutations are the most commonly present in LGSOC
* Hormone therapy is sometimes used due to the presence of estrogen/progesterone receptors.
* CNAs in p53 are rare, and brca inactivation seem to unrelated to LGSOC

### Summary

Just worked my way through the above summary today. Will continue and complete reading this lit tomorrow.


***

## 28

### TODO


```{r}
ID Age Project        Tags           Description                            Urg
-- --- -------------- -------------- -------------------------------------- ----
77 4d  starrlab       computer w     make custom csl                        1.92
79 3d  starrlab       computer n     GCP Human RSRCH Biomed Studies         1.92
80 3d  starrlab       computer n     CITI research ethics training          1.92
81 3d  starrlab       computer n     University Information Security        1.92
                                     Awareness Training
78 4d  starrlab                      agenda items [1]                       1.82
91 1d  starrlab.ocpmi computer lit n read papalexi 2018 [2]                 2.91
92 17h starrlab.ocpmi computer lit w read macintyre 2018 [2]                 2.9
68 7d  starrlab.ocpmi computer lit n find ovca review re cur treatment      2.04
69 7d  starrlab.ocpmi computer lit w read starr paper re 1st pt             2.04
70 7d  starrlab.ocpmi computer lit w read ocpmi summary                     2.04
72 7d  starrlab.ocpmi computer lit w read single cell paper zheng 2017      2.04
73 7d  starrlab.ocpmi computer lit w find and read copy num variation paper 2.04
71 7d  starrlab.ocpmi computer w     run surat tutorial w/ pt 59            1.94

13 tasks
```


### [@Lheureux2019]

#### Evolution of Management in the Era of Precision Medicine (2/2)

* OCCC or clear cell carcinoma represents about 5% of cases.
* The pathogenesis of OCCC is not well understood.
* A proposed mechanism is related to the level of iron content, promoting oxidative stress.
* Mutations in the SWI/SNF and PI3K pathway are represented in large proportions of cases.
* Activation of the PI3K or RTK pathway has resulted in better OS
* OCCCs are found to have considerably fewer CNAs and mutations in p53 or brca genes.
* Endometrioid cancers represent 10% of EOC cases.
* Current thought is that endometrioid and OCCC arise from similar pathways.
* Deregulation of WNT/b-catenin and PI3K pathways are essential in mouse models.
* Mucinous OC is a rare subtype.
* Could be classified as non-invasive and infiltrative.
* KRAS mutations are a common feature of this subtype.
* p53 and brca mutations do not seem to play a role.
* Treatment
* To date, there has been no randomized assessment of debulking surgery, platinum therapy, or radiation therapy.
* Surgery has been the bedrock of therapy.
* Surgery involves many approaches including primary, interval, and secondary debulking
* Chemotherapy
* carboplatin is as effective as cisplatin and is better tolerated.
* dose is important.
* pairing drugs does not improve outcomes.
* IP chemo is often used due to the tight localization of ovca in the abdomen
* hyperthermic IP chemo is currently being explored with reasonable results
* Treatment after recurrence follows several judgments of benefit.
* Recurrence after a period of over 6mo suggests platinum sensitivity where platinum based chemo would be reapplied
* Shorter intervals of recurrence employ non-platinum based therapies
* Current consensus is the use of carboplatin and paclitaxel together every 3wks
* Targeted therapy has been gradually introduced in recent years.
* Bevacizumab has shown the greatest promise among antiangiogenics being the only one in clinical use.
* PARP inhibitors
* Used as maintenance with platinum therapy at the time of recurrence.
* A trial showed that the key benefit in PFS was driven by the parp inhibitor maintenance therapy.
* Trial from 2018 led to first-line maintenance in brca mutated cases
* Stratified approach to systemic therapy can be used with bevacizumab after surgery with a switch to maintenance on the basis of brca mutations.
* Immunotherapy is currently being explored on several fronts though none are in clinical practice.
* ovca has a high expression of PD-L1.
* Checkpoint inhibitors targeting PD-L1/PD1 or CTLA-4 have shown only modest results.
* Autologous cell therapy is also being utilized.
* A variation of this therapy is exploring car-t cells targeting mesothelin
* Vaccines have not worked well due to the highly immunosuppressive cancer environment
* A vaccine targeting survivin is currently being explored
* Additionally, the concept of the patient-specific mutanome is being researched in an effort to develop patient-specific cancer vaccines.
* OCCC and LGSOC are relatively chemoresistant.
* Prevention and screening are currently limited.
* There is no validated screening method for standard practice.
* Ultrasound has produced false-negatives and false-positives and is not recommended.
* Screening high-risk populations has been shown to be effective but requires prior knowledge of family history.
* Some biomarkers have been utilized such as HE4 in combination with CA125.
* Another approach has been to evaluate DNA methylation in cfDNA.










### Summary

I have shifted focus to begin a comprehensive literature review. I've started by taking a look at a review of the current landscape of treatment and outcomes for ovca [@Lheureux2019]. This was helpful to fill many of the holes I have in my general knowledge of cancer biology with the more specific case studies of ovca. Much of this reinforced the review that Tim has given in our meetings, which was helpful. Along with this, I began to work my way through the Seurat tutorial. I was able to locate the v3 tutorial and pbmc data set and made it about half way through the written guide. I plan on getting fully through this tomorrow as well as meeting with Tim to discuss some questions I developed from the ovca review.

#### Completed


```{r}
Created    Completed  Age Project        Description
---------- ---------- --- -------------- ---------------------------------
2019-05-20 2019-05-28 7d  starrlab.ocpmi find ovca review re cur treatment

1 task
```

***

## 29

### TODO

```{r}
ID Age  Project        Tags           Description                           Urg
-- ---- -------------- -------------- ------------------------------------- ----
75 5d   starrlab                      agenda items [3]                      2.03
74 6d   starrlab       computer w     make custom csl                       1.93
76 5d   starrlab       computer n     GCP Human RSRCH Biomed Studies        1.93
77 5d   starrlab       computer n     CITI research ethics training         1.93
78 5d   starrlab       computer n     University Information Security       1.93
                                      Awareness Training
90 1d   starrlab       computer n     find alternative to jabref            1.91
92 22h  starrlab       computer n     fix god damn git issue                 1.9
97 1min starrlab       computer n     fix hostname issue w/ mesabi           1.9
88 2d   starrlab.ocpmi computer lit n read papalexi 2018 [2]                2.91
89 1d   starrlab.ocpmi computer lit w read macintyre 2018 [2]               2.91
91 1d   starrlab.ocpmi lit s          read sahin 2017 [1]                   2.71
66 8d   starrlab.ocpmi computer lit w read starr paper re 1st pt            2.04
67 8d   starrlab.ocpmi computer lit w read ocpmi summary                    2.04
69 8d   starrlab.ocpmi computer lit w read single cell paper zheng 2017     2.04
70 8d   starrlab.ocpmi computer lit w read venteicher - CNV paper           2.04
68 8d   starrlab.ocpmi computer w     run surat tutorial w/ pt 59           1.94
96 2min starrlab.ocpmi computer n     run seurat with pbmc data              1.9

17 tasks
```


### wkly mtg w/ Tim

present: chris, tim

```{r}


Description
--------------------------------------------------------------------------------
agenda items
  2019-05-27 validation in diff between lumpectomy and mastectomy
  2019-05-28 elaborate on lack of randomized controls for ovca treatments
  2019-05-28 discuss vacation
  2019-05-28 discuss chemotherapy at large
  2019-05-28 review parp inhibitor mechanism
  2019-05-28 are occc and lgsoc our general targets given their chemoresistance?
  2019-05-28 how do our questions differ from the findings of kondrashova 2017 -
RAD51 mutations acquired resistance

1 task
```

#### mtg notes

* mysql reference: single cell rnaseq analysis protocol
* figure out mesabi hostname issue
* explore starrt2 MSI path
    + [/home/starrt2/data_release/umgc/hiseq/170627_D00635_0259_ACB8R1ANXX/Starr_Project_037/Analysis/cellranger-Starr_037/filtered_gene_bc_matrices/GRCh38]
* There is no specific focus in the pathology we're focusing on. OCCC and LGSOC are being considered just as readily as other subtypes.
* Kondrashova 2017 (RAD51 mutations leading to chemoresistance) is potentially a good example of what our research questions will be.
* Tim's focus is more basic science oriented. Potential applications would be to identify when a pt would be resistant and that they should pursue other treatment from the outset.
* All of our findings from this data set will require wetlab validation
* This data set is incredibly unique. We should be able to discover something nobody else has.
* Data validation has generally been handled on the project so far.
    + Objective: determine cell type.
* There's no way to call copy number variation from this data. That would be novel.

### MSI Project Directory

I logged into the MSI account for the first time today. There is a relatively deep file structure which is composed of a combination of processed data from umgc/MSI and other collaborator work. I have a personal directory in this space along with access to other shared directories.

The root project directory is:

```{r}
/home/starrt2/
```

I have a personal directory under the root. At the same level, there is also a shared directory which includes some content that is under active collaboration.

```{r}
/home/starrt2/tasta005
```

Processed data can be found in the hiseq folder under data_release. This data is mirrored in the MSI file structure. Each file set is named by a unique ID which follows a naming convention given by umgc.

* Look for the leading 6 digit integer as a point of reference for the data_release folder.
* The Starr_Project sub folder is given a number which is considered the MSI and sample ID.

```{r}
/home/starrt2/data_release/umgc/hiseq
```


The Analysis directory contains the source data

* illumina-basicQC # quality control data
* cellranger-`*`
    + analysis # ?
    + filtered_gene_bc_matricies # data source that has been filtered by "true cell content"
    + raw_gene_bc_matrices # matrix data that includes unfiltered reads or all read content
    + web_summary.html # interactive cell ranger summary

An example of a complete data source location is below:

```{r}
/home/starrt2/data_release/umgc/hiseq/170627_D00635_0259_ACB8R1ANXX/Starr_Project_037/Analysis/cellranger-Starr_037/filtered_gene_bc_matrices/GRCh38
```

### Summary

I met with Tim today for our weekly meeting. We discussed some aspects of ovca content that I had been reading to develop my general knowledge. I major element of this conversation that will adjust what my intentions were going forward dealt with the nature of statistically validating the results of our pipeline. My original thought was that there was some need to dig into the backend of the Seurat/cellranger packages to both confirm and optimize the output. It sounds as though this has been done to a degree that Tim is satisfied with and that we can move on to explore questions that are Biological in nature.

Additionally, I did a little house keeping by establishing the first part of the methods page in my notebook. I explored the MSI project directory for the first time today and recorded some documentation about the file structure and data locations. My expectation is to pull what would be method oriented documentation from Notes -> Methods or Notes -> Results as these things need to be pinned for repeat use and reference.

#### Completed

```{r}
Created    Completed  Age Project  Description
---------- ---------- --- -------- ----------------------------
2019-05-28 2019-05-29 1d  starrlab fix god damn git issue
2019-05-29 2019-05-29 2h  starrlab fix hostname issue w/ mesabi

2 tasks
```

***

## 30

### TODO


```{r}
ID Age Project        Tags           Description                            Urg
-- --- -------------- -------------- -------------------------------------- ----
75 6d  starrlab                      agenda items [3]                       2.03
74 6d  starrlab       computer w     make custom csl                        1.93
76 5d  starrlab       computer n     GCP Human RSRCH Biomed Studies         1.93
77 5d  starrlab       computer n     CITI research ethics training          1.93
78 5d  starrlab       computer n     University Information Security        1.93
                                     Awareness Training
90 1d  starrlab       computer n     find alternative to jabref             1.91
88 3d  starrlab.ocpmi computer lit n read papalexi 2018 [2]                 2.92
89 2d  starrlab.ocpmi computer lit w read macintyre 2018 [2]                2.91
91 1d  starrlab.ocpmi lit s          read sahin 2017 [1]                    2.71
66 9d  starrlab.ocpmi computer lit w read starr paper re 1st pt             2.05
67 9d  starrlab.ocpmi computer lit w read ocpmi summary                     2.05
69 9d  starrlab.ocpmi computer lit w read single cell paper zheng 2017      2.05
70 9d  starrlab.ocpmi computer lit w read venteicher - CNV paper            2.05
68 9d  starrlab.ocpmi computer w     run surat tutorial w/ pt 59            1.95
95 20h starrlab.ocpmi computer n     run seurat with pbmc data               1.9

15 tasks
```

### Seurat Tutorial pbmc3k

I ran the v3 Seurat tutorial [@2019d] to completion using the pbmc dataset. This outline was very intuitive and worked through the pipeline seamlessly. There are several points that warrant further exploration.

The output of the v3 PBMC data tutorial can be found [here](/home/chris/starr_lab/ocpmi/code/seurat_tutorial/pbmc3k_tutorial.html).


### Summary

Briefly, I wrapped up safety training today, so that is all taken care of. The rest of my afternoon was dedicated to working my way through the default v3 Seurat tutorial. The tutorial comes packaged in a markdown file, so the presentation and documentation is very effective. To some extent, the documentation is so good I am a little concerned it encourages over use of the default settings. I haven't looked at any other processing that's been completed by Tim et al, so I'll have to keep an eye out for how adjustments have been made. Tim mentioned there has been a limited amount of customization, so this may stand as a short term area of focus. Generally, I think can consider today's work a reasonable summary of the process and syntax associated with the toolset. With this, I would like to spend some time digging into the nuances of the pipeline's analysis. I will probably spend most of tomorrow doing that. Much of the default analysis hinges on PCA dimensionality, so I feel I need to brush up on my knowledge in this area. Going forward I need to run the same pipeline using the new pt59 data and evaluate results.

#### Completed


```{r}
Created    Completed  Age Project        Description
---------- ---------- --- -------------- ---------------------------------------
2019-05-24 2019-05-30 6d  starrlab       GCP Human RSRCH Biomed Studies
2019-05-24 2019-05-30 6d  starrlab       CITI research ethics training
2019-05-24 2019-05-30 6d  starrlab       University Information Security
                                         Awareness Training
2019-05-29 2019-05-30 1d  starrlab.ocpmi run seurat with pbmc data

4 tasks
```

***

## 31

### TODO

```{r}
ID Age Project        Tags       Description                                Urg
-- --- -------------- ---------- ------------------------------------------ ----
73 8d  starrlab                  agenda items [3]                           2.04
72 8d  starrlab       computer w make custom csl                            1.94
83 3d  starrlab       computer n find alternative to jabref                 1.92
91 1d  starrlab       computer n review pca concepts                        1.91
92 1d  starrlab       computer n compare default seurat tutorial to         1.91
                                 starrlab protocol
93 1d  starrlab       computer n deep dive into default seurat analysis     1.91
94 1d  starrlab       computer n update notes regarding citation tools re   1.91
                                 mendeley
81 4d  starrlab.ocpmi lit n      read papalexi 2018 [2]                     2.82
82 4d  starrlab.ocpmi lit w      read macintyre 2018 [2]                    2.82
84 3d  starrlab.ocpmi lit s      read sahin 2017 [1]                        2.72
65 11d starrlab.ocpmi lit w      read starr paper re 1st pt                 1.96
66 11d starrlab.ocpmi lit w      read ocpmi summary                         1.96
67 11d starrlab.ocpmi computer w run surat tutorial w/ pt 59                1.96
68 11d starrlab.ocpmi lit w      read single cell paper zheng 2017          1.96
69 11d starrlab.ocpmi lit w      read venteicher - CNV paper                1.96

15 tasks
```

### Mendeley

At the risk of seeming indecisive, I've decided to switch citation managers. Given my use if the BibTex file format, it is effectively painless to switch. JabRef showed some promise but presented a few bugs related to database calls which would render much of the value of using it worthless. Mendeley has both a native linux application along with a web import tool for firefox. This is the only major citation tool I've come across that combines those two things. Here are a few key features of my use:

* All content will still be synced back to a BibTex file name `library.bib` located under the `starr_lab/ocpmi/notebook/` subdirectory
* There are watched folders which will draw sources from any pdf put into them at `starr_lab/ocpmi/ref/`.
* I've decided to use the IEEE with url csl style taken from the zotero style repository. [@2019]

### [@Winterhoff2019]

#### Single-Cell Sequencing in Ovarian Cancer: a new frontier in precision Medicine

Key Points

* Intratumor heterogeneity is an inherent property of ovarian cancer, which impacts clinical outcomes
* SC sequencing tech enables characterization of intratumor heterogeneity in ovarian cancer
* To date, single-cell studies on ovarian cancers are scarce, which limits the opportunity to investigate effects on clinical outcomes
* Larger data sets are needed to establish potential associations between the unique information captured by SC sequencing and clinically relevant outcomes in ovca

Notes

* They claim they found genes associated with ovca stem cells.
* When they evaluated based on molecular eoc cancer subtypes, all subtypes were present.
* Studying the immune microenvironment of a tumor becomes more feasible with scRNA sequencing
* Evaluation of cell plasticity as it relates to cell type transitions may provide insight into the development of chemoresistance
* Understanding the lineage of the cellular qualities of metastatic dissemination is made possible with scRNA data

### Using Ancestry to find the Stem Cell

Question: can we use sequence-based lineage tracing to identify a common progenitor cell expression profile?

A thought I had while looking at some lit tonight led me to wonder if it's worth exploring sequence-based cell lineage across individual cells rather than using differential expression patterns as a general means of characterization. I will need to do a dedicated lit search to explore this concept, but I imagine the use of single cell data for lineage tracing is relatively limited. At a minimum, Tim has alluded to the notion that our data set is unique among the ovca model, which would lend itself to a novel application of single cell lineage tracing. Regardless, this could be a relatively significant deviation from their current pipeline, so I will have to discuss it with Tim.

### Updated create_site.R Script

I made a minor update to the script I use to automate total site rendering to clean up the directory a bit. I don't have any use in the parent notebook/ directory for individual html files since they are generated in the site render. The `ifelse` at the end of the script clears these html files from the notebook directory.

```{r}
rmarkdown::render("index.Rmd")
rmarkdown::render("notes.Rmd")
rmarkdown::render("results.Rmd")
rmarkdown::render("methods.Rmd")
rmarkdown::render("references.Rmd")
rmarkdown::render_site("index.Rmd")

fl <- Sys.glob("*.html")

ifelse(file.exists(fl), file.remove(fl))
```

### Summary

I was able to explore a few different fronts today. I put some more work in to polishing my citation management. This is an area I have been woefully lacking in from past experience, so I do not discount the extra work applied to make sure this is at a publication-ready level.

Additionally, I worked my way through the earliest publication related to the ocpmi project [@Winterhoff2019]. Through this I was able to develop some insight to the early considerations this group gave to this data as they were only working with 66 individual cells. They offered a broad scope of the advantages single cell tech offers along with pointing me to a few sources that establish the current state of the field.

Last, I can't say what generated this thought specifically, but I came across what my limited knowledge of the field would call a good idea. All of our discussions on this data set up to this point have centered on the use of UMI counts and gene expression levels as a means to develop differentiated cell profiles. A consideration I would potentially like to explore is the possibility that we look at sequence level data to identify cell lineage in search of a common progenitor. I have little basis to say if this is a reasonable or even realistic scientific question, but I feel this could be a substantive, novel insight given the nature of our data set. It's quite possible this approach would require too great an investment in shifting the informatics work toward raw sequence analysis, but I will discuss it with Tim nonetheless.

#### Completed

```{r}
Created    Completed  Age Project        Description
---------- ---------- --- -------------- --------------------------
2019-05-20 2019-06-01 11d starrlab.ocpmi read starr paper re 1st pt
2019-05-28 2019-05-31 3d  starrlab       find alternative to jabref
2019-05-30 2019-05-31 1d  starrlab       update notes regarding citation tools re
                                         mendeley
2019-05-23 2019-05-31 8d  starrlab       make custom csl

4 tasks
```


***

# June

## 3

### TODO

```{r}
ID Age Project        Tags       Description                                Urg
-- --- -------------- ---------- ------------------------------------------ ----
52 10d starrlab                  agenda items [2]                           1.95
74 2d  starrlab       computer n add notes about cancer stem cell database  1.91
49 13d starrlab.ocpmi lit w      read tirosh 2016 [2]                       2.87
56 7d  starrlab.ocpmi lit n      read papalexi 2018 [2]                     2.84
57 6d  starrlab.ocpmi lit w      read macintyre 2018 [2]                    2.83
58 5d  starrlab.ocpmi lit s      read sahin 2017 [1]                        2.73
67 2d  starrlab.ocpmi lit n      do lit search on cell ancestry [1]         2.71
46 13d starrlab.ocpmi lit w      read ocpmi summary                         1.97
47 13d starrlab.ocpmi computer w run surat tutorial w/ pt 59                1.97
48 13d starrlab.ocpmi lit w      read single cell paper zheng 2017          1.97
63 3d  starrlab.ocpmi computer n review pca concepts                        1.92
64 3d  starrlab.ocpmi computer n compare default seurat tutorial to         1.92
                                 starrlab protocol
65 3d  starrlab.ocpmi computer w deep dive into default seurat analysis     1.92
69 2d  starrlab.ocpmi lit n      find cancer stem cell review               1.91

14 tasks
```

### Cancer Stem Cell Database

http://www.csctt.org/ [@csctt2019]

Tim sent me a link to a possible resource last week that could be useful in building references for exploring the cancer stem cell concept.

### LitSrch re Cancer Stem Cell

I've done some extremely high level review from the wikipedia article on CSCs [@csc_wiki]. This pointed me to several papers of relevance that I hope to look at in future exploration of the topic.

#### Notes

* While there has been significant research around the topic, CSCs remain a debated idea.
* There are currently a few models for CSC tumor propagation
    + The CSC model
        - This is also known as the hierarchical model
        - CSCs are a tumorigenic and biologically distinct subpopulation [@Beck2013].
        - Two defining features: ability to self-renew, capacity to generate pro-growth progeny
        - This model suggests only certain cells have the ability to drive the progression of cancer
    + Stochastic model
        - Also known as clonal evolution model
        - This model suggests the necessary mutations to achieve CSC status can happen to any cell at random.
        - All tumor cells retain the ability to develop plasticity and self-renew/differentiate [@Kreso2014].
        - Elements of resistance and fitness accumulate over time [@Beck2013].
        - CSCs with a growth advantage outperform other cells [@Barabe2007].
    + CSC Immunology model
        - Both stem cell and CSCs possess qualities that make them resistant to immune surveillance [@MacCalli2014].
        - CSCs may exist at all times but only propagate once 1) they shed dependence on niche factors or 2) they develop a means to escape immune surveillance or 3) the immune system loses tumorsuppressive capacity [@Bruttel2014].
* Debate
    + The existence of CSCs is still under debate because no consistently definable characteristics have been found [@Gupta2009].
    + The origin of CSCs is also a major topic.
        - Are these disregulated normal SCs
        - Are they instead a more specialized population that gained the ability to self-renew.
        - Is it something else entirely due to the modular plasticity cancer cells can show [@Nouri2017].
* Modeling
    + There has been some mathematical modeling taken to the growth of these cells.
    + The cell compartment method is one approach [@Ganguly2006].
    + Other models may exist and are worth exploring in the context of highlighting CSC mutation lineage.
* Origin
    + stem cell mutation
        - stem cell niche populations develop mutations which derail regular progression and can be past to progeny [@Wang2009].
    + adult stem cells
        - Stem cell populations in tissue with high rates of turnover experience frequent cell divisions. Increased likelihood of mutations leads to accumulated genetic defects that drive cancer initiation. [@Lopez-Lazaro2018]
    + De-differentiation
        - Cells are driven back to a state of stemness through some external factor.

### Summary

I focused mostly on lit search efforts today. I want to develop a better understanding of the current landscape of the CSC hypothesis, so I have taken a broad overview of different findings. I started with the wikipedia article on the subject and have branched out to several other major papers in the field. A few points of interest are the controversies around origin and characterization. There seems to be a relatively wide interpretation of CSCs in terms of typing and behavior. This is obviously due to the relatively abstract nature of stemness and cell development. Talking with Tim today, there are some limitations since we only have expression data. One consideration is that our pipeline is using a known reference sequence database to process the output. Given this, there may be unknown mutations that could serve as markers of both CSC identity and possible lineage association. Screening for these in a cell specific way could bring about interesting findings.

#### Completed

```{r}
Created    Completed  Age Project        Description
---------- ---------- --- -------------- ---------------------------------------
2019-06-01 2019-06-03 2d  starrlab       add notes about cancer stem cell
                                         database
2019-06-01 2019-06-03 2d  starrlab.ocpmi litsrch re cancer stem cell review

2 tasks
```

***

## 4

### TODO


```{r}
ID Age Project Tags   Description                                           Urg
-- --- ------- ------ ----------------------------------------------------- ----
49 2w  lab     lit    read tirosh 2016 [2]                                  2.78
55 8d  lab     lit    read papalexi 2018 [2]                                2.74
56 7d  lab     lit    read macintyre 2018 [2]                               2.74
48 2w  lab     lit    read zheng 2017 [1]                                   2.68
57 7d  lab     lit    read sahin 2017 [1]                                   2.64
65 3d  lab     lit    litsrch re genetic lineage tracing method [1]         2.62
52 11d lab            agenda items [3]                                      2.06
47 2w  lab     comp w run surat tutorial w/ pt 59                           1.98
62 5d  lab     comp n review pca concepts                                   1.93
63 4d  lab     comp n compare default seurat tutorial to starrlab protocol  1.92
64 4d  lab     comp w deep dive into default seurat analysis                1.92
46 2w  lab     lit    read ocpmi summary                                    1.88
77 20h lab     lit    find list of CSC markers                               1.8

13 tasks
```

### Summary

I didn't get a lot done today. I began to apply the vanilla Seurat analysis to the pt59 data set and found myself encountering many of the setup hurdles that come with using a remote system. I have to do some retooling so my environment can seamlessly run on MSI.

***

## 5

### TODO

```{r}
ID Age  Project Tags   Description                                          Urg
-- ---- ------- ------ ---------------------------------------------------- ----
49 2w   lab     lit    read tirosh 2016 [2]                                 2.78
55 9d   lab     lit    read papalexi 2018 [2]                               2.75
56 8d   lab     lit    read macintyre 2018 [2]                              2.74
76 2d   lab     lit    read konrashova 2017 [2]                             2.71
48 2w   lab     lit    read zheng 2017 [1]                                  2.68
57 8d   lab     lit    read sahin 2017 [1]                                  2.64
65 4d   lab     lit    litsrch re genetic lineage tracing method [1]        2.62
47 2w   lab     comp w run surat tutorial w/ pt 59                          1.98
62 6d   lab     comp n review pca concepts                                  1.93
63 5d   lab     comp n compare default seurat tutorial to starrlab protocol 1.93
64 5d   lab     comp w deep dive into default seurat analysis               1.93
84 20h  lab     comp n figure out remote vim setup                           1.9
89 1min lab     comp n send vacation dates to tim                            1.9
46 2w   lab     lit    read ocpmi summary                                   1.88
77 1d   lab     lit    find list of CSC markers                             1.81
52 12d  lab            agenda items                                         1.07

16 tasks
```


### wkly mtg w/ Tim

present: chris, tim

```{r}
Description
--------------------------------------------------------------------------------
agenda items
  2019-05-28 discuss vacation
  2019-06-01 can we use sequence-based lineage tracing to identify a common
progenitor cell expression profile
  2019-06-03 is the JGO article just a couple poster abstracts?
  2019-06-05 discuss seurat output

1 task
```

#### mtg notes

* We discussed the nature of the Seurat output to some degree
    + I had been wondering how Tim came about the specifications to the metrics he used in his Seurat analysis file.
    + He has largely used the default params other outside of a few specific cases
    + These specifications and their justifications are all detailed in the Single Cell RNAseq Analysis protocol document in my `methods` folder.
    + One thing he mentioned would be an area I could assist him with is the way we fine tune these params.
    + Isaac is working on this to some extent by evaluating the 3 methods available for `Identification of highly variable features`.
        - So far, he's found that there is a notable difference in the genes identified, but the clustering output is relatively similar.
    + Tim has created a custom analysis specific to our data set that applied cell typing to clusters.
    + Additionally, we discussed the broad strokes of concepts behind the idea of CSCs. There is a great deal of ambiguity here, and we didn't really arrive at any objective conclusions. To some extent, I just needed to work through the ideas with someone else. I will need to really formulate a sense of this for any comprehensive experimentation which leverages this concept.
    + This was a subpoint but an important realization: **the 10x do not include full coverage.** This was a mistaken assumption of mine in thinking we had more complete expression coverage. Instead, we are working with ~100bp lengths with poly-A ranges. It may be worth digging into this to make sure I understand the specifics.

### Remote MSI Setup

I've reached somewhat of a hang up in my development environment. I've yet to work in centos, and I'm generally unfamiliar with the MSI setup. So far, I've been able to make sense of my home directory in my personal instance and load the R console. Additionally, I've been able to get the installation of packages going and have a process running over night in tmux.

The challenge I see stems from the combination of developing in an Rstudio-like environment that allows for the execution of code snippets while being executed in the MSI environment. What I am looking at as I wrap the day is using Rserve on my MSI instance to connect to my local dev environment.




### Rserve

I've found the package `Rserve` which can establish a servable instance of R on a remote server. This holds some potential to allow me to develop locally and execute any part of my code in the MSI instance. There is documentation at the link below. One necessary element of making this work would be to find a suitable Rserve client that can run locally and communicate with the server.

https://www.rforge.net/Rserve/doc.html



https://www.msi.umn.edu/support/faq/how-can-i-install-r-packages-my-home-directory

https://satijalab.org/seurat/install.html

### Summary

I met with Tim today for our weekly meeting. We were able to discuss things that were mostly conceptual to aid in my continued content exploration. The rest of my time was spent getting a sense for the MSI environment and how I will be interacting with it. My hope is to actually develop in this space full time rather than just schedule jobs. My own computer is limited in its processing power, and using my MSI instance would be highly preferred. Still, there are some hurdles to doing this that I haven't dealt with previously, so I have been working to link my current setup to the operational environment of MSI. This will probably continue to take some time, but I hope to be fully ready tomorrow.

***

## 6

### TODO


```{r}
ID Age Project Tags   Description                                           Urg
-- --- ------- ------ ----------------------------------------------------- ----
49 2w  lab     lit    read tirosh 2016 [2]                                  2.79
54 10d lab     lit    read papalexi 2018 [2]                                2.75
55 9d  lab     lit    read macintyre 2018 [2]                               2.75
74 3d  lab     lit    read konrashova 2017 [2]                              2.72
48 2w  lab     lit    read zheng 2017 [1]                                   2.69
56 8d  lab     lit    read sahin 2017 [1]                                   2.64
63 5d  lab     lit    litsrch re genetic lineage tracing method [1]         2.63
47 2w  lab     comp w run surat tutorial w/ pt 59                           1.99
60 6d  lab     comp n review pca concepts                                   1.93
61 6d  lab     comp n compare default seurat tutorial to starrlab protocol  1.93
62 6d  lab     comp w deep dive into default seurat analysis                1.93
82 1d  lab     comp n figure out remote vim setup                           1.91
87 22h lab     comp n send vacation dates to tim                             1.9
88 18h lab     comp n update methods on msi dev setup                        1.9
89 18h lab     comp n spell check notebook                                   1.9
46 2w  lab     lit    read ocpmi summary                                    1.89
75 2d  lab     lit    find list of CSC markers                              1.81
51 13d lab            agenda items                                          1.07

18 tasks
```


### MSI Workflow

After a considerable amount of effort to create a synchronous workflow with my MSI instance, I've generally given up and settled on scripting locally and batching with MSI. I've mounted my home directory with sshfs and clone my ocpmi git repo so I can work out of my MSI instance for the most part. My only concern is if I run into resource intensive cases that require more ram. It does seem that MSI kills unbatched jobs that exceed a threshold of resource use since my process was killed each time it went over 10gb of ram.

### Notes to add

* git repo info
* sshfs setup and bash script
* msi setup info
* seurat install and docs methods

### Summary

I've wasted a large amount of time over the past two days. I spent a solid 10 hours trying to get my workflow in MSI established all to essentially go with what would have been my setup had I done nothing at all. I will generally resort to developing on my local machine with a connection to my MSI directory. Beyond this, I will need to take a look at the batching process for MSI and possible explore some of their training resources to gain a better understanding.

#### Completed

```{r}
Created    Completed  Age Project Description
---------- ---------- --- ------- -------------------------------
2019-06-05 2019-06-06 1d  lab     send vacation dates to tim
2019-06-04 2019-06-06 1d  lab     figure out remote vim setup
2019-06-06 2019-06-06 2h  lab     clone git repo to msi directory
2019-06-06 2019-06-06 2h  lab     exchange ssh keys with msi

4 tasks
```

***

## 10

### TODO

```{r}
ID Age Project Tags     Description                                         Urg
-- --- ------- -------- --------------------------------------------------- ----
48 2w  lab     lit work read tirosh 2016 [2]                                2.91
53 2w  lab     lit work read papalexi 2018 [2]                              2.88
54 13d lab     lit work read macintyre 2018 [2]                             2.87
70 7d  lab     lit work read konrashova 2017 [2]                            2.84
47 2w  lab     lit work read zheng 2017 [1]                                 2.81
55 13d lab     lit work read sahin 2017 [1]                                 2.77
61 9d  lab     lit work litsrch re genetic lineage tracing method [1]       2.75
45 2w  lab     lit work read ocpmi summary                                  2.01
46 2w  lab     w work   run surat tutorial w/ pt 59                         2.01
58 11d lab     n work   review pca concepts                                 1.96
59 10d lab     n work   compare default seurat tutorial to starrlab         1.95
                        protocol
60 10d lab     w work   deep dive into default seurat analysis              1.95
71 6d  lab     lit work find list of CSC markers                            1.93
80 4d  lab     n work   update methods on msi dev setup                     1.92
81 4d  lab     n work   spell check notebook                                1.92
83 3d  lab     n work   add notes re sshfs bash script                      1.92
50 2w  lab     work     agenda items                                        1.89

17 tasks
```

### Non-contemporaneous Notes - 6/6

#### Load/Install R pkgs

* In order to run R in my MSI instance, I need to load the module and install applications.

```{r}
module load R/3.6.0
R
```

* To install packages in my local home directory, need to define a path.

```{r}
.libPaths(new='~/R/x86_64-pc-linux-gnu-library/3.6.0')
dir.create('~/R/x86_64-pc-linux-gnu-library/3.6.0', showWarnings = FALSE, recursive = TRUE)
```

* Confirm the path is correct.

```{r}
.libPaths()
```

* Choose as CRAN mirror.

```{r}
getCRANmirrors()
chooseCRANmirror(ind=2)
```

* Packages should be able to be installed and loaded normally now.

#### sshfs Setup

I'm using `sshfs` to mount my MSI home folder on my local machine to make it more readily accessible. I already have ssh keys exchanged with MSI, so I can create an executable auto-mount bash script. This is located at `/home/chris/starr_lab/auto_mnt_sshfs` on my local machine.

* Auto-mount MSI sshfs

```{bash eval=F}
#!/bin/bash

sshfs tasta005@login.msi.umn.edu: ~/starr_lab/msi_mnt
```

### Summary

I made some more progress on developing a base implementation of the default Seurat analysis pipeline when evaluating the pt59 dataset. To a degree, I am mostly taking default params using the PBMC tutorial and running them with the new dataset. Outside of the default params, I have been using the file located at `/ocpmi/code/tim/seurat_analysis.R` as a reference to see what Tim has modified thus far. There are certain points where he has taken the base value and others where his code deviates heavily. The general flow of his implementation differs from the tutorial in the sense that his code is less step wise and more focused on the output while the tutorial is a bit more exploratory to exhibit features. This makes sense of course. An aspect of my own implementation will have to consider how distributed each analysis process to provide modularity.

A major point I came across today was that I forgot to review the RNAseq protocol located at `/ocpmi/methods/`. Tim and I had discussed this in our meeting last week, but I forgot to make a note about this. There is a good deal of documentation for his code and methods there, so I probably should have remembered to look at this more. I will dedicate some time tomorrow to do this.

Overall, I will spend most of tomorrow continuing to review the documentation and code for Tim's pipeline along with getting a sense of the veracity of my pt59 output.

#### Completed

```{r}
Created    Completed  Age Project Description
---------- ---------- --- ------- -------------------------------
2019-06-05 2019-06-10 4d  lab     spell check notebook
2019-06-05 2019-06-10 4d  lab     update methods on msi dev setup
2019-06-06 2019-06-10 3d  lab     add notes re sshfs bash script

3 tasks
```

***

## 11

### TODO

```{r}
ID Age Project Tags     Description                                         Urg
-- --- ------- -------- --------------------------------------------------- ----
48 3w  lab     lit work read tirosh 2016 [2]                                2.92
50 2w  lab     work     agenda items [3]                                     2.9
53 2w  lab     lit work read papalexi 2018 [2]                              2.88
54 2w  lab     lit work read macintyre 2018 [2]                             2.88
70 7d  lab     lit work read konrashova 2017 [2]                            2.84
47 3w  lab     lit work read zheng 2017 [1]                                 2.82
55 13d lab     lit work read sahin 2017 [1]                                 2.77
61 10d lab     lit work litsrch re genetic lineage tracing method [1]       2.75
45 3w  lab     lit work read ocpmi summary                                  2.02
46 3w  lab     w work   run surat tutorial w/ pt 59                         2.02
58 11d lab     n work   review pca concepts                                 1.96
59 11d lab     n work   compare default seurat tutorial to starrlab         1.96
                        protocol
60 11d lab     w work   deep dive into default seurat analysis              1.96
71 7d  lab     lit work find list of CSC markers                            1.94
89 18h lab     n work   evaluate list of highly var features pt59            1.9
91 18h lab     errand n bring monitor to work                                1.9
92 17h lab     n work   review rnaseq analysis protocol                      1.9
95 20s lab     n work   email tim re mtg resched                             1.9

18 tasks
```

### RNA Velocity

I came a cross a [thread](https://www.reddit.com/r/bioinformatics/comments/bz292v/_/) on the bioinformatics subreddit discussing the issues of the day for single cell analysis. One of the points that was discussed was the challenge around merging data sets and the way this might regress some relevant artifacts of the data. The concept of RNA velocity was brought up as a means to address this, and it may hold some relevance to how we analyze our data. [@LaManno2018]

### Render .R -> .html

I made an amazing discovery today. It is possible to render markdown output straight from a .R file. I'm not sure if I can apply certain yaml or formatting settings to this process yet, but this function alone will save the need to create two separate files when running code and producing a formatted output.

https://rmarkdown.rstudio.com/articles_report_from_r_script.html

A markdown report can be compiled from an R script by:

```{r}
rmarkdown::render("analysis.R") # will generate an html output
rmarkdown::render("analysis.R", "pdf_document")
```

The metadata and output can be modified through special comments.

```{r}
#' ---
#' title: "Crop Analysis Q3 2013"
#' author: "John Smith"
#' date: "May 3rd, 2014"
#' output: pdf_document
#' ---
```

### Summary

My day consisted of a continued focus on developing a complete working model of the general Seurat pipeline through the lens of the pt59 data. I haven't done much in terms of evaluation for the output and will need to dig into this more. The highly variable features and linear dimensional reduction are items I need to both confirm for efficacy and make sense of in their findings. Additionally, I merely clustered the data along 13 dimensions and did not apply any further analysis such as the exploration of highly variable features in the umap'd data.

I did make some unintended progress on formatting and organization of code after discovering that I can render markdown directly from an R script file. This is what I was striving for to some degree in my MSI setup from last week, so it is a relief to arrive at a workable solution.

Last, I plan on taking a bit of a dive into the concept of RNA velocity to see if it holds any relevance to evaluating our data set. As I have become acclimated to the content of this field, I have felt there is some aspect of single cell analysis that the group has not considered to explore concepts beyond cell typing such as CNV, cell lineage, and stem cell character. I don't say that with the authority to know if it is the case, but the adolescence of this field draws the general assumption.

#### Completed

```{r}
Created    Completed  Age Project Description
---------- ---------- --- ------- ---------------------------
2019-06-11 2019-06-11 5h  lab     email tim re mtg resched
2019-05-20 2019-06-11 3w  lab     run surat tutorial w/ pt 59

2 tasks
```

***

## 12

### TODO

```{r}


ID Age Project Tags     Description                                         Urg
-- --- ------- -------- --------------------------------------------------- ----
46 3w  lab     lit work read tirosh 2016 [2]                                2.92
48 2w  lab     work     agenda items [3]                                     2.9
51 2w  lab     lit work read papalexi 2018 [2]                              2.89
52 2w  lab     lit work read macintyre 2018 [2]                             2.88
68 9d  lab     lit work read konrashova 2017 [2]                            2.85
45 3w  lab     lit work read zheng 2017 [1]                                 2.82
53 2w  lab     lit work read sahin 2017 [1]                                 2.78
59 11d lab     lit work litsrch re genetic lineage tracing method [1]       2.76
90 1d  lab     lit work read manno 2018 [1]                                 2.71
44 3w  lab     lit work read ocpmi summary                                  2.02
56 13d lab     n work   review pca concepts                                 1.97
57 12d lab     n work   compare default seurat tutorial to starrlab         1.97
                        protocol
58 12d lab     w work   deep dive into default seurat analysis              1.97
69 8d  lab     lit work find list of CSC markers                            1.94
85 1d  lab     n work   evaluate list of highly var features pt59           1.91
87 1d  lab     errand n bring monitor to work                               1.91
88 1d  lab     n work   review rnaseq analysis protocol                     1.91
93 19h lab     n work   perform diff exp analysis on non-lin red pt59        1.9
                        output

18 tasks
```


### sshfs Unmount

I've run into the case where a server disconnect causes a hang any time I navigate to this mount point. To reset the mount point use `fusermount` to force the unmount first.

```{r}
fusermount -uz /data
```

### MSI Interactive Computing

Running all of my work locally has shown to be more resource consuming than originally anticipated. This has developed somewhat of a bottleneck, so I'm hoping I can transfer the larger rendering jobs to MSI so they can process in the background. A limitation in doing this is developing the ability to generate markdown since this would be the best output. It isn't necessary, but I would prefer this. The hangup seems to be the lack of a graphics environment. Obviously, a headless server will not have something like this loaded by default, but I have to see if I can get that working.

### RNA Velocity

I began to gloss over this paper today [@LaManno2018]. The concept is both relatively simple in general terms yet includes a good deal of complexity since it involves RNA splicing. I need to read further, but what I've read so far lends itself to the potential for some application in our data set. I'll have to discuss with Tim more tomorrow.

### pt59 analysis

I continued to round out my general analysis of the pt59 data. This analysis was furthered with my original approach of being blind to the Biological relevance. I've been able to generate outputs that follow the base pipeline but will need to evaluate markers specific to our cell types and genetic markers of interest.

### Summary

Today included some light work around finishing off my pt59 analysis. The code I've put together generally mirrors the PBMC tutorial with some minor variations. Key to this general approach is that I haven't customized the marker selection. This would be a good discussion with Tim in how he has identified this in the past in a consistent manner. Additionally, my computer is becoming more limited with every process I run. It may be worth breaking these things into smaller chunks of analysis to account for my resource limitations, but I'm going to return to getting things working on MSI first.

#### Completed

```{r}
Created    Completed  Age Project Description
---------- ---------- --- ------- ----------------------------------------------
2019-06-10 2019-06-12 2d  lab     bring monitor to work
2019-05-30 2019-06-12 13d lab     compare default seurat tutorial to starrlab
                                  protocol
2019-06-11 2019-06-12 1d  lab     perform diff exp analysis on non-lin red pt59
                                  output

3 tasks
```

***

## 13

### TODO

```{r}

ID Age Project Tags     Description                                   Urg
-- --- ------- -------- --------------------------------------------- ----
46 3w  lab     lit work read tirosh 2016 [2]                          2.93
48 2w  lab     work     agenda items [6]                              2.91
51 2w  lab     lit work read papalexi 2018 [2]                        2.89
52 2w  lab     lit work read macintyre 2018 [2]                       2.89
66 9d  lab     lit work read konrashova 2017 [2]                      2.85
45 3w  lab     lit work read zheng 2017 [1]                           2.83
53 2w  lab     lit work read sahin 2017 [1]                           2.78
58 12d lab     lit work litsrch re genetic lineage tracing method [1] 2.77
84 1d  lab     lit work read manno 2018 [1]                           2.71
44 3w  lab     lit work read ocpmi summary                            2.03
56 13d lab     n work   review pca concepts                           1.97
57 13d lab     w work   deep dive into default seurat analysis        1.97
67 9d  lab     lit work find list of CSC markers                      1.95
81 2d  lab     n work   evaluate list of highly var features pt59     1.91
83 2d  lab     n work   review rnaseq analysis protocol               1.91
88 20h lab     n work   clean up github tracked files                  1.9
89 17h lab     n work   put pt59 output on results page                1.9

17 tasks
```

### wkly mtg w/ Tim

present: chris, tim

```{r}

Description
------------------------------------------------------------------------------
agenda items
  2019-06-10 discuss application of different R files
  2019-06-10 compare pt59 data
  2019-06-11 job funding
  2019-06-12 discuss selection of genetic markers and highly variable features
  2019-06-13 assess current progress and direction
  2019-06-13 tirosh 2016 CNV method
  2019-06-13 how are you determining dimensionality

1 task
```

#### Notes

Direction and project selection

* When do I want my thesis to come together?
    + September is the goal right now.
* The CSC concept is probably not a suitable project since it is relatively hard to define without some type of in situ validation, which I won't have time for.
* The RNA velocity concept is interesting but probably outside my scope right now.
* The CNV project
    + Has been started to some degree by jason and would be a good candidate project.
    + He has a preliminary version of this that I should discuss with him.
    + They did not have the bulk exome data at this point, so there is potential for a greater reference in analysis.
    + I should get the exome data and develop a process to establish a CNV baseline.
    + I can use @Tirosh2016 as a general method reference in SC CNV analysis, but developing a novel, standardized approach would be valuable according to Tim.

Analysis of genetic markers and variable features

* Tim gave me a list of cell type gene reference lists which will come in handy when identifying clusters and variable features.
* We discussed Tim's R scripts. The `seurat_analysis.R` is meant to be a bit of a customization analysis after the general pipeline has run. The others serve their dedicated purpose, which is outlined in the RNAseq analysis protocol.

Selection of dimensionality

* Tim gave me an overview of his method for selection the dimensionality of the data sets. He uses the option to take a csv output of the PCA and rank by statistical significance. Currently, he uses the relatively arbitrary cutoff of 1e-10.

Cell Ranger Status

* It sounds like the Cell Ranger processing is a work in progress that is being conducted by Ying. They have 5 pt done, and the data is located at the root directory below.

```{r}
/scratch.global/zhan2142/starr/
```


ACTIONS:

* reach out to jason re his efforts so far.
* get exome data.
* read @Zare2017 for a comparison of CNV analysis in whole exome data.
* email yuk and explore grant opportunities.
* compare Tim's pt59 output.
* continue pt59 analysis by applying boolean script


### CNV

#### Points of contact

* Juan Abrahante <abrah023@umn.edu>
* Christy Henzler <chenzler@umn.edu>
* Jinhua Wang <wangjh@umn.edu>
* Ying Zhang <zhan2142@umn.edu>
* Andrew Nelson <nels2055@umn.edu>

#### Varscan2

Tim wants me to try to use Varscan2 as a method to assess the exome data [@Koboldt2012]. Jinhua had tried this in some fashion previously.

#### Contra

Juan had tried using contra in the past which did not work.

##### from: Juan 6/13/2019

Sure thing. In a nutshell back in 2013 I ran Contra with Tumor/Normal Samples and it worked, I got results... fast forward 6 years and the test files are not even working for me.


First, I attempted via Python 3, 2.7 and the original 2.6 version using the test set in my local computer. Then, I moved into the "Matrix" or in this case Mesabi.

With Help from Master Jedi Christy (Thank you Christy!) I created a conda environment for Contra via:

```{r}
conda create -n contra python=2.6
source activate /home/umgc-staff/abrah023/bin/miniconda3/envs/contra
```

change directory to folder containing Contra script (tried both Version 2.0.8 and 1.0.3).

```{r}

python contra.py --target ../0247401_D_BED_20090724_hg19_MERGED.bed --test ../P0667N_GATKrealigned_duplicates_marked.bam --control ../P0667T_GATKrealigned_duplicates_marked.bam --fasta human_g1k_v37.fasta --outfolder ../output2/
```

That was using the files that came with Contra and got an error:

```{r}


(contra) abrah023@ln0003 [~/AndyTemp/CONTRA.v2.0.8] % python contra.py --target ../0247401_D_BED_20090724_hg19_MERGED.bed --test ../P0667N_GATKrealigned_duplicates_marked.bam --control ../P0667T_GATKrealigned_duplicates_marked.bam --fasta human_g1k_v37.fasta --o ../outputr/

target		: ../0247401_D_BED_20090724_hg19_MERGED.bed

test		: ../P0667N_GATKrealigned_duplicates_marked.bam

control		: ../P0667T_GATKrealigned_duplicates_marked.bam

outfolder	: ../outputr/

numBin		: [20]

minreaddepth	: 10

minNBases	: 10

sam		: False

pval		: 0.05

sampleName	: No-SampleName

nomultimapped	: False

plot		: False

bedInput		: False

minExon		: 2000

largeDeletion	: False

removeDups	: False

Creating Output Folder :  Done.

Traceback (most recent call last):

  File "contra.py", line 625, in <module>

    main()

  File "contra.py", line 570, in main

    get_genome(params.TEST, genomeFile)

  File "/panfs/roc/groups/8/umgc-staff/abrah023/AndyTemp/CONTRA.v2.0.8/scripts/get_chr_length.py", line 31, in get_genome

    raw_header = subprocess.Popen(args, stdout = subprocess.PIPE).communicate()[0]

  File "/home/umgc-staff/abrah023/bin/miniconda3/envs/contra/lib/python2.6/subprocess.py", line 623, in __init__

    errread, errwrite)

  File "/home/umgc-staff/abrah023/bin/miniconda3/envs/contra/lib/python2.6/subprocess.py", line 1141, in _execute_child

    raise child_exception

OSError: [Errno 2] No such file or directory

```

### Summary

I had my weekly meeting with Tim today. We had a productive conversation where we were able to assess and re-gear my progress. We came to the conclusion that I would focus on the CNV approach and were able to put together a large ranger of reference points around getting this started. There are several actions that came from our meeting, and I will have a good amount to focus on. The primary focus beyond content review will be developing a method for evaluating CNV within the whole exome data.

I was also able to get some house keeping done around documentation.

#### Completed

```{r}
Created    Completed  Age  Project Description
---------- ---------- ---- ------- ---------------------------------------------
2019-06-11 2019-06-13 2d   lab     read manno 2018
                                     2019-06-11 rna velocity of single cells
2019-06-03 2019-06-13 9d   lab     find list of CSC markers
2019-06-01 2019-06-13 12d  lab     litsrch re genetic lineage tracing method
                                     2019-06-01 can we determine a common
                                   ancestor among cell ancestry
2019-06-12 2019-06-13 1d   lab     clean up github tracked files
2019-06-13 2019-06-13 2min lab     organize cnv reference content

5 tasks
```

***

## 14

### TODO

```{r}

ID Age  Project Tags     Description                                        Urg
-- ---- ------- -------- -------------------------------------------------- ----
44 3w   lab     lit work read tirosh 2016 [2]                               2.94
49 2w   lab     lit work read papalexi 2018 [2]                              2.9
50 2w   lab     lit work read macintyre 2018 [2]                             2.9
60 11d  lab     lit work read konrashova 2017 [2]                           2.86
43 3w   lab     lit work read zheng 2017 [1]                                2.84
80 1d   lab     lit work read zare 2017 [1]                                 2.71
42 3w   lab     lit work read ocpmi summary                                 2.04
53 2w   lab     n work   review pca concepts                                1.98
54 2w   lab     w work   deep dive into default seurat analysis             1.98
69 4d   lab     n work   evaluate list of highly var features pt59          1.92
70 4d   lab     n work   review rnaseq analysis protocol                    1.92
46 3w   lab     work     agenda items                                       1.92
71 2d   lab     n work   put pt59 output on results page                    1.91
78 1d   lab     n work   reach out to jason re his CNV method               1.91
79 1d   lab     w work   get exome data                                     1.91
81 1d   lab     n work   email yuk re grant opportunities                   1.91
82 1d   lab     n work   explore grant opportunities                        1.91
83 1d   lab     n work   compare tims pt59 output                          1.91
84 1d   lab     n work   continue pt59 analysis with cell type selection    1.91
                         boolean script
85 1d   lab     n work   get graphics function to work on msi               1.91
93 3min lab     n work   send reminder to tim re rhel access                 1.9

21 tasks

```

### More CNV refernce

#### from: Jinhua 6/14/19

If it is for single cell exome DNA copy number call, you may try this tool: SCOPE,   https://www.biorxiv.org/content/10.1101/594267v1.full

For low coverage whole exome CNV,  you may try: EXCAVATOR,  http://sourceforge.net/projects/excavatortool/,  and also VARscan 2.

#### from: Andrew 6/14/19

You should now have access to nelsona2, where the somatic DNA exome data is located.  Rebecca and Christy have done the primary work on these samples, so please reach out to them if you have questions about the exome data.  Also, there are directories on there of sequencing from the Illumina TST170 panel, a focused DNAseq panel that has copy number calling built into the stock pipeline (using an algorithm called Craft).

Tim has cross walk files; as you will find out there are multiple different sample identifiers for each surgical specimen.

***

## 17

### TODO

```{r}

ID  Age  Project Tags     Description                                       Urg
--- ---- ------- -------- ------------------------------------------------- ----
 44 3w   lab     lit work read tirosh 2016 [2]                              2.95
 49 3w   lab     lit work read papalexi 2018 [2]                            2.92
 50 2w   lab     lit work read macintyre 2018 [2]                           2.91
 60 2w   lab     lit work read konrashova 2017 [2]                          2.88
 43 3w   lab     lit work read zheng 2017 [1]                               2.85
 80 4d   lab     lit work read zare 2017 [1]                                2.72
 42 3w   lab     lit work read ocpmi summary                                2.05
 53 2w   lab     n work   review pca concepts                                  2
 54 2w   lab     w work   deep dive into default seurat analysis            1.99
 69 6d   lab     n work   evaluate list of highly var features pt59         1.93
 70 6d   lab     n work   review rnaseq analysis protocol                   1.93
 46 3w   lab     work     agenda items                                      1.93
 71 4d   lab     n work   put pt59 output on results page                   1.92
 78 4d   lab     n work   reach out to jason re his CNV method              1.92
 79 4d   lab     w work   get exome data                                    1.92
 81 4d   lab     n work   email yuk re grant opportunities                  1.92
 82 4d   lab     n work   explore grant opportunities                       1.92
 83 4d   lab     n work   compare tim's pt59 output                         1.92
 84 4d   lab     n work   continue pt59 analysis with cell type selection   1.92
                          boolean script
 85 3d   lab     n work   get graphics function to work on msi              1.92
 86 3d   lab     w work   get scripts from jason re cnv analysis            1.92
 93 2d   lab     n work   send reminder to tim re rhel access               1.91
 94 2d   lab     n work   explore nelsona2 directory                        1.91
103 4min lab     n work   email tim re short mtg to discuss updated grad     1.9
                          plan

24 tasks
```

### @Tirosh2016

This paper is the only other case Tim is aware of where a group was able to effectively show CNV in single cell expression data.

* This group took human oligodendrogliomas and isolated ~4300 cells for sequencing.
* They constructed a developmental program from genome-wide expression signatures.
* The analysis produced the conclusion that these cells display 2 distinct developmental programs.
* Subclonal point mutation and insitu hybridization was also done as verification of these results.

#### CNV Method

* Their CNV approach was to define a window along a chromosomal mapping. By doing this, they effectively created "bins" of expression level to define relative changes in expression.
* They identified two cell types that aligned with a genetic character (high values at 1p and 19q) which they defined as baseline expression.
* This was used to define the baseline for microglia and oligdendrocytes.
* PCA comprised the main statistical approach.

### Minor Notebook Restructure

I realized I have a home page for no reason. I've replaced what is currently a blank placeholder page with what is my notes page since that is effectively the main page of my notebook. In doing so, I added a preamble to the begging of this document as well.

### Jason's CNV Method

Jason sent a very detailed overview over the weekend of his work on the CNV project. This includes code, references, and some analysis. I will need to look at this in greater depth and potentially set time to discuss it with him further. The files are located at the directory below.

* `/starr_lab/ocpmi/methods/jason_cnv`

### Summary

I did some house keeping on my documentation today. I've received a considerable amount of data and references regarding the CNV project, so I put some effort into organizing and clarifying my direction with this. My primary goals right now are covering the literature re the comparison for CNV techniques in whole exome data (@Zare2017) and then getting up to speed on Jason's past work. I hope to have a better mastery of the content to attack the project by the time I return from France.

#### Completed

```{r}

Created    Completed  Age Project Description
---------- ---------- --- ------- ----------------------------------------------
2019-06-13 2019-06-17 4d  lab     email yuk re grant opportunities
2019-06-15 2019-06-17 2d  lab     send reminder to tim re rhel access
2019-06-17 2019-06-17 2h  lab     email tim re short mtg to discuss updated grad
                                  plan
2019-05-20 2019-06-17 3w  lab     read tirosh 2016
                                    2019-06-03
                                  https://doi.org/10.1038/nature20123
                                    2019-06-03 CNV analysis in glioma
2019-06-13 2019-06-17 4d  lab     reach out to jason re his CNV method
2019-06-13 2019-06-17 4d  lab     get exome data
2019-06-13 2019-06-17 4d  lab     explore grant opportunities
2019-06-13 2019-06-17 4d  lab     get scripts from jason re cnv analysis

8 tasks
```

## 18

### mtg w/ Tim

present: tim, chris

I met briefly with Tim before leaving on my trip. We discussed the potential change in trajectory given the update that I will need to take 3 credits of directed research instead of 1 in the fall. He felt this generally won't adjust what our current plan is, and I tend to aggree.

Additionally, I introduced a concept that Yuk had brought up regarding the possiblity of a "long-distance, part-time PhD" given that I may move in the fall and secure a full time job. Tim was open to this, and I will keep it on the backburner for when that time comes after I've completed my masters degree.

Last, we discussed some of the smaller CNV analysis papers I've found that take a gene prep platform approach to evaluating CNV in single cells. The key distinction here is that these are likely done with DNA, but we both agreed that they may lack significance since I haven't found one published in a journal of significant impact. It may be something to explore for expanding my knowledge of methods though.

## 19

### @Zare2017

This paper explored a comparative analysis between current CNV analysis tools that operate in the context of whole exome sequencing which in wide use in the field. The group narrowed the landscape down to just 5 tools that meet a few criteria.

* There are 3 approaches to identifying CNVs
    + read count (read depth)
    + paired-end
    + assembly
* All tools the group evaluated used the read depth method since the pair-end and assembly methods are rendered ineffective with low read coverage like is seen in exome data. All CNV detection tools for WES are based on this method as a result.
* Ambiguities in alignment can create bias in the RD approach.
* The hybridization process and an uneven distribution of reads in exonic regions leads to bias.
* There are reads in the WES data that will be overlapping noise which would otherwise be delineated in the scRNA data. I will have to account for this.
* As of 2016, there are 15 seq-based CNV tools.
* B allele frequencies (BAF) is used to address tumor complexity in some of these tools.
* Group used both read and synthesized data.
* Look at table 1 for feature breakdown
* Look at table 4 for performance summary

ADTEx

* Newest of the tools they evaluated
* Has ability to leverage BAF
* Uses Discrete Wavelet Transform as a preprocessing step to reduce noise.
* Two Hidden Markov Models are used in detection.
* CBS is used in segmentation

Contra

* used for very small target regions.
* normalizes read count and log ratios for each base.
* better for reducing effect of GC bias.
* addresses problem of very low coverage and GC bias.

cn.MOPS

* Can use multiple samples for each genomics region to create a better estimate
* Uses a non-overlapping sliding window.
* Creates a null-hypothesis to measure distance to observed value.
* Can reduce false discovery rate

ExomeCNV

* Specifically design for WES data.
* Can use BAF data
* Counts overlapping reads from exons by using these counts for tumor and normal.
* Can also detect loss of heterzygosity.

VarScan2

* Also specifically designed for WES data from tumor-normal.
* Does not use a sliding window.
* Calculates tumor to normal read count ratios of high quality base that fulfill minimum coverage requirements.
* Using Fischers test, consecutive bases do not change significantly are binned together.
* Does not utilize a segmentation algorithm, CBS is advised.

Results

* The tools show low consistency.
* only about 25% of the true amplified genes and deleted genes were identified by all tools in the gene-based comparison.
* In the segment-based comparison, the sensitivities and FDRs were all comparably similar to the gene-based comparison.
* There was no consistency among tools for the size and number of detected CNVs as well.
* Due to contra's approach, it excels with clean data but may not be suitable for messy cancer data.
* Mis-detection of poor coverage sequences is a central cause of the poor performance.
* Because of the poor performance across the tools, there is a great need for improvement in preprocessing techniques.
* A point made that may hold relevance to my project is that tumor subclonality has not been addressed by any tool. The granularity offered by our scRNA data could help with this.

### items to clarify

* More specifically, what is the source of the exome data?
    + Is it at the site of the tumor?
    + Is it the same sample as the scRNA?
    + Is it general pt seq data?
* Difference between CBS and HMM algorithms.
* GC bias
* Do lit search on newer tools.
* B allele frequency (BAF). Does our dataset have these values.
* What is phred base quality
* VarSimlab, for generating synthetic CNV
* cghMCR to identify CNV genes using Refseq gene identifications.
* GenomicRanges package to detect overlapping regions.

***

# July

## 8

### TODO

```{r}

ID Age  Project  Tags     Description                                       Urg
-- ---- -------- -------- ------------------------------------------------- ----
46 6w   lab      lit work read papalexi 2018 [2]                            3.03
47 5w   lab      lit work read macintyre 2018 [2]                           3.02
72 2w   lab      w work   what is the source of the exome data [3]             3
79 8min lab      w work   items to clarify [6]                               2.9
44 6w   lab      work     agenda items [1]                                  2.85
41 6w   lab      w work   review frameworks for ccgd database               2.16
42 6w   lab      lit work read ocpmi summary                                2.16
43 6w   lab      w work   explore shiny framework for database              2.16
45 6w   lab      w work   get admin access to server                        2.14
48 5w   lab      n work   review pca concepts                               2.11
49 5w   lab      w work   deep dive into default seurat analysis            2.11
57 3w   lab      n work   evaluate list of highly var features pt59         2.05
58 3w   lab      n work   review rnaseq analysis protocol                   2.05
59 3w   lab      n work   put pt59 output on results page                   2.04
61 3w   lab      n work   compare tim's pt59 output                         2.04
62 3w   lab      n work   continue pt59 analysis with cell type selection   2.04
                          boolean script
63 3w   lab      n work   get graphics function to work on msi              2.04
66 3w   lab      n work   explore nelsona2 directory                        2.03
70 2w   lab      n work   begin comparison of cnv tools                     2.01
71 2w   lab      n work   clear uncached files from git                     2.01
73 2w   lab      w work   look up version tracking in document editing for     2
                          rmarkdown
74 2w   lab      n work   go back to june 19 and review items to clarify       2
69 2w   lab.ccgd n work   email oit points of contact to get rhel server    2.81
                          access [1]

23 tasks
```

### CCGD upgrade

I reached out to oialinux@umn.edu today regarding getting server access to the ccgd application. From looking at the ccgd project document, I found a few oit points of contact: Sonya Sustacek (ssustace), John Trammell (tram0004), and Tom Krell (sp?).


### Summary

I'm back. Feeling pretty rusty, so I've been reacclimating myself to the work I left 3wks ago. I started up the process of getting ccgd upgrade going and should be able to work on that in tandem with the cnv work. Additionally, I need to begin the process of really testing and applying the cnv protocols I reviewed on 6/19.

#### Completed

```{r}
Created    Completed  Age P Project  Description
---------- ---------- --- - -------- -------------------------------------------
2019-06-13 2019-07-08 3w  H lab      read zare 2017
                                       2019-06-13 comparison of CNV analysis in
                                     whole exome data
2019-06-17 2019-07-08 2w    lab      clear uncached files from git
2019-06-17 2019-07-08 3w    lab.ccgd email oit points of contact to get rhel
                                     server access
3 tasks
```

***

## 9

### TODO

```{r}
ID Age Project Tags     Description                                         Urg
-- --- ------- -------- --------------------------------------------------- ----
45 6w  lab     lit work read papalexi 2018 [2]                              3.04
46 6w  lab     lit work read macintyre 2018 [2]                             3.03
69 2w  lab     w work   what is the source of the exome data [3]            3.01
75 21h lab     w work   items to clarify [6]                                 2.9
44 6w  lab     work     agenda items [1]                                    2.85
41 7w  lab     w work   review frameworks for ccgd database                 2.17
42 7w  lab     lit work read ocpmi summary                                  2.17
43 7w  lab     w work   explore shiny framework for database                2.17
47 5w  lab     n work   review pca concepts                                 2.11
48 5w  lab     w work   deep dive into default seurat analysis              2.11
56 4w  lab     n work   evaluate list of highly var features pt59           2.05
57 4w  lab     n work   review rnaseq analysis protocol                     2.05
58 3w  lab     n work   put pt59 output on results page                     2.04
60 3w  lab     n work   compare tim's pt59 output                           2.04
61 3w  lab     n work   continue pt59 analysis with cell type selection     2.04
                        boolean script
62 3w  lab     n work   get graphics function to work on msi                2.04
65 3w  lab     n work   explore nelsona2 directory                          2.03
68 3w  lab     n work   begin comparison of cnv tools                       2.02
70 2w  lab     n work   go back to june 19 and review items to clarify      2.01
76 10s lab     n work   contact hst rep re ccgd access                       1.9

20 tasks
```

### CCGD access

I was informed by OIT that the server is actually owned by HST (Health Science Technology?) and that I need to reach out to them. The contacts I was given are:

* Colby Reese reese007@umn.edu
* Tom Kell kell@umn.edu
* Nathan Huff nrhuff@umn.edu
* Jeffrey Sik sikxx005@umn.edu

I emailed Tom since he was the only one listed in the original CCGD documentation.

### Summary

I am still going through the process of reviewing my work to reestablish the momentum and direction I had 3 weeks ago. I have been looking at Jason's code to make sense of his approach. I will probably reach out to him depending on what comes from my meeting with Tim tomorrow so that we can discuss his method. I need to clarify the picture of how I will implement the proper tool sets for both the exome and scRNA data.

#### Completed

```{r}
Created    Completed  Age Project Description
---------- ---------- --- ------- ----------------------------------------------
2019-07-09 2019-07-09 5h  lab     contact hst rep re ccgd access
2019-06-19 2019-07-09 2w  lab     go back to june 19 and review items to clarify

2 tasks
```

***

## 10

### TODO

```{r}
ID Age P Project Tags     Description                                  Urg
-- --- - ------- -------- -------------------------------------------- ----
72 18h H lab     n work   review zare notes before meeting              7.9
45 6w    lab     lit work read papalexi 2018 [2]                       3.04
46 6w    lab     lit work read macintyre 2018 [2]                      3.04
63 3w    lab     w work   what is the source of the exome data [3]     3.02
68 1d    lab     w work   items to clarify [6]                         2.91
44 6w    lab     work     agenda items [1]                             2.86
41 7w    lab     w work   review frameworks for ccgd database          2.17
42 7w    lab     lit work read ocpmi summary                           2.17
43 7w    lab     w work   explore shiny framework for database         2.17
47 5w    lab     n work   review pca concepts                          2.12
56 3w    lab     n work   get graphics function to work on msi         2.04
59 3w    lab     n work   explore nelsona2 directory                   2.04
62 3w    lab     n work   clarify direction for cnv tool selection     2.02

13 tasks
```

### wkly mtg w/ Tim

present: chris, tim

* Since I have been away, I did not have any work to discuss with Tim.
* We realigned my direction around the CNV project approach.
* We have variant allele freq (var). I should look into this.
* Reach out to people who processed wes data re ability get baf data. Talk to Rebecca Larue first.
  - Ask Rebecca which pipeline she used for methods purposes.
* Read papers on excavator (@Magi2013), varscan2, scope (@Wang2019) and assess which would be best. Run on one dataset.
* We discussed the source exome data since I wanted to clarify my understanding of this. The wes tissue can effectively be assumed to be the same source as the scRNA for each pt.

### Summary

The conversation with Tim today was helpful in regaining some direction on the CNV work. Right now, my immediate goals are to establish the toolset for wes analysis, develop a reliable wes output for reference, and then evaluate the method I use to approach the scRNA cnv analysis. I need more detail about the development of the wes data, so reaching out to Rebecca will be important at this point. Identifying supporting information for the wes data such as BAFs and VAFs could be determining factors in the pipeline I use.

#### Completed

```{r}
Created    Completed  Age P Project Description
---------- ---------- --- - ------- --------------------------------------------
2019-06-19 2019-07-10 3w    lab     what is the source of the exome data
                                      2019-06-19 is it at the tumor site?
                                      2019-06-19 is it the same sample as the
                                    scRNA?
                                      2019-06-19 is it just general pt exome
                                    data?
2019-07-09 2019-07-10 20h H lab     review zare notes before meeting
2019-06-17 2019-07-10 3w    lab     clarify direction for cnv tool selection

3 tasks
```

***

## 11

### TODO

```{r}
ID Age P Project Tags     Description                                       Urg
-- --- - ------- -------- ------------------------------------------------- ----
67 23h H lab     n work   contact rebecca re wes processing                  7.9
68 23h M lab     n work   create summarizing outline cnv project direction   5.8
63 2d    lab     w work   items to clarify [6]                              2.91
65 23h   lab     n work   read wang 2019 [1]                                 2.7
66 23h   lab     n work   read magi 2013 [1]                                 2.7
41 7w    lab     lit work read ocpmi summary                                2.18
46 6w    lab     n work   review pca concepts                               2.13
44 6w  L lab     lit work read papalexi 2018 [2]                            2.05
57 3w    lab     n work   explore nelsona2 directory                        2.04
45 6w  L lab     lit work read macintyre 2018 [2]                           2.04
69 23h   lab     w work   ask rebecca about wes processing method            1.9
43 6w  L lab     work     agenda items [1]                                  1.86
40 7w  L lab     w work   review frameworks for ccgd database               1.18
42 7w  L lab     w work   explore shiny framework for database              1.18
54 3w  L lab     n work   get graphics function to work on msi              1.05

15 tasks
```

### Summary

I reached out to Rebecca re a conversation about the background of the exome data processing. While I can investigate different tool sets to get a sense of what I will use, There are a few things she may be able to help me with that could be determining factors. I also began investigating the SCOPE tool today at the suggestion of Jinhua. I didn't have any background on this pipeline and was surprised to find it's purpose is to address single cell data, albeit with DNA. They indicate they did some validation with RNA data from 10x, so it remains to be seen if I will be able to use this.

#### Completed

```{r}
Created    Completed  Age P Project Description
---------- ---------- --- - ------- ---------------------------------
2019-07-10 2019-07-11 1d  H lab     contact rebecca re wes processing

1 task
```

***

## 12

### TODO

```{r}
ID Age P Project Tags     Description                                       Urg
-- --- - ------- -------- ------------------------------------------------- ----
67 1d  M lab     n work   create summarizing outline cnv project direction  5.81
63 3d    lab     w work   items to clarify [6]                              2.92
65 1d    lab     n work   read wang 2019 [1]                                2.71
66 1d    lab     n work   read magi 2013 [1]                                2.71
41 7w    lab     lit work read ocpmi summary                                2.18
46 6w    lab     n work   review pca concepts                               2.13
44 6w  L lab     lit work read papalexi 2018 [2]                            2.05
57 3w    lab     n work   explore nelsona2 directory                        2.05
45 6w  L lab     lit work read macintyre 2018 [2]                           2.05
68 19h   lab     w work   email other hst staff                              1.9
69 1h    lab     n work   take a look at genomespace                         1.9
43 7w  L lab     work     agenda items [1]                                  1.87
40 7w  L lab     w work   review frameworks for ccgd database               1.18
42 7w  L lab     w work   explore shiny framework for database              1.18
54 4w  L lab     n work   get graphics function to work on msi              1.05

15 tasks
```


### Collaborator mtg

present: josh, andy, isaac, shobhana, borris, jason, tim, ying

* exome data
    + freebase was used to call variants
    + there are some issues in variant filtering for calls in processing the wes data.
    + Andy took pt pm1 (pt54) and did a cross refernce across the 4 bams available
        - Coverage variability is high
        - Filtering poor calls stands as a significant hurdle
        - Currently, we need to establish a base line for "common" mutation calls
        - the tst170 was thought to be a higher standard, but the consistency does not seem to be there
        - Andy and Tim are going to create a simplified list of common genes to identify mutations
* scRNA
    + all single cell data has been rerun through cell ranger v3 and seurat.
    + we are switching to nova seq from hiseq.
    + the location and naming of this are somewhat ambiguous
    + the new v3 cell ranger data might be named "v2"...

### CNV tool lit search

#### EXCAVATOR

#### SCOPE

#### Awesome single cell repo

I found a really comprehensive list of single cell analysis tools at the github below.

* https://github.com/seandavi/awesome-single-cell

#### InferCNV

I also found a tool that was inspired by the @Tirosh2016 paper which explicitly aims to identify cnv in scRNA data. This looks very promising, and I will probably attempt this tool set first when getting into the sc data. The tool look like it could potentially be very simple. As this may help in getting started, it may be a drawback with the added complexity of our data set.

* https://github.com/broadinstitute/infercnv/wiki

### Summary

I spent the day evaluating CNV tools. While this was relatively productive, a brief conversation with Tim did draw attention to the fact I need to stop aiming and shoot. With the general info I've gathered, I can begin to run some analysis. My first step will be to run EXCAVATOR with a single pt data set followed by VarScan2 with the same data. My hope is that these pipelines come together with some ease and offer some level of a tutorial or test case. If not, I will focus on which of the two I can fully execute more quickly.

Additionally, my content search led me to an application I have an early excitement for, and that is InferCNV. I have really only glanced at the wiki for this tool, but it seems as if it was made specifically for our purposes. It looks relatively simple to use, and I am hoping that it can allow me to generate some data right away.

#### Completed

```{r}
Created    Completed  Age Project Description
---------- ---------- --- ------- ----------------------------------------------
2019-07-11 2019-07-12 1d  lab     email other hst staff
2019-07-08 2019-07-12 4d  lab     items to clarify
                                    2019-07-08 varsimlab, for generating
                                  synthetic cnv
                                    2019-07-08 cghmcr to identify cnv genes
                                  using refseq gene id
                                    2019-07-08 genomicranges package to detect
                                  overlapping regions
2019-07-12 2019-07-12 8h  lab     take a look at genomespace
2019-05-20 2019-07-12 7w  lab     read ocpmi summary
2019-07-10 2019-07-12 2d  lab     read wang 2019
                                    2019-07-10 scope wes tool
2019-07-10 2019-07-12 2d  lab     read magi 2013
                                    2019-07-10 excavator wes tool

7 tasks
```

***

## 15

### TODO

```{r}
ID Age P Project Tags   Description                                         Urg
-- --- - ------- ------ --------------------------------------------------- ----
60 2d  H lab     n work run excavator                                       7.91
59 4d  M lab     n work create summarizing outline cnv project direction    5.82
61 2d    lab     n work summarize value of excavator                        1.91
62 2d    lab     n work summarize features of scope                         1.91
42 7w  L lab     work   agenda items [1]                                    1.88
40 7w  L lab     w work review frameworks for ccgd database                  1.2
41 7w  L lab     w work explore shiny framework for database                 1.2
50 4w  L lab     n work get graphics function to work on msi                1.07

8 tasks
```

### WES analysis with EXCAVATOR/XCAVATOR

I will be taking the first wes cnv approach using the excavator/xcavator tool. This group published EXCAVATOR in @Magi2013 and have since expanded on it @Magi2017. I will be looking at XCAVATOR first, given its recency to see if it can bring anything to fruition.

Excavator leverages perl, bash, R, and fortran (of all things), and xcavator runs off of the same underlying architecture. I've installed the source files for the application using the scripts they provided at the location below.

* `/home/chris/starr_lab/ocpmi/methods/code/xcavator/source/lib`

Documentation can be found here:

* `/home/chris/starr_lab/ocpmi/methods/code/xcavator/source/docs/EXCAVATOR Manual.pdf`

#### XCAVATOR Implementation

As stated in @Magi2017

> All the computational approaches described in this paper have been packaged in the XCAVATOR software tool. XCAVATOR is a collection of perl, bash, R and fortran codes and its computational architecture has been derived from the EXCAVATOR tool that we published in 2013 [34] for the detection of CNVs/sCNA from whole-exome sequencing data. Our tool takes as input WGS data as BAM files and gives as output plots reporting raw, normalized, segmented and called data and a list of detected CNVs in tab-delimited and VCF format.

>The tool allows to analyze WGS data with three different experimental designs: “pooling”, “paired” and “nocontrol”. In “paired” mode each test sample is compared with its matched control and it is the best scheme to detect sCNA from pairs of tumor and matched normal samples. The “paired” mode has been used to analyze the TCGA Benchmark 4 dataset. In “nocontrol” mode the RCs of each test sample are normalized to two copies and this scheme is best suited for population genomics studies and has been used to analyze the 929 WGS experiments sequenced by 1000GP consortium. In “pooling” scheme, each test sample is compared to a pool control samples by summing the RC of each window across all the control samples. XCAVATOR can run on any unix system (desktop and servers) and allows the user to set the number of processor to analyze multiple samples in parallel. On a desktop computer with a 2.5 GHz cpu and 8 GB of ram, by using four cores, it takes four hours to perform the analysis of ten WGS samples sequenced at 60x. The XCAVATOR tool is freely available at https://sourceforge.net/projects/xcavator/.

### Summary

I put time into getting xcavator implemented. I am going with the updated version of excavator that was introduced in 2017. I had not noticed in my initial glance at the tool that it utilized perl and fortran. I don't believe this will impact any of the coding necessary since the pipeline seems to rely on modification of reference files, so my limited knowledge of perl and fortran shouldn't be an issue.

***

## 16

### TODO

```{r}
ID Age P Project Tags   Description                                         Urg
-- --- - ------- ------ --------------------------------------------------- ----
60 3d  H lab     n work run excavator                                       7.92
59 5d  M lab     n work create summarizing outline cnv project direction    5.83
61 3d    lab     n work summarize value of excavator                        1.92
62 3d    lab     n work summarize features of scope                         1.92
42 7w  L lab     work   agenda items [1]                                    1.89
40 8w  L lab     w work review frameworks for ccgd database                 1.21
41 8w  L lab     w work explore shiny framework for database                1.21
50 4w  L lab     n work get graphics function to work on msi                1.08

8 tasks
```

### Interactive Computing

It is possible to run an interactive session with the MSI lab space using the `isub` command. Additionally, `qsub` allows for scheduling of jobs in the lab space.

This is useful in the event I want to do development inside MSI rather than just batch jobs. More documentation at the link below:

* https://www.msi.umn.edu/content/interactive-queue-use-isub

***

## 17

### TODO

```{r}
ID Age P Project Tags    Description                                        Urg
-- --- - ------- ------- -------------------------------------------------- ----
60 4d  H lab     n work  run excavator                                      7.92
59 6d  M lab     n work  create summarizing outline cnv project direction   5.83
71 23h M lab     n phone call any hst number                                 5.8
61 4d    lab     n work  summarize value of excavator                       1.92
62 4d    lab     n work  summarize features of scope                        1.92
42 7w  L lab     work    agenda items [1]                                    1.9
40 8w  L lab     w work  review frameworks for ccgd database                1.21
41 8w  L lab     w work  explore shiny framework for database               1.21
50 4w  L lab     n work  get graphics function to work on msi               1.08

9 tasks
```

### wkly mtg w/ Tim

present: tim, chris

We had a limited discussion today. I updated Tim on progress with getting started in xcavator and getting access to the ccgd server space. Tim suggested I approach sonya given her past involvement with the project.

### XCAVATOR Progress

I'm continuing the process of getting xcavator setup and running. As a pipeline, xcavator is less mature than something like Seurat, especially with its documentation. I need to discuss a handful of items with either Rebecca or Christy regarding the processing of the wes data.

### Summary

Most of my time today was spent making sense of the xcavator workflow. There are some gaps in the documentation that I may have to parse out, but I am getting a sense for the parameters.

#### Completed

```{r}
Created    Completed  Age Project Description
---------- ---------- --- ------- ----------------------------------------------
2019-07-17 2019-07-17 23h lab     reach out to ssustace@umn.edu re ccgd server
                                  access

1 task
```

***

## 18

### TODO

```{r}
ID Age P Project Tags   Description                                         Urg
-- --- - ------- ------ --------------------------------------------------- ----
59 5d  H lab     n work run excavator                                       7.93
58 7d  M lab     n work create summarizing outline cnv project direction    5.84
73 21h   lab     w work mtg with rebecca [3]                                 2.9
60 5d    lab     n work summarize value of excavator                        1.93
61 5d    lab     n work summarize features of scope                         1.93
42 7w  L lab     work   agenda items [1]                                     1.9
75 17h   lab     n work find all sfconflicts                                 1.9
40 8w  L lab     w work review frameworks for ccgd database                 1.22
41 8w  L lab     w work explore shiny framework for database                1.22
50 4w  L lab     n work get graphics function to work on msi                1.09

10 tasks
```

### create_site to notebook_knit

I've renamed my `create_site` file to `notebook_knit` due to a conflict with a different executable. The location of this file is now at `/usr/bin` on my local machine. The updated script is below.

```{r}
#!/usr/bin/env Rscript

cur_dir <- getwd()
setwd("/home/chris/starr_lab/ocpmi/notebook")

rmarkdown::render("index.Rmd")
rmarkdown::render("results.Rmd")
rmarkdown::render("methods.Rmd")
rmarkdown::render("references.Rmd")
rmarkdown::render_site("index.Rmd")

fl <- Sys.glob("*.html")

ifelse(file.exists(fl), file.remove(fl))

setwd(cur_dir)

```

A key element to make this executable was to include it in an environment directory (i.e. `/usr/bin`) and also make it executable for all users with the command below.

```{r}
sudo chmod a+x notebook_knit.R
```

### XCAVATOR Window Initialize

The first command seems to take too much memory, so I installed the pipeline on MSI today. The setup should mirror my local installation for now. The goal right now is just to get the code to run. I'm attempting to execute the `ReferenceWindowInitialize.pl` command.

Overall, I'm using the default params and referencing the hg19.fa fasta file I downloaded from here:

* http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/

Here is the `SourceTarget.txt` file on the MSI instance:

```{r}
/panfs/roc/groups/11/starrt2/tasta005/ocpmi/methods/XCAVATOR/data/ucsc.hg19.bw /panfs/roc/groups/11/starrt2/tasta005/ocpmi/methods/XCAVATOR/data/hg19.fa

```

Here is the command itself:

```{r}
perl ReferenceWindowInitialize.pl -v SourceTarget.txt test_window1 100 hg19
```


#### Error

I've encountered an error attempting to execute `ReferenceWindowInitialize.pl`.

I am initially just putting any data that will work into the pipeline to get the code running. As far as I can tell, I should be able to use a random fasta file. I chose hg19 since that is identified as a reference genome.

The output of the error is below. The key issues seems to be the creation of the test directory and whatever `*_chromosome.txt` is.

```{r}
Program folder is: /panfs/roc/groups/11/starrt2/tasta005/ocpmi/methods/XCAVATOR
Creating Consecutive and non-overlapping windows of 100 bp Size
sh: R: command not found
Calculating Mappability and GC content...
cat: /panfs/roc/groups/11/starrt2/tasta005/ocpmi/methods/XCAVATOR/data/targets/hg19/test_window1/*_chromosome.txt: No such file or directory
...done!

```

### Summary

Thinking about the test data I've chosen, I might need to reevaluate. The hg19 dataset isn't wes data, and I don't know if that is an issue for the pipeline. It's also entirely possible I'm missing a setting that is essential to the initialization step. Another point of consideration is that I need a dataset that also has associated bam files, which I am not aware of for hg19. I could take a pt from the ocpmi data set, but they do not have basic fasta files that I am aware of yet, and I don't know if the fastq files we do actually have are compatible with the pipeline.

#### Completed

```{r}
Created    Completed  Age P Project Description
---------- ---------- --- - ------- --------------------
2019-07-16 2019-07-18 2d  M lab     call any hst number
2019-07-17 2019-07-18 23h   lab     find all sfconflicts

2 tasks
```

***

## 21

### TODO

```{r}
ID Age P Project Tags   Description                                         Urg
-- --- - ------- ------ --------------------------------------------------- ----
58 8d  H lab     n work run excavator                                       7.94
57 11d M lab     n work create summarizing outline cnv project direction    5.86
72 4d    lab     w work mtg with rebecca [5]                                2.92
59 8d    lab     n work summarize value of excavator                        1.94
60 8d    lab     n work summarize features of scope                         1.94
68 5d    lab     w work get ahold of someone from hst re ccgd database      1.93
81 2d    lab     n work test fastq compatiblity with xcavator               1.91
82 2d    lab     n work trouble shoot xcavator error from june 18           1.91
86 44s   lab     n work access and explore ccgd server                       1.9
40 8w  L lab     w work review frameworks for ccgd database                 0.24
41 8w  L lab     w work explore shiny framework for database                0.23
42 8w  L lab            agenda [1]                                          0.12

12 tasks
```

### CCGD Access

I was finally able to get through on the access portion of the ccgd project. Sonya played an important role in initiating access, which was ultimately completed by Satish.

#### Server site

* [ccgd-starrlab.oit.umn.edu](ccgd-starrlab.oit.umn.edu)

The working directory for the website seems to be located at:

* `/swadm/var/www/html`

***

## 22

### TODO

```{r}
ID Age P Project Tags   Description                                         Urg
-- --- - ------- ------ --------------------------------------------------- ----
58 9d  H lab     n work run excavator                                       7.95
57 11d M lab     n work create summarizing outline cnv project direction    5.86
71 4d    lab     w work mtg with rebecca [5]                                2.92
59 9d    lab     n work summarize value of excavator                        1.95
60 9d    lab     n work summarize features of scope                         1.95
79 3d    lab     w work check status of access to ccgd server               1.92
80 3d    lab     n work test fastq compatiblity with xcavator               1.92
81 3d    lab     n work trouble shoot xcavator error from june 18           1.92
85 21h   lab     n work put satish on contact list                           1.9
87 19h   lab     n work enter daily summary from yesterday                   1.9
40 8w  L lab     w work review frameworks for ccgd database                 0.24
41 8w  L lab     w work explore shiny framework for database                0.24
42 8w  L lab            agenda [1]                                          0.12

13 tasks
```


### Folding elements

Following an example from the Seurat pbmc tutorial, I am incorporating folding page elements in my notebook to reduce length and clutter.

```{r eval=F}
<details>
  <summary>****</summary>
</details>
```

### XCAVATOR vs EXCAVATOR2

I've been working through the error I'm getting in running the `ReferenceWindowInitialize.pl` script to no avail. There is some ambiguity in the file setup for the script arguments. It asks for a bed file but does not specify where that is supposed to be referenced. As a result, I'm going to see if I can easily transition to using EXCAVATOR instead.

* https://sourceforge.net/projects/excavator2tool/

### Summary

The XCAVATOR package has been giving me some difficulty with the inconsistencies in its documentation. Part of the issue could stem from my lack of knowledge regarding the file types necessary, but there is not a clear indication that I am wrong either. There script requirements do not fully line up, and the lack of a complete test dataset compounds the issue. I'm going to try my luck with the EXCAVATOR2 package since it is probably more mature in its function and support.

#### Completed

```{r}
Created    Completed  Age Project Description
---------- ---------- --- ------- -------------------------------------
2019-07-18 2019-07-22 4d  lab     check status of access to ccgd server
2019-07-21 2019-07-22 1d  lab     put satish on contact list
2019-07-21 2019-07-22 1d  lab     enter daily summary from yesterday
2019-07-18 2019-07-22 3d  lab     trouble shoot xcavator error from june 18

3 tasks
```

***

## 23

### TODO

```{r}
ID Age  P Project Tags   Description                                        Urg
-- ---- - ------- ------ -------------------------------------------------- ----
66 5d     lab     w work mtg with rebecca [5]                               2.93
75 1min   lab     n work find vcf files in dna_project                       1.9
76 1min   lab     n work begin pilot with data from RL                       1.9
78 18s    lab     n work explore application of vcf files in cnv pipeline    1.9
40 9w   L lab     w work review frameworks for ccgd database                0.25
41 9w   L lab     w work explore shiny framework for database               0.25
42 8w   L lab            agenda [1]                                         0.13

7 tasks
```

### Call w/ Rebecca

Topic: wes data processing and prep

I was able to finally connect with Rebecca and discuss the nature of the wes data processing that took place earlier this year. It sounds like there has been some distance between when they were actually working on this and now. Nonetheless, she was able to share some general information on their process, which will be very helpful.

```{r}
Description
------------------------------------------------------------
mtg with rebecca
  2019-07-17 what is the reference assembly for the wes data
  2019-07-17 file naming conventions
  2019-07-17 what is a reasonable window size
  2019-07-18 what is the nature of our controls
  2019-07-18 do we have fasta files

1 task
```

#### Notes

* All processing was done in MSI scratch due to the large resource use.
* The data is stored in 2nd tier storage right now as a result.
  BAM files might also be in 2nd tier storage.
* The location of the data is `/nelsona2/shared/ovarian_Jan2019/DNA_project`.
* The directory `/ovarian_Jan2019/singlesamples`, which I had thought was the directory of interest, contains the data for the RNAseq project.
* The sc and wes data were mapped using different reference genomes. sc - GR38 and wes - GR37. **This could be a problem.**
    - We might need to remap the wes data to GR38.
* I need to do a search of `/DNA_project` to see if there are any usable sequence files in there.
* RL is going to move a small dataset from aws so that I can begin a pilot analysis.
* The Freebayes params may be relevant, I should look into those.

#### Process

*Pipeline (in-house) &rarr; Mapping (GR37) &rarr; Preprocessing (`ScanIndel-ovarian.py`) &rarr; Freebayes (variance calling) &rarr; Selective Capture (ovarian specific) &rarr; VCF file output &rarr; Selective Filtering (`fb_VCF_filter.R`) &rarr; ClinVar (Christy)*

Project dir: `/nelsona2/shared/ovarian_Jan2019/DNA_project`

* Important scripts are located in `/DNA_project/scanindel_mod`.
* The probe pull down inclued some non-coding regions.
    - The assumption was that there may be a relevant impact on the coding regions in some cases, so these elements should be recovered and selectively filtered out.
* ClinVar was used to apply annotation to variance that matches existing clinical reference.

***

## 24

### TODO

```{r}
ID Age P Project Tags   Description                                         Urg
-- --- - ------- ------ --------------------------------------------------- ----
74 21h   lab     n work find vcf files in dna_project                        1.9
75 21h   lab     n work begin pilot with data from RL                        1.9
77 21h   lab     n work explore application of vcf files in cnv pipeline     1.9
78 20h   lab     n work explore freebayes params                             1.9
42 8w  L lab            agenda [5]                                          0.33
40 9w  L lab     w work review scripts for ccgd database                    0.25
41 9w  L lab     w work explore shiny framework for database                0.25

7 tasks
```

### wkly mtg w/ Tim

present: chris, tim

```{r}
Description
---------------------------------------------------------------
agenda
  2019-07-08 status of ccgd project
  2019-07-23 expand on the rnaseq project
  2019-07-23 discuss call with rebecca
  2019-07-24 cosmic update
  2019-07-24 can we use the preprocessed data for cnv pipelines

1 task
```

#### Notes

Our meeting covered the two major branches of the ccgd and ocpmi project. First, I updated Tim on my new access to the ccgd server.

* Looking at the current status, the next steps are to determine the process for updating the server os.
    - I need to reach out to OIT to determine their policy on upgrades.
* Also, I need to begin the backup process to make sure all elements can be recovered in the event of any issues in the upgrade.
    - This is also a conversation to be had with OIT.

The updates on the ocpmi project centered on my conversation with Rebecca.

* As I had highlighted, the processing work done by RL/CH on the wes data will not be relevant to my cnv pipeline.
    - I need to reach out to them and clarify where their raw data is currently since I assume this is also on aws.
* With the roadblocks I've experienced with E/Xcavator, Tim suggested I reach out to the developers. Which I will do after a trial of Excavator2
    - The plan for my approach in the event I continue to see issues can be seen below
* A point of clarification is that hg19 and GRCh37 are the same. This means that any data I use with upstream processing done by Rebecca should follow with hg19 as the reference.

### CNV Plan

Fallback strategy for cnv pipeline:

*Try Excavator2 &rarr; Contact Devs &rarr; Resort to VarScan2*

### Excavator2 Troubleshooting

#### BED and FASTA

Continuing from the error on 7/18, I found I was not referencing the bed file correctly. This was not as clear in the Xcavator documentation. I took generic bed file from ucsc genome browser.

* http://genome.ucsc.edu/cgi-bin/hgTables?command=start
* Dir: `/EXCAVATOR2_Package_v1.1.2/data/hg19.bed`

Looking at the background on fasta reference, I decided to take the g1k fasta file from 1000 genome project.

* ftp://ftp-trace.ncbi.nih.gov/1000genomes/ftp/technical/reference/
* Dir: `/EXCAVATOR2_Package_v1.1.2/data/human_g1k_v37.fasta`

The `TargetPerla.pl` script executes after I arranged these bed and fasta files.

#### libpng12.deb

This produced an error re a missing object

```{r}
/home/chris/Seafile/library/work/research/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/lib/OtherLibrary/./bigWigAverageOverBed: error while loading shared libraries: libpng12.so.0: cannot open shared object file: No such file or directory
Error in file(file, "rt") : cannot open the connection
Calls: read.table -> file
In addition: Warning message:
In file(file, "rt") :
  cannot open file '/home/chris/Seafile/library/work/research/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/targets/hg19/test_target2/MAP/Mapout.txt': No such file or directory
Execution halted
```

* Needed to install legacy version of `libpng12.deb`
* https://askubuntu.com/questions/978294/how-to-fix-libpng12-so-0-cannot-open-shared-object-file-no-such-file-or-direc/978338

```{r}
wget -q -O /tmp/libpng12.deb http://mirrors.kernel.org/ubuntu/pool/main/libp/libpng/libpng12-0_1.2.54-1ubuntu1_amd64.deb
sudo dpkg -i /tmp/libpng12.deb
sudo rm /tmp/libpng12.deb
```

#### BEDTools

Was getting another error re another missing object: bedtools. The documentation says nothing about it, but I needed to install `bedtools`.

```{r}
sudo apt install bedtools
```

#### Unresolved

Current error. Not sure what this is yet. My current thought is that there is an issue with the formatting between the fasta and bed file. I believe the g1k fasta I downloaded is a multi fasta, and I don't know if that's an issue.

```{r}
WARNING. chromosome (chr22) was not found in the FASTA file. Skipping.
```

### Summary

I made some progress working through the Excavator2 pipeline today. I am still encountering hang ups that are probably related to file formatting and binary file setup. I need to reach out to RL/CH regarding the change in source files I need since we didn't fully cover this when I spoke to Rebecca. Given the quick jump I made in troubleshooting today, I hope to be able to continue that progress with the pipeline tomorrow.

***

## 25

### TODO

```{r}
ID Age  P Project Tags   Description                                        Urg
-- ---- - ------- ------ -------------------------------------------------- ----
73 1d     lab     n work find vcf files in dna_project                      1.91
74 1d     lab     n work begin pilot with data from RL                      1.91
76 1d     lab     n work explore application of vcf files in cnv pipeline   1.91
77 1d     lab     n work explore freebayes params                           1.91
79 21h    lab     n work add tasks from wkly mtg                             1.9
81 2min   lab     n work email rebecca re bam files                          1.9
82 1min   lab     n work contact oit re update policy and process            1.9
83 37s    lab     n work start back up process for ccgd server               1.9
42 8w   L lab            agenda [5]                                         0.34
40 9w   L lab     w work review scripts for ccgd database                   0.26
41 9w   L lab     w work explore shiny framework for database               0.26

11 tasks
```

### BAM file search

I contacted Rebecca re the location of the wes bam files today since we missed that point in our conversation.

### Excavator2 Trbl (cont.)

#### Unresolved - Solved

I was able to address the last error given on 7/24 by using a different fasta reference file. I believe the issue was that the fasta and bed files contained different chromosome notation as indicated by section 1.6 "Before you start" of the Excavator2 docs. Probably should have read that.

As a result, the fasta file I'm using now also comes from UCSC

* http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/
* Dir: `/EXCAVATOR2_Package_v1.1.2/data/hg19.fa`

### ExcavatorDataPrepare.pl

The next step in running the Excavator2 pipeline is setting up `EXCAVATORDataPrepare.pl` to process sample BAM files. I don't have a clear set of files I know will work at this point, so I will need to get that sorted out. The BAMs are referenced using the `ExperimentalFilePrepare.window.txt` file which includes 1) an abs path to the BAMs 2) the path to the sample output 3) the sample name.

### Summary

I was able to finally run the first of 3 execution scripts with the Excavator2 pipeline. Having lined all the necessary items up for that first step, I need to arrange a set of pilot BAM files that will allow me to continue with the test run of the pipeline. Currently, I'm looking at pulling a 1kgp dataset that was referenced in the Excavator2 (@DAurizio2016). This may be an issue since the 1kgp has restructured their ftp since the Excavator2 publication.

***

## 26

### TODO

```{r}
ID Age  P Project Tags   Description                                        Urg
-- ---- - ------- ------ -------------------------------------------------- ----
69 2d     lab     n work find vcf files in dna_project                      1.91
70 2d     lab     n work begin pilot with data from RL                      1.91
72 2d     lab     n work explore application of vcf files in cnv pipeline   1.91
73 2d     lab     n work explore freebayes params                           1.91
75 1d     lab     n work add tasks from wkly mtg                            1.91
78 1d     lab     n work contact oit re update policy and process           1.91
79 1d     lab     n work start back up process for ccgd server              1.91
83 1min   lab     n work continue to troubleshoot excavator2 pipeline        1.9
84 1min   lab     n work find pilot bam files for excavator2 test run        1.9
42 9w   L lab            agenda [5]                                         0.35
40 9w   L lab     w work review scripts for ccgd database                   0.26
41 9w   L lab     w work explore shiny framework for database               0.26

12 tasks
```

### 1kgp ftp

The 1kgp ftp has moved to a new site.

* ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000_genomes_project/data

### BAM Selection

Currently I am taking 2 of the genome samples referenced in Supplementary Table 2 of @DAurizio2016.

* NA19131
* NA10847

Both BAMs were downloaded as the phase 1 1kgp sample from the IGSR

* http://www.internationalgenome.org/data-portal/sample

### Excavator2 Trbl (cont.)

#### Missing BAM Index

I realized I was missing the BAM index included on the IGSR.

```{r}
[main_samview] random alignment retrieval only works for indexed BAM or CRAM files.
```

#### Chr notation

It seems like the chromosome notation has become an issue again. The UCSC and 1kgp gene notations use "chr1" and "1" respectively making them incompatible without a variety of data wrangling.

```{r}
[main_samview] region "chr1" specifies an unknown reference name. Continue anyway.
```

### Summary

I hit another formatting snag today that will require me to go back and find different reference files. I believe I generally have the `ExcavatorDataPrepare.pl` script setup correctly, but the chromosome naming scheme needs to be consistent across the reference and sample files. This should be relatively easy to fix. I just need to find fasta, bed, and bam files that match.

***

## 27

### New fasta and bed

To address the chromosome naming discrepancy, I retrieved new fasta and bed files from the 1kgp so that the source will be consistent.

* http://www.internationalgenome.org/category/grch37/

More specifically:

fasta

* http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/
  Dir: `/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/human_g1k_v37.fasta`

phase 1/2 bed files

* http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/exome_pull_down_targets_phases1_and_2/
* Dir: `/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/20120518.consensus.annotation.bed`

***

## 28

### New fasta and bed again

In addition, I reran the pipeline with a new reference due to the chromosome annotation issue. This includes some BAMs from UCSC that are smaller in size.

* http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeUwRepliSeq/
* Dir: `/home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc`

Here's the updated `SourceTarget.txt`

```{r}
/home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bPctSignalRep1.bigWig /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/hg19.fa

```


### ExcavatorPrepareData.pl

I've was able to run the `ExcavatorPrepareData.pl` script with the following parameters. I'm not entirely sure this worked based on the output, but it doesn't seem explicitly that it didn't work either.

#### ExperimentalFilePrepare.50000.txt

This is the tab delimited file that identifies the bam files, of which were

```{r}
/home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/pilot/test_bam test_bam

```


Final `ExperimentalFilePrepare.50000.txt`

```{r}
perl EXCAVATORDataPrepare.pl ExperimentalFilePrepare.50000.txt --processors 4 --target test_target2 --assembly hg19
```


<details>
  <summary>**ExcavatorDataPrepare.pl Output**</summary>

```{r}
perl EXCAVATORDataPrepare.pl ExperimentalFilePrepare.50000.txt --processors 4 --target test_target2 --assembly hg19
Program folder is: /home/chris/Seafile/library/work/research/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2
Preparing Multiprocessor Analysis...
Starting Multiprocessor Analysis!
Creating '/home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/pilot/test_bam/RC' folder...
Creating '/home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/pilot/test_bam/RCNorm' folder...
Creating '/home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/pilot/test_bam/Images' folder...
Working on sample test_bam.
Creating Read Count Data...
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
[W::hts_idx_load2] The index file is older than the data file: /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/gene_ref/ucsc/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam.bai
Removing temporary files...
Read Count done!
Normalizing Read Count Data...
Loading required package: lattice
Loading required package: survival
Loading required package: Formula
Loading required package: ggplot2

Attaching package: ‘Hmisc’

The following objects are masked from ‘package:base’:

    format.pval, units

null device
          1
null device
          1
null device
          1
null device
          1
null device
          1
Normalization done!
Performed RC calculations and Normalization on samples:
```

</details>

***

## 29

### TODO

```{r}
ID Age P Project Tags   Description                                         Urg
-- --- - ------- ------ --------------------------------------------------- ----
66 5d    lab     n work begin pilot with data from RL                       1.93
67 5d    lab     n work reach out to christy re her wes work                1.93
68 5d    lab     n work explore application of vcf files in cnv pipeline    1.93
69 5d    lab     n work explore freebayes params                            1.93
72 3d    lab     n work contact oit re update policy and process            1.92
73 3d    lab     n work start back up process for ccgd server               1.92
76 2d    lab     n work continue to troubleshoot excavator2 pipeline        1.91
77 2d    lab     n work find pilot bam files for excavator2 test run        1.91
78 2d    lab     w work summarize excv2 process in methods.Rmd              1.91
40 9w  L lab     w work review scripts for ccgd database                    0.28
41 9w  L lab     w work explore shiny framework for database                0.28
42 9w  L lab            agenda [1]                                          0.16

12 tasks
```

### Excv2 trbl (cont.)

I've run the final script for the Excavator2 (excv2) pipeline to no avail. I'm at a bit of a loss and will likely transition to VarScan2 today unless I get a decent lead from reaching out the Excv2 authors.

```{r}
perl EXCAVATORDataAnalysis.pl ExperimentalFileAnalysis.50000.txt --processors 4 --target test_target --assembly hg19 --output /home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/results --mode pooling
```

<details>
  <summary>**ExcavatorDataAnalysis Failed Output**</summary>

```{r}
Checking output folders...
'/home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/results' folder ready!
Checking output subfolders...
'/home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/results/Results' folder ready!
'/home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/results/Plots' folder ready!
'/home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/results/.tmp' folder ready!
Creating Pooling Control!
Warning message:
In dir.create(file.path(DataFolder, "Control")) :
  '/home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/results/Control' already exists
Warning message:
In dir.create(file.path(DataFolder, "Control", "RCNorm")) :
  '/home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/results/Control/RCNorm' already exists
Pooling Control Created!
Preparing Multiprocessor Analysis...
Multiprocessor Analysis Complete
Starting Multiprocessor Analysis!
My Children: 10574
Starting Segmentation and Calling...
Read 23 items
Error in JointSegIn(DataSeq1, muk, mi, smu, sepsilon, Pos1, omega, eta,  :
  NA/NaN/Inf in foreign function call (arg 3)
Execution halted
Segmentation and Calling Complete
Error in file(file, "rt") : cannot open the connection
Calls: read.table -> file
In addition: Warning message:
In file(file, "rt") :
  cannot open file '/home/chris/starr_lab/ocpmi/methods/EXCAVATOR2_Package_v1.1.2/data/output/results/Results/g1b/HSLMResults_g1b.txt': No such file or directory
Execution halted
Multiprocessor Analysis Complete!

```

</details>

### VarScan2

While waiting to hear back from the excv2 authors, I've decided to see where VarScan2 can take me.

Project Home

* https://dkoboldt.github.io/varscan/index.html
* git: https://github.com/dkoboldt/varscan
* dir: `/home/chris/starr_lab/ocpmi/methods/varscan2`
* man: https://dkoboldt.github.io/varscan/using-varscan.html

I need to build a pileup file with `samtools`. I don't know how long that will take, so I'm letting that run on msi. Currently, I am planning to use the UCSC hg19 fasta reference with BAMs from the 1kgp (NA10847, NA19131).

### Summary

I've hit an impasse with excv2 and am putting it on hold without new information from the authors. I've reached out to them for a pilot dataset since I believe file inconsistencies are playing a role in the pipeline not running to completion. So as to not waste time, I've started the process of testing out VarScan2. I've downloaded the .jar file and am getting setup to run on msi. Taking a cursory glance, it seems like VarScan2 has a similar executional structure as excv2 in that it is also implemented through command line arguments. I expect to spend all of tomorrow focused on seeing how far I can get with this.


#### Completed

```{r}
Created    Completed  Age Project Description
---------- ---------- --- ------- --------------------------------------------
2019-07-25 2019-07-29 4d  lab     email rebecca re bam files
2019-07-23 2019-07-29 6d  lab     reach out to christy re her wes work
2019-07-26 2019-07-29 3d  lab     find pilot bam files for excavator2 test run
2019-07-29 2019-07-29 6h  lab     reach out to excv2 devs re test dataset
2019-07-26 2019-07-29 2d  lab     summarize excv2 process in methods.Rmd
2019-07-26 2019-07-29 3d  lab     continue to troubleshoot excavator2 pipeline

6 tasks
```

***

## 30

### TODO

```{r}
ID Age P Project Tag    Description                                      Urg
-- --- - ------- ------ ------------------------------------------------ ----
65 6d    lab     n work begin pilot with data from RL                    1.93
66 6d    lab     n work explore application of vcf files in cnv pipeline 1.93
67 6d    lab     n work explore freebayes params                         1.93
70 4d    lab     n work contact oit re update policy and process         1.92
71 4d    lab     n work start back up process for ccgd server            1.92
76 18h   lab     n work begin varscan2 testing                            1.9
40 10w L lab     w work review scripts for ccgd database                 0.28
41 10w L lab     w work explore shiny framework for database             0.28

8 tasks
```

### vs2 pileup file

VarScan2 (vs2) requires a pileup to operate. I'm running that on both msi and my personal machine as it seems to take a long time.

```{r}
samtools mpileup -f [reference sequence] [BAM file(s)]
```

***

## 31

### TODO

```{r}
ID Age P Project Tags   Description                                         Urg
-- --- - ------- ------ --------------------------------------------------- ----
64 7d    lab     n work begin pilot with data from RL                       1.94
65 7d    lab     n work explore application of vcf files in cnv pipeline    1.94
66 7d    lab     n work explore freebayes params                            1.94
69 5d    lab     n work contact oit re update policy and process            1.93
70 5d    lab     n work start back up process for ccgd server               1.93
75 1d    lab     n work begin varscan2 testing                              1.91
40 10w L lab     w work review scripts for ccgd database                    0.29
41 10w L lab     w work explore shiny framework for database                0.29

8 tasks
```

### Collaborator mtg

* Speaking with Ying, it sounds like RL and CH may not have saved bam files in their wes processing. I may need to generate my own files from the raw data
    - looking at the directory, Tim was able to find fastq files for this
* Tim has generally completed the cellranger3 reprocessing.
    - Apparently, the updated pipeline identifies a different resolution of rna content in immune cells. As a result, the new processing gives an elevated immune cell content in the Seurat output.

### excv2 w/ 1kgp GR38

**excv2 pilot**

* samples: `NA10847`, `NA19131` via 1kgp phase3
* gene_ref: `hg38.fa`, `hg38.bed` via UCSC; `GCA_000001405.15_GRCh38.bw` via excv2

*msi install &rarr; GR38 ref &rarr; `TargetPerla.pl` &rarr; **`EXCAVATORDataPrepare.pl`** &rarr; `EXCAVATORDataAnalysis.pl` &rarr; cnv output*

### vs2 pileup creation

To get started on the vs2 pipeline, I've decided to work with the same 1kgp bam files I have available for excv2. In the event both work, I'll hopefully be able to compare the results.

**vs2 pilot**

* samples: `NA10847`, `NA19131` via 1kgp phase3
* gene_ref: `hg38.fa`, via UCSC

*BAMs &rarr; **`pileup` (samtools)** &rarr; [bam + pileup] `copynumber` (vs2) &rarr; cnv output?*

### BAM Creation

I need to create new bam files from our source wes data since these files were likely lost to scratch. I'll be doing a pilot run with two files that I will then run through each of excv2 and vs2.

**First Samples**

* src: `/home/nelsona2/data_release/umgc/hiseq/181205_D00635_0425_BCD07EANXX/Winterhoff_Project_001_SI`
* dir: `~/ocpmi/bam/ocpmi/pilot`
* samples: `PM-2-T17-0307-3_S9_R1_001.fastq`, `PM-15-T17-0910-1_S2_R1_001.fastq`
* gene_ref: `hg38.fa`, via UCSC

*fastq &rarr; **`index` (bwa bwtsw)** &rarr; align (bwa mem) + `sort` (samtools) &rarr; output.bam*

#### bwa

I'll be using bwa for sequence alignment.

* man: http://bio-bwa.sourceforge.net/bwa.shtml

Command for reference index creation using the bwtsw algorithm:

```{r}
bwa index -a bwtsw reference.fa
```

Command for direct bwa to bam output using the mem algorithm:

```{r}
bwa mem genome.fa reads.fastq | samtools sort -o output.bam -

# multiple threads

bwa mem -t 8 genome.fa reads.fastq | samtools sort -@8 -o output.bam -
```

### Summary

Currently, I have the vs2 and excv2 pipelines running in parallel. I've adjusted the approach of the excv2 pipeline to use the GR38 reference. I learned I will need to create new bam files since these were not preserved in scratch. To some extent this frees me to use the newer reference since I won't be relying on RL/CH's outputs.

***

# Aug

## 1

### TODO

```{r}
ID Age  P Project Tags   Description                                   Urg
-- ---- - ------- ------ --------------------------------------------- ----
64 8d     lab     w work explore freebayes params                      1.94
79 2min   lab     n work run pm2 and pm15 bams through vs2/excv2        1.9
80 1min   lab     n work evaluate excv2 output                          1.9
81 1min   lab     n work run vs2 copy number function on 1kgp bams      1.9
40 10w  L lab     w work review scripts for ccgd database              0.29
41 10w  L lab     w work explore shiny framework for database          0.29

6 tasks
```

### BAM Creation

I belive I've finished the process of converting a couple fastq files to bam files. I'm going to run these through the vs2 and excv2 pipelines to see if they will take.

* src: `/home/nelsona2/data_release/umgc/hiseq/181205_D00635_0425_BCD07EANXX/Winterhoff_Project_001_SI`
* dir: `~/ocpmi/bam/ocpmi/pilot`
* samples: `PM-2-T17-0307-3_S9_R1_001.fastq`, `PM-15-T17-0910-1_S2_R1_001.fastq`
* gene_ref: `hg38.fa`, via UCSC
* bam: `pm2.bam`, `pm15.bam`

**Process**

*fastq &rarr; `index` (bwa bwtsw) &rarr; align (bwa mem) + `sort` (samtools) &rarr;*  ***output.bam***

### vs2 Pilot

* samples: `NA10847`, `NA19131` via 1kgp phase3
* gene_ref: `hg38.fa`, via UCSC

*BAMs &rarr; `pileup` (samtools) &rarr; **[bam + pileup] `copynumber` (vs2)** &rarr; cnv output?*

### excv2 w/ 1kgp GR38

**excv2 pilot**

* samples: `NA10847`, `NA19131` via 1kgp phase3
* gene_ref: `hg38.fa`, `hg38.bed` via UCSC; `GCA_000001405.15_GRCh38.bw` via excv2

*msi install &rarr; GR38 ref &rarr; `TargetPerla.pl` &rarr; `EXCAVATORDataPrepare.pl` &rarr; **`EXCAVATORDataAnalysis.pl`** &rarr; cnv output

***

# References



