---
title: "Notes"
author: "Christopher Tastad"
date: \today
bibliography: library.bib
output:
    html_document:
        toc: yes
        toc_float: yes
        theme: paper
---

```{r}
library(devtools)
library(Seurat)
library(knitr)
library(kableExtra)

```

# May 2019

## 20

### Background mtg w/ Tim

* present: chris, tim

We discussed the outlook for my role in the lab. This included a considerable amount of background on the ocpmi project and potential directions I could take.

Focus Options

* ID of ovca stem cell
* what is the basis for clustering
* single cell seq methods paper
* compare mouse to human
* copy number variation on single cell
* rerun new version of cell ranger
* RHEL update


## 21

### TODO




ID  Age   Project        Tags       R Due        Description                Urg
--- ----- -------------- ---------- - ---------- -------------------------- ----
326 10h   starrlab       computer n R 2019-05-21 write daily summary        10.5
327 27min starrlab       computer n              find umn git and hosting    1.9
312 20h   starrlab.ocpmi computer n              organize lab notebook [1]   2.7
315 18h   starrlab.ocpmi computer n              find ovca review re cur     1.9
                                                 treatment
316 18h   starrlab.ocpmi computer n              read starr paper re 1st pt  1.9
317 18h   starrlab.ocpmi computer n              read ocpmi summary          1.9
318 18h   starrlab.ocpmi computer n              install surat               1.9
319 18h   starrlab.ocpmi computer w              run surat tutorial w/ pt    1.9
                                                 59
320 18h   starrlab.ocpmi computer n              read single cell paper      1.9
                                                 zheng 2017
321 18h   starrlab.ocpmi computer n              find and read copy num      1.9
                                                 variation paper
322 18h   starrlab.ocpmi computer n              organize refs in email      1.9
                                                 inbox
323 18h   starrlab.ocpmi computer n              compile bibtex ref page     1.9

12 tasks


### Citation packages

#### Rmarkdown Default

This is the intrinsic formatting style for Rmarkdown citation management. It has the ability to call a BibTex file and carries the least effort to get it working. It's what I will probably go for across my whole setup. [@citations]

#### knitcitations

The knitcitations package might provide the best balance of feature set to ease of use. The only constraining factor is what seems like a few bugs that limit intended functionality. The ref() function doesn't work for instance. [@cboettig_2019]

#### RefManageR

This is a fully featured R package that seems to attempt to replace a complete citation manager. It comes across as very comprehensive but more complicated than I would like the approach to be. [@refmanager]

### Summary

#### Completed

* 327, 312:progress

The majority of the day was spent exploring the setup for my lab notebook. My intention is to create a living document using the Rmarkdown website knit feature [@rmarkdown_cheatsheet]. This will allow for a dynamic storage of both records and code work inline with one comprehensive document. A large portion of this time was spent giving consideration to how I will handle citations both in the document and for future publication. After significant exploration, my plan is to use the intrinsic Rmarkdown system [@citations] through use of a bibtex file. This will allow for simple reference in-place with the notebook along with using a standardized format for publication ready references.
Additionally, I plan on establishing a git repo to manage version control and historical record with the notebook as well using the umn github server [citation needed].

***

## 22

### TODO



ID  Age Project        Tags       R Due        Description                  Urg
--- --- -------------- ---------- - ---------- ---------------------------- ----
326 15h starrlab       computer n R 2019-05-22 write daily summary          10.5
312 1d  starrlab.ocpmi computer n              organize lab notebook [1]    2.71
315 1d  starrlab.ocpmi computer n              find ovca review re cur      1.91
                                               treatment
316 1d  starrlab.ocpmi computer n              read starr paper re 1st pt   1.91
317 1d  starrlab.ocpmi computer n              read ocpmi summary           1.91
318 1d  starrlab.ocpmi computer n              install surat                1.91
319 1d  starrlab.ocpmi computer w              run surat tutorial w/ pt 59  1.91
320 1d  starrlab.ocpmi computer n              read single cell paper zheng 1.91
                                               2017
321 1d  starrlab.ocpmi computer n              find and read copy num       1.91
                                               variation paper
322 1d  starrlab.ocpmi computer n              organize refs in email inbox 1.91
323 1d  starrlab.ocpmi computer n              compile bibtex ref page      1.91

11 tasks


### wkly mtg w/ Tim

present: chris, tim

* Make sure to utilize notes field in bibtex file. Having an ongoing record of each paper read is critical.
* Consider separating data and results in file organization.
* Change notebook structure to follow layout of paper. Organize the work with a mindset toward writing the paper from the start, not the finish.
* Currently, we are still working to establish the standardized parameters of the data set. The goal is to arrive at a place where we can make biological conclusions.
    + How do we draw contrast from a data set of this nature?

### Notebook Structure

In my conversation with Tim, I decided I need to take an alternate structure to my lab notebook. My original intent was to emulate a development oriented documentation process. Instead, I will go a route that better follows the academic writing process with pages dedicated to methods, results, etc.

### Reference Management

While I would prefer to take a lean, text based approach to this, I am clearly beginning to waste some time in gear the system. I came across jabref, which seems to leverage a unix philosophy and can be found in the ubuntu repos.

### Summary

#### Completed

* 312:progress, 323:progress

My work today included continued focus on getting organized an taking a very deliberate approaching to my workflow before I begin. I met with Tim to begin our weekly meetings, and we covered some aspects of this workflow. Major take aways include the importance of annotation when reviewing literature. I had originally assumed I could handle references processing merely in a bare bibtex file. My feeling now is that I will begin to use a dedicated application for greater consistency in documentation.

***

## 23

### TODO



ID  Age Project        Tags       R Due        Description                  Urg
--- --- -------------- ---------- - ---------- ---------------------------- ----
334 11h starrlab       computer n R 2019-05-23 write daily summary          10.6
327 23h starrlab       computer n              reorg notebook to follow      1.9
                                               paper structure
329 19h starrlab       computer n              setup jabref                  1.9
330 19h starrlab       lab n                   setup desk                    1.9
331 19h starrlab       computer n              create agenda field           1.9
332 19h starrlab       computer n              create lit tag                1.9
333 19h starrlab       computer n              create lab context            1.9
336 41s starrlab       computer n              create website knit script    1.9
312 2d  starrlab.ocpmi computer n              organize lab notebook [1]    2.71
315 2d  starrlab.ocpmi computer n              find ovca review re cur      1.91
                                               treatment
316 2d  starrlab.ocpmi computer n              read starr paper re 1st pt   1.91
317 2d  starrlab.ocpmi computer n              read ocpmi summary           1.91
318 2d  starrlab.ocpmi computer n              install surat                1.91
319 2d  starrlab.ocpmi computer w              run surat tutorial w/ pt 59  1.91
320 2d  starrlab.ocpmi computer n              read single cell paper zheng 1.91
                                               2017
321 2d  starrlab.ocpmi computer n              find and read copy num       1.91
                                               variation paper
322 2d  starrlab.ocpmi computer n              organize refs in email inbox 1.91
323 2d  starrlab.ocpmi computer n              compile bibtex ref page      1.91

18 tasks


### Citations

#### Custom CSL

At some point, I would like to setup a custom csl file to format my references. The goal here would be to add formatted features that are note readily available in some existing format style. [@2019], [@BibEntry2019May]

#### JabRef

I've started to figure out jabref. I think I will still look to my bibtex file as the primary source for citation management, but will make use of the interface and some of the features of jabref. Additionally dio2bib will play a central role in extracting bibtex content. [@2019b]

### GitHub

The setup for my git server is complete. It's a private server host by the university a /tasta005/ocpmi [@ocpmi_git]

### Summary

I was able to round out much of my organizational setup today. I feel confident about the citation management strategy I plan to use going forward which combines the use of dio2bib for bibtex creation, BibItNow for website extraction, and JabRef to centralize management. Additionally, I was able to connect my relevant directories to my git server, which will account for both a measure of backup and version control. Having gotten settled into these systems I think I will be able to begin incorporating my literature search into my daily activities.

#### Completed


Created    Completed  Age Project        R Due        Description
---------- ---------- --- -------------- - ---------- --------------------------
2019-05-22 2019-05-23 1d  starrlab       R 2019-05-22 write daily summary
2019-05-22 2019-05-23 1d  starrlab                    setup desk
2019-05-23 2019-05-23 16h starrlab       R 2019-05-23 write daily summary
2019-05-20 2019-05-23 3d  starrlab.ocpmi              organize lab notebook
                                                        2019-05-21 record,
                                                      master, comms, refs, home,
                                                      docs
2019-05-22 2019-05-23 1d  starrlab                    create lab context
2019-05-22 2019-05-23 1d  starrlab                    create lit tag
2019-05-22 2019-05-23 1d  starrlab                    setup jabref
2019-05-20 2019-05-23 3d  starrlab.ocpmi              compile bibtex ref page
2019-05-20 2019-05-23 3d  starrlab.ocpmi              organize refs in email
                                                      inbox
2019-05-22 2019-05-23 1d  starrlab                    setup umn git
2019-05-22 2019-05-23 1d  starrlab                    create agenda field

11 tasks

***

## 24

### TODO


ID  Age Project        Tag            Description                           Urg
--- --- -------------- -------------- ------------------------------------- ----
314 3d  starrlab.ocpmi computer lit n find ovca review re cur treatment     2.02
315 3d  starrlab.ocpmi computer lit n read starr paper re 1st pt            2.02
316 3d  starrlab.ocpmi computer lit n read ocpmi summary                    2.02
319 3d  starrlab.ocpmi computer lit n read single cell paper zheng 2017     2.02
320 3d  starrlab.ocpmi computer lit n find and read copy num variation      2.02
                                      paper
317 3d  starrlab.ocpmi computer n     install surat                         1.92
318 3d  starrlab.ocpmi computer w     run surat tutorial w/ pt 59           1.92
325 1d  starrlab       computer n     create website knit script            1.91
327 22h starrlab       comp w         make custom csl                        1.9
328 22h starrlab       lab n          complete lab training                  1.9
330 20h starrlab       w              agenda items                           1.8

11 tasks

### create_site Script

I created a small R script that will effectively knit each page of my lab notebook. This can be executed in the R console or in bash, but each file has to be saved. The save -> knit function that's performed by the IDE does not apply to the console executed render command.

```{r eval=F}
rmarkdown::render("index.Rmd")
rmarkdown::render("notes.Rmd")
rmarkdown::render("results.Rmd")
rmarkdown::render("methods.Rmd")
rmarkdown::render("references.Rmd")
rmarkdown::render_site("index.Rmd")
```


### Summary

Today was filled with primarily administrative work. A large bulk of my time was taken by lab safety training, which I was unable to fully complete. Along with this, I was finally able to install Seurat and tinker with the package tutorial. It seems like the version I downloaded may not be compatible with the tutorial code, so I will have to explore that more.

#### Completed


Created    Completed  Age Project        Description
---------- ---------- --- -------------- ---------------------------------------
2019-05-23 2019-05-24 1d  starrlab       create website knit script
2019-05-24 2019-05-24 2h  starrlab       DEHS Radiation Safety Orientation
                                         Training
2019-05-24 2019-05-24 2h  starrlab       Implementation of NIH Guidelines
2019-05-24 2019-05-24 2h  starrlab       Bloodborne Pathogens: Advanced
2019-05-24 2019-05-24 2h  starrlab       DEHS Chemical Safety Training
2019-05-24 2019-05-24 2h  starrlab       Biological Safety in the Laboratory
2019-05-24 2019-05-24 2h  starrlab       Bloodborne Pathogens: Introduction and
                                         Advanced
2019-05-24 2019-05-24 2h  starrlab       DEHS Introduction to Research Safety
                                         Training
2019-05-24 2019-05-24 2h  starrlab       Animal Use Tutorial
2019-05-23 2019-05-24 1d  starrlab       complete lab training
2019-05-20 2019-05-24 4d  starrlab.ocpmi install surat

11 tasks

***

## 27

### TODO


ID Age Project        Tags           Description                            Urg
-- --- -------------- -------------- -------------------------------------- ----
77 4d  starrlab       computer w     make custom csl                        1.92
79 3d  starrlab       computer n     GCP Human RSRCH Biomed Studies         1.92
80 3d  starrlab       computer n     CITI research ethics training          1.92
81 3d  starrlab       computer n     University Information Security        1.92
                                     Awareness Training
78 4d  starrlab                      agenda items                           1.02
91 15h starrlab.ocpmi computer lit n read papalexi 2018 [2]                  2.9
68 6d  starrlab.ocpmi computer lit n find ovca review re cur treatment      2.03
69 6d  starrlab.ocpmi computer lit w read starr paper re 1st pt             2.03
70 6d  starrlab.ocpmi computer lit w read ocpmi summary                     2.03
72 6d  starrlab.ocpmi computer lit w read single cell paper zheng 2017      2.03
73 6d  starrlab.ocpmi computer lit w find and read copy num variation paper 2.03
71 6d  starrlab.ocpmi computer w     run surat tutorial w/ pt 59            1.93

12 tasks


###[@Lheureux2019]

#### Evolution of Management in the Era of Precision Medicine  (1/2)

* State of field for ovarian cancer
* Current treatments and outcomes
* Ovca presents very late. The internal location of the cancer and general lack of screening options lead to most cases being diagnosed at a later stage than many other cancers would.
* Treatment options have largely focused on surgical debulking and platinum based chemotherapy.
* Risk factors are inherited: germ line mutations in breast cancer susceptible genes (e.g. brca), nulliparity, infertility, endometriosis, obesity, age, potentially perineal talc.
* Reducing factors that have been found to be potentially significant are gravidity, oral contraceptive use, regular use of aspirin, and nsaids.
* Currently, population screening in ovca has not been found to be effective.
* Epithelial ovarian cancer accounts for 95% of ovarian malignancies
* 5 eoc subtypes: high-grade serous, low-grade serous, clear cell, endometrioid, and mucinous
* HGSOC has the highest incidence with 70% of eoc cases.
* It is believed fallopian tube epithelium is the possible site for tumor initiation.
* Mutations in p53 are seen in +90% cases of HGSOC
* p53 mutations and brca inactivation lead to high numbers of copy variation.
* HGSOC has 4 prognostic subtypes: differentiated, immunoreactive, mesenchymal, and proliferative
* Immunoreactive is considered to have the best overall survival while mesenchymal has the worst.
* Ovca cells also over express folate and mesothelin which have been targets for antibody therapy.
* LGSOC is seen in about 10% of cases
* LGSOC is considered to develop in a pathway which is very distinct from HGSOC
* B-Raf and KRAS mutations are the most commonly present in LGSOC
* Hormone therapy is sometimes used due to the presence of estrogen/progesterone receptors.
* CNAs in p53 are rare, and brca inactivation seem to unrelated to LGSOC

### Summary

Just worked my way through the above summary today. Will continue and complete reading this lit tomorrow.


***

## 28

### TODO


ID Age Project        Tags           Description                            Urg
-- --- -------------- -------------- -------------------------------------- ----
77 4d  starrlab       computer w     make custom csl                        1.92
79 3d  starrlab       computer n     GCP Human RSRCH Biomed Studies         1.92
80 3d  starrlab       computer n     CITI research ethics training          1.92
81 3d  starrlab       computer n     University Information Security        1.92
                                     Awareness Training
78 4d  starrlab                      agenda items [1]                       1.82
91 1d  starrlab.ocpmi computer lit n read papalexi 2018 [2]                 2.91
92 17h starrlab.ocpmi computer lit w read macintyre 2018 [2]                 2.9
68 7d  starrlab.ocpmi computer lit n find ovca review re cur treatment      2.04
69 7d  starrlab.ocpmi computer lit w read starr paper re 1st pt             2.04
70 7d  starrlab.ocpmi computer lit w read ocpmi summary                     2.04
72 7d  starrlab.ocpmi computer lit w read single cell paper zheng 2017      2.04
73 7d  starrlab.ocpmi computer lit w find and read copy num variation paper 2.04
71 7d  starrlab.ocpmi computer w     run surat tutorial w/ pt 59            1.94

13 tasks


### [@Lheureux2019]

#### Evolution of Management in the Era of Precision Medicine (2/2)

* OCCC or clear cell carcinoma represents about 5% of cases.
* The pathogenesis of OCCC is not well understood.
* A proposed mechanism is related to the level of iron content, promoting oxidative stress.
* Mutations in the SWI/SNF and PI3K pathway are represented in large proportions of cases.
* Activation of the PI3K or RTK pathway has resulted in better OS
* OCCCs are found to have considerably fewer CNAs and mutations in p53 or brca genes.
* Endometrioid cancers represent 10% of EOC cases.
* Current thought is that endometrioid and OCCC arise from similar pathways.
* Deregulation of WNT/b-catenin and PI3K pathways are essential in mouse models.
* Mucinous OC is a rare subtype.
* Could be classified as non-invasive and infiltrative.
* KRAS mutations are a common feature of this subtype.
* p53 and brca mutations do not seem to play a role.
* Treatment
* To date, there has been no randomized assessment of debulking surgery, platinum therapy, or radiation therapy.
* Surgery has been the bedrock of therapy.
* Surgery involves many approaches including primary, interval, and secondary debulking
* Chemotherapy
* carboplatin is as effective as cisplatin and is better tolerated.
* dose is important.
* pairing drugs does not improve outcomes.
* IP chemo is often used due to the tight localization of ovca in the abdomen
* hyperthermic IP chemo is currently being explored with reasonable results
* Treatment after recurrence follows several judgments of benefit.
* Recurrence after a period of over 6mo suggests platinum sensitivity where platinum based chemo would be reapplied
* Shorter intervals of recurrence employ non-platinum based therapies
* Current consensus is the use of carboplatin and paclitaxel together every 3wks
* Targeted therapy has been gradually introduced in recent years.
* Bevacizumab has shown the greatest promise among antiangiogenics being the only one in clinical use.
* PARP inhibitors
* Used as maintenance with platinum therapy at the time of recurrence.
* A trial showed that the key benefit in PFS was driven by the parp inhibitor maintenance therapy.
* Trial from 2018 led to first-line maintenance in brca mutated cases
* Stratified approach to systemic therapy can be used with bevacizumab after surgery with a switch to maintenance on the basis of brca mutations.
* Immunotherapy is currently being explored on several fronts though none are in clinical practice.
* ovca has a high expression of PD-L1.
* Checkpoint inhibitors targeting PD-L1/PD1 or CTLA-4 have shown only modest results.
* Autologous cell therapy is also being utilized.
* A variation of this therapy is exploring car-t cells targeting mesothelin
* Vaccines have not worked well due to the highly immunosuppressive cancer environment
* A vaccine targeting survivin is currently being explored
* Additionally, the concept of the patient-specific mutanome is being researched in an effort to develop patient-specific cancer vaccines.
* OCCC and LGSOC are relatively chemoresistant.
* Prevention and screening are currently limited.
* There is no validated screening method for standard practice.
* Ultrasound has produced false-negatives and false-positives and is not recommended.
* Screening high-risk populations has been shown to be effective but requires prior knowledge of family history.
* Some biomarkers have been utilized such as HE4 in combination with CA125.
* Another approach has been to evaluate DNA methylation in cfDNA.










### Summary

I have shifted focus to begin a comprehensive literature review. I've started by taking a look at a review of the current landscape of treatment and outcomes for ovca [@Lheureux2019]. This was helpful to fill many of the holes I have in my general knowledge of cancer biology with the more specific case studies of ovca. Much of this reinforced the review that Tim has given in our meetings, which was helpful. Along with this, I began to work my way through the Seurat tutorial. I was able to locate the v3 tutorial and pbmc data set and made it about half way through the written guide. I plan on getting fully through this tomorrow as well as meeting with Tim to discuss some questions I developed from the ovca review.

#### Completed


Created    Completed  Age Project        Description
---------- ---------- --- -------------- ---------------------------------
2019-05-20 2019-05-28 7d  starrlab.ocpmi find ovca review re cur treatment

1 task

***

## 29

### TODO

ID Age  Project        Tags           Description                           Urg
-- ---- -------------- -------------- ------------------------------------- ----
75 5d   starrlab                      agenda items [3]                      2.03
74 6d   starrlab       computer w     make custom csl                       1.93
76 5d   starrlab       computer n     GCP Human RSRCH Biomed Studies        1.93
77 5d   starrlab       computer n     CITI research ethics training         1.93
78 5d   starrlab       computer n     University Information Security       1.93
                                      Awareness Training
90 1d   starrlab       computer n     find alternative to jabref            1.91
92 22h  starrlab       computer n     fix god damn git issue                 1.9
97 1min starrlab       computer n     fix hostname issue w/ mesabi           1.9
88 2d   starrlab.ocpmi computer lit n read papalexi 2018 [2]                2.91
89 1d   starrlab.ocpmi computer lit w read macintyre 2018 [2]               2.91
91 1d   starrlab.ocpmi lit s          read sahin 2017 [1]                   2.71
66 8d   starrlab.ocpmi computer lit w read starr paper re 1st pt            2.04
67 8d   starrlab.ocpmi computer lit w read ocpmi summary                    2.04
69 8d   starrlab.ocpmi computer lit w read single cell paper zheng 2017     2.04
70 8d   starrlab.ocpmi computer lit w read venteicher - CNV paper           2.04
68 8d   starrlab.ocpmi computer w     run surat tutorial w/ pt 59           1.94
96 2min starrlab.ocpmi computer n     run seurat with pbmc data              1.9

17 tasks


### wkly mtg w/ Tim

present: chris, tim

```{r eval=F}


Description
--------------------------------------------------------------------------------
agenda items
  2019-05-27 validation in diff between lumpectomy and mastectomy
  2019-05-28 elaborate on lack of randomized controls for ovca treatments
  2019-05-28 discuss vacation
  2019-05-28 discuss chemotherapy at large
  2019-05-28 review parp inhibitor mechanism
  2019-05-28 are occc and lgsoc our general targets given their chemoresistance?
  2019-05-28 how do our questions differ from the findings of kondrashova 2017 -
RAD51 mutations acquired resistance

1 task
```

#### mtg notes

* mysql reference: single cell rnaseq analysis protocol
* figure out mesabi hostname issue
* explore starrt2 MSI path
    + [/home/starrt2/data_release/umgc/hiseq/170627_D00635_0259_ACB8R1ANXX/Starr_Project_037/Analysis/cellranger-Starr_037/filtered_gene_bc_matrices/GRCh38]
* There is no specific focus in the pathology we're focusing on. OCCC and LGSOC are being considered just as readily as other subtypes.
* Kondrashova 2017 (RAD51 mutations leading to chemoresistance) is potentially a good example of what our research questions will be.
* Tim's focus is more basic science oriented. Potential applications would be to identify when a pt would be resistant and that they should pursue other treatment from the outset.
* All of our findings from this data set will require wetlab validation
* This data set is incredibly unique. We should be able to discover something nobody else has.
* Data validation has generally been handled on the project so far.
    + Objective: determine cell type.
* There's no way to call copy number variation from this data. That would be novel.

### MSI Project Directory

I logged into the MSI account for the first time today. There is a relatively deep file structure which is composed of a combination of processed data from umgc/MSI and other collaborator work. I have a personal directory in this space along with access to other shared directories.

The root project directory is:

```{r eval=F}
/home/starrt2/
```

I have a personal directory under the root. At the same level, there is also a shared directory which includes some content that is under active collaboration.

```{r eval=F}
/home/starrt2/tasta005
```

Processed data can be found in the hiseq folder under data_release. This data is mirrored in the MSI file structure. Each file set is named by a unique ID which follows a naming convention given by umgc.

* Look for the leading 6 digit integer as a point of reference for the data_release folder.
* The Starr_Project sub folder is given a number which is considered the MSI and sample ID.

```{r eval=F}
/home/starrt2/data_release/umgc/hiseq
```


The Analysis directory contains the source data

* illumina-basicQC # quality control data
* cellranger-`*`
    + analysis # ?
    + filtered_gene_bc_matricies # data source that has been filtered by "true cell content"
    + raw_gene_bc_matrices # matrix data that includes unfiltered reads or all read content
    + web_summary.html # interactive cell ranger summary

An example of a complete data source location is below:

```{r eval=F}
/home/starrt2/data_release/umgc/hiseq/170627_D00635_0259_ACB8R1ANXX/Starr_Project_037/Analysis/cellranger-Starr_037/filtered_gene_bc_matrices/GRCh38
```

### Summary

I met with Tim today for our weekly meeting. We discussed some aspects of ovca content that I had been reading to develop my general knowledge. I major element of this conversation that will adjust what my intentions were going forward dealt with the nature of statistically validating the results of our pipeline. My original thought was that there was some need to dig into the backend of the Seurat/cellranger packages to both confirm and optimize the output. It sounds as though this has been done to a degree that Tim is satisfied with and that we can move on to explore questions that are Biological in nature.

Additionally, I did a little house keeping by establishing the first part of the methods page in my notebook. I explored the MSI project directory for the first time today and recorded some documentation about the file structure and data locations. My expectation is to pull what would be method oriented documentation from Notes -> Methods or Notes -> Results as these things need to be pinned for repeat use and reference.

#### Completed

Created    Completed  Age Project  Description
---------- ---------- --- -------- ----------------------------
2019-05-28 2019-05-29 1d  starrlab fix god damn git issue
2019-05-29 2019-05-29 2h  starrlab fix hostname issue w/ mesabi

2 tasks

***

## 30

### TODO


ID Age Project        Tags           Description                            Urg
-- --- -------------- -------------- -------------------------------------- ----
75 6d  starrlab                      agenda items [3]                       2.03
74 6d  starrlab       computer w     make custom csl                        1.93
76 5d  starrlab       computer n     GCP Human RSRCH Biomed Studies         1.93
77 5d  starrlab       computer n     CITI research ethics training          1.93
78 5d  starrlab       computer n     University Information Security        1.93
                                     Awareness Training
90 1d  starrlab       computer n     find alternative to jabref             1.91
88 3d  starrlab.ocpmi computer lit n read papalexi 2018 [2]                 2.92
89 2d  starrlab.ocpmi computer lit w read macintyre 2018 [2]                2.91
91 1d  starrlab.ocpmi lit s          read sahin 2017 [1]                    2.71
66 9d  starrlab.ocpmi computer lit w read starr paper re 1st pt             2.05
67 9d  starrlab.ocpmi computer lit w read ocpmi summary                     2.05
69 9d  starrlab.ocpmi computer lit w read single cell paper zheng 2017      2.05
70 9d  starrlab.ocpmi computer lit w read venteicher - CNV paper            2.05
68 9d  starrlab.ocpmi computer w     run surat tutorial w/ pt 59            1.95
95 20h starrlab.ocpmi computer n     run seurat with pbmc data               1.9

15 tasks

### Seurat Tutorial pbmc3k

I ran the v3 Seurat tutorial [@2019d] to completion using the pbmc dataset. This outline was very intuitive and worked through the pipeline seamlessly. There are several points that warrant further exploration.

The output of the v3 PBMC data tutorial can be found [here](/home/chris/starr_lab/ocpmi/code/seurat_tutorial/pbmc3k_tutorial.html).


### Summary

Briefly, I wrapped up safety training today, so that is all taken care of. The rest of my afternoon was dedicated to working my way through the default v3 Seurat tutorial. The tutorial comes packaged in a markdown file, so the presentation and documentation is very effective. To some extent, the documentation is so good I am a little concerned it encourages over use of the default settings. I haven't looked at any other processing that's been completed by Tim et al, so I'll have to keep an eye out for how adjustments have been made. Tim mentioned there has been a limited amount of customization, so this may stand as a short term area of focus. Generally, I think can consider today's work a reasonable summary of the process and syntax associated with the toolset. With this, I would like to spend some time digging into the nuances of the pipeline's analysis. I will probably spend most of tomorrow doing that. Much of the default analysis hinges on PCA dimensionality, so I feel I need to brush up on my knowledge in this area. Going forward I need to run the same pipeline using the new pt59 data and evaluate results.

#### Completed


Created    Completed  Age Project        Description
---------- ---------- --- -------------- ---------------------------------------
2019-05-24 2019-05-30 6d  starrlab       GCP Human RSRCH Biomed Studies
2019-05-24 2019-05-30 6d  starrlab       CITI research ethics training
2019-05-24 2019-05-30 6d  starrlab       University Information Security
                                         Awareness Training
2019-05-29 2019-05-30 1d  starrlab.ocpmi run seurat with pbmc data

4 tasks

***

## 31

### TODO

ID Age Project        Tags       Description                                Urg
-- --- -------------- ---------- ------------------------------------------ ----
73 8d  starrlab                  agenda items [3]                           2.04
72 8d  starrlab       computer w make custom csl                            1.94
83 3d  starrlab       computer n find alternative to jabref                 1.92
91 1d  starrlab       computer n review pca concepts                        1.91
92 1d  starrlab       computer n compare default seurat tutorial to         1.91
                                 starrlab protocol
93 1d  starrlab       computer n deep dive into default seurat analysis     1.91
94 1d  starrlab       computer n update notes regarding citation tools re   1.91
                                 mendeley
81 4d  starrlab.ocpmi lit n      read papalexi 2018 [2]                     2.82
82 4d  starrlab.ocpmi lit w      read macintyre 2018 [2]                    2.82
84 3d  starrlab.ocpmi lit s      read sahin 2017 [1]                        2.72
65 11d starrlab.ocpmi lit w      read starr paper re 1st pt                 1.96
66 11d starrlab.ocpmi lit w      read ocpmi summary                         1.96
67 11d starrlab.ocpmi computer w run surat tutorial w/ pt 59                1.96
68 11d starrlab.ocpmi lit w      read single cell paper zheng 2017          1.96
69 11d starrlab.ocpmi lit w      read venteicher - CNV paper                1.96

15 tasks

### Mendeley

At the risk of seeming indecisive, I've decided to switch citation managers. Given my use if the BibTex file format, it is effectively painless to switch. JabRef showed some promise but presented a few bugs related to database calls which would render much of the value of using it worthless. Mendeley has both a native linux application along with a web import tool for firefox. This is the only major citation tool I've come across that combines those two things. Here are a few key features of my use:

* All content will still be synced back to a BibTex file name `library.bib` located under the `starr_lab/ocpmi/notebook/` subdirectory
* There are watched folders which will draw sources from any pdf put into them at `starr_lab/ocpmi/ref/`.
* I've decided to use the IEEE with url csl style taken from the zotero style repository. [@2019]

### [@Winterhoff2019]

#### Single-Cell Sequencing in Ovarian Cancer: a new frontier in precision Medicine

Key Points

* Intratumor heterogeneity is an inherent property of ovarian cancer, which impacts clinical outcomes
* SC sequencing tech enables characterization of intratumor heterogeneity in ovarian cancer
* To date, single-cell studies on ovarian cancers are scarce, which limits the opportunity to investigate effects on clinical outcomes
* Larger data sets are needed to establish potential associations between the unique information captured by SC sequencing and clinically relevant outcomes in ovca

Notes

* They claim they found genes associated with ovca stem cells.
* When they evaluated based on molecular eoc cancer subtypes, all subtypes were present.
* Studying the immune microenvironment of a tumor becomes more feasible with scRNA sequencing
* Evaluation of cell plasticity as it relates to cell type transitions may provide insight into the development of chemoresistance
* Understanding the lineage of the cellular qualities of metastatic dissemination is made possible with scRNA data

### Using Ancestry to find the Stem Cell

Question: can we use sequence-based lineage tracing to identify a common progenitor cell expression profile?

A thought I had while looking at some lit tonight led me to wonder if it's worth exploring sequence-based cell lineage across individual cells rather than using differential expression patterns as a general means of characterization. I will need to do a dedicated lit search to explore this concept, but I imagine the use of single cell data for lineage tracing is relatively limited. At a minimum, Tim has alluded to the notion that our data set is unique among the ovca model, which would lend itself to a novel application of single cell lineage tracing. Regardless, this could be a relatively significant deviation from their current pipeline, so I will have to discuss it with Tim.

### Updated create_site.R Script

I made a minor update to the script I use to automate total site rendering to clean up the directory a bit. I don't have any use in the parent notebook/ directory for individual html files since they are generated in the site render. The `ifelse` at the end of the script clears these html files from the notebook directory.

```{r eval=F}
rmarkdown::render("index.Rmd")
rmarkdown::render("notes.Rmd")
rmarkdown::render("results.Rmd")
rmarkdown::render("methods.Rmd")
rmarkdown::render("references.Rmd")
rmarkdown::render_site("index.Rmd")

fl <- Sys.glob("*.html")

ifelse(file.exists(fl), file.remove(fl))
```

### Summary

I was able to explore a few different fronts today. I put some more work in to polishing my citation management. This is an area I have been woefully lacking in from past experience, so I do not discount the extra work applied to make sure this is at a publication-ready level.

Additionally, I worked my way through the earliest publication related to the ocpmi project [@Winterhoff2019]. Through this I was able to develop some insight to the early considerations this group gave to this data as they were only working with 66 individual cells. They offered a broad scope of the advantages single cell tech offers along with pointing me to a few sources that establish the current state of the field.

Last, I can't say what generated this thought specifically, but I came across what my limited knowledge of the field would call a good idea. All of our discussions on this data set up to this point have centered on the use of UMI counts and gene expression levels as a means to develop differentiated cell profiles. A consideration I would potentially like to explore is the possibility that we look at sequence level data to identify cell lineage in search of a common progenitor. I have little basis to say if this is a reasonable or even realistic scientific question, but I feel this could be a substantive, novel insight given the nature of our data set. It's quite possible this approach would require too great an investment in shifting the informatics work toward raw sequence analysis, but I will discuss it with Tim nonetheless.

#### Completed

Created    Completed  Age Project        Description
---------- ---------- --- -------------- --------------------------
2019-05-20 2019-06-01 11d starrlab.ocpmi read starr paper re 1st pt
2019-05-28 2019-05-31 3d  starrlab       find alternative to jabref
2019-05-30 2019-05-31 1d  starrlab       update notes regarding citation tools re
                                         mendeley
2019-05-23 2019-05-31 8d  starrlab       make custom csl

4 tasks


***

# June

## 3

### TODO

ID Age Project        Tags       Description                                Urg
-- --- -------------- ---------- ------------------------------------------ ----
52 10d starrlab                  agenda items [2]                           1.95
74 2d  starrlab       computer n add notes about cancer stem cell database  1.91
49 13d starrlab.ocpmi lit w      read tirosh 2016 [2]                       2.87
56 7d  starrlab.ocpmi lit n      read papalexi 2018 [2]                     2.84
57 6d  starrlab.ocpmi lit w      read macintyre 2018 [2]                    2.83
58 5d  starrlab.ocpmi lit s      read sahin 2017 [1]                        2.73
67 2d  starrlab.ocpmi lit n      do lit search on cell ancestry [1]         2.71
46 13d starrlab.ocpmi lit w      read ocpmi summary                         1.97
47 13d starrlab.ocpmi computer w run surat tutorial w/ pt 59                1.97
48 13d starrlab.ocpmi lit w      read single cell paper zheng 2017          1.97
63 3d  starrlab.ocpmi computer n review pca concepts                        1.92
64 3d  starrlab.ocpmi computer n compare default seurat tutorial to         1.92
                                 starrlab protocol
65 3d  starrlab.ocpmi computer w deep dive into default seurat analysis     1.92
69 2d  starrlab.ocpmi lit n      find cancer stem cell review               1.91

14 tasks

### Cancer Stem Cell Database

http://www.csctt.org/ [@csctt2019]

Tim sent me a link to a possible resource last week that could be useful in building references for exploring the cancer stem cell concept.

### LitSrch re Cancer Stem Cell

I've done some extremely high level review from the wikipedia article on CSCs [@csc_wiki]. This pointed me to several papers of relevance that I hope to look at in future exploration of the topic.

#### Notes

* While there has been significant research around the topic, CSCs remain a debated idea.
* There are currently a few models for CSC tumor propagation
    + The CSC model
        - This is also known as the hierarchical model
        - CSCs are a tumorigenic and biologically distinct subpopulation [@Beck2013].
        - Two defining features: ability to self-renew, capacity to generate pro-growth progeny
        - This model suggests only certain cells have the ability to drive the progression of cancer
    + Stochastic model
        - Also known as clonal evolution model
        - This model suggests the necessary mutations to achieve CSC status can happen to any cell at random.
        - All tumor cells retain the ability to develop plasticity and self-renew/differentiate [@Kreso2014].
        - Elements of resistance and fitness accumulate over time [@Beck2013].
        - CSCs with a growth advantage outperform other cells [@Barabe2007].
    + CSC Immunology model
        - Both stem cell and CSCs possess qualities that make them resistant to immune surveillance [@MacCalli2014].
        - CSCs may exist at all times but only propagate once 1) they shed dependence on niche factors or 2) they develop a means to escape immune surveillance or 3) the immune system loses tumorsuppressive capacity [@Bruttel2014].
* Debate
    + The existence of CSCs is still under debate because no consistently definable characteristics have been found [@Gupta2009].
    + The origin of CSCs is also a major topic.
        - Are these disregulated normal SCs
        - Are they instead a more specialized population that gained the ability to self-renew.
        - Is it something else entirely due to the modular plasticity cancer cells can show [@Nouri2017].
* Modeling
    + There has been some mathematical modeling taken to the growth of these cells.
    + The cell compartment method is one approach [@Ganguly2006].
    + Other models may exist and are worth exploring in the context of highlighting CSC mutation lineage.
* Origin
    + stem cell mutation
        - stem cell niche populations develop mutations which derail regular progression and can be past to progeny [@Wang2009].
    + adult stem cells
        - Stem cell populations in tissue with high rates of turnover experience frequent cell divisions. Increased likelihood of mutations leads to accumulated genetic defects that drive cancer initiation. [@Lopez-Lazaro2018]
    + De-differentiation
        - Cells are driven back to a state of stemness through some external factor.

### Summary

I focused mostly on lit search efforts today. I want to develop a better understanding of the current landscape of the CSC hypothesis, so I have taken a broad overview of different findings. I started with the wikipedia article on the subject and have branched out to several other major papers in the field. A few points of interest are the controversies around origin and characterization. There seems to be a relatively wide interpretation of CSCs in terms of typing and behavior. This is obviously due to the relatively abstract nature of stemness and cell development. Talking with Tim today, there are some limitations since we only have expression data. One consideration is that our pipeline is using a known reference sequence database to process the output. Given this, there may be unknown mutations that could serve as markers of both CSC identity and possible lineage association. Screening for these in a cell specific way could bring about interesting findings.

#### Completed

Created    Completed  Age Project        Description
---------- ---------- --- -------------- ---------------------------------------
2019-06-01 2019-06-03 2d  starrlab       add notes about cancer stem cell
                                         database
2019-06-01 2019-06-03 2d  starrlab.ocpmi litsrch re cancer stem cell review

2 tasks

***

## 4

### TODO


ID Age Project Tags   Description                                           Urg
-- --- ------- ------ ----------------------------------------------------- ----
49 2w  lab     lit    read tirosh 2016 [2]                                  2.78
55 8d  lab     lit    read papalexi 2018 [2]                                2.74
56 7d  lab     lit    read macintyre 2018 [2]                               2.74
48 2w  lab     lit    read zheng 2017 [1]                                   2.68
57 7d  lab     lit    read sahin 2017 [1]                                   2.64
65 3d  lab     lit    litsrch re genetic lineage tracing method [1]         2.62
52 11d lab            agenda items [3]                                      2.06
47 2w  lab     comp w run surat tutorial w/ pt 59                           1.98
62 5d  lab     comp n review pca concepts                                   1.93
63 4d  lab     comp n compare default seurat tutorial to starrlab protocol  1.92
64 4d  lab     comp w deep dive into default seurat analysis                1.92
46 2w  lab     lit    read ocpmi summary                                    1.88
77 20h lab     lit    find list of CSC markers                               1.8

13 tasks

### Summary

I didn't get a lot done today. I began to apply the vanilla Seurat analysis to the pt59 data set and found myself encountering many of the setup hurdles that come with using a remote system. I have to do some retooling so my environment can seamlessly run on MSI.

***

## 5

### TODO

ID Age  Project Tags   Description                                          Urg
-- ---- ------- ------ ---------------------------------------------------- ----
49 2w   lab     lit    read tirosh 2016 [2]                                 2.78
55 9d   lab     lit    read papalexi 2018 [2]                               2.75
56 8d   lab     lit    read macintyre 2018 [2]                              2.74
76 2d   lab     lit    read konrashova 2017 [2]                             2.71
48 2w   lab     lit    read zheng 2017 [1]                                  2.68
57 8d   lab     lit    read sahin 2017 [1]                                  2.64
65 4d   lab     lit    litsrch re genetic lineage tracing method [1]        2.62
47 2w   lab     comp w run surat tutorial w/ pt 59                          1.98
62 6d   lab     comp n review pca concepts                                  1.93
63 5d   lab     comp n compare default seurat tutorial to starrlab protocol 1.93
64 5d   lab     comp w deep dive into default seurat analysis               1.93
84 20h  lab     comp n figure out remote vim setup                           1.9
89 1min lab     comp n send vacation dates to tim                            1.9
46 2w   lab     lit    read ocpmi summary                                   1.88
77 1d   lab     lit    find list of CSC markers                             1.81
52 12d  lab            agenda items                                         1.07

16 tasks


### wkly mtg w/ Tim

present: chris, tim

```{r eval=F}
Description
--------------------------------------------------------------------------------
agenda items
  2019-05-28 discuss vacation
  2019-06-01 can we use sequence-based lineage tracing to identify a common
progenitor cell expression profile
  2019-06-03 is the JGO article just a couple poster abstracts?
  2019-06-05 discuss seurat output

1 task
```

#### mtg notes

* We discussed the nature of the Seurat output to some degree
    + I had been wondering how Tim came about the specifications to the metrics he used in his Seurat analysis file.
    + He has largely used the default params other outside of a few specific cases
    + These specifications and their justifications are all detailed in the Single Cell RNAseq Analysis protocol document in my `methods` folder.
    + One thing he mentioned would be an area I could assist him with is the way we fine tune these params.
    + Isaac is working on this to some extent by evaluating the 3 methods available for `Identification of highly variable features`.
        - So far, he's found that there is a notable difference in the genes identified, but the clustering output is relatively similar.
    + Tim has created a custom analysis specific to our data set that applied cell typing to clusters.
    + Additionally, we discussed the broad strokes of concepts behind the idea of CSCs. There is a great deal of ambiguity here, and we didn't really arrive at any objective conclusions. To some extent, I just needed to work through the ideas with someone else. I will need to really formulate a sense of this for any comprehensive experimentation which leverages this concept.
    + This was a subpoint but an important realization: **the 10x do not include full coverage.** This was a mistaken assumption of mine in thinking we had more complete expression coverage. Instead, we are working with ~100bp lengths with poly-A ranges. It may be worth digging into this to make sure I understand the specifics.

### Remote MSI Setup

I've reached somewhat of a hang up in my development environment. I've yet to work in centos, and I'm generally unfamiliar with the MSI setup. So far, I've been able to make sense of my home directory in my personal instance and load the R console. Additionally, I've been able to get the installation of packages going and have a process running over night in tmux.

The challenge I see stems from the combination of developing in an Rstudio-like environment that allows for the execution of code snippets while being executed in the MSI environment. What I am looking at as I wrap the day is using Rserve on my MSI instance to connect to my local dev environment.




### Rserve

I've found the package `Rserve` which can establish a servable instance of R on a remote server. This holds some potential to allow me to develop locally and execute any part of my code in the MSI instance. There is documentation at the link below. One necessary element of making this work would be to find a suitable Rserve client that can run locally and communicate with the server.

https://www.rforge.net/Rserve/doc.html



https://www.msi.umn.edu/support/faq/how-can-i-install-r-packages-my-home-directory

https://satijalab.org/seurat/install.html

### Summary

I met with Tim today for our weekly meeting. We were able to discuss things that were mostly conceptual to aid in my continued content exploration. The rest of my time was spent getting a sense for the MSI environment and how I will be interacting with it. My hope is to actually develop in this space full time rather than just schedule jobs. My own computer is limited in its processing power, and using my MSI instance would be highly preferred. Still, there are some hurdles to doing this that I haven't dealt with previously, so I have been working to link my current setup to the operational environment of MSI. This will probably continue to take some time, but I hope to be fully ready tomorrow.

***

## 6

### TODO


ID Age Project Tags   Description                                           Urg
-- --- ------- ------ ----------------------------------------------------- ----
49 2w  lab     lit    read tirosh 2016 [2]                                  2.79
54 10d lab     lit    read papalexi 2018 [2]                                2.75
55 9d  lab     lit    read macintyre 2018 [2]                               2.75
74 3d  lab     lit    read konrashova 2017 [2]                              2.72
48 2w  lab     lit    read zheng 2017 [1]                                   2.69
56 8d  lab     lit    read sahin 2017 [1]                                   2.64
63 5d  lab     lit    litsrch re genetic lineage tracing method [1]         2.63
47 2w  lab     comp w run surat tutorial w/ pt 59                           1.99
60 6d  lab     comp n review pca concepts                                   1.93
61 6d  lab     comp n compare default seurat tutorial to starrlab protocol  1.93
62 6d  lab     comp w deep dive into default seurat analysis                1.93
82 1d  lab     comp n figure out remote vim setup                           1.91
87 22h lab     comp n send vacation dates to tim                             1.9
88 18h lab     comp n update methods on msi dev setup                        1.9
89 18h lab     comp n spell check notebook                                   1.9
46 2w  lab     lit    read ocpmi summary                                    1.89
75 2d  lab     lit    find list of CSC markers                              1.81
51 13d lab            agenda items                                          1.07

18 tasks


### MSI Workflow

After a considerable amount of effort to create a synchronous workflow with my MSI instance, I've generally given up and settled on scripting locally and batching with MSI. I've mounted my home directory with sshfs and clone my ocpmi git repo so I can work out of my MSI instance for the most part. My only concern is if I run into resource intensive cases that require more ram. It does seem that MSI kills unbatched jobs that exceed a threshold of resource use since my process was killed each time it went over 10gb of ram.

### Notes to add

* git repo info
* sshfs setup and bash script
* msi setup info
* seurat install and docs methods

### Summary

I've wasted a large amount of time over the past two days. I spent a solid 10 hours trying to get my workflow in MSI established all to essentially go with what would have been my setup had I done nothing at all. I will generally resort to developing on my local machine with a connection to my MSI directory. Beyond this, I will need to take a look at the batching process for MSI and possible explore some of their training resources to gain a better understanding.

#### Completed

Created    Completed  Age Project Description
---------- ---------- --- ------- -------------------------------
2019-06-05 2019-06-06 1d  lab     send vacation dates to tim
2019-06-04 2019-06-06 1d  lab     figure out remote vim setup
2019-06-06 2019-06-06 2h  lab     clone git repo to msi directory
2019-06-06 2019-06-06 2h  lab     exchange ssh keys with msi

4 tasks

***

## 10

### TODO

ID Age Project Tags     Description                                         Urg
-- --- ------- -------- --------------------------------------------------- ----
48 2w  lab     lit work read tirosh 2016 [2]                                2.91
53 2w  lab     lit work read papalexi 2018 [2]                              2.88
54 13d lab     lit work read macintyre 2018 [2]                             2.87
70 7d  lab     lit work read konrashova 2017 [2]                            2.84
47 2w  lab     lit work read zheng 2017 [1]                                 2.81
55 13d lab     lit work read sahin 2017 [1]                                 2.77
61 9d  lab     lit work litsrch re genetic lineage tracing method [1]       2.75
45 2w  lab     lit work read ocpmi summary                                  2.01
46 2w  lab     w work   run surat tutorial w/ pt 59                         2.01
58 11d lab     n work   review pca concepts                                 1.96
59 10d lab     n work   compare default seurat tutorial to starrlab         1.95
                        protocol
60 10d lab     w work   deep dive into default seurat analysis              1.95
71 6d  lab     lit work find list of CSC markers                            1.93
80 4d  lab     n work   update methods on msi dev setup                     1.92
81 4d  lab     n work   spell check notebook                                1.92
83 3d  lab     n work   add notes re sshfs bash script                      1.92
50 2w  lab     work     agenda items                                        1.89

17 tasks

### Non-contemporaneous Notes - 6/6

#### Load/Install R pkgs

* In order to run R in my MSI instance, I need to load the module and install applications.

```{r eval=F}
module load R/3.6.0
R
```

* To install packages in my local home directory, need to define a path.

```{r eval=F}
.libPaths(new='~/R/x86_64-pc-linux-gnu-library/3.6.0')
dir.create('~/R/x86_64-pc-linux-gnu-library/3.6.0', showWarnings = FALSE, recursive = TRUE)
```

* Confirm the path is correct.

```{r eval=F}
.libPaths()
```

* Choose as CRAN mirror.

```{r eval=F}
getCRANmirrors()
chooseCRANmirror(ind=2)
```

* Packages should be able to be installed and loaded normally now.

#### sshfs Setup

I'm using `sshfs` to mount my MSI home folder on my local machine to make it more readily accessible. I already have ssh keys exchanged with MSI, so I can create an executable auto-mount bash script. This is located at `/home/chris/starr_lab/auto_mnt_sshfs` on my local machine.

* Auto-mount MSI sshfs

```{bash eval=F}
#!/bin/bash

sshfs tasta005@login.msi.umn.edu: ~/starr_lab/msi_mnt
```

### Summary

I made some more progress on developing a base implementation of the default Seurat analysis pipeline when evaluating the pt59 dataset. To a degree, I am mostly taking default params using the PBMC tutorial and running them with the new dataset. Outside of the default params, I have been using the file located at `/ocpmi/code/tim/seurat_analysis.R` as a reference to see what Tim has modified thus far. There are certain points where he has taken the base value and others where his code deviates heavily. The general flow of his implementation differs from the tutorial in the sense that his code is less step wise and more focused on the output while the tutorial is a bit more exploratory to exhibit features. This makes sense of course. An aspect of my own implementation will have to consider how distributed each analysis process to provide modularity.

A major point I came across today was that I forgot to review the RNAseq protocol located at `/ocpmi/methods/`. Tim and I had discussed this in our meeting last week, but I forgot to make a note about this. There is a good deal of documentation for his code and methods there, so I probably should have remembered to look at this more. I will dedicate some time tomorrow to do this.

Overall, I will spend most of tomorrow continuing to review the documentation and code for Tim's pipeline along with getting a sense of the veracity of my pt59 output.

#### Completed

Created    Completed  Age Project Description
---------- ---------- --- ------- -------------------------------
2019-06-05 2019-06-10 4d  lab     spell check notebook
2019-06-05 2019-06-10 4d  lab     update methods on msi dev setup
2019-06-06 2019-06-10 3d  lab     add notes re sshfs bash script

3 tasks

***

## 11

### TODO

ID Age Project Tags     Description                                         Urg
-- --- ------- -------- --------------------------------------------------- ----
48 3w  lab     lit work read tirosh 2016 [2]                                2.92
50 2w  lab     work     agenda items [3]                                     2.9
53 2w  lab     lit work read papalexi 2018 [2]                              2.88
54 2w  lab     lit work read macintyre 2018 [2]                             2.88
70 7d  lab     lit work read konrashova 2017 [2]                            2.84
47 3w  lab     lit work read zheng 2017 [1]                                 2.82
55 13d lab     lit work read sahin 2017 [1]                                 2.77
61 10d lab     lit work litsrch re genetic lineage tracing method [1]       2.75
45 3w  lab     lit work read ocpmi summary                                  2.02
46 3w  lab     w work   run surat tutorial w/ pt 59                         2.02
58 11d lab     n work   review pca concepts                                 1.96
59 11d lab     n work   compare default seurat tutorial to starrlab         1.96
                        protocol
60 11d lab     w work   deep dive into default seurat analysis              1.96
71 7d  lab     lit work find list of CSC markers                            1.94
89 18h lab     n work   evaluate list of highly var features pt59            1.9
91 18h lab     errand n bring monitor to work                                1.9
92 17h lab     n work   review rnaseq analysis protocol                      1.9
95 20s lab     n work   email tim re mtg resched                             1.9

18 tasks

### RNA Velocity

I came a cross a [thread](https://www.reddit.com/r/bioinformatics/comments/bz292v/_/) on the bioinformatics subreddit discussing the issues of the day for single cell analysis. One of the points that was discussed was the challenge around merging data sets and the way this might regress some relevant artifacts of the data. The concept of RNA velocity was brought up as a means to address this, and it may hold some relevance to how we analyze our data. [@LaManno2018]

### Render .R -> .html

I made an amazing discovery today. It is possible to render markdown output straight from a .R file. I'm not sure if I can apply certain yaml or formatting settings to this process yet, but this function alone will save the need to create two separate files when running code and producing a formatted output.

https://rmarkdown.rstudio.com/articles_report_from_r_script.html

A markdown report can be compiled from an R script by:

```{r eval=F}
rmarkdown::render("analysis.R") # will generate an html output
rmarkdown::render("analysis.R", "pdf_document")
```

The metadata and output can be modified through special comments.

```{r eval=F}
#' ---
#' title: "Crop Analysis Q3 2013"
#' author: "John Smith"
#' date: "May 3rd, 2014"
#' output: pdf_document
#' ---
```

### Summary

My day consisted of a continued focus on developing a complete working model of the general Seurat pipeline through the lens of the pt59 data. I haven't done much in terms of evaluation for the output and will need to dig into this more. The highly variable features and linear dimensional reduction are items I need to both confirm for efficacy and make sense of in their findings. Additionally, I merely clustered the data along 13 dimensions and did not apply any further analysis such as the exploration of highly variable features in the umap'd data.

I did make some unintended progress on formatting and organization of code after discovering that I can render markdown directly from an R script file. This is what I was striving for to some degree in my MSI setup from last week, so it is a relief to arrive at a workable solution.

Last, I plan on taking a bit of a dive into the concept of RNA velocity to see if it holds any relevance to evaluating our data set. As I have become acclimated to the content of this field, I have felt there is some aspect of single cell analysis that the group has not considered to explore concepts beyond cell typing such as CNV, cell lineage, and stem cell character. I don't say that with the authority to know if it is the case, but the adolescence of this field draws the general assumption.

#### Completed

Created    Completed  Age Project Description
---------- ---------- --- ------- ---------------------------
2019-06-11 2019-06-11 5h  lab     email tim re mtg resched
2019-05-20 2019-06-11 3w  lab     run surat tutorial w/ pt 59

2 tasks

***

## 12

### TODO

ID Age Project Tags     Description                                         Urg
-- --- ------- -------- --------------------------------------------------- ----
46 3w  lab     lit work read tirosh 2016 [2]                                2.92
48 2w  lab     work     agenda items [3]                                     2.9
51 2w  lab     lit work read papalexi 2018 [2]                              2.89
52 2w  lab     lit work read macintyre 2018 [2]                             2.88
68 9d  lab     lit work read konrashova 2017 [2]                            2.85
45 3w  lab     lit work read zheng 2017 [1]                                 2.82
53 2w  lab     lit work read sahin 2017 [1]                                 2.78
59 11d lab     lit work litsrch re genetic lineage tracing method [1]       2.76
90 1d  lab     lit work read manno 2018 [1]                                 2.71
44 3w  lab     lit work read ocpmi summary                                  2.02
56 13d lab     n work   review pca concepts                                 1.97
57 12d lab     n work   compare default seurat tutorial to starrlab         1.97
                        protocol
58 12d lab     w work   deep dive into default seurat analysis              1.97
69 8d  lab     lit work find list of CSC markers                            1.94
85 1d  lab     n work   evaluate list of highly var features pt59           1.91
87 1d  lab     errand n bring monitor to work                               1.91
88 1d  lab     n work   review rnaseq analysis protocol                     1.91
93 19h lab     n work   perform diff exp analysis on non-lin red pt59        1.9
                        output

18 tasks


### sshfs Unmount

I've run into the case where a server disconnect causes a hang any time I navigate to this mount point. To reset the mount point use `fusermount` to force the unmount first.

```{r eval=F}
fusermount -uz /data
```

### MSI Interactive Computing

Running all of my work locally has shown to be more resource consuming than originally anticipated. This has developed somewhat of a bottleneck, so I'm hoping I can transfer the larger rendering jobs to MSI so they can process in the background. A limitation in doing this is developing the ability to generate markdown since this would be the best output. It isn't necessary, but I would prefer this. The hangup seems to be the lack of a graphics environment. Obviously, a headless server will not have something like this loaded by default, but I have to see if I can get that working.

### RNA Velocity

I began to gloss over this paper today [@LaManno2018]. The concept is both relatively simple in general terms yet includes a good deal of complexity since it involves RNA splicing. I need to read further, but what I've read so far lends itself to the potential for some application in our data set. I'll have to discuss with Tim more tomorrow.

### pt59 analysis

I continued to round out my general analysis of the pt59 data. This analysis was furthered with my original approach of being blind to the Biological relevance. I've been able to generate outputs that follow the base pipeline but will need to evaluate markers specific to our cell types and genetic markers of interest.

### Summary

Today included some light work around finishing off my pt59 analysis. The code I've put together generally mirrors the PBMC tutorial with some minor variations. Key to this general approach is that I haven't customized the marker selection. This would be a good discussion with Tim in how he has identified this in the past in a consistent manner. Additionally, my computer is becoming more limited with every process I run. It may be worth breaking these things into smaller chunks of analysis to account for my resource limitations, but I'm going to return to getting things working on MSI first.

#### Completed

Created    Completed  Age Project Description
---------- ---------- --- ------- ----------------------------------------------
2019-06-10 2019-06-12 2d  lab     bring monitor to work
2019-05-30 2019-06-12 13d lab     compare default seurat tutorial to starrlab
                                  protocol
2019-06-11 2019-06-12 1d  lab     perform diff exp analysis on non-lin red pt59
                                  output

3 tasks

***

## 13

### TODO

ID Age Project Tags     Description                                   Urg
-- --- ------- -------- --------------------------------------------- ----
46 3w  lab     lit work read tirosh 2016 [2]                          2.93
48 2w  lab     work     agenda items [6]                              2.91
51 2w  lab     lit work read papalexi 2018 [2]                        2.89
52 2w  lab     lit work read macintyre 2018 [2]                       2.89
66 9d  lab     lit work read konrashova 2017 [2]                      2.85
45 3w  lab     lit work read zheng 2017 [1]                           2.83
53 2w  lab     lit work read sahin 2017 [1]                           2.78
58 12d lab     lit work litsrch re genetic lineage tracing method [1] 2.77
84 1d  lab     lit work read manno 2018 [1]                           2.71
44 3w  lab     lit work read ocpmi summary                            2.03
56 13d lab     n work   review pca concepts                           1.97
57 13d lab     w work   deep dive into default seurat analysis        1.97
67 9d  lab     lit work find list of CSC markers                      1.95
81 2d  lab     n work   evaluate list of highly var features pt59     1.91
83 2d  lab     n work   review rnaseq analysis protocol               1.91
88 20h lab     n work   clean up github tracked files                  1.9
89 17h lab     n work   put pt59 output on results page                1.9

17 tasks


***

# References


