---
title: "Notes"
author: "Christopher Tastad"
date: \today
bibliography: library.bib
output:
    html_document:
        toc: yes
        toc_float: yes
        theme: paper
---

```{r}
library(devtools)
library(Seurat)

```

# May 2019

## 20

### Background mtg w/ Tim

* present: chris, tim

We discussed the outlook for my role in the lab. This included a considerable amount of background on the ocpmi project and potential directions I could take.

Focus Options

* ID of ovca stem cell
* what is the basis for clustering
* single cell seq methods paper
* compare mouse to human
* copy number variation on single cell
* rerun new version of cell ranger
* RHEL update


## 21

### TODO




ID  Age   Project        Tags       R Due        Description                Urg
--- ----- -------------- ---------- - ---------- -------------------------- ----
326 10h   starrlab       computer n R 2019-05-21 write daily summary        10.5
327 27min starrlab       computer n              find umn git and hosting    1.9
312 20h   starrlab.ocpmi computer n              organize lab notebook [1]   2.7
315 18h   starrlab.ocpmi computer n              find ovca review re cur     1.9
                                                 treatment
316 18h   starrlab.ocpmi computer n              read starr paper re 1st pt  1.9
317 18h   starrlab.ocpmi computer n              read ocpmi summary          1.9
318 18h   starrlab.ocpmi computer n              install surat               1.9
319 18h   starrlab.ocpmi computer w              run surat tutorial w/ pt    1.9
                                                 59
320 18h   starrlab.ocpmi computer n              read single cell paper      1.9
                                                 zheng 2017
321 18h   starrlab.ocpmi computer n              find and read copy num      1.9
                                                 variation paper
322 18h   starrlab.ocpmi computer n              organize refs in email      1.9
                                                 inbox
323 18h   starrlab.ocpmi computer n              compile bibtex ref page     1.9

12 tasks


### Citation packages

#### Rmarkdown Default

This is the intrinsic formatting style for Rmarkdown citation management. It has the ability to call a BibTex file and carries the least effort to get it working. It's what I will probably go for across my whole setup. [@citations]

#### knitcitations

The knitcitations package might provide the best balance of feature set to ease of use. The only constraining factor is what seems like a few bugs that limit intended functionality. The ref() function doesn't work for instance. [@cboettig_2019]

#### RefManageR

This is a fully featured R package that seems to attempt to replace a complete citation manager. It comes across as very comprehensive but more complicated than I would like the approach to be. [@refmanager]

### Summary

#### Completed

* 327, 312:progress

The majority of the day was spent exploring the setup for my lab notebook. My intention is to create a living document using the Rmarkdown website knit feature [@rmarkdown_cheatsheet]. This will allow for a dynamic storage of both records and code work inline with one comprehensive document. A large portion of this time was spent giving consideration to how I will handle citations both in the document and for future publication. After significant exploration, my plan is to use the intrinsic Rmarkdown system [@citations] through use of a bibtex file. This will allow for simple reference in-place with the notebook along with using a standardized format for publication ready references.
Additionally, I plan on establishing a git repo to manage version control and historical record with the notebook as well using the umn github server [citation needed].

***

## 22

### TODO



ID  Age Project        Tags       R Due        Description                  Urg
--- --- -------------- ---------- - ---------- ---------------------------- ----
326 15h starrlab       computer n R 2019-05-22 write daily summary          10.5
312 1d  starrlab.ocpmi computer n              organize lab notebook [1]    2.71
315 1d  starrlab.ocpmi computer n              find ovca review re cur      1.91
                                               treatment
316 1d  starrlab.ocpmi computer n              read starr paper re 1st pt   1.91
317 1d  starrlab.ocpmi computer n              read ocpmi summary           1.91
318 1d  starrlab.ocpmi computer n              install surat                1.91
319 1d  starrlab.ocpmi computer w              run surat tutorial w/ pt 59  1.91
320 1d  starrlab.ocpmi computer n              read single cell paper zheng 1.91
                                               2017
321 1d  starrlab.ocpmi computer n              find and read copy num       1.91
                                               variation paper
322 1d  starrlab.ocpmi computer n              organize refs in email inbox 1.91
323 1d  starrlab.ocpmi computer n              compile bibtex ref page      1.91

11 tasks


### wkly mtg w/ Tim

present: chris, tim

* Make sure to utilize notes field in bibtex file. Having an ongoing record of each paper read is critical.
* Consider separating data and results in file organization.
* Change notebook structure to follow layout of paper. Organize the work with a mindset toward writing the paper from the start, not the finish.
* Currently, we are still working to establish the standardized parameters of the data set. The goal is to arrive at a place where we can make biological conclusions.
    + How do we draw contrast from a data set of this nature?

### Notebook Structure

In my conversation with Tim, I decided I need to take an alternate structure to my lab notebook. My original intent was to emulate a development oriented documentation process. Instead, I will go a route that better follows the academic writing process with pages dedicated to methods, results, etc.

### Reference Management

While I would prefer to take a lean, text based approach to this, I am clearly beginning to waste some time in gear the system. I came across jabref, which seems to leverage a unix philosophy and can be found in the ubuntu repos.

### Summary

#### Completed

* 312:progress, 323:progress

My work today included continued focus on getting organized an taking a very deliberate approaching to my workflow before I begin. I met with Tim to begin our weekly meetings, and we covered some aspects of this workflow. Major take aways include the importance of annotation when reviewing literature. I had originally assumed I could handle references processing merely in a bare bibtex file. My feeling now is that I will begin to use a dedicated application for greater consistency in documentation.

***

## 23

### TODO



ID  Age Project        Tags       R Due        Description                  Urg
--- --- -------------- ---------- - ---------- ---------------------------- ----
334 11h starrlab       computer n R 2019-05-23 write daily summary          10.6
327 23h starrlab       computer n              reorg notebook to follow      1.9
                                               paper structure
329 19h starrlab       computer n              setup jabref                  1.9
330 19h starrlab       lab n                   setup desk                    1.9
331 19h starrlab       computer n              create agenda field           1.9
332 19h starrlab       computer n              create lit tag                1.9
333 19h starrlab       computer n              create lab context            1.9
336 41s starrlab       computer n              create website knit script    1.9
312 2d  starrlab.ocpmi computer n              organize lab notebook [1]    2.71
315 2d  starrlab.ocpmi computer n              find ovca review re cur      1.91
                                               treatment
316 2d  starrlab.ocpmi computer n              read starr paper re 1st pt   1.91
317 2d  starrlab.ocpmi computer n              read ocpmi summary           1.91
318 2d  starrlab.ocpmi computer n              install surat                1.91
319 2d  starrlab.ocpmi computer w              run surat tutorial w/ pt 59  1.91
320 2d  starrlab.ocpmi computer n              read single cell paper zheng 1.91
                                               2017
321 2d  starrlab.ocpmi computer n              find and read copy num       1.91
                                               variation paper
322 2d  starrlab.ocpmi computer n              organize refs in email inbox 1.91
323 2d  starrlab.ocpmi computer n              compile bibtex ref page      1.91

18 tasks


### Citations

#### Custom CSL

At some point, I would like to setup a custom csl file to format my references. The goal here would be to add formatted features that are note readily available in some existing format style. [@2019], [@BibEntry2019May]

#### JabRef

I've started to figure out jabref. I think I will still look to my bibtex file as the primary source for citation management, but will make use of the interface and some of the features of jabref. Additionally dio2bib will play a central role in extracting bibtex content. [@2019b]

### GitHub

The setup for my git server is complete. It's a private server host by the university a /tasta005/ocpmi [@ocpmi_git]

### Summary

I was able to round out much of my organizational setup today. I feel confident about the citation management strategy I plan to use going forward which combines the use of dio2bib for bibtex creation, BibItNow for website extraction, and JabRef to centralize management. Additionally, I was able to connect my relevant directories to my git server, which will account for both a measure of backup and version control. Having gotten settled into these systems I think I will be able to begin incorporating my literature search into my daily activities.

#### Completed


Created    Completed  Age Project        R Due        Description
---------- ---------- --- -------------- - ---------- --------------------------
2019-05-22 2019-05-23 1d  starrlab       R 2019-05-22 write daily summary
2019-05-22 2019-05-23 1d  starrlab                    setup desk
2019-05-23 2019-05-23 16h starrlab       R 2019-05-23 write daily summary
2019-05-20 2019-05-23 3d  starrlab.ocpmi              organize lab notebook
                                                        2019-05-21 record,
                                                      master, comms, refs, home,
                                                      docs
2019-05-22 2019-05-23 1d  starrlab                    create lab context
2019-05-22 2019-05-23 1d  starrlab                    create lit tag
2019-05-22 2019-05-23 1d  starrlab                    setup jabref
2019-05-20 2019-05-23 3d  starrlab.ocpmi              compile bibtex ref page
2019-05-20 2019-05-23 3d  starrlab.ocpmi              organize refs in email
                                                      inbox
2019-05-22 2019-05-23 1d  starrlab                    setup umn git
2019-05-22 2019-05-23 1d  starrlab                    create agenda field

11 tasks

***

## 24

### TODO


ID  Age Project        Tag            Description                           Urg
--- --- -------------- -------------- ------------------------------------- ----
314 3d  starrlab.ocpmi computer lit n find ovca review re cur treatment     2.02
315 3d  starrlab.ocpmi computer lit n read starr paper re 1st pt            2.02
316 3d  starrlab.ocpmi computer lit n read ocpmi summary                    2.02
319 3d  starrlab.ocpmi computer lit n read single cell paper zheng 2017     2.02
320 3d  starrlab.ocpmi computer lit n find and read copy num variation      2.02
                                      paper
317 3d  starrlab.ocpmi computer n     install surat                         1.92
318 3d  starrlab.ocpmi computer w     run surat tutorial w/ pt 59           1.92
325 1d  starrlab       computer n     create website knit script            1.91
327 22h starrlab       comp w         make custom csl                        1.9
328 22h starrlab       lab n          complete lab training                  1.9
330 20h starrlab       w              agenda items                           1.8

11 tasks

### create_site Script

I created a small R script that will effectively knit each page of my lab notebook. This can be executed in the R console or in bash, but each file has to be saved. The save -> knit function that's performed by the IDE does not apply to the console executed render command.

```{r eval=F}
rmarkdown::render("index.Rmd")
rmarkdown::render("notes.Rmd")
rmarkdown::render("results.Rmd")
rmarkdown::render("methods.Rmd")
rmarkdown::render("references.Rmd")
rmarkdown::render_site("index.Rmd")
```


### Summary

Today was filled with primarily administrative work. A large bulk of my time was taken by lab safety training, which I was unable to fully complete. Along with this, I was finally able to install Seurat and tinker with the package tutorial. It seems like the version I downloaded may not be compatible with the tutorial code, so I will have to explore that more.

#### Completed


Created    Completed  Age Project        Description
---------- ---------- --- -------------- ---------------------------------------
2019-05-23 2019-05-24 1d  starrlab       create website knit script
2019-05-24 2019-05-24 2h  starrlab       DEHS Radiation Safety Orientation
                                         Training
2019-05-24 2019-05-24 2h  starrlab       Implementation of NIH Guidelines
2019-05-24 2019-05-24 2h  starrlab       Bloodborne Pathogens: Advanced
2019-05-24 2019-05-24 2h  starrlab       DEHS Chemical Safety Training
2019-05-24 2019-05-24 2h  starrlab       Biological Safety in the Laboratory
2019-05-24 2019-05-24 2h  starrlab       Bloodborne Pathogens: Introduction and
                                         Advanced
2019-05-24 2019-05-24 2h  starrlab       DEHS Introduction to Research Safety
                                         Training
2019-05-24 2019-05-24 2h  starrlab       Animal Use Tutorial
2019-05-23 2019-05-24 1d  starrlab       complete lab training
2019-05-20 2019-05-24 4d  starrlab.ocpmi install surat

11 tasks

***

## 27

### TODO


ID Age Project        Tags           Description                            Urg
-- --- -------------- -------------- -------------------------------------- ----
77 4d  starrlab       computer w     make custom csl                        1.92
79 3d  starrlab       computer n     GCP Human RSRCH Biomed Studies         1.92
80 3d  starrlab       computer n     CITI research ethics training          1.92
81 3d  starrlab       computer n     University Information Security        1.92
                                     Awareness Training
78 4d  starrlab                      agenda items                           1.02
91 15h starrlab.ocpmi computer lit n read papalexi 2018 [2]                  2.9
68 6d  starrlab.ocpmi computer lit n find ovca review re cur treatment      2.03
69 6d  starrlab.ocpmi computer lit w read starr paper re 1st pt             2.03
70 6d  starrlab.ocpmi computer lit w read ocpmi summary                     2.03
72 6d  starrlab.ocpmi computer lit w read single cell paper zheng 2017      2.03
73 6d  starrlab.ocpmi computer lit w find and read copy num variation paper 2.03
71 6d  starrlab.ocpmi computer w     run surat tutorial w/ pt 59            1.93

12 tasks


###[@Lheureux2019]

#### Evolution of Management in the Era of Precision Medicine  (1/2)

* State of field for ovarian cancer
* Current treatments and outcomes
* Ovca presents very late. The internal location of the cancer and general lack of screening options lead to most cases being diagnosed at a later stage than many other cancers would.
* Treatment options have largely focused on surgical debulking and platinum based chemotherapy.
* Risk factors are inherited: germ line mutations in breast cancer susceptible genes (e.g. brca), nulliparity, infertility, endometriosis, obesity, age, potentially perineal talc.
* Reducing factors that have been found to be potentially significant are gravidity, oral contraceptive use, regular use of aspirin, and nsaids.
* Currently, population screening in ovca has not been found to be effective.
* Epithelial ovarian cancer accounts for 95% of ovarian malignancies
* 5 eoc subtypes: high-grade serous, low-grade serous, clear cell, endometrioid, and mucinous
* HGSOC has the highest incidence with 70% of eoc cases.
* It is believed fallopian tube epithelium is the possible site for tumor initiation.
* Mutations in p53 are seen in +90% cases of HGSOC
* p53 mutations and brca inactivation lead to high numbers of copy variation.
* HGSOC has 4 prognostic subtypes: differentiated, immunoreactive, mesenchymal, and proliferative
* Immunoreactive is considered to have the best overall survival while mesenchymal has the worst.
* Ovca cells also over express folate and mesothelin which have been targets for antibody therapy.
* LGSOC is seen in about 10% of cases
* LGSOC is considered to develop in a pathway which is very distinct from HGSOC
* B-Raf and KRAS mutations are the most commonly present in LGSOC
* Hormone therapy is sometimes used due to the presence of estrogen/progesterone receptors.
* CNAs in p53 are rare, and brca inactivation seem to unrelated to LGSOC

### Summary

Just worked my way through the above summary today. Will continue and complete reading this lit tomorrow.


***

## 28

### TODO


ID Age Project        Tags           Description                            Urg
-- --- -------------- -------------- -------------------------------------- ----
77 4d  starrlab       computer w     make custom csl                        1.92
79 3d  starrlab       computer n     GCP Human RSRCH Biomed Studies         1.92
80 3d  starrlab       computer n     CITI research ethics training          1.92
81 3d  starrlab       computer n     University Information Security        1.92
                                     Awareness Training
78 4d  starrlab                      agenda items [1]                       1.82
91 1d  starrlab.ocpmi computer lit n read papalexi 2018 [2]                 2.91
92 17h starrlab.ocpmi computer lit w read macintyre 2018 [2]                 2.9
68 7d  starrlab.ocpmi computer lit n find ovca review re cur treatment      2.04
69 7d  starrlab.ocpmi computer lit w read starr paper re 1st pt             2.04
70 7d  starrlab.ocpmi computer lit w read ocpmi summary                     2.04
72 7d  starrlab.ocpmi computer lit w read single cell paper zheng 2017      2.04
73 7d  starrlab.ocpmi computer lit w find and read copy num variation paper 2.04
71 7d  starrlab.ocpmi computer w     run surat tutorial w/ pt 59            1.94

13 tasks


### [@Lheureux2019]

#### Evolution of Management in the Era of Precision Medicine (2/2)

* OCCC or clear cell carcinoma represents about 5% of cases.
* The pathogenesis of OCCC is not well understood.
* A proposed mechanism is related to the level of iron content, promoting oxidative stress.
* Mutations in the SWI/SNF and PI3K pathway are represented in large proportions of cases.
* Activation of the PI3K or RTK pathway has resulted in better OS
* OCCCs are found to have considerably fewer CNAs and mutations in p53 or brca genes.
* Endometrioid cancers represent 10% of EOC cases.
* Current thought is that endometrioid and OCCC arise from similar pathways.
* Deregulation of WNT/b-catenin and PI3K pathways are essential in mouse models.
* Mucinous OC is a rare subtype.
* Could be classified as non-invasive and infiltrative.
* KRAS mutations are a common feature of this subtype.
* p53 and brca mutations do not seem to play a role.
* Treatment
* To date, there has been no randomized assessment of debulking surgery, platinum therapy, or radiation therapy.
* Surgery has been the bedrock of therapy.
* Surgery involves many approaches including primary, interval, and secondary debulking
* Chemotherapy
* carboplatin is as effective as cisplatin and is better tolerated.
* dose is important.
* pairing drugs does not improve outcomes.
* IP chemo is often used due to the tight localization of ovca in the abdomen
* hyperthermic IP chemo is currently being explored with reasonable results
* Treatment after recurrence follows several judgments of benefit.
* Recurrence after a period of over 6mo suggests platinum sensitivity where platinum based chemo would be reapplied
* Shorter intervals of recurrence employ non-platinum based therapies
* Current consensus is the use of carboplatin and paclitaxel together every 3wks
* Targeted therapy has been gradually introduced in recent years.
* Bevacizumab has shown the greatest promise among antiangiogenics being the only one in clinical use.
* PARP inhibitors
* Used as maintenance with platinum therapy at the time of recurrence.
* A trial showed that the key benefit in PFS was driven by the parp inhibitor maintenance therapy.
* Trial from 2018 led to first-line maintenance in brca mutated cases
* Stratified approach to systemic therapy can be used with bevacizumab after surgery with a switch to maintenance on the basis of brca mutations.
* Immunotherapy is currently being explored on several fronts though none are in clinical practice.
* ovca has a high expression of PD-L1.
* Checkpoint inhibitors targeting PD-L1/PD1 or CTLA-4 have shown only modest results.
* Autologous cell therapy is also being utilized.
* A variation of this therapy is exploring car-t cells targeting mesothelin
* Vaccines have not worked well due to the highly immunosuppressive cancer environment
* A vaccine targeting survivin is currently being explored
* Additionally, the concept of the patient-specific mutanome is being researched in an effort to develop patient-specific cancer vaccines.
* OCCC and LGSOC are relatively chemoresistant.
* Prevention and screening are currently limited.
* There is no validated screening method for standard practice.
* Ultrasound has produced false-negatives and false-positives and is not recommended.
* Screening high-risk populations has been shown to be effective but requires prior knowledge of family history.
* Some biomarkers have been utilized such as HE4 in combination with CA125.
* Another approach has been to evaluate DNA methylation in cfDNA.










### Summary

I have shifted focus to begin a comprehensive literature review. I've started by taking a look at a review of the current landscape of treatment and outcomes for ovca [@Lheureux2019]. This was helpful to fill many of the holes I have in my general knowledge of cancer biology with the more specific case studies of ovca. Much of this reinforced the review that Tim has given in our meetings, which was helpful. Along with this, I began to work my way through the Seurat tutorial. I was able to locate the v3 tutorial and pbmc data set and made it about half way through the written guide. I plan on getting fully through this tomorrow as well as meeting with Tim to discuss some questions I developed from the ovca review.

#### Completed


Created    Completed  Age Project        Description
---------- ---------- --- -------------- ---------------------------------
2019-05-20 2019-05-28 7d  starrlab.ocpmi find ovca review re cur treatment

1 task

***

## 29

### TODO

ID Age  Project        Tags           Description                           Urg
-- ---- -------------- -------------- ------------------------------------- ----
75 5d   starrlab                      agenda items [3]                      2.03
74 6d   starrlab       computer w     make custom csl                       1.93
76 5d   starrlab       computer n     GCP Human RSRCH Biomed Studies        1.93
77 5d   starrlab       computer n     CITI research ethics training         1.93
78 5d   starrlab       computer n     University Information Security       1.93
                                      Awareness Training
90 1d   starrlab       computer n     find alternative to jabref            1.91
92 22h  starrlab       computer n     fix god damn git issue                 1.9
97 1min starrlab       computer n     fix hostname issue w/ mesabi           1.9
88 2d   starrlab.ocpmi computer lit n read papalexi 2018 [2]                2.91
89 1d   starrlab.ocpmi computer lit w read macintyre 2018 [2]               2.91
91 1d   starrlab.ocpmi lit s          read sahin 2017 [1]                   2.71
66 8d   starrlab.ocpmi computer lit w read starr paper re 1st pt            2.04
67 8d   starrlab.ocpmi computer lit w read ocpmi summary                    2.04
69 8d   starrlab.ocpmi computer lit w read single cell paper zheng 2017     2.04
70 8d   starrlab.ocpmi computer lit w read venteicher - CNV paper           2.04
68 8d   starrlab.ocpmi computer w     run surat tutorial w/ pt 59           1.94
96 2min starrlab.ocpmi computer n     run seurat with pbmc data              1.9

17 tasks


### wkly mtg w/ Tim

present: chris, tim

Description
--------------------------------------------------------------------------------
agenda items
  2019-05-27 validation in diff between lumpectomy and mastectomy
  2019-05-28 elaborate on lack of randomized controls for ovca treatments
  2019-05-28 discuss vacation
  2019-05-28 discuss chemotherapy at large
  2019-05-28 review parp inhibitor mechanism
  2019-05-28 are occc and lgsoc our general targets given their chemoresistance?
  2019-05-28 how do our questions differ from the findings of kondrashova 2017 -
RAD51 mutations acquired resistance

1 task

#### mtg notes

* mysql reference: single cell rnaseq analysis protocol
* figure out mesabi hostname issue
* explore starrt2 msi path
    + [/home/starrt2/data_release/umgc/hiseq/170627_D00635_0259_ACB8R1ANXX/Starr_Project_037/Analysis/cellranger-Starr_037/filtered_gene_bc_matrices/GRCh38]
* There is no specific focus in the pathology we're focusing on. OCCC and LGSOC are being considered just as readily as other subtypes.
* Kondrashova 2017 (RAD51 mutations leading to chemoresistance) is potentially a good example of what our research questions will be.
* Tim's focus is more basic science oriented. Potential applications would be to identify when a pt would be resistant and that they should pursue other treatment from the outset.
* All of our findings from this data set will require wetlab validation
* This data set is incredibly unique. We should be able to discover something nobody else has.
* Data validation has generally been handled on the project so far.
    + Objective: determine cell type.
* There's no way to call copy number variation from this data. That would be novel.

### MSI Project Directory

I logged into the MSI account for the first time today. There is a relatively deep file structure which is composed of a combination of processed data from umgc/msi and other collaborator work. I have a personal directory in this space along with access to other shared directories.

The root project directory is:

```{r eval=F}
/home/starrt2/
```

I have a personal directory under the root. At the same level, there is also a shared directory which includes some content that is under active collaboration.

```{r eval=F}
/home/starrt2/tasta005
```

Processed data can be found in the hiseq folder under data_release. This data is mirrored in the msi file structure. Each file set is named by a unique ID which follows a naming convention given by umgc.

* Look for the leading 6 digit integer as a point of reference for the data_release folder.
* The Starr_Project sub folder is given a number which is considered the MSI and sample ID.

```{r eval=F}
/home/starrt2/data_release/umgc/hiseq
```


The Analysis directory contains the source data

* illumina-basicQC # quality control data
* cellranger-`*`
    + analysis # ?
    + filtered_gene_bc_matricies # data source that has been filtered by "true cell content"
    + raw_gene_bc_matrices # matrix data that includes unfiltered reads or all read content
    + web_summary.html # interactive cell ranger summary

An example of a complete data source location is below:

```{r eval=F}
/home/starrt2/data_release/umgc/hiseq/170627_D00635_0259_ACB8R1ANXX/Starr_Project_037/Analysis/cellranger-Starr_037/filtered_gene_bc_matrices/GRCh38
```

### Summary

I met with Tim today for our weekly meeting. We discussed some aspects of ovca content that I had been reading to develop my general knowledge. I major element of this conversation that will adjust what my intentions were going forward dealt with the nature of statistically validating the results of our pipeline. My original thought was that there was some need to dig into the backend of the Seurat/cellranger packages to both confirm and optimize the output. It sounds as though this has been done to a degree that Tim is satisfied with and that we can move on to explore questions that are Biological in nature.

Additionally, I did a little house keeping by establishing the first part of the methods page in my notebook. I explored the MSI project directory for the first time today and recorded some documentation about the file structure and data locations. My expectation is to pull what would be method oriented documentation from Notes -> Methods or Notes -> Results as these things need to be pinned for repeat use and reference.

#### Completed

Created    Completed  Age Project  Description
---------- ---------- --- -------- ----------------------------
2019-05-28 2019-05-29 1d  starrlab fix god damn git issue
2019-05-29 2019-05-29 2h  starrlab fix hostname issue w/ mesabi

2 tasks

***

## 30

### TODO


ID Age Project        Tags           Description                            Urg
-- --- -------------- -------------- -------------------------------------- ----
75 6d  starrlab                      agenda items [3]                       2.03
74 6d  starrlab       computer w     make custom csl                        1.93
76 5d  starrlab       computer n     GCP Human RSRCH Biomed Studies         1.93
77 5d  starrlab       computer n     CITI research ethics training          1.93
78 5d  starrlab       computer n     University Information Security        1.93
                                     Awareness Training
90 1d  starrlab       computer n     find alternative to jabref             1.91
88 3d  starrlab.ocpmi computer lit n read papalexi 2018 [2]                 2.92
89 2d  starrlab.ocpmi computer lit w read macintyre 2018 [2]                2.91
91 1d  starrlab.ocpmi lit s          read sahin 2017 [1]                    2.71
66 9d  starrlab.ocpmi computer lit w read starr paper re 1st pt             2.05
67 9d  starrlab.ocpmi computer lit w read ocpmi summary                     2.05
69 9d  starrlab.ocpmi computer lit w read single cell paper zheng 2017      2.05
70 9d  starrlab.ocpmi computer lit w read venteicher - CNV paper            2.05
68 9d  starrlab.ocpmi computer w     run surat tutorial w/ pt 59            1.95
95 20h starrlab.ocpmi computer n     run seurat with pbmc data               1.9

15 tasks

### Seurat Tutorial pbmc3k

I ran the v3 Seurat tutorial [@2019d] to completion using the pbmc dataset. This outline was very intuitive and worked through the pipeline seamlessly. There are several points that warrant further exploration.

The output of the v3 PBMC data tutorial can be found [here](/home/chris/starr_lab/ocpmi/code/seurat_tutorial/pbmc3k_tutorial.html).


### Summary

Briefly, I wrapped up safety training today, so that is all taken care of. The rest of my afternoon was dedicated to working my way through the default v3 Seurat tutorial. The tutorial comes packaged in a markdown file, so the presentation and documentation is very effective. To some extent, the documentation is so good I am a little concerned it encourages over use of the default settings. I haven't looked at any other processing that's been completed by Tim et al, so I'll have to keep an eye out for how adjustments have been made. Tim mentioned there has been a limited amount of customization, so this may stand as a short term area of focus. Generally, I think can consider today's work a reasonable summary of the process and syntax associated with the toolset. With this, I would like to spend some time digging into the nuances of the pipeline's analysis. I will probably spend most of tomorrow doing that. Much of the default analysis hinges on PCA dimensionality, so I feel I need to brush up on my knowledge in this area. Going forward I need to run the same pipeline using the new pt59 data and evaluate results.

#### Completed


Created    Completed  Age Project        Description
---------- ---------- --- -------------- ---------------------------------------
2019-05-24 2019-05-30 6d  starrlab       GCP Human RSRCH Biomed Studies
2019-05-24 2019-05-30 6d  starrlab       CITI research ethics training
2019-05-24 2019-05-30 6d  starrlab       University Information Security
                                         Awareness Training
2019-05-29 2019-05-30 1d  starrlab.ocpmi run seurat with pbmc data

4 tasks

***

## 31

### TODO

ID Age Project        Tags       Description                                Urg
-- --- -------------- ---------- ------------------------------------------ ----
73 8d  starrlab                  agenda items [3]                           2.04
72 8d  starrlab       computer w make custom csl                            1.94
83 3d  starrlab       computer n find alternative to jabref                 1.92
91 1d  starrlab       computer n review pca concepts                        1.91
92 1d  starrlab       computer n compare default seurat tutorial to         1.91
                                 starrlab protocol
93 1d  starrlab       computer n deep dive into default seurat analysis     1.91
94 1d  starrlab       computer n update notes regarding citation tools re   1.91
                                 mendeley
81 4d  starrlab.ocpmi lit n      read papalexi 2018 [2]                     2.82
82 4d  starrlab.ocpmi lit w      read macintyre 2018 [2]                    2.82
84 3d  starrlab.ocpmi lit s      read sahin 2017 [1]                        2.72
65 11d starrlab.ocpmi lit w      read starr paper re 1st pt                 1.96
66 11d starrlab.ocpmi lit w      read ocpmi summary                         1.96
67 11d starrlab.ocpmi computer w run surat tutorial w/ pt 59                1.96
68 11d starrlab.ocpmi lit w      read single cell paper zheng 2017          1.96
69 11d starrlab.ocpmi lit w      read venteicher - CNV paper                1.96

15 tasks

### Mendeley

At the risk of seeming indecisive, I've decided to switch citation managers. Given my use if the BibTex file format, it is effectively painless to switch. JabRef showed some promise but presented a few bugs related to database calls which would render much of the value of using it worthless. Mendeley has both a native linux application along with a web import tool for firefox. This is the only major citation tool I've come across that combines those two things. Here are a few key features of my use:

* All content will still be synced back to a BibTex file name `library.bib` located under the `starr_lab/ocpmi/notebook/` subdirectory
* There are watched folders which will draw sources from any pdf put into them at `starr_lab/ocpmi/ref/`.
* I've decided to use the IEEE with url csl style taken from the zotero style repository. [@2019]

### [@Winterhoff2019]

#### Single-Cell Sequencing in Ovarian Cancer: a new frontier in precision Medicine

Key Points

* Intratumor heterogeneity is an inherent property of ovarian cancer, which impacts clinical outcomes
* SC sequencing tech enables characterization of intratumor heterogeneity in ovarian cancer
* To date, single-cell studies on ovarian cancers are scarce, which limits the opportunity to investigate effects on clinical outcomes
* Larger data sets are needed to establish potential associations between the unique information captured by SC sequencing and clinically relevant outcomes in ovca

Notes

* They claim they found genes associated with ovca stem cells.
* When they evaluated based on molecular eoc cancer subtypes, all subtypes were present.
* Studying the immune microenvironment of a tumor becomes more feasible with scRNA sequencing
* Evaluation of cell plasticity as it relates to cell type transitions may provide insight into the development of chemoresistance
* Understanding the lineage of the cellular qualities of metastatic dissemination is made possible with scRNA data

### Using Ancestry to find the Stem Cell

Question: can we use sequence-based lineage tracing to identify a common progenitor cell expression profile?

A thought I had while looking at some lit tonight led me to wonder if it's worth exploring sequence-based cell lineage across individual cells rather than using differential expression patterns as a general means of characterization. I will need to do a dedicated lit search to explore this concept, but I imagine the use of single cell data for lineage tracing is relatively limited. At a minimum, Tim has alluded to the notion that our data set is unique among the ovca model, which would lend itself to a novel application of single cell lineage tracing. Regardless, this could be a relatively significant deviation from their current pipeline, so I will have to discuss it with Tim.

### Updated create_site.R Script

I made a minor update to the script I use to automate total site rendering to clean up the directory a bit. I don't have any use in the parent notebook/ directory for individual html files since they are generated in the site render. The `ifelse` at the end of the script clears these html files from the notebook directory.

```{r eval=F}
rmarkdown::render("index.Rmd")
rmarkdown::render("notes.Rmd")
rmarkdown::render("results.Rmd")
rmarkdown::render("methods.Rmd")
rmarkdown::render("references.Rmd")
rmarkdown::render_site("index.Rmd")

fl <- Sys.glob("*.html")

ifelse(file.exists(fl), file.remove(fl))
```

### Summary

I was able to explore a few different fronts today. I put some more work in to polishing my citation management. This is an area I have been woefully lacking in from past experience, so I do not discount the extra work applied to make sure this is at a publication-ready level.

Additionally, I worked my way through the earliest publication related to the ocpmi project [@Winterhoff2019]. Through this I was able to develop some insight to the early considerations this group gave to this data as they were only working with 66 individual cells. They offered a broad scope of the advantages single cell tech offers along with pointing me to a few sources that establish the current state of the field.

Last, I can't say what generated this thought specifically, but I came across what my limited knowledge of the field would call a good idea. All of our discussions on this data set up to this point have centered on the use of UMI counts and gene expression levels as a means to develop differentiated cell profiles. A consideration I would potentially like to explore is the possibility that we look at sequence level data to identify cell lineage in search of a common progenitor. I have little basis to say if this is a reasonable or even realistic scientific question, but I feel this could be a substantive, novel insight given the nature of our data set. It's quite possible this approach would require too great an investment in shifting the informatics work toward raw sequence analysis, but I will discuss it with Tim nonetheless.

#### Completed

Created    Completed  Age Project        Description
---------- ---------- --- -------------- --------------------------
2019-05-20 2019-06-01 11d starrlab.ocpmi read starr paper re 1st pt
2019-05-28 2019-05-31 3d  starrlab       find alternative to jabref
2019-05-30 2019-05-31 1d  starrlab       update notes regarding citation tools re
                                         mendeley
2019-05-23 2019-05-31 8d  starrlab       make custom csl

4 tasks


***

***

# References



